<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Brian M. Schilder" />


<title>echolocatoR: full pipeline vignette</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">echolocatoR: full pipeline vignette</h1>
<h4 class="author">Brian M. Schilder</h4>
<h4 class="date"><br>Most Recent Update:<br> 2020-06-17</h4>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(echolocatoR) </a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#&gt;   method from   </span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">#&gt;   +.gg   ggplot2</span></a></code></pre></div>
<div id="full-pipeline" class="section level1">
<h1>Full pipeline</h1>
<div id="prepare-top_snps-data.frame" class="section level2">
<h2>Prepare <code>top_SNPs</code> data.frame</h2>
<ul>
<li>To enable rapid fine-mapping of many loci,<br />
you can create a <code>top_SNPs</code> data.frame which contains the position<br />
of the lead/index SNP within each locus.</li>
<li><code>finemap_loci()</code> (see next step) will then use this info to extract subsets of the<br />
full GWAS/QTL summary statistics using windows centered on each lead/index SNP.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">data</span>(<span class="st">&quot;Nalls_top_SNPs&quot;</span>);</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">top_SNPs &lt;-<span class="st"> </span><span class="kw">import_topSNPs</span>(</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">  <span class="co"># topSS = &quot;~/Desktop/Fine_Mapping/Data/GWAS/Nalls23andMe_2019/Nalls2019_TableS2.xlsx&quot;,</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">  <span class="dt">topSS =</span> Nalls_top_SNPs,</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">  <span class="dt">chrom_col =</span> <span class="st">&quot;CHR&quot;</span>, <span class="dt">position_col =</span> <span class="st">&quot;BP&quot;</span>, <span class="dt">snp_col=</span><span class="st">&quot;SNP&quot;</span>,</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">  <span class="dt">pval_col=</span><span class="st">&quot;P, all studies&quot;</span>, <span class="dt">effect_col=</span><span class="st">&quot;Beta, all studies&quot;</span>, <span class="dt">gene_col=</span><span class="st">&quot;Nearest Gene&quot;</span>,</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">  <span class="dt">group_by_locus =</span> T,</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">  <span class="dt">locus_col =</span> <span class="st">&quot;Locus Number&quot;</span>,</a>
<a class="sourceLine" id="cb2-9" data-line-number="9">  <span class="dt">remove_variants =</span> <span class="st">&quot;rs34637584&quot;</span>) </a></code></pre></div>
</div>
<div id="run-fine-mapping-pipeline" class="section level2">
<h2>Run fine-mapping pipeline</h2>
<p>For a full description of all arguments, see <code>?finemap_loci</code>.</p>
<p>Here are some key arguments:<br />
* <em>results_dir</em>: Where you want to store all of your results.<br />
* <em>finemap_methods</em>: Which fine-mapping methods you want to run (currently includes ABF, FINEMAP, SUSIE, POLYFUN_SUSIE, and COJO).<br />
* <em>bp_distance</em>: Controls window size. Specifically, <code>bp_distance</code> is the number of basepairs upstream/downstream you want to extract for each locus. For example, if you want a 2Mb window (+/- 1Mb from the lead/index SNP in <code>top_SNPs</code>), set <code>bp_distance=1e+06</code>.<br />
* <em>plot.window</em>: Zoom in/out from the center of each locus when producing the multiview plot.<br />
You can adjust this separately from <code>bp_distance</code> so that you don’t have rerun the whole pipeline each time (locus subsets, LD matrices, and fine-mapping results are all automatically saved in locus-specific folders).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">Nalls23andMe_<span class="fl">2019.</span>results &lt;-<span class="st"> </span><span class="kw">finemap_loci</span>(<span class="co"># GENERAL ARGUMENTS </span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">                                          <span class="dt">top_SNPs =</span> top_SNPs, </a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                                          <span class="dt">results_dir =</span> <span class="st">&quot;~/Desktop/results&quot;</span>,</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">                                          <span class="dt">loci =</span> <span class="kw">c</span>(<span class="st">&quot;BST1&quot;</span>,<span class="st">&quot;MEX3C&quot;</span>),<span class="co">#top_SNPs$Gene, </span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5">                                          <span class="dt">dataset_name =</span> <span class="st">&quot;Nalls23andMe_2019&quot;</span>,</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">                                          <span class="dt">dataset_type =</span> <span class="st">&quot;GWAS&quot;</span>,  </a>
<a class="sourceLine" id="cb3-7" data-line-number="7">                                          <span class="dt">force_new_subset =</span> F,</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">                                          <span class="dt">force_new_LD =</span> F,</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">                                          <span class="dt">force_new_finemap =</span> F,</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">                                          <span class="dt">remove_tmps =</span> T,</a>
<a class="sourceLine" id="cb3-11" data-line-number="11">                                          </a>
<a class="sourceLine" id="cb3-12" data-line-number="12">                 <span class="co"># SUMMARY STATS ARGUMENTS</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13">                 <span class="dt">fullSS_path =</span> <span class="st">&quot;~/Desktop/nallsEtAl2019_allSamples_allVariants.mod.txt.gz&quot;</span>,</a>
<a class="sourceLine" id="cb3-14" data-line-number="14">                 <span class="dt">query_by =</span><span class="st">&quot;tabix&quot;</span>,</a>
<a class="sourceLine" id="cb3-15" data-line-number="15">                 <span class="dt">chrom_col =</span> <span class="st">&quot;CHR&quot;</span>, <span class="dt">position_col =</span> <span class="st">&quot;POS&quot;</span>, <span class="dt">snp_col =</span> <span class="st">&quot;RSID&quot;</span>,</a>
<a class="sourceLine" id="cb3-16" data-line-number="16">                 <span class="dt">pval_col =</span> <span class="st">&quot;p&quot;</span>, <span class="dt">effect_col =</span> <span class="st">&quot;beta&quot;</span>, <span class="dt">stderr_col =</span> <span class="st">&quot;se&quot;</span>,</a>
<a class="sourceLine" id="cb3-17" data-line-number="17">                 <span class="dt">freq_col =</span> <span class="st">&quot;freq&quot;</span>, <span class="dt">MAF_col =</span> <span class="st">&quot;calculate&quot;</span>,</a>
<a class="sourceLine" id="cb3-18" data-line-number="18">                 <span class="dt">A1_col =</span> <span class="st">&quot;A1&quot;</span>,</a>
<a class="sourceLine" id="cb3-19" data-line-number="19">                 <span class="dt">A2_col =</span> <span class="st">&quot;A2&quot;</span>,</a>
<a class="sourceLine" id="cb3-20" data-line-number="20">                 </a>
<a class="sourceLine" id="cb3-21" data-line-number="21">                 <span class="co"># FILTERING ARGUMENTS</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22">                 <span class="dt">bp_distance =</span> <span class="dv">500000</span><span class="op">*</span><span class="dv">2</span>,</a>
<a class="sourceLine" id="cb3-23" data-line-number="23">                 <span class="dt">min_MAF =</span> <span class="fl">0.001</span>, </a>
<a class="sourceLine" id="cb3-24" data-line-number="24">                 <span class="dt">trim_gene_limits =</span> F,</a>
<a class="sourceLine" id="cb3-25" data-line-number="25">                 </a>
<a class="sourceLine" id="cb3-26" data-line-number="26">                 <span class="co"># FINE-MAPPING ARGUMENTS</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27">                 <span class="dt">finemap_methods =</span> <span class="kw">c</span>(<span class="st">&quot;ABF&quot;</span>,<span class="st">&quot;SUSIE&quot;</span>,<span class="st">&quot;FINEMAP&quot;</span>),</a>
<a class="sourceLine" id="cb3-28" data-line-number="28">                 <span class="dt">n_causal =</span> <span class="dv">5</span>,</a>
<a class="sourceLine" id="cb3-29" data-line-number="29">                 <span class="dt">PP_threshold =</span> <span class="fl">.95</span>,</a>
<a class="sourceLine" id="cb3-30" data-line-number="30">                 </a>
<a class="sourceLine" id="cb3-31" data-line-number="31">                 <span class="co"># LD ARGUMENTS </span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32">                 <span class="dt">LD_reference =</span> <span class="st">&quot;UKB&quot;</span>,<span class="co">#&quot;1KG_Phase1&quot;,</span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33">                 <span class="dt">superpopulation =</span> <span class="st">&quot;EUR&quot;</span>,</a>
<a class="sourceLine" id="cb3-34" data-line-number="34">                 <span class="dt">LD_download_method =</span> <span class="st">&quot;axel&quot;</span>,</a>
<a class="sourceLine" id="cb3-35" data-line-number="35">                 </a>
<a class="sourceLine" id="cb3-36" data-line-number="36">                 <span class="co"># PLOT ARGUMENTS </span></a>
<a class="sourceLine" id="cb3-37" data-line-number="37">                 <span class="co">## general   </span></a>
<a class="sourceLine" id="cb3-38" data-line-number="38">                 <span class="dt">plot.types=</span><span class="kw">c</span>(<span class="st">&quot;fancy&quot;</span>),</a>
<a class="sourceLine" id="cb3-39" data-line-number="39">                 <span class="dt">plot.window =</span> <span class="dv">100000</span>,</a>
<a class="sourceLine" id="cb3-40" data-line-number="40">                 <span class="co">## XGR</span></a>
<a class="sourceLine" id="cb3-41" data-line-number="41">                 <span class="co"># plot.XGR_libnames=c(&quot;ENCODE_TFBS_ClusteredV3_CellTypes&quot;), </span></a>
<a class="sourceLine" id="cb3-42" data-line-number="42">                 <span class="co">## Roadmap</span></a>
<a class="sourceLine" id="cb3-43" data-line-number="43">                 <span class="dt">plot.Roadmap =</span> F,</a>
<a class="sourceLine" id="cb3-44" data-line-number="44">                 <span class="dt">plot.Roadmap_query =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb3-45" data-line-number="45">                 <span class="co"># Nott et al. (2019)</span></a>
<a class="sourceLine" id="cb3-46" data-line-number="46">                 <span class="dt">plot.Nott_epigenome =</span> T, </a>
<a class="sourceLine" id="cb3-47" data-line-number="47">                 <span class="dt">plot.Nott_binwidth =</span> <span class="dv">100</span></a>
<a class="sourceLine" id="cb3-48" data-line-number="48">                 )</a>
<a class="sourceLine" id="cb3-49" data-line-number="49"><span class="co">#&gt; 🦇 🦇 🦇 BST1 🦇 🦇 🦇</span></a>
<a class="sourceLine" id="cb3-50" data-line-number="50"><span class="co">#&gt; ------------------ Step 1: Query ---------------</span></a>
<a class="sourceLine" id="cb3-51" data-line-number="51"><span class="co">#&gt; [1] &quot;+ Extracting relevant variants from fullSS...&quot;</span></a>
<a class="sourceLine" id="cb3-52" data-line-number="52"><span class="co">#&gt; [1] &quot;++ Query Method: 'tabix&quot;</span></a>
<a class="sourceLine" id="cb3-53" data-line-number="53"><span class="co">#&gt; [1] &quot;---Min snp position: 14737348 ---&quot;</span></a>
<a class="sourceLine" id="cb3-54" data-line-number="54"><span class="co">#&gt; [1] &quot;---Max snp position: 16737348 ---&quot;</span></a>
<a class="sourceLine" id="cb3-55" data-line-number="55"><span class="co">#&gt; [1] &quot;TABIX:: Existing indexed tabix file detected&quot;</span></a>
<a class="sourceLine" id="cb3-56" data-line-number="56"><span class="co">#&gt; [1] &quot;TABIX:: Extracting subset of sum stats&quot;</span></a>
<a class="sourceLine" id="cb3-57" data-line-number="57"><span class="co">#&gt; [1] &quot;++ Returning 6282 x 11 data.table&quot;</span></a>
<a class="sourceLine" id="cb3-58" data-line-number="58"><span class="co">#&gt; [1] &quot;++ Saving query ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/BST1_Nalls23andMe_2019_subset.tsv.gz&quot;</span></a>
<a class="sourceLine" id="cb3-59" data-line-number="59"><span class="co">#&gt; [1] &quot;&quot;</span></a>
<a class="sourceLine" id="cb3-60" data-line-number="60"><span class="co">#&gt; ---------------- Step 1.5: Standardize ----------</span></a>
<a class="sourceLine" id="cb3-61" data-line-number="61"><span class="co">#&gt; [1] &quot;+Inferring MAF from frequency column...&quot;</span></a>
<a class="sourceLine" id="cb3-62" data-line-number="62"><span class="co">#&gt; [1] &quot;+ Standardize:: Calculating proportion of cases.&quot;</span></a>
<a class="sourceLine" id="cb3-63" data-line-number="63"><span class="co">#&gt; [1] &quot;+ Calculating t-statistic from Effect and StdErr...&quot;</span></a>
<a class="sourceLine" id="cb3-64" data-line-number="64"><span class="co">#&gt; [1] &quot;+ leadSNP missing. Assigning new one by min p-value.&quot;</span></a>
<a class="sourceLine" id="cb3-65" data-line-number="65"><span class="co">#&gt; [1] &quot;+ Extraction completed in 5.88 seconds&quot;</span></a>
<a class="sourceLine" id="cb3-66" data-line-number="66"><span class="co">#&gt; [1] &quot;+ 6273 SNPs x  15 columns&quot;</span></a>
<a class="sourceLine" id="cb3-67" data-line-number="67"><span class="co">#&gt; [1] &quot;+ Removing subset tmp...&quot;</span></a>
<a class="sourceLine" id="cb3-68" data-line-number="68"><span class="co">#&gt; --- Step 2: Calculate Linkage Disequilibrium ---</span></a>
<a class="sourceLine" id="cb3-69" data-line-number="69"><span class="co">#&gt; [1] &quot;LD:: Using UK Biobank LD reference panel...&quot;</span></a>
<a class="sourceLine" id="cb3-70" data-line-number="70"><span class="co">#&gt; [1] &quot;+ UKB LD file name: chr4_14000001_17000001&quot;</span></a>
<a class="sourceLine" id="cb3-71" data-line-number="71"><span class="co">#&gt; [1] &quot;+ LD:: Pre-existing UKB_LD.RDS file detected. Importing...&quot;</span></a>
<a class="sourceLine" id="cb3-72" data-line-number="72"><span class="co">#&gt; -------------- Step 3: Filter SNPs -------------</span></a>
<a class="sourceLine" id="cb3-73" data-line-number="73"><span class="co">#&gt; [1] &quot;++ Post-filtered data: 6273 x 15&quot;</span></a>
<a class="sourceLine" id="cb3-74" data-line-number="74"><span class="co">#&gt; -------- Step 4: Statistically Fine-map --------</span></a>
<a class="sourceLine" id="cb3-75" data-line-number="75"><span class="co">#&gt; [1] &quot;++ Previously multi-fine-mapped results identified. Importing...&quot;</span></a>
<a class="sourceLine" id="cb3-76" data-line-number="76"><span class="co">#&gt; [1] &quot;++ Fine-mapping with ' ABF, SUSIE, FINEMAP ' completed in  0.08&quot;</span></a>
<a class="sourceLine" id="cb3-77" data-line-number="77"><span class="co">#&gt; [1] &quot;+ Identifying Consensus SNPs...&quot;</span></a>
<a class="sourceLine" id="cb3-78" data-line-number="78"><span class="co">#&gt; [1] &quot;++ support_thresh = 2&quot;</span></a>
<a class="sourceLine" id="cb3-79" data-line-number="79"><span class="co">#&gt; [1] &quot;+ Calculating mean Posterior Probability (mean.PP)...&quot;</span></a>
<a class="sourceLine" id="cb3-80" data-line-number="80"><span class="co">#&gt; [1] &quot;++ 3 fine-mapping methods used.&quot;</span></a>
<a class="sourceLine" id="cb3-81" data-line-number="81"><span class="co">#&gt; [1] &quot;++ 3 Credible Set SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb3-82" data-line-number="82"><span class="co">#&gt; [1] &quot;++ 0 Consensus SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb3-83" data-line-number="83"><span class="co">#&gt; --------------- Step 7: Visualize --------------</span></a>
<a class="sourceLine" id="cb3-84" data-line-number="84"><span class="co">#&gt; [1] &quot;+ Identifying Consensus SNPs...&quot;</span></a>
<a class="sourceLine" id="cb3-85" data-line-number="85"><span class="co">#&gt; [1] &quot;++ support_thresh = 2&quot;</span></a>
<a class="sourceLine" id="cb3-86" data-line-number="86"><span class="co">#&gt; [1] &quot;+ Calculating mean Posterior Probability (mean.PP)...&quot;</span></a>
<a class="sourceLine" id="cb3-87" data-line-number="87"><span class="co">#&gt; [1] &quot;++ 3 fine-mapping methods used.&quot;</span></a>
<a class="sourceLine" id="cb3-88" data-line-number="88"><span class="co">#&gt; [1] &quot;++ 3 Credible Set SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb3-89" data-line-number="89"><span class="co">#&gt; [1] &quot;++ 0 Consensus SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb3-90" data-line-number="90"><span class="co">#&gt; [1] &quot;LD Matrix dimensions: 6216 x 6216&quot;</span></a>
<a class="sourceLine" id="cb3-91" data-line-number="91"><span class="co">#&gt; [1] &quot;Extracting LD subset for lead SNP: rs4698412&quot;</span></a>
<a class="sourceLine" id="cb3-92" data-line-number="92"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-93" data-line-number="93"><span class="co">#&gt; [1] &quot;++ GGBIO:: ABF Track&quot;</span></a>
<a class="sourceLine" id="cb3-94" data-line-number="94"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-95" data-line-number="95"><span class="co">#&gt; [1] &quot;++ GGBIO:: SUSIE Track&quot;</span></a>
<a class="sourceLine" id="cb3-96" data-line-number="96"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-97" data-line-number="97"><span class="co">#&gt; [1] &quot;++ GGBIO:: FINEMAP Track&quot;</span></a>
<a class="sourceLine" id="cb3-98" data-line-number="98"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-99" data-line-number="99"><span class="co">#&gt; [1] &quot;++ GGBIO:: mean Track&quot;</span></a>
<a class="sourceLine" id="cb3-100" data-line-number="100"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-101" data-line-number="101"><span class="co">#&gt; [1] &quot;++ GGBIO:: Gene Model Track&quot;</span></a>
<a class="sourceLine" id="cb3-102" data-line-number="102"><span class="co">#&gt; [1] &quot;+ Annotating at transcript-level.&quot;</span></a>
<a class="sourceLine" id="cb3-103" data-line-number="103"><span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span></a>
<a class="sourceLine" id="cb3-104" data-line-number="104"><span class="co">#&gt; [1] &quot;Importing Astrocyte enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb3-105" data-line-number="105"><span class="co">#&gt; [1] &quot;Importing Astrocyte promoters ...&quot;</span></a>
<a class="sourceLine" id="cb3-106" data-line-number="106"><span class="co">#&gt; [1] &quot;Importing Neuronal enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb3-107" data-line-number="107"><span class="co">#&gt; [1] &quot;Importing Neuronal promoters ...&quot;</span></a>
<a class="sourceLine" id="cb3-108" data-line-number="108"><span class="co">#&gt; [1] &quot;Importing Oligo enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb3-109" data-line-number="109"><span class="co">#&gt; [1] &quot;Importing Oligo promoters ...&quot;</span></a>
<a class="sourceLine" id="cb3-110" data-line-number="110"><span class="co">#&gt; [1] &quot;Importing Microglia enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb3-111" data-line-number="111"><span class="co">#&gt; [1] &quot;Importing Microglia promoters ...&quot;</span></a>
<a class="sourceLine" id="cb3-112" data-line-number="112"><span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span></a>
<a class="sourceLine" id="cb3-113" data-line-number="113"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-114" data-line-number="114"><span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span></a>
<a class="sourceLine" id="cb3-115" data-line-number="115"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-116" data-line-number="116"><span class="co">#&gt; [1] &quot;++ Nott sn-epigenomics:: Returning PLAC-seq track.&quot;</span></a>
<a class="sourceLine" id="cb3-117" data-line-number="117"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-118" data-line-number="118"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-119" data-line-number="119"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-120" data-line-number="120"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-121" data-line-number="121"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-122" data-line-number="122"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-123" data-line-number="123"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-124" data-line-number="124"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-125" data-line-number="125"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-126" data-line-number="126"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-127" data-line-number="127"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-128" data-line-number="128"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-129" data-line-number="129"><span class="co">#&gt; [1] &quot;+ GGBIO:: Saving plot ==&gt; /Users/schilder/Desktop/results/GWAS/Nalls23andMe_2019/BST1/multiview_BST1_100kb.png&quot;</span></a>
<a class="sourceLine" id="cb3-130" data-line-number="130"><span class="co">#&gt; Warning: Removed 98 rows containing non-finite values (stat_smooth).</span></a>
<a class="sourceLine" id="cb3-131" data-line-number="131"><span class="co">#&gt; Warning: Removed 6212 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-132" data-line-number="132"><span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-133" data-line-number="133"><span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-134" data-line-number="134"></a>
<a class="sourceLine" id="cb3-135" data-line-number="135"><span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-136" data-line-number="136"><span class="co">#&gt; Warning: Removed 10 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-137" data-line-number="137"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-138" data-line-number="138"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-139" data-line-number="139"></a>
<a class="sourceLine" id="cb3-140" data-line-number="140"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-141" data-line-number="141"><span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span></a>
<a class="sourceLine" id="cb3-142" data-line-number="142"><span class="co">#&gt; Warning: Removed 98 rows containing non-finite values (stat_smooth).</span></a>
<a class="sourceLine" id="cb3-143" data-line-number="143"><span class="co">#&gt; Warning: Removed 6212 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-144" data-line-number="144"><span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-145" data-line-number="145"><span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-146" data-line-number="146"></a>
<a class="sourceLine" id="cb3-147" data-line-number="147"><span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-148" data-line-number="148"><span class="co">#&gt; Warning: Removed 10 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-149" data-line-number="149"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-150" data-line-number="150"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-151" data-line-number="151"></a>
<a class="sourceLine" id="cb3-152" data-line-number="152"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-153" data-line-number="153"><span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span></a>
<a class="sourceLine" id="cb3-154" data-line-number="154"><span class="co">#&gt; Error in grid.Call(C_convert, x, as.integer(whatfrom), as.integer(whatto),  : </span></a>
<a class="sourceLine" id="cb3-155" data-line-number="155"><span class="co">#&gt;   Viewport has zero dimension(s)</span></a>
<a class="sourceLine" id="cb3-156" data-line-number="156"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-157" data-line-number="157"><span class="co">#&gt; Fine-mappping complete:BST1</span></a>
<a class="sourceLine" id="cb3-158" data-line-number="158"><span class="co">#&gt; Time difference of 1.2 mins</span></a>
<a class="sourceLine" id="cb3-159" data-line-number="159"><span class="co">#&gt; 🦇 🦇 🦇 MEX3C 🦇 🦇 🦇</span></a>
<a class="sourceLine" id="cb3-160" data-line-number="160"><span class="co">#&gt; ------------------ Step 1: Query ---------------</span></a>
<a class="sourceLine" id="cb3-161" data-line-number="161"><span class="co">#&gt; [1] &quot;+ Extracting relevant variants from fullSS...&quot;</span></a>
<a class="sourceLine" id="cb3-162" data-line-number="162"><span class="co">#&gt; [1] &quot;++ Query Method: 'tabix&quot;</span></a>
<a class="sourceLine" id="cb3-163" data-line-number="163"><span class="co">#&gt; [1] &quot;---Min snp position: 47683589 ---&quot;</span></a>
<a class="sourceLine" id="cb3-164" data-line-number="164"><span class="co">#&gt; [1] &quot;---Max snp position: 49683589 ---&quot;</span></a>
<a class="sourceLine" id="cb3-165" data-line-number="165"><span class="co">#&gt; [1] &quot;TABIX:: Existing indexed tabix file detected&quot;</span></a>
<a class="sourceLine" id="cb3-166" data-line-number="166"><span class="co">#&gt; [1] &quot;TABIX:: Extracting subset of sum stats&quot;</span></a>
<a class="sourceLine" id="cb3-167" data-line-number="167"><span class="co">#&gt; [1] &quot;++ Returning 5405 x 11 data.table&quot;</span></a>
<a class="sourceLine" id="cb3-168" data-line-number="168"><span class="co">#&gt; [1] &quot;++ Saving query ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/MEX3C/MEX3C_Nalls23andMe_2019_subset.tsv.gz&quot;</span></a>
<a class="sourceLine" id="cb3-169" data-line-number="169"><span class="co">#&gt; [1] &quot;&quot;</span></a>
<a class="sourceLine" id="cb3-170" data-line-number="170"><span class="co">#&gt; ---------------- Step 1.5: Standardize ----------</span></a>
<a class="sourceLine" id="cb3-171" data-line-number="171"><span class="co">#&gt; [1] &quot;+Inferring MAF from frequency column...&quot;</span></a>
<a class="sourceLine" id="cb3-172" data-line-number="172"><span class="co">#&gt; [1] &quot;+ Standardize:: Calculating proportion of cases.&quot;</span></a>
<a class="sourceLine" id="cb3-173" data-line-number="173"><span class="co">#&gt; [1] &quot;+ Calculating t-statistic from Effect and StdErr...&quot;</span></a>
<a class="sourceLine" id="cb3-174" data-line-number="174"><span class="co">#&gt; [1] &quot;+ leadSNP missing. Assigning new one by min p-value.&quot;</span></a>
<a class="sourceLine" id="cb3-175" data-line-number="175"><span class="co">#&gt; [1] &quot;+ Extraction completed in 5.36 seconds&quot;</span></a>
<a class="sourceLine" id="cb3-176" data-line-number="176"><span class="co">#&gt; [1] &quot;+ 5399 SNPs x  15 columns&quot;</span></a>
<a class="sourceLine" id="cb3-177" data-line-number="177"><span class="co">#&gt; [1] &quot;+ Removing subset tmp...&quot;</span></a>
<a class="sourceLine" id="cb3-178" data-line-number="178"><span class="co">#&gt; --- Step 2: Calculate Linkage Disequilibrium ---</span></a>
<a class="sourceLine" id="cb3-179" data-line-number="179"><span class="co">#&gt; [1] &quot;LD:: Using UK Biobank LD reference panel...&quot;</span></a>
<a class="sourceLine" id="cb3-180" data-line-number="180"><span class="co">#&gt; [1] &quot;+ UKB LD file name: chr18_47000001_50000001&quot;</span></a>
<a class="sourceLine" id="cb3-181" data-line-number="181"><span class="co">#&gt; [1] &quot;+ LD:: Pre-existing UKB_LD.RDS file detected. Importing...&quot;</span></a>
<a class="sourceLine" id="cb3-182" data-line-number="182"><span class="co">#&gt; -------------- Step 3: Filter SNPs -------------</span></a>
<a class="sourceLine" id="cb3-183" data-line-number="183"><span class="co">#&gt; [1] &quot;++ Post-filtered data: 5399 x 15&quot;</span></a>
<a class="sourceLine" id="cb3-184" data-line-number="184"><span class="co">#&gt; -------- Step 4: Statistically Fine-map --------</span></a>
<a class="sourceLine" id="cb3-185" data-line-number="185"><span class="co">#&gt; [1] &quot;++ Previously multi-fine-mapped results identified. Importing...&quot;</span></a>
<a class="sourceLine" id="cb3-186" data-line-number="186"><span class="co">#&gt; [1] &quot;++ Fine-mapping with ' ABF, SUSIE, FINEMAP ' completed in  0.06&quot;</span></a>
<a class="sourceLine" id="cb3-187" data-line-number="187"><span class="co">#&gt; [1] &quot;+ Identifying Consensus SNPs...&quot;</span></a>
<a class="sourceLine" id="cb3-188" data-line-number="188"><span class="co">#&gt; [1] &quot;++ support_thresh = 2&quot;</span></a>
<a class="sourceLine" id="cb3-189" data-line-number="189"><span class="co">#&gt; [1] &quot;+ Calculating mean Posterior Probability (mean.PP)...&quot;</span></a>
<a class="sourceLine" id="cb3-190" data-line-number="190"><span class="co">#&gt; [1] &quot;++ 3 fine-mapping methods used.&quot;</span></a>
<a class="sourceLine" id="cb3-191" data-line-number="191"><span class="co">#&gt; [1] &quot;++ 1 Credible Set SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb3-192" data-line-number="192"><span class="co">#&gt; [1] &quot;++ 0 Consensus SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb3-193" data-line-number="193"><span class="co">#&gt; --------------- Step 7: Visualize --------------</span></a>
<a class="sourceLine" id="cb3-194" data-line-number="194"><span class="co">#&gt; [1] &quot;+ Identifying Consensus SNPs...&quot;</span></a>
<a class="sourceLine" id="cb3-195" data-line-number="195"><span class="co">#&gt; [1] &quot;++ support_thresh = 2&quot;</span></a>
<a class="sourceLine" id="cb3-196" data-line-number="196"><span class="co">#&gt; [1] &quot;+ Calculating mean Posterior Probability (mean.PP)...&quot;</span></a>
<a class="sourceLine" id="cb3-197" data-line-number="197"><span class="co">#&gt; [1] &quot;++ 3 fine-mapping methods used.&quot;</span></a>
<a class="sourceLine" id="cb3-198" data-line-number="198"><span class="co">#&gt; [1] &quot;++ 1 Credible Set SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb3-199" data-line-number="199"><span class="co">#&gt; [1] &quot;++ 0 Consensus SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb3-200" data-line-number="200"><span class="co">#&gt; [1] &quot;LD Matrix dimensions: 5368 x 5368&quot;</span></a>
<a class="sourceLine" id="cb3-201" data-line-number="201"><span class="co">#&gt; [1] &quot;Extracting LD subset for lead SNP: rs8087969&quot;</span></a>
<a class="sourceLine" id="cb3-202" data-line-number="202"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-203" data-line-number="203"><span class="co">#&gt; [1] &quot;++ GGBIO:: ABF Track&quot;</span></a>
<a class="sourceLine" id="cb3-204" data-line-number="204"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-205" data-line-number="205"><span class="co">#&gt; [1] &quot;++ GGBIO:: SUSIE Track&quot;</span></a>
<a class="sourceLine" id="cb3-206" data-line-number="206"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-207" data-line-number="207"><span class="co">#&gt; [1] &quot;++ GGBIO:: FINEMAP Track&quot;</span></a>
<a class="sourceLine" id="cb3-208" data-line-number="208"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-209" data-line-number="209"><span class="co">#&gt; [1] &quot;++ GGBIO:: mean Track&quot;</span></a>
<a class="sourceLine" id="cb3-210" data-line-number="210"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-211" data-line-number="211"><span class="co">#&gt; [1] &quot;++ GGBIO:: Gene Model Track&quot;</span></a>
<a class="sourceLine" id="cb3-212" data-line-number="212"><span class="co">#&gt; [1] &quot;+ Annotating at transcript-level.&quot;</span></a>
<a class="sourceLine" id="cb3-213" data-line-number="213"><span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span></a>
<a class="sourceLine" id="cb3-214" data-line-number="214"><span class="co">#&gt; [1] &quot;Importing Astrocyte enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb3-215" data-line-number="215"><span class="co">#&gt; [1] &quot;Importing Astrocyte promoters ...&quot;</span></a>
<a class="sourceLine" id="cb3-216" data-line-number="216"><span class="co">#&gt; [1] &quot;Importing Neuronal enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb3-217" data-line-number="217"><span class="co">#&gt; [1] &quot;Importing Neuronal promoters ...&quot;</span></a>
<a class="sourceLine" id="cb3-218" data-line-number="218"><span class="co">#&gt; [1] &quot;Importing Oligo enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb3-219" data-line-number="219"><span class="co">#&gt; [1] &quot;Importing Oligo promoters ...&quot;</span></a>
<a class="sourceLine" id="cb3-220" data-line-number="220"><span class="co">#&gt; [1] &quot;Importing Microglia enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb3-221" data-line-number="221"><span class="co">#&gt; [1] &quot;Importing Microglia promoters ...&quot;</span></a>
<a class="sourceLine" id="cb3-222" data-line-number="222"><span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span></a>
<a class="sourceLine" id="cb3-223" data-line-number="223"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-224" data-line-number="224"><span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span></a>
<a class="sourceLine" id="cb3-225" data-line-number="225"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-226" data-line-number="226"><span class="co">#&gt; [1] &quot;++ Nott sn-epigenomics:: Returning PLAC-seq track.&quot;</span></a>
<a class="sourceLine" id="cb3-227" data-line-number="227"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-228" data-line-number="228"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-229" data-line-number="229"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-230" data-line-number="230"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-231" data-line-number="231"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-232" data-line-number="232"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-233" data-line-number="233"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-234" data-line-number="234"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-235" data-line-number="235"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-236" data-line-number="236"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-237" data-line-number="237"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-238" data-line-number="238"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-239" data-line-number="239"><span class="co">#&gt; [1] &quot;+ GGBIO:: Saving plot ==&gt; /Users/schilder/Desktop/results/GWAS/Nalls23andMe_2019/MEX3C/multiview_MEX3C_100kb.png&quot;</span></a>
<a class="sourceLine" id="cb3-240" data-line-number="240"><span class="co">#&gt; Warning: Removed 303 rows containing non-finite values (stat_smooth).</span></a>
<a class="sourceLine" id="cb3-241" data-line-number="241"><span class="co">#&gt; Warning: Removed 5364 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-242" data-line-number="242"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-243" data-line-number="243"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-244" data-line-number="244"></a>
<a class="sourceLine" id="cb3-245" data-line-number="245"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-246" data-line-number="246"><span class="co">#&gt; Warning: Removed 9 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-247" data-line-number="247"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-248" data-line-number="248"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-249" data-line-number="249"></a>
<a class="sourceLine" id="cb3-250" data-line-number="250"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-251" data-line-number="251"><span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span></a>
<a class="sourceLine" id="cb3-252" data-line-number="252"><span class="co">#&gt; Warning: Removed 303 rows containing non-finite values (stat_smooth).</span></a>
<a class="sourceLine" id="cb3-253" data-line-number="253"><span class="co">#&gt; Warning: Removed 5364 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-254" data-line-number="254"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-255" data-line-number="255"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-256" data-line-number="256"></a>
<a class="sourceLine" id="cb3-257" data-line-number="257"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-258" data-line-number="258"><span class="co">#&gt; Warning: Removed 9 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-259" data-line-number="259"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb3-260" data-line-number="260"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-261" data-line-number="261"></a>
<a class="sourceLine" id="cb3-262" data-line-number="262"><span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_label_repel).</span></a>
<a class="sourceLine" id="cb3-263" data-line-number="263"><span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span></a>
<a class="sourceLine" id="cb3-264" data-line-number="264"><span class="co">#&gt; Error in grid.Call(C_convert, x, as.integer(whatfrom), as.integer(whatto),  : </span></a>
<a class="sourceLine" id="cb3-265" data-line-number="265"><span class="co">#&gt;   Viewport has zero dimension(s)</span></a>
<a class="sourceLine" id="cb3-266" data-line-number="266"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-267" data-line-number="267"><span class="co">#&gt; Fine-mappping complete:MEX3C</span></a>
<a class="sourceLine" id="cb3-268" data-line-number="268"><span class="co">#&gt; Time difference of 1.1 mins</span></a>
<a class="sourceLine" id="cb3-269" data-line-number="269"><span class="co">#&gt; &lt;div id=&quot;htmlwidget-a680a1fcd62a0dd1ba32&quot; style=&quot;width:960px;height:500px;&quot; class=&quot;datatables html-widget&quot;&gt;&lt;/div&gt;</span></a>
<a class="sourceLine" id="cb3-270" data-line-number="270"><span class="co">#&gt; &lt;script type=&quot;application/json&quot; data-for=&quot;htmlwidget-a680a1fcd62a0dd1ba32&quot;&gt;{&quot;x&quot;:{&quot;filter&quot;:&quot;none&quot;,&quot;extensions&quot;:[&quot;Buttons&quot;],&quot;caption&quot;:&quot;&lt;caption&gt;&lt;\/caption&gt;&quot;,&quot;data&quot;:[[&quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;],[&quot;BST1&quot;,&quot;BST1&quot;,&quot;BST1&quot;,&quot;MEX3C&quot;],[&quot;rs34559912&quot;,&quot;rs4389574&quot;,&quot;rs4541502&quot;,&quot;rs1893379&quot;],[4,4,4,18],[15730146,15730398,15712787,48718536],[1.213e-27,3.141e-17,8.309e-22,2.847e-08],[0.103,-0.0977,-0.0897,0.0558],[0.0095,0.0116,0.0093,0.01],[&quot;T&quot;,&quot;A&quot;,&quot;T&quot;,&quot;A&quot;],[&quot;G&quot;,&quot;G&quot;,&quot;G&quot;,&quot;G&quot;],[0.5526,0.4443,0.4749,0.4542],[0.4474,0.4443,0.4749,0.4542],[56306,42598,56306,49053],[1417791,1322509,1417791,1411006],[0.0381969436203995,0.0312048799105125,0.0381969436203995,0.0335965875351613],[10.8421052631579,-8.42241379310345,-9.64516129032258,5.58],[false,false,false,false],[null,null,null,null],[null,null,null,null],[3,1,2,1],[1,1,1,1],[0,0,null,null],[0,0,null,null],[1,1,1,1],[false,false,false,false],[0.333333333333333,0.333333333333333,0.333333333333333,0.333333333333333],[0,0,0,0]],&quot;container&quot;:&quot;&lt;table class=\&quot;display\&quot;&gt;\n  &lt;thead&gt;\n    &lt;tr&gt;\n      &lt;th&gt; &lt;\/th&gt;\n      &lt;th&gt;Locus&lt;\/th&gt;\n      &lt;th&gt;SNP&lt;\/th&gt;\n      &lt;th&gt;CHR&lt;\/th&gt;\n      &lt;th&gt;POS&lt;\/th&gt;\n      &lt;th&gt;P&lt;\/th&gt;\n      &lt;th&gt;Effect&lt;\/th&gt;\n      &lt;th&gt;StdErr&lt;\/th&gt;\n      &lt;th&gt;A1&lt;\/th&gt;\n      &lt;th&gt;A2&lt;\/th&gt;\n      &lt;th&gt;Freq&lt;\/th&gt;\n      &lt;th&gt;MAF&lt;\/th&gt;\n      &lt;th&gt;N_cases&lt;\/th&gt;\n      &lt;th&gt;N_controls&lt;\/th&gt;\n      &lt;th&gt;proportion_cases&lt;\/th&gt;\n      &lt;th&gt;t_stat&lt;\/th&gt;\n      &lt;th&gt;leadSNP&lt;\/th&gt;\n      &lt;th&gt;ABF.CS&lt;\/th&gt;\n      &lt;th&gt;ABF.PP&lt;\/th&gt;\n      &lt;th&gt;SUSIE.CS&lt;\/th&gt;\n      &lt;th&gt;SUSIE.PP&lt;\/th&gt;\n      &lt;th&gt;FINEMAP.CS&lt;\/th&gt;\n      &lt;th&gt;FINEMAP.PP&lt;\/th&gt;\n      &lt;th&gt;Support&lt;\/th&gt;\n      &lt;th&gt;Consensus_SNP&lt;\/th&gt;\n      &lt;th&gt;mean.PP&lt;\/th&gt;\n      &lt;th&gt;mean.CS&lt;\/th&gt;\n    &lt;\/tr&gt;\n  &lt;\/thead&gt;\n&lt;\/table&gt;&quot;,&quot;options&quot;:{&quot;dom&quot;:&quot;Bfrtip&quot;,&quot;buttons&quot;:[&quot;copy&quot;,&quot;csv&quot;,&quot;excel&quot;,&quot;pdf&quot;,&quot;print&quot;],&quot;scrollY&quot;:400,&quot;scrollX&quot;:true,&quot;scrollCollapse&quot;:true,&quot;paging&quot;:false,&quot;columnDefs&quot;:[{&quot;className&quot;:&quot;dt-center&quot;,&quot;targets&quot;:&quot;_all&quot;},{&quot;orderable&quot;:false,&quot;targets&quot;:0}],&quot;order&quot;:[],&quot;autoWidth&quot;:false,&quot;orderClasses&quot;:false}},&quot;evals&quot;:[],&quot;jsHooks&quot;:[]}&lt;/script&gt;</span></a></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
