---
title: "Plotting vignette"
knit: (function(input_file, encoding) {
  out_dir <- 'doc';
  rmarkdown::render(input_file,
  encoding=encoding,
  output_file=file.path(dirname(input_file), out_dir, 'index.html'))})
author: "Brian M. Schilder"
date: "Most recent update:<br> `r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{plotting_vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE, error = TRUE}
root.dir <- "~/Desktop"
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  root.dir = root.dir, 
  fig.height = 12,
  fig.width = 10
)  
knitr::opts_knit$set(
  root.dir = root.dir
  )
# knitr::opts_chunk$get("root.dir")
# devtools::build_vignettes(quiet = F)
```

```{r setup, root.dir="~/Desktop"}
library(echolocatoR) 
```

# Plotting only

*echolocatoR* contains various functions that can be used separately  
from the comprehensive `finemap_loci()` pipeline.  

Generate a multiview plot using `GGBIO.plot()`.  

* You can mix and match different tracks and annotations using the different arguments 
(see `?GGBIO.plot` for details).  
* Zoom in/out by adjusting the `plot.window` size (in units of basepairs).  
The plot is centered on the lead/index SNP.  
* `GGBIO.plot()` returns a `ggbio` track object. 
  + Further modify this object using `ggplot2` functions like `+ theme()`.
  + The modifications will be applied to all tracks at once.  
* Save a high-resolution *.png* of the plot by setting `save_plot=T`  
  + Further increase resolution by adjusting the `dpi` argument (*default=300*).
  + Adjust the `height` and `width` of the saved plot using these respective arguments.
  + The plot will be automatically saved in the locus-specific directory as:  
  *multiview_<locus>_<plot.window>.png*.
   
## Load example data   

Load example dataset of the results from fine-mapping the BST1 locus with `finemap_loci()`.
Original data comes from the recent Nalls et al. (2019) Parkinson's disease GWAS (see `?BST1` for details).

```{r}
library(ggplot2)
data("BST1"); data("LD_matrix"); data("locus_dir");
locus_dir <- file.path("~/Desktop",locus_dir)
finemap_DT <- BST1 
show_plot <- T


# locus <- "DNAH17"
# locus_dir <- file.path("~/Desktop/Fine_Mapping/Data/GWAS/Nalls23andMe_2019",locus)
# LD_matrix <- readRDS(file.path(locus_dir,"plink/UKB_LD.RDS"))
# finemap_DT <- data.table::fread(file.path(locus_dir,"Multi-finemap/Multi-finemap_results.txt"))
# finemap_DT <- find_consensus_SNPs(finemap_DT) 
# finemap_DT <- update_CS_cols(finemap_DT)
```

## Using NO annotations  

```{r trk_plot}
trk_plot <- GGBIO.plot(finemap_dat=finemap_DT, 
                       LD_matrix=LD_matrix, 
                       locus_dir=locus_dir, 
                       XGR_libnames=NULL, 
                       save_plot=F,
                       show_plot=show_plot)
trk_plot <- trk_plot + theme(strip.background = element_rect(fill="gray"),
                             text = element_text(size=3, color="blue"))
```

## Using only XGR annotations   

Whenever you use annotation arguments (e.g. `XGR_libnames`,`Roadmap`,`Nott_epigenome`)
the annotations that overlap with your locus will automatically be saved as `GRanges` objects  
in a locus-specific subdirectory:  
*results/<dataset_type>/<dataset_name>/<locus>/annotation* 

```{r trk_plot.xgr}
trk_plot.xgr <- GGBIO.plot(finemap_dat=finemap_DT, 
                           LD_matrix=LD_matrix, 
                           locus_dir=locus_dir, 
                           XGR_libnames=c("ENCODE_TFBS_ClusteredV3_CellTypes"), 
                           save_plot=F,
                           show_plot=show_plot)
```

## Using only Roadmap annotations

```{r trk_plot.roadmap} 
trk_plot.roadmap <- GGBIO.plot(finemap_dat=finemap_DT, 
                               LD_matrix=LD_matrix, 
                               locus_dir=locus_dir, 
                               XGR_libnames=NULL, 
                               Roadmap=T, 
                               Roadmap_query="monocyte", 
                               save_plot=F, 
                               show_plot=show_plot)
```


## Using only Nott_2019 annotations   

* Query and plot brain cell type-specific epigenomic assays from Nott et al. (2019)  
(see `?NOTT_2019.bigwig_metadata` for details).

```{r trk_plot.nott_2019} 
trk_plot.nott_2019 <- GGBIO.plot(finemap_dat=finemap_DT,  
                                 LD_matrix=LD_matrix, 
                                 locus_dir=locus_dir,
                                 label_snps=F,
                                 XGR_libnames=NULL, 
                                 Nott_epigenome=T, 
                                 plot.window = NULL,
                                 Nott_binwidth = 100,
                                 Nott_regulatory_rects = T, 
                                 Nott_show_placseq = T,
                                 save_plot=T,
                                 show_plot=show_plot)
# ggbio::ggsave("~/Desktop/Fine_Mapping/Data/GWAS/Nalls23andMe_2019/LRRK2/multiview_LRRK2_.svg",
#               trk_plot.nott_2019, 
#               height=12, width=10)
```


## Using QTL datasets  

* Plot multiple QTL p-value columns (or really P-value columns from any kind of datset).  
* Each QTL dataset will be plotted as a new track.

```{r trk_plot.QTL}
# Make fake QTL P-values for the sake a demonstration
finemap_DT$fake_eQTL.P <- finemap_DT$P  * c(1,.9,.7)
finemap_DT$fake_sQTL.P <- finemap_DT$P  * c(1,.8,.5)

trk_plot.qtl <- GGBIO.plot(finemap_dat=finemap_DT, 
                           LD_matrix=LD_matrix, 
                           locus_dir=locus_dir,
                           QTL_prefixes=c("fake_eQTL.","fake_sQTL."),
                           XGR_libnames=NULL, 
                           save_plot=F,
                           show_plot=show_plot)
```


