<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Automated genomic fine-mapping â€¢ echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Automated genomic fine-mapping">
<meta property="og:description" content="Automated statistical and functional fine-mapping pipeline with extensive API access to datasets.  ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/QTL_pipeline_vignette.html">echolocatoR: QTL pipeline vignette</a>
    </li>
    <li>
      <a href="articles/full_pipeline_vignette.html">echolocatoR: full pipeline vignette</a>
    </li>
    <li>
      <a href="articles/plotting_vignette.html">echolocatoR: plotting vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<center>
<div class="page-header"><h1>
) ) ) ) ))) ðŸ¦‡ echolocatoR ðŸ¦‡ ((( ( ( ( (
</h1></div>
</center>
<p><strong><em>echolocatoR</em></strong> is an R package for automating statistical and functional fine-mapping with extensive API access to datasets.</p>
<div id="documentation" class="section level2">
<h2 class="hasAnchor">
<a href="#documentation" class="anchor"></a>Documentation</h2>
<div id="documentation-website" class="section level3">
<h3 class="hasAnchor">
<a href="#documentation-website" class="anchor"></a><a href="https://rajlabmssm.github.io/echolocatoR/">Documentation website</a>
</h3>
</div>
<div id="full-pipeline-vignette" class="section level3">
<h3 class="hasAnchor">
<a href="#full-pipeline-vignette" class="anchor"></a><a href="https://rajlabmssm.github.io/echolocatoR/articles/full_pipeline_vignette.html">Full pipeline vignette</a>
</h3>
</div>
<div id="plotting-vignette" class="section level3">
<h3 class="hasAnchor">
<a href="#plotting-vignette" class="anchor"></a><a href="https://rajlabmssm.github.io/echolocatoR/articles/plotting_vignette.html">Plotting vignette</a>
</h3>
</div>
</div>
<div id="workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h2>
<p><img src="./images/PD_Finemapping_Flowchart_plus.png" alt="echoFlow"></p>
</div>
<div id="quick-installation" class="section level2">
<h2 class="hasAnchor">
<a href="#quick-installation" class="anchor"></a>Quick installation</h2>
<p>In R:</p>
<div class="sourceCode" id="cb1"><pre class="r"><span class="kw">if</span>(!<span class="st">"devtools"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span>()){<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)}
<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"RajLabMSSM/echolocatoR"</span>)</pre></div>
<p><em>NOTE</em>: While this GitHub repo is still private, you need to use a token to install <strong><em>echolocatoR</em></strong> using the <code>auth_token</code> argument (see <a href="http://devtools.r-lib.org/reference/install_github">here for details</a>).</p>
</div>
<div id="robust-installation-conda" class="section level2">
<h2 class="hasAnchor">
<a href="#robust-installation-conda" class="anchor"></a>Robust installation (<em>conda</em>)</h2>
<p>As with most softwares, installation is half the battle. The easiest way to install all of <strong><em>echolocatoR</em></strong>â€™s dependencies (which include R, Python, and command line tools) and make sure they play well together is to create a <a href="https://docs.conda.io/en/latest/"><em>conda</em></a> environment.</p>
<ol>
<li><p>If you havenâ€™t done so already, install <a href="https://docs.conda.io/en/latest/"><em>conda</em></a>.</p></li>
<li><p>Download the echoR.yml file found <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/inst/conda/echoR.yml">here</a> (this file tells <em>conda</em> what to install).</p></li>
<li>In command line, create the env from the <em>.yml</em> file:<br>
</li>
</ol>
<pre><code>conda env create -f &lt;path_to_file&gt;/echoR.yml</code></pre>
<ol>
<li>Activate the new env:<br>
</li>
</ol>
<pre><code>conda activate echoR</code></pre>
<ol>
<li>In R, install <strong><em>echolocatoR</em></strong>:<br>
</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="r"><span class="kw">if</span>(!<span class="st">"devtools"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span>()){<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)}
<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"RajLabMSSM/echolocatoR"</span>)</pre></div>
<p>To make sure <strong><em>echolocatoR</em></strong> uses the packages in this env (esp.Â if using from RStudio), you can then supply the env name to the <code><a href="reference/finemap_loci.html">finemap_loci()</a></code> function using <code>conda_env="echoR"</code>.</p>
</div>
<div id="dependencies" class="section level2">
<h2 class="hasAnchor">
<a href="#dependencies" class="anchor"></a>Dependencies</h2>
<p>* = <em>optional</em></p>
<div id="r" class="section level3">
<h3 class="hasAnchor">
<a href="#r" class="anchor"></a>R</h3>
<ul>
<li>magrittr<br>
</li>
<li>R.utils<br>
</li>
<li>dplyr<br>
</li>
<li>BiocManager</li>
<li>tidyverse</li>
<li>knitr</li>
<li>rmarkdown<br>
</li>
<li>data.table<br>
</li>
<li>foreign<br>
</li>
<li>reticulate<br>
</li>
<li>ggplot2<br>
</li>
<li>ggrepel<br>
</li>
<li>coloc<br>
</li>
<li>RColorBrewer<br>
</li>
<li>patchwork<br>
</li>
<li>htmltools<br>
</li>
<li>stringr<br>
</li>
<li>openxlsx<br>
</li>
<li>EnsDb.Hsapiens.v75<br>
</li>
<li>ensembldb<br>
</li>
<li>ggbio<br>
</li>
<li>BSgenome<br>
</li>
<li>Ckmeans.1d.dp<br>
</li>
<li>refGenome<br>
</li>
<li><ul>
<li><em>For a full list of suggested packages, see <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/DESCRIPTION">DESCRIPTION</a>.</em></li>
</ul></li>
</ul>
</div>
<div id="python" class="section level3">
<h3 class="hasAnchor">
<a href="#python" class="anchor"></a>Python</h3>
<ul>
<li>python&gt;=3.6.1<br>
</li>
<li>pandas&gt;=0.25.0<br>
</li>
<li>pandas-plink<br>
</li>
<li>pyarrow<br>
</li>
<li>fastparquet<br>
</li>
<li>scipy<br>
</li>
<li>scikit-learn<br>
</li>
<li>tqdm<br>
</li>
<li>bitarray<br>
</li>
<li>networkx<br>
</li>
<li>rpy2<br>
</li>
<li>requests</li>
</ul>
</div>
<div id="command-line" class="section level3">
<h3 class="hasAnchor">
<a href="#command-line" class="anchor"></a>Command line</h3>
<ul>
<li>
<a href="http://www.htslib.org/doc/tabix.html"><strong>Tabix</strong></a>:
<ul>
<li>Rapid querying of summary stats files.<br>
</li>
<li>To use it, specify <code>download_method = "axel"</code> in <code><a href="reference/finemap_loci.html">finemap_loci()</a></code> .</li>
</ul>
</li>
<li>
<a href="http://samtools.github.io/bcftools/bcftools.html"><strong>bcftools</strong></a>:
<ul>
<li>Used here for filtering populations in vcf files.<br>
</li>
</ul>
</li>
<li>
<a href="https://github.com/axel-download-accelerator/axel"><strong>Axel</strong></a>:*
<ul>
<li>Rapid multi-core downloading of large files (e.g.Â LD matrices from UK Biobank).<br>
</li>
</ul>
</li>
<li>
<a href="https://hbctraining.github.io/Intro-to-ChIPseq/lessons/05_peak_calling_macs.html"><strong>MACS2</strong></a>:*
<ul>
<li>Epigenomic peak calling.</li>
</ul>
</li>
</ul>
</div>
</div>
<div id="fine-mapping-tools" class="section level2">
<h2 class="hasAnchor">
<a href="#fine-mapping-tools" class="anchor"></a>Fine-mapping Tools</h2>
<p>All methods require colums: <code>SNP</code>, <code>CHR</code>, <code>POS</code>.</p>
<ul>
<li>
<a href="https://cran.r-project.org/web/packages/coloc/vignettes/vignette.html">ABF</a>
<ul>
<li>Required columns: <code>proportion_cases</code>,<code>MAF</code>,<code>Effect</code>,<code>StdErr</code>
</li>
</ul>
</li>
<li>
<a href="https://github.com/stephenslab/susieR">SuSiE</a><br>
</li>
<li>
<a href="http://www.christianbenner.com">FINEMAP</a><br>
</li>
<li><a href="https://github.com/omerwe/polyfun">PolyFun</a></li>
<li><a href="https://cnsgenomics.com/software/gcta/#COJO">GCTA-COJO</a></li>
<li>
<a href="https://github.com/gkichaev/PAINTOR_V3.0">PAINTOR</a><br>
</li>
<li><a href="https://cran.r-project.org/web/packages/coloc/vignettes/vignette.html">coloc</a></li>
</ul>
<p><br></p>
</div>
<div id="datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#datasets" class="anchor"></a>Datasets</h2>
<p>For more detailed information about each dataset, use <code>?</code>:<br><code>R   library(echolocatoR)   ?NOTT_2019.interactome # example dataset</code></p>
<div id="epigenomic-genome-wide-annotations" class="section level3">
<h3 class="hasAnchor">
<a href="#epigenomic-genome-wide-annotations" class="anchor"></a>Epigenomic &amp; Genome-wide Annotations</h3>
<div id="nott-et-al--2019" class="section level4">
<h4 class="hasAnchor">
<a href="#nott-et-al--2019" class="anchor"></a><a href="https://science.sciencemag.org/content/366/6469/1134.abstract">Nott et al. (2019)</a>
</h4>
<ul>
<li><p>Data from this publication contains results from cell type-specific (neurons, oligodendrocytes, astrocytes, microglia, &amp; peripheral myeloid cells) epigenomic assays (H3K27ac, ATAC, H3K4me3) from human brain tissue.</p></li>
<li>
<p>For detailed metadata, see:</p>
<div class="sourceCode" id="cb5"><pre class="r"><span class="kw pkg">echolocatoR</span><span class="kw ns">::</span><span class="no"><a href="reference/NOTT_2019.bigwig_metadata.html">NOTT_2019.bigwig_metadata</a></span>
<span class="co"># Or </span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"NOTT_2019.bigwig_metadata"</span>)</pre></div>
</li>
<li>Built-in datasets:
<ul>
<li>Enhancer/promoter coordinates (as <em>GenomicRanges</em>)<br>
</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"NOTT_2019.interactome"</span>)
<span class="co"># Examples of the data nested in "NOTT_2019.interactome" object:</span>
<span class="no">NOTT_2019.interactome</span>$<span class="no">`Neuronal promoters`</span>
<span class="no">NOTT_2019.interactome</span>$<span class="no">`Neuronal enhancers`</span>
<span class="no">NOTT_2019.interactome</span>$<span class="no">`Microglia promoters`</span>
<span class="no">NOTT_2019.interactome</span>$<span class="no">`Microglia enhancers`</span>
<span class="no">...</span>
<span class="no">...</span></pre></div>
<ul>
<li>PLAC-seq enhancer-promoter interactome coordinates<br>
</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="r"><span class="no">NOTT_2019.interactome</span>$<span class="no">H3K4me3_around_TSS_annotated_pe</span>
<span class="no">NOTT_2019.interactome</span>$<span class="no">`Microglia interactome`</span>
<span class="no">NOTT_2019.interactome</span>$<span class="no">`Neuronal interactome`</span>
<span class="no">NOTT_2019.interactome</span>$<span class="no">`Oligo interactome`</span>
<span class="no">...</span>
<span class="no">...</span></pre></div>
</li>
<li>API access to full bigWig files on UCSC Genome Browser, which includes
<ul>
<li>Epigenomic reads (as <em>GenomicRanges</em>)<br>
</li>
<li>Aggregate epigenomic <em>score</em> for each cell type - assay combination</li>
</ul>
</li>
</ul>
</div>
<div id="corces-et-al--2020" class="section level4">
<h4 class="hasAnchor">
<a href="#corces-et-al--2020" class="anchor"></a><a href="https://www.biorxiv.org/content/10.1101/2020.01.06.896159v1">Corces et al. (2020)</a>
</h4>
<ul>
<li>
<p>Data from this preprint contains results from single-cell chromatin accessibility epigenomic assays in from 39 human brains.</p>
<div class="sourceCode" id="cb8"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"CORCES_2020.bulkATACseq_peaks"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"CORCES_2020.cicero_coaccessibility"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"CORCES_2020.HiChIP_FitHiChIP_loop_calls"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"CORCES_2020.scATACseq_celltype_peaks"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"CORCES_2020.scATACseq_peaks"</span>)</pre></div>
</li>
</ul>
</div>
<div id="xgr" class="section level4">
<h4 class="hasAnchor">
<a href="#xgr" class="anchor"></a><a href="http://xgr.r-forge.r-project.org">XGR</a>
</h4>
<ul>
<li>API access to a diverse library of cell type/line-specific epigenomic (e.g.Â ENCODE) and other genome-wide annotations.</li>
</ul>
</div>
<div id="roadmap" class="section level4">
<h4 class="hasAnchor">
<a href="#roadmap" class="anchor"></a><a href="http://www.roadmapepigenomics.org">Roadmap</a>
</h4>
<ul>
<li>API access to cell type-specific epigenomic data.</li>
</ul>
</div>
<div id="biomart" class="section level4">
<h4 class="hasAnchor">
<a href="#biomart" class="anchor"></a><a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html">biomaRt</a>
</h4>
<ul>
<li>API access to various genome-wide SNP annotations (e.g.Â missense, nonsynonmous, intronic, enhancer).</li>
</ul>
</div>
<div id="haplor" class="section level4">
<h4 class="hasAnchor">
<a href="#haplor" class="anchor"></a><a href="https://cran.r-project.org/web/packages/haploR/vignettes/haplor-vignette.html">HaploR</a>
</h4>
<ul>
<li>API access to known per-SNP QTL and epigenomic data hits.</li>
</ul>
</div>
</div>
<div id="qtls" class="section level3">
<h3 class="hasAnchor">
<a href="#qtls" class="anchor"></a>QTLs</h3>
<div id="eqtl-catalogue" class="section level4">
<h4 class="hasAnchor">
<a href="#eqtl-catalogue" class="anchor"></a><a href="https://www.ebi.ac.uk/eqtl/">eQTL Catalogue</a>
</h4>
<ul>
<li>API access to full summary statistics from many standardized e/s/t-QTL datasets.<br>
</li>
<li>Data access and colocalization tests facilitated through the <a href="https://github.com/RajLabMSSM/catalogueR">catalogueR</a> R package.</li>
</ul>
<p><br></p>
</div>
</div>
</div>
<div id="enrichment-tools" class="section level2">
<h2 class="hasAnchor">
<a href="#enrichment-tools" class="anchor"></a>Enrichment Tools</h2>
<div id="xgr-1" class="section level3">
<h3 class="hasAnchor">
<a href="#xgr-1" class="anchor"></a><a href="http://xgr.r-forge.r-project.org">XGR</a>
</h3>
<ul>
<li>Binomial enrichment tests between customisable foreground and background SNPs.</li>
</ul>
</div>
<div id="goshifter" class="section level3">
<h3 class="hasAnchor">
<a href="#goshifter" class="anchor"></a><a href="https://github.com/immunogenomics/goshifter">GoShifter</a>
</h3>
<ul>
<li>LD-informed iterative enrichment analysis.</li>
</ul>
</div>
<div id="s-ldsc" class="section level3">
<h3 class="hasAnchor">
<a href="#s-ldsc" class="anchor"></a><a href="https://www.nature.com/articles/ng.3954">S-LDSC</a>
</h3>
<ul>
<li>Genome-wide stratified LD score regression.</li>
<li>Inlcudes 187-annotation baseline model from <a href="https://www.nature.com/articles/s41588-018-0231-8">Gazal et al.Â 2018</a>.<br>
</li>
<li>You can alternatively supply a custom annotations matrix.</li>
</ul>
</div>
<div id="garfield-under-construction" class="section level3">
<h3 class="hasAnchor">
<a href="#garfield-under-construction" class="anchor"></a><a href="https://www.bioconductor.org/packages/release/bioc/html/garfield.html">GARFIELD</a> (<strong>under construction</strong>)</h3>
<ul>
<li>Genomic enrichment with LD-informed heuristics.</li>
</ul>
<p><br></p>
</div>
</div>
<div id="ld-reference-panels" class="section level2">
<h2 class="hasAnchor">
<a href="#ld-reference-panels" class="anchor"></a>LD Reference Panels</h2>
<div id="uk-biobank" class="section level3">
<h3 class="hasAnchor">
<a href="#uk-biobank" class="anchor"></a><a href="https://www.ukbiobank.ac.uk">UK Biobank</a>
</h3>
</div>
<div id="genomes-phase-1" class="section level3">
<h3 class="hasAnchor">
<a href="#genomes-phase-1" class="anchor"></a><a href="https://www.internationalgenome.org">1000 Genomes Phase 1</a>
</h3>
</div>
<div id="genomes-phase-3" class="section level3">
<h3 class="hasAnchor">
<a href="#genomes-phase-3" class="anchor"></a><a href="https://www.internationalgenome.org">1000 Genomes Phase 3</a>
</h3>
<hr>
<hr>
</div>
</div>
<div id="author" class="section level2">
<h2 class="hasAnchor">
<a href="#author" class="anchor"></a>Author</h2>
<p><a href="https://bschilder.github.io/BMSchilder/" target="_blank">Brian M. Schilder, Bioinformatician II</a><br><a href="https://rajlab.org" target="_blank">Raj Lab</a><br><a href="https://icahn.mssm.edu/about/departments/neuroscience" target="_blank">Department of Neuroscience, Icahn School of Medicine at Mount Sinai</a><br><img src="./inst/images/sinai.png" alt="Sinai"></p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/RajLabMSSM/echolocatoR/">https://â€‹github.com/â€‹RajLabMSSM/â€‹echolocatoR/â€‹</a>
</li>
<li>Report a bug at <br><a href="https://github.com/RajLabMSSM/echolocatoR/issues">https://â€‹github.com/â€‹RajLabMSSM/â€‹echolocatoR/â€‹issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Brian Schilder <br><small class="roles"> Author, maintainer </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
