<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Automated genomic fine-mapping â€¢ echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Automated genomic fine-mapping">
<meta property="og:description" content="Automated statistical and functional fine-mapping with extensive access to genome-wide datasets.  ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/echolocatoR.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<img src="https://github.com/RajLabMSSM/echolocatoR/raw/%20%20dev/inst/hex/echolocatoR.png" height="300"><br><br><a href="https://github.com/RajLabMSSM/echolocatoR" class="external-link"><img src="https://img.shields.io/badge/devel%20version-0.2.2-black.svg"></a> <a href="https://github.com/RajLabMSSM/echolocatoR/actions" class="external-link"><img src="https://github.com/RajLabMSSM/echolocatoR/workflows/R-CMD-check-bioc/badge.svg" alt="R build status"></a> <a href="https://github.com/RajLabMSSM/echolocatoR/commits/master" class="external-link"><img src="https://img.shields.io/github/last-commit/RajLabMSSM/echolocatoR.svg"></a> <a href="https://codecov.io/gh/RajLabMSSM/echolocatoR" class="external-link"><img src="https://codecov.io/gh/RajLabMSSM/echolocatoR/branch/master/graph/badge.svg"></a> <a href="https://cran.r-project.org/web/licenses/MIT" class="external-link"><img src="https://img.shields.io/badge/license-MIT-blue.svg" alt="License: MIT"></a> <a href="https://doi.org/10.1093/bioinformatics/btab658" class="external-link"><img src="https://img.shields.io/badge/doi-10.1093/bioinformatics/btab658-blue.svg"></a>
<h5>
Author: <i>Brian M. Schilder</i>
</h5>
<h5>
README updated: <i>Mar-04-2022</i>
</h5>
<center>
<div class="page-header"><h1>
) ) ) ) ))) <span class="emoji" data-emoji="bat">ðŸ¦‡</span> echolocatoR <span class="emoji" data-emoji="bat">ðŸ¦‡</span> ((( ( ( ( (
</h1></div>
</center>
<h3>
Automated statistical and functional fine-mapping with extensive access to genome-wide datasets
</h3>
<hr>
<p>If you use <code>echolocatoR</code>, please cite:</p>
<blockquote>
<p>Brian M Schilder, Jack Humphrey, Towfique Raj (2021) echolocatoR: an automated end-to-end statistical and functional genomic fine-mapping pipeline, <em>Bioinformatics</em>; btab658, <a href="https://doi.org/10.1093/bioinformatics/btab658" class="external-link uri">https://doi.org/10.1093/bioinformatics/btab658</a></p>
</blockquote>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Fine-mapping methods are a powerful means of identifying causal variants underlying a given phenotype, but are underutilized due to the technical challenges of implementation. <strong><em>echolocatoR</em></strong> is an R package that automates end-to-end genomics fine-mapping, annotation, and plotting in order to identify the most probable causal variants associated with a given phenotype.</p>
<p>It requires minimal input from users (a GWAS or QTL summary statistics file), and includes a suite of statistical and functional fine-mapping tools. It also includes extensive access to datasets (linkage disequilibrium panels, epigenomic and genome-wide annotations, QTL).</p>
<p>The elimination of data gathering and preprocessing steps enables rapid fine-mapping of many loci in any phenotype, complete with locus-specific publication-ready figure generation. All results are merged into a single per-SNP summary file for additional downstream analysis and results sharing. Therefore <strong><em>echolocatoR</em></strong> drastically reduces the barriers to identifying causal variants by making the entire fine-mapping pipeline rapid, robust and scalable.</p>
<div class="figure">
<img src="./images/echolocatoR_Fig1.png" alt=""><p class="caption">echoFlow</p>
</div>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<div class="section level3">
<h3 id="website">
<a href="https://rajlabmssm.github.io/echolocatoR" class="external-link">Website</a><a class="anchor" aria-label="anchor" href="#website"></a>
</h3>
</div>
<div class="section level3">
<h3 id="getting-started">
<a href="https://rajlabmssm.github.io/echolocatoR/articles/echolocatoR" class="external-link">Getting started</a><a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h3>
</div>
<div class="section level3">
<h3 id="bugsrequests">Bugs/requests<a class="anchor" aria-label="anchor" href="#bugsrequests"></a>
</h3>
<p>Please report any bugs/requests on <a href="https://github.com/RajLabMSSM/echolocatoR/issues" class="external-link">GitHub Issues</a>:</p>
<p><a href="https://github.com/RajLabMSSM/echolocatoR/pulls" class="external-link">Contributions</a> are welcome!:</p>
</div>
</div>
<div class="section level2">
<h2 id="literature">Literature<a class="anchor" aria-label="anchor" href="#literature"></a>
</h2>
<div class="section level3">
<h3 id="for-applications-of-echolocator-in-the-literature-please-see">For applications of <strong><em>echolocatoR</em></strong> in the literature, please see:<a class="anchor" aria-label="anchor" href="#for-applications-of-echolocator-in-the-literature-please-see"></a>
</h3>
<blockquote>
<ol style="list-style-type: decimal">
<li>E Navarro, E Udine, K de Paiva Lopes, M Parks, G Riboldi, BM Schilderâ€¦T Raj (2020) Dysregulation of mitochondrial and proteo-lysosomal genes in Parkinsonâ€™s disease myeloid cells. Nature Genetics. <a href="https://doi.org/10.1101/2020.07.20.212407" class="external-link uri">https://doi.org/10.1101/2020.07.20.212407</a>
</li>
<li>BM Schilder, T Raj (2021) Fine-Mapping of Parkinsonâ€™s Disease Susceptibility Loci Identifies Putative Causal Variants. Human Molecular Genetics, ddab294, <a href="https://doi.org/10.1093/hmg/ddab294" class="external-link uri">https://doi.org/10.1093/hmg/ddab294</a><br>
</li>
<li>K de Paiva Lopes, G JL Snijders, J Humphrey, A Allan, M Sneeboer, E Navarro, BM Schilderâ€¦T Raj (2022) Genetic analysis of the human microglial transcriptome across brain regions, aging and disease pathologies. Nature Genetics, <a href="https://doi.org/10.1038/s41588-021-00976-y" class="external-link uri">https://doi.org/10.1038/s41588-021-00976-y</a>
</li>
</ol>
</blockquote>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="general-tips">General tips<a class="anchor" aria-label="anchor" href="#general-tips"></a>
</h3>
<ul>
<li>We generally recommend users upgrading to R&gt;=4.0.0 before trying to install <em>echolocatoR.</em> While <em>echolocatoR</em> should technically be able to run in R&gt;=3.6.0, some additional challenges with getting dependency versions not to conflict with one another.</li>
</ul>
</div>
<div class="section level3">
<h3 id="quick-installation">Quick installation<a class="anchor" aria-label="anchor" href="#quick-installation"></a>
</h3>
<p>In R:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://remotes.r-lib.org" class="external-link">"remotes"</a></span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span>

<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"RajLabMSSM/echolocatoR"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="robust-installation-conda">Robust installation (<em>conda</em>)<a class="anchor" aria-label="anchor" href="#robust-installation-conda"></a>
</h3>
<p>As with most softwares, installation is half the battle. The easiest way to install all of <strong><em>echolocatoR</em></strong>â€™s dependencies (which include R, Python, and command line tools) and make sure they play well together is to create a <a href="https://docs.conda.io/en/latest/" class="external-link"><em>conda</em></a> environment.</p>
<ol style="list-style-type: decimal">
<li><p>If you havenâ€™t done so already, install <a href="https://docs.conda.io/en/latest/" class="external-link"><em>conda</em></a>.</p></li>
<li><p>In command line, create the env from the <em>.yml</em> file (this file tells <em>conda</em> what to install): <code>conda env create -f https://github.com/RajLabMSSM/echolocatoR/raw/master/inst/conda/echoR.yml</code></p></li>
<li><p>Activate the new env:<br><code>conda activate echoR</code></p></li>
<li><p>Install <em>echolocatoR</em> from command line so that it installs <strong>within</strong> the <em>conda</em> env:</p></li>
<li>
<p>Open Rstudio from the command line interface (not by clicking the Rstudio icon). This helps to ensure Rstudio can find the paths to the packages in the conda env:<br><code>open model_celltype_conservation.Rproj</code></p>
<p>Alternatively, the <em>conda</em> env also comes with <a href="https://github.com/randy3k/radian" class="external-link"><em>radian</em></a>, which is a convenient R console thatâ€™s much more advanced than the default R console, but doesnâ€™t require access to a GUI. This can be especially useful on computing clusters that donâ€™t support RStudio or other IDEs.<br><code>radian</code></p>
</li>
<li><p>Finally, to make extra sure <strong><em>echolocatoR</em></strong> uses the packages in this env (esp.Â if using from RStudio), you can then supply the env name to the <code><a href="reference/finemap_loci.html">finemap_loci()</a></code> function (and many other <strong><em>echolocatoR</em></strong> functions) using <code>conda_env="echoR"</code>.</p></li>
</ol>
</div>
<div class="section level3">
<h3 id="binary-installation">Binary installation<a class="anchor" aria-label="anchor" href="#binary-installation"></a>
</h3>
<p>You can also install <strong><em>echolocatoR</em></strong> using the compressed binary file available on <a href="https://drive.google.com/drive/folders/1oS6DIBcHWkDULAh_KHudvpzVab5NkWI9?usp=sharing" class="external-link">Google Drive</a>. See the README within the folder for further instructions.</p>
</div>
<div class="section level3">
<h3 id="clone-installation-rstudio">Clone installation (<em>Rstudio</em>)<a class="anchor" aria-label="anchor" href="#clone-installation-rstudio"></a>
</h3>
<p>Lastly, if youâ€™d like (or if for some reason none of the other installation methods are working for you), you can alternatively clone and then build <strong><em>echolocatoR</em></strong>:</p>
<ol style="list-style-type: decimal">
<li>Clone <em>echolocatoR:<br><code>git clone https://github.com/RajLabMSSM/echolocatoR.git</code></em>
</li>
<li>Open <em>echolocatoR.Rproj</em> within the echolocatoR folder.</li>
<li>Then, within <em>Rstudio</em>, build <strong><em>echolocatoR</em></strong> by clicking the following drop down menu items: <code>Build --&gt; Install and Restart</code> (or pressing the keys <code>CMD + SHIFT + B</code> on a Mac).</li>
</ol>
</div>
<div class="section level3">
<h3 id="section"></h3>
<p><br></p>
</div>
<div class="section level3">
<h3 id="dependencies">Dependencies<a class="anchor" aria-label="anchor" href="#dependencies"></a>
</h3>
<div class="section level4">
<h4 id="r">R<a class="anchor" aria-label="anchor" href="#r"></a>
</h4>
<p>For a full list of required and suggested packages, see <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/DESCRIPTION" class="external-link">DESCRIPTION</a>.</p>
<p>Additionally, thereâ€™s some optional R dependencies (e.g. <a href="https://github.com/hfang-bristol/XGR" class="external-link">XGR</a>, <a href="https://www.bioconductor.org/packages/release/bioc/html/Rgraphviz.html" class="external-link">Rgraphviz</a>) that can be a bit tricky to install, so weâ€™ve removed them as requirements and instead provided a separate R function that helps users to install them afterwards if needed:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echolocatoR" class="external-link">echolocatoR</a></span><span class="op">)</span>
<span class="fu"><a href="reference/extra_installs.html">extra_installs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="python">Python<a class="anchor" aria-label="anchor" href="#python"></a>
</h4>
<p>For a full list of required python packages, see the <em>conda</em> env <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/inst/conda/echoR.yml" class="external-link"><em>echoR.yml</em></a>. But here are some of the key ones.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> python<span class="sc">&gt;=</span><span class="dv">3</span>.<span class="fl">6.1</span>  </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> pandas<span class="sc">&gt;=</span><span class="dv">0</span>.<span class="fl">25.0</span>   </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> pandas<span class="sc">-</span>plink  </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> pyarrow  </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> fastparquet  </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> scipy  </span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> scikit<span class="sc">-</span>learn  </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> tqdm  </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> bitarray  </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> networkx  </span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> rpy2  </span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> requests  </span></code></pre></div>
</div>
<div class="section level4">
<h4 id="command-line">Command line<a class="anchor" aria-label="anchor" href="#command-line"></a>
</h4>
<div class="section level5">
<h5 id="tabix">
<a href="http://www.htslib.org/doc/tabix.html" class="external-link">tabix</a><a class="anchor" aria-label="anchor" href="#tabix"></a>
</h5>
<ul>
<li>Rapid querying of summary stats files.</li>
<li>To use it, specify <code>query_by="tabix"</code> in <code><a href="reference/finemap_loci.html">finemap_loci()</a></code>.</li>
<li>If you encounter difficulties using a <em>conda</em> distribution of tabix, we recommend you uninstall it from the env and instead install its parent package, <a href="https://anaconda.org/bioconda/htslib" class="external-link"><em>htslib</em></a> as this should be more up to date. <em>htslib</em> is now included in the echoR <em>conda</em> env by default.</li>
<li>Alternatively, you may install <em>htslib</em> to your machine globally via <a href="https://formulae.brew.sh/formula/htslib" class="external-link"><em>Brew</em></a> (for Mac users) or from <a href="http://www.htslib.org/download" class="external-link">source</a>.</li>
</ul>
</div>
<div class="section level5">
<h5 id="bcftools">
<a href="http://samtools.github.io/bcftools/bcftools.html" class="external-link">bcftools</a><a class="anchor" aria-label="anchor" href="#bcftools"></a>
</h5>
<ul>
<li>Used here for filtering populations in vcf files.</li>
<li>Can be installed via <a href="https://formulae.brew.sh/formula/bcftools" class="external-link"><em>Brew</em></a> (for Mac users) or <a href="https://anaconda.org/bioconda/bcftools" class="external-link"><em>conda</em></a>.</li>
</ul>
</div>
<div class="section level5">
<h5 id="axel">
<a href="https://github.com/axel-download-accelerator/axel" class="external-link">axel</a><a class="anchor" aria-label="anchor" href="#axel"></a>
</h5>
<ul>
<li><p>Rapid multi-core downloading of large files (e.g.Â LD matrices from UK Biobank).</p></li>
<li><p>To use it, specify <code>download_method="axel"</code> in <code><a href="reference/finemap_loci.html">finemap_loci()</a></code>.</p></li>
<li><p><strong>Update</strong>: A conda version of <em>axel</em> has been kindly provided by <a href="https://github.com/RajLabMSSM/echolocatoR/pull/23" class="external-link">@jdblischak</a>, no longer requiring a separate installation.</p></li>
<li>
<p>However, if you want to use <em>axel</em> without the conda env, see this <a href="https://www.tecmint.com/axel-commandline-download-accelerator-for-linux/" class="external-link">tutorial</a> for more info on installation. Hereâ€™s a quick overview:</p>
<ul>
<li>
<strong>Mac</strong>: Install <a href="https://brew.sh/" class="external-link">brew</a>, then: <code>brew install axel</code>
</li>
<li>
<strong>CentOS/RHEL 7</strong>: <code>yum install epel-release; yum install axel</code>
</li>
<li>
<strong>Fedora</strong>: <code>yum install axel; dnf install axel</code>
</li>
<li>
<strong>Debian Jessie (e.g.Â Ubuntu, Linux Mint)</strong>: <code>aptitude install axel</code>
</li>
</ul>
</li>
</ul>
<p><br></p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="fine-mapping-tools">Fine-mapping tools<a class="anchor" aria-label="anchor" href="#fine-mapping-tools"></a>
</h2>
<p><strong><em>echolocatoR</em></strong> will automatically check whether you have the necessary columns to run each tool you selected in <code>finemap_loci(finemap_methods=...)</code>. It will remove any tools that for which there are missing necessary columns, and produces a message letting you know which columns are missing. Note that some columns (e.g. <code>MAF</code>,<code>N</code>,<code>t-stat</code>) can be automatically inferred if missing.<br>
For easy reference, we list the necessary columns here as well.<br>
See <code>?finemap_loci()</code> for descriptions of these columns.<br>
All methods require the columns: <code>SNP</code>,<code>CHR</code>,<code>POS</code>,<code>Effect</code>,<code>StdErr</code></p>
<p>Additional required columns:</p>
<div class="section level3">
<h3 id="abf">
<a href="https://cran.r-project.org/web/packages/coloc/vignettes/vignette.html" class="external-link">ABF</a><a class="anchor" aria-label="anchor" href="#abf"></a>
</h3>
<div class="section level4">
<h4 id="proportion_casesmaf">
<code>proportion_cases</code>,<code>MAF</code>
<a class="anchor" aria-label="anchor" href="#proportion_casesmaf"></a>
</h4>
</div>
</div>
<div class="section level3">
<h3 id="finemap">
<a href="http://www.christianbenner.com" class="external-link">FINEMAP</a><a class="anchor" aria-label="anchor" href="#finemap"></a>
</h3>
<div class="section level4">
<h4 id="a1a2mafn">
<code>A1</code>,<code>A2</code>,<code>MAF</code>,<code>N</code>
<a class="anchor" aria-label="anchor" href="#a1a2mafn"></a>
</h4>
</div>
</div>
<div class="section level3">
<h3 id="susie">
<a href="https://github.com/stephenslab/susieR" class="external-link">SuSiE</a><a class="anchor" aria-label="anchor" href="#susie"></a>
</h3>
<div class="section level4">
<h4 id="n">
<code>N</code><a class="anchor" aria-label="anchor" href="#n"></a>
</h4>
</div>
</div>
<div class="section level3">
<h3 id="polyfun">
<a href="https://github.com/omerwe/polyfun" class="external-link">PolyFun</a><a class="anchor" aria-label="anchor" href="#polyfun"></a>
</h3>
<div class="section level4">
<h4 id="a1a2pn">
<code>A1</code>,<code>A2</code>,<code>P</code>,<code>N</code>
<a class="anchor" aria-label="anchor" href="#a1a2pn"></a>
</h4>
</div>
</div>
<div class="section level3">
<h3 id="paintor">
<a href="https://github.com/gkichaev/PAINTOR_V3.0" class="external-link">PAINTOR</a><a class="anchor" aria-label="anchor" href="#paintor"></a>
</h3>
<div class="section level4">
<h4 id="a1a2t-stat">
<code>A1</code>,<code>A2</code>,<code>t-stat</code>
<a class="anchor" aria-label="anchor" href="#a1a2t-stat"></a>
</h4>
</div>
</div>
<div class="section level3">
<h3 id="gcta-cojo">
<a href="https://cnsgenomics.com/software/gcta/#COJO" class="external-link">GCTA-COJO</a><a class="anchor" aria-label="anchor" href="#gcta-cojo"></a>
</h3>
<div class="section level4">
<h4 id="a1a2freqpn">
<code>A1</code>,<code>A2</code>,<code>Freq</code>,<code>P</code>,<code>N</code>
<a class="anchor" aria-label="anchor" href="#a1a2freqpn"></a>
</h4>
</div>
</div>
<div class="section level3">
<h3 id="coloc">
<a href="https://cran.r-project.org/web/packages/coloc/vignettes/vignette.html" class="external-link">coloc</a><a class="anchor" aria-label="anchor" href="#coloc"></a>
</h3>
<div class="section level4">
<h4 id="nmaf">
<code>N</code>,<code>MAF</code>
<a class="anchor" aria-label="anchor" href="#nmaf"></a>
</h4>
<p><br></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="multi-finemap-results-files">Multi-finemap results files<a class="anchor" aria-label="anchor" href="#multi-finemap-results-files"></a>
</h2>
<p>The main output of <strong><em>echolocatoR</em></strong> are the multi-finemap files (for example, <code>data("BST1")</code>). They are stored in the locus-specific <em>Multi-finemap</em> subfolders.</p>
<div class="section level3">
<h3 id="column-descriptions">Column descriptions<a class="anchor" aria-label="anchor" href="#column-descriptions"></a>
</h3>
<ul>
<li>
<strong>Standardized GWAS/QTL summary statistics</strong>: e.g. <code>SNP</code>,<code>CHR</code>,<code>POS</code>,<code>Effect</code>,<code>StdErr</code>. See <code>?finemap_loci()</code> for descriptions of each.<br>
</li>
<li>
<strong>leadSNP</strong>: The designated proxy SNP per locus, which is the SNP with the smallest p-value by default.</li>
<li>
<strong>&lt;tool&gt;.CS</strong>: The 95% probability Credible Set (CS) to which a SNP belongs within a given fine-mapping toolâ€™s results. If a SNP is not in any of the toolâ€™s CS, it is assigned <code>NA</code> (or <code>0</code> for the purposes of plotting).<br>
</li>
<li>
<strong>&lt;tool&gt;.PP</strong>: The posterior probability that a SNP is causal for a given GWAS/QTL trait.<br>
</li>
<li>
<strong>Support</strong>: The total number of fine-mapping tools that include the SNP in its CS.</li>
<li>
<strong>Consensus_SNP</strong>: By default, defined as a SNP that is included in the CS of more than <code>N</code> fine-mapping tool(s), i.e.Â <code>Support&gt;1</code> (default: <code>N=1</code>).<br>
</li>
<li>
<strong>mean.PP</strong>: The mean SNP-wise PP across all fine-mapping tools used.</li>
<li>
<strong>mean.CS</strong>: If mean PP is greater than the 95% probability threshold (<code>mean.PP&gt;0.95</code>) then <code>mean.CS</code> is 1, else 0. This tends to be a very stringent threshold as it requires a high degree of agreement between fine-mapping tools.</li>
</ul>
</div>
<div class="section level3">
<h3 id="notes">Notes<a class="anchor" aria-label="anchor" href="#notes"></a>
</h3>
<ul>
<li><p>Separate multi-finemap files are generated for each LD reference panel used, which is included in the file name (e.g. <em>UKB_LD.Multi-finemap.tsv.gz</em>).</p></li>
<li><p>Each fine-mapping tool defines its CS and PP slightly differently, so please refer to the associated original publications for the exact details of how these are calculated (links provided above).</p></li>
</ul>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="datasets">Datasets<a class="anchor" aria-label="anchor" href="#datasets"></a>
</h2>
<p>For more detailed information about each dataset, use <code>?</code>:<br><code>R   library(echolocatoR)   ?NOTT_2019.interactome # example dataset</code></p>
<div class="section level3">
<h3 id="epigenomic--genome-wide-annotations">Epigenomic &amp; genome-wide annotations<a class="anchor" aria-label="anchor" href="#epigenomic--genome-wide-annotations"></a>
</h3>
<div class="section level4">
<h4 id="nott-et-al-2019">
<a href="https://science.sciencemag.org/content/366/6469/1134.abstract" class="external-link">Nott et al.Â (2019)</a><a class="anchor" aria-label="anchor" href="#nott-et-al-2019"></a>
</h4>
<ul>
<li><p>Data from this publication contains results from cell type-specific (neurons, oligodendrocytes, astrocytes, microglia, &amp; peripheral myeloid cells) epigenomic assays (H3K27ac, ATAC, H3K4me3) from human brain tissue.</p></li>
<li>
<p>For detailed metadata, see:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"NOTT_2019.bigwig_metadata"</span><span class="op">)</span></code></pre></div>
</li>
<li>
<p>Built-in datasets:</p>
<ul>
<li>Enhancer/promoter coordinates (as <em>GenomicRanges</em>)</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"NOTT_2019.interactome"</span><span class="op">)</span>
<span class="co"># Examples of the data nested in "NOTT_2019.interactome" object:</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Neuronal promoters`</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Neuronal enhancers`</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Microglia promoters`</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Microglia enhancers`</span>
<span class="va">...</span>
<span class="va">...</span></code></pre></div>
<ul>
<li>PLAC-seq enhancer-promoter interactome coordinates</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">H3K4me3_around_TSS_annotated_pe</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Microglia interactome`</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Neuronal interactome`</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Oligo interactome`</span>
<span class="va">...</span>
<span class="va">...</span></code></pre></div>
</li>
<li>
<p>API access to full bigWig files on UCSC Genome Browser, which includes</p>
<ul>
<li>Epigenomic reads (as <em>GenomicRanges</em>)<br>
</li>
<li>Aggregate epigenomic <em>score</em> for each cell type - assay combination</li>
</ul>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="corces-et-al-2020">
<a href="https://www.biorxiv.org/content/10.1101/2020.01.06.896159v1" class="external-link">Corces et al.Â (2020)</a><a class="anchor" aria-label="anchor" href="#corces-et-al-2020"></a>
</h4>
<ul>
<li>
<p>Data from this preprint contains results from bulk and single-cell chromatin accessibility epigenomic assays in 39 human brains.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CORCES_2020.bulkATACseq_peaks"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CORCES_2020.cicero_coaccessibility"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CORCES_2020.HiChIP_FitHiChIP_loop_calls"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CORCES_2020.scATACseq_celltype_peaks"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CORCES_2020.scATACseq_peaks"</span><span class="op">)</span></code></pre></div>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="xgr">
<a href="http://xgr.r-forge.r-project.org" class="external-link">XGR</a><a class="anchor" aria-label="anchor" href="#xgr"></a>
</h4>
<ul>
<li>API access to a diverse library of cell type/line-specific epigenomic (e.g.Â ENCODE) and other genome-wide annotations.</li>
</ul>
</div>
<div class="section level4">
<h4 id="roadmap">
<a href="http://www.roadmapepigenomics.org" class="external-link">Roadmap</a><a class="anchor" aria-label="anchor" href="#roadmap"></a>
</h4>
<ul>
<li>API access to cell type-specific epigenomic data.</li>
</ul>
</div>
<div class="section level4">
<h4 id="biomart">
<a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html" class="external-link">biomaRt</a><a class="anchor" aria-label="anchor" href="#biomart"></a>
</h4>
<ul>
<li>API access to various genome-wide SNP annotations (e.g.Â missense, nonsynonmous, intronic, enhancer).</li>
</ul>
</div>
<div class="section level4">
<h4 id="haplor">
<a href="https://cran.r-project.org/web/packages/haploR/vignettes/haplor-vignette.html" class="external-link">HaploR</a><a class="anchor" aria-label="anchor" href="#haplor"></a>
</h4>
<ul>
<li>API access to known per-SNP QTL and epigenomic data hits.</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="qtls">QTLs<a class="anchor" aria-label="anchor" href="#qtls"></a>
</h3>
<div class="section level4">
<h4 id="eqtl-catalogue">
<a href="https://www.ebi.ac.uk/eqtl/" class="external-link">eQTL Catalogue</a><a class="anchor" aria-label="anchor" href="#eqtl-catalogue"></a>
</h4>
<ul>
<li>API access to full summary statistics from many standardized e/s/t-QTL datasets.<br>
</li>
<li>Data access and colocalization tests facilitated through the <a href="https://github.com/RajLabMSSM/catalogueR" class="external-link">catalogueR</a> R package.</li>
</ul>
<p><br></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="enrichment-tools">Enrichment tools<a class="anchor" aria-label="anchor" href="#enrichment-tools"></a>
</h2>
<div class="section level3">
<h3 id="xgr-1">
<a href="http://xgr.r-forge.r-project.org" class="external-link">XGR</a><a class="anchor" aria-label="anchor" href="#xgr-1"></a>
</h3>
<ul>
<li>Binomial enrichment tests between customisable foreground and background SNPs.</li>
</ul>
</div>
<div class="section level3">
<h3 id="goshifter">
<a href="https://github.com/immunogenomics/goshifter" class="external-link">GoShifter</a><a class="anchor" aria-label="anchor" href="#goshifter"></a>
</h3>
<ul>
<li>LD-informed iterative enrichment analysis.</li>
</ul>
</div>
<div class="section level3">
<h3 id="s-ldsc">
<a href="https://www.nature.com/articles/ng.3954" class="external-link">S-LDSC</a><a class="anchor" aria-label="anchor" href="#s-ldsc"></a>
</h3>
<ul>
<li>Genome-wide stratified LD score regression.</li>
<li>Inlccles 187-annotation baseline model from <a href="https://www.nature.com/articles/s41588-018-0231-8" class="external-link">Gazal et al. 2018</a>.<br>
</li>
<li>You can alternatively supply a custom annotations matrix.</li>
</ul>
</div>
<div class="section level3">
<h3 id="motifbreakr">
<a href="https://github.com/Simon-Coetzee/motifBreakR" class="external-link">motifbreakR</a><a class="anchor" aria-label="anchor" href="#motifbreakr"></a>
</h3>
<ul>
<li>Identification of transcript factor binding motifs (TFBM) and prediction of SNP disruption to said motifs.</li>
<li>Includes a comprehensive list of TFBM databases via <a href="https://bioconductor.org/packages/release/bioc/html/MotifDb.html" class="external-link">MotifDB</a> (9,900+ annotated position frequency matrices from 14 public sources, for multiple organisms).</li>
</ul>
</div>
<div class="section level3">
<h3 id="garfield-under-construction">
<a href="https://www.bioconductor.org/packages/release/bioc/html/garfield.html" class="external-link">GARFIELD</a> (<strong>under construction</strong>)<a class="anchor" aria-label="anchor" href="#garfield-under-construction"></a>
</h3>
<ul>
<li>Genomic enrichment with LD-informed heuristics.</li>
</ul>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="ld-reference-panels">LD reference panels<a class="anchor" aria-label="anchor" href="#ld-reference-panels"></a>
</h2>
<div class="section level3">
<h3 id="uk-biobank">
<a href="https://www.ukbiobank.ac.uk" class="external-link">UK Biobank</a><a class="anchor" aria-label="anchor" href="#uk-biobank"></a>
</h3>
</div>
<div class="section level3">
<h3 id="1000-genomes-phase-1">
<a href="https://www.internationalgenome.org" class="external-link">1000 Genomes Phase 1</a><a class="anchor" aria-label="anchor" href="#1000-genomes-phase-1"></a>
</h3>
</div>
<div class="section level3">
<h3 id="1000-genomes-phase-3">
<a href="https://www.internationalgenome.org" class="external-link">1000 Genomes Phase 3</a><a class="anchor" aria-label="anchor" href="#1000-genomes-phase-3"></a>
</h3>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="creator">Creator<a class="anchor" aria-label="anchor" href="#creator"></a>
</h2>
<p><a href="https://bschilder.github.io/BMSchilder/" target="_blank" class="external-link">Brian M. Schilder, Bioinformatician II</a><br><a href="https://rajlab.org" target="_blank" class="external-link">Raj Lab</a><br><a href="https://icahn.mssm.edu/about/departments/neuroscience" target="_blank" class="external-link">Department of Neuroscience, Icahn School of Medicine at Mount Sinai</a><br><img src="./images/sinai.png" alt="Sinai"></p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/RajLabMSSM/echolocatoR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/RajLabMSSM/echolocatoR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing echolocatoR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Brian Schilder <br><small class="roles"> Maintainer </small> <a href="https://orcid.org/0000-0001-5949-2191" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Jack Humphrey <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-6274-6620" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Towfique Raj <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-9355-5704" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
