<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Automated genomic fine-mapping â€¢ echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Automated genomic fine-mapping">
<meta property="og:description" content="Automated statistical and functional fine-mapping with extensive access to genome-wide datasets.  ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="articles/full_pipeline_vignette.html">Full pipeline vignette</a>
    </li>
    <li>
      <a href="articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<center>
<div class="page-header"><h1>
) ) ) ) ))) <span class="emoji" data-emoji="bat">ðŸ¦‡</span> echolocatoR <span class="emoji" data-emoji="bat">ðŸ¦‡</span> ((( ( ( ( (
</h1></div>
</center>
<h3>
Automated statistical and functional fine-mapping with extensive access to genome-wide datasets
</h3>
<hr>
<div id="if-you-use-echolocator-please-cite-our-preprint-thanks" class="section level4">
<h4 class="hasAnchor">
<a href="#if-you-use-echolocator-please-cite-our-preprint-thanks" class="anchor"></a>If you use <strong><em>echolocatoR</em></strong>, please cite our preprint (thanks!):</h4>
<blockquote>
<p>BM Schilder, J Humphrey &amp; T Raj (2020) echolocatoR: an automated end-to-end statistical and functional genomic fine-mapping pipeline. <em>bioRxiv</em> 2020.10.22.351221; doi: <a href="https://doi.org/10.1101/2020.10.22.351221" class="uri">https://doi.org/10.1101/2020.10.22.351221</a></p>
</blockquote>
</div>
<div id="for-applications-of-echolocator-in-the-literature-please-see" class="section level4">
<h4 class="hasAnchor">
<a href="#for-applications-of-echolocator-in-the-literature-please-see" class="anchor"></a>For applications of <em>echolocatoR</em> in the literature, please see:</h4>
<blockquote>
<ol>
<li>E Navarro, E Udine, K de Paiva Lopes, M Parks, G Riboldi, BM Schilderâ€¦T Raj (2020) Discordant transcriptional signatures of mitochondrial genes in Parkinsonâ€™s disease human myeloid cells. <em>bioRxiv</em>; <a href="https://doi.org/10.1101/2020.07.20.212407" class="uri">https://doi.org/10.1101/2020.07.20.212407</a>
</li>
<li>BM Schilder &amp; T Raj (2020) Fine-Mapping of Parkinsonâ€™s Disease Susceptibility Loci Identifies Putative Causal Variants. <em>bioRxiv</em>, 0â€“30;Â <a href="https://doi.org/10.1101/2020.10.22.340158" class="uri">https://doi.org/10.1101/2020.10.22.340158</a>.</li>
<li>K de Paiva Lopes, GJL Snijders, J Humphrey, A Allan, M Sneeboer, E Navarro, BM Schilderâ€¦T Raj (2020) Atlas of Genetic Effects in Human Microglia Transcriptome across Brain Regions, Aging and Disease Pathologies. <em>bioRxiv</em>;Â <a href="https://doi.org/10.1101/2020.10.27.356113" class="uri">https://doi.org/10.1101/2020.10.27.356113</a>.</li>
</ol>
</blockquote>
<hr>
</div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Fine-mapping methods are a powerful means of identifying causal variants underlying a given phenotype, but are underutilized due to the technical challenges of implementation. <strong><em>echolocatoR</em></strong> is an R package that automates end-to-end genomics fine-mapping, annotation, and plotting in order to identify the most probable causal variants associated with a given phenotype.</p>
<p>It requires minimal input from users (a GWAS or QTL summary statistics file), and includes a suite of statistical and functional fine-mapping tools. It also includes extensive access to datasets (linkage disequilibrium panels, epigenomic and genome-wide annotations, QTL).</p>
<p>The elimination of data gathering and preprocessing steps enables rapid fine-mapping of many loci in any phenotype, complete with locus-specific publication-ready figure generation. All results are merged into a single per-SNP summary file for additional downstream analysis and results sharing. Therefore <strong><em>echolocatoR</em></strong> drastically reduces the barriers to identifying causal variants by making the entire fine-mapping pipeline rapid, robust and scalable.</p>
<p><img src="./images/echolocatoR_Fig1.png" alt="echoFlow"></p>
</div>
<div id="documentation" class="section level2">
<h2 class="hasAnchor">
<a href="#documentation" class="anchor"></a>Documentation</h2>
<div id="documentation-website" class="section level3">
<h3 class="hasAnchor">
<a href="#documentation-website" class="anchor"></a><a href="https://rajlabmssm.github.io/echolocatoR/">Documentation website</a>
</h3>
</div>
<div id="full-pipeline-vignette" class="section level3">
<h3 class="hasAnchor">
<a href="#full-pipeline-vignette" class="anchor"></a><a href="https://rajlabmssm.github.io/echolocatoR/articles/full_pipeline_vignette.html">Full pipeline vignette</a>
</h3>
</div>
<div id="plotting-vignette" class="section level3">
<h3 class="hasAnchor">
<a href="#plotting-vignette" class="anchor"></a><a href="https://rajlabmssm.github.io/echolocatoR/articles/plotting_vignette.html">Plotting vignette</a>
</h3>
<p><br></p>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div id="general-tips" class="section level3">
<h3 class="hasAnchor">
<a href="#general-tips" class="anchor"></a>General tips</h3>
<ul>
<li>We generally recommend users upgrading to R&gt;=4.0.0 before trying to install <em>echolocatoR.</em> While <em>echolocatoR</em> should technically be able to run in R&gt;=3.6.0, some additional challenges with getting dependency versions not to conflict with one another.</li>
</ul>
</div>
<div id="quick-installation" class="section level3">
<h3 class="hasAnchor">
<a href="#quick-installation" class="anchor"></a>Quick installation</h3>
<p>In R:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"remotes"</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span><span class="op">}</span>

<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"RajLabMSSM/echolocatoR"</span><span class="op">)</span></code></pre></div>
</div>
<div id="robust-installation-conda" class="section level3">
<h3 class="hasAnchor">
<a href="#robust-installation-conda" class="anchor"></a>Robust installation (<em>conda</em>)</h3>
<p>As with most softwares, installation is half the battle. The easiest way to install all of <strong><em>echolocatoR</em></strong>â€™s dependencies (which include R, Python, and command line tools) and make sure they play well together is to create a <a href="https://docs.conda.io/en/latest/"><em>conda</em></a> environment.</p>
<ol>
<li><p>If you havenâ€™t done so already, install <a href="https://docs.conda.io/en/latest/"><em>conda</em></a>.</p></li>
<li><p>Download the <em>echoR.yml</em> file found <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/inst/conda/echoR.yml">here</a> (this file tells <em>conda</em> what to install).</p></li>
<li><p>In command line, create the env from the <em>.yml</em> file:<br><code>conda env create -f &lt;path_to_file&gt;/echoR.yml</code></p></li>
<li><p>Activate the new env:<br><code>conda activate echoR</code></p></li>
<li><p>Install <em>echolocatoR</em> from command line so that it installs <strong>within</strong> the <em>conda</em> env:</p></li>
<li>
<p>Open Rstudio from the command line interface (not by clicking the Rstudio icon). This helps to ensure Rstudio can find the paths to the packages in the conda env:<br><code>open model_celltype_conservation.Rproj</code></p>
<p>Alternatively, the <em>conda</em> env also comes with <a href="https://github.com/randy3k/radian"><em>radian</em></a>, which is a convenient R console thatâ€™s much more advanced than the default R console, but doesnâ€™t require access to a GUI. This can be especially useful on computing clusters that donâ€™t support RStudio or other IDEs.<br><code>radian</code></p>
</li>
<li><p>Finally, to make extra sure <strong><em>echolocatoR</em></strong> uses the packages in this env (esp.Â if using from RStudio), you can then supply the env name to the <code><a href="reference/finemap_loci.html">finemap_loci()</a></code> function (and many other <em>echolocatoR</em> functions) using <code>conda_env="echoR"</code>.</p></li>
</ol>
</div>
<div id="binary-installation" class="section level3">
<h3 class="hasAnchor">
<a href="#binary-installation" class="anchor"></a>Binary installation</h3>
<p>You can also install <strong><em>echolocatoR</em></strong> using the compressed binary file available on <a href="https://drive.google.com/drive/folders/1oS6DIBcHWkDULAh_KHudvpzVab5NkWI9?usp=sharing">Google Drive</a>. See the README within the folder for further instructions.</p>
</div>
<div id="clone-installation-rstudio" class="section level3">
<h3 class="hasAnchor">
<a href="#clone-installation-rstudio" class="anchor"></a>Clone installation (<em>Rstudio</em>)</h3>
<p>Lastly, if youâ€™d like (or if for some reason none of the other installation methods are working for you), you can alternatively clone and then build <em>echolocatoR</em>:</p>
<ol>
<li>Clone <em>echolocatoR:<br><code>git clone https://github.com/RajLabMSSM/echolocatoR.git</code></em>
</li>
<li>Open <em>echolocatoR.Rproj</em> within the echolocatoR folder.</li>
<li>Then, within <em>Rstudio</em>, build <em>echolocatoR</em> by clicking the following drop down menu items: <code>Build --&gt; Install and Restart</code> (or pressing the keys <code>CMD + SHIFT + B</code> on a Mac).</li>
</ol>
</div>
<div id="section" class="section level3">
<h3></h3>
<p><br></p>
</div>
<div id="dependencies" class="section level3">
<h3 class="hasAnchor">
<a href="#dependencies" class="anchor"></a>Dependencies</h3>
<div id="r" class="section level4">
<h4 class="hasAnchor">
<a href="#r" class="anchor"></a>R</h4>
<p>For a full list of required and suggested packages, see <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/DESCRIPTION">DESCRIPTION</a>.</p>
<p>Additionally, thereâ€™s some optional R dependencies (e.g. <a href="https://github.com/hfang-bristol/XGR">XGR</a>, <a href="https://www.bioconductor.org/packages/release/bioc/html/Rgraphviz.html">Rgraphviz</a>) that can be a bit tricky to install, so weâ€™ve removed them as requirements and instead provided a separate R function that helps users to install them afterwards if needed:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echolocatoR">echolocatoR</a></span><span class="op">)</span>
<span class="fu"><a href="reference/extra_installs.html">extra_installs</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><strong>data.table</strong>: <span class="math display">â€…*â€…<em>d</em><em>a</em><em>t</em><em>a</em>.<em>t</em><em>a</em><em>b</em><em>l</em><em>e</em>*</span>(<a href="https://github.com/Rdatatable/data.table)" class="uri">https://github.com/Rdatatable/data.table)</a> is a great R package for reading/writing/transforming tables far faster than base R methods. Unfortunately, in <em>data.table v1.14</em> they made some changes that cause errors reading/writing with compressed files (<em>.gz</em>) without additional setup steps on your computer. Therefore, I recommend you use versions v1.13 (the current CRAN binary distribution, NOT the source distribution, as of Feb 2021).</p>
</div>
<div id="python" class="section level4">
<h4 class="hasAnchor">
<a href="#python" class="anchor"></a>Python</h4>
<p>For a full list of required python packages, see the <em>conda</em> env <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/inst/conda/echoR.yml"><em>echoR.yml</em></a>. But here are some of the key ones.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> python<span class="sc">&gt;=</span><span class="dv">3</span>.<span class="fl">6.1</span>  </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> pandas<span class="sc">&gt;=</span><span class="dv">0</span>.<span class="fl">25.0</span>   </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> pandas<span class="sc">-</span>plink  </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> pyarrow  </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> fastparquet  </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> scipy  </span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> scikit<span class="sc">-</span>learn  </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> tqdm  </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> bitarray  </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> networkx  </span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> rpy2  </span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> requests  </span></code></pre></div>
</div>
<div id="command-line" class="section level4">
<h4 class="hasAnchor">
<a href="#command-line" class="anchor"></a>Command line</h4>
<div id="tabix" class="section level5">
<h5 class="hasAnchor">
<a href="#tabix" class="anchor"></a><a href="http://www.htslib.org/doc/tabix.html">tabix</a>
</h5>
<ul>
<li>Rapid querying of summary stats files.</li>
<li>To use it, specify <code>query_by="tabix"</code> in <code><a href="reference/finemap_loci.html">finemap_loci()</a></code>.</li>
<li>If you encounter difficulties using a <em>conda</em> distribution of tabix, we recommend you uninstall it from the env and instead install its parent package, <a href="https://anaconda.org/bioconda/htslib"><em>htslib</em></a> as this should be more up to date. <em>htslib</em> is now included in the echoR <em>conda</em> env by default.</li>
<li>Alternatively, you may install <em>htslib</em> to your machine globally via <a href="https://formulae.brew.sh/formula/htslib"><em>Brew</em></a> (for Mac users) or from <a href="http://www.htslib.org/download">source</a>.</li>
</ul>
</div>
<div id="bcftools" class="section level5">
<h5 class="hasAnchor">
<a href="#bcftools" class="anchor"></a><a href="http://samtools.github.io/bcftools/bcftools.html">bcftools</a>
</h5>
<ul>
<li>Used here for filtering populations in vcf files.</li>
<li>Can be installed via <a href="https://formulae.brew.sh/formula/bcftools"><em>Brew</em></a> (for Mac users) or <a href="https://anaconda.org/bioconda/bcftools"><em>conda</em></a>.</li>
</ul>
</div>
<div id="axel" class="section level5">
<h5 class="hasAnchor">
<a href="#axel" class="anchor"></a><a href="https://github.com/axel-download-accelerator/axel">axel</a>
</h5>
<ul>
<li><p>Rapid multi-core downloading of large files (e.g.Â LD matrices from UK Biobank).</p></li>
<li><p>To use it, specify <code>download_method="axel"</code> in <code><a href="reference/finemap_loci.html">finemap_loci()</a></code>.</p></li>
<li><p><strong>Update</strong>: A conda version of <em>axel</em> has been kindly provided by <a href="https://github.com/RajLabMSSM/echolocatoR/pull/23">@jdblischak</a>, no longer requiring a separate installation.</p></li>
<li>
<p>However, if you want to use <em>axel</em> without the conda env, see this <a href="https://www.tecmint.com/axel-commandline-download-accelerator-for-linux/">tutorial</a> for more info on installation. Hereâ€™s a quick overview:</p>
<ul>
<li>
<strong>Mac</strong>: Install <a href="https://brew.sh/">brew</a>, then: <code>brew install axel</code>
</li>
<li>
<strong>CentOS/RHEL 7</strong>: <code>yum install epel-release; yum install axel</code>
</li>
<li>
<strong>Fedora</strong>: <code>yum install axel; dnf install axel</code>
</li>
<li>
<strong>Debian Jessie (e.g.Â Ubuntu, Linux Mint)</strong>: <code>aptitude install axel</code>
</li>
</ul>
</li>
</ul>
<p><br></p>
</div>
</div>
</div>
</div>
<div id="fine-mapping-tools" class="section level2">
<h2 class="hasAnchor">
<a href="#fine-mapping-tools" class="anchor"></a>Fine-mapping tools</h2>
<p><strong><em>echolocatoR</em></strong> will automatically check whether you have the necessary columns to run each tool you selected in <code><a href="reference/finemap_loci.html">finemap_loci(finemap_methods=...)</a></code>. It will remove any tools that for which there are missing necessary columns, and produces a message letting you know which columns are missing. Note that some columns (e.g. <code>MAF</code>,<code>N</code>,<code>t-stat</code>) can be automatically inferred if missing.<br>
For easy reference, we list the necessary columns here as well.<br>
See <code>?finemap_loci()</code> for descriptions of these columns.<br>
All methods require the columns: <code>SNP</code>,<code>CHR</code>,<code>POS</code>,<code>Effect</code>,<code>StdErr</code></p>
<p>Additional required columns:</p>
<div id="abf" class="section level3">
<h3 class="hasAnchor">
<a href="#abf" class="anchor"></a><a href="https://cran.r-project.org/web/packages/coloc/vignettes/vignette.html">ABF</a>
</h3>
<div id="proportion_casesmaf" class="section level4">
<h4 class="hasAnchor">
<a href="#proportion_casesmaf" class="anchor"></a><code>proportion_cases</code>,<code>MAF</code>
</h4>
</div>
</div>
<div id="finemap" class="section level3">
<h3 class="hasAnchor">
<a href="#finemap" class="anchor"></a><a href="http://www.christianbenner.com">FINEMAP</a>
</h3>
<div id="a1a2mafn" class="section level4">
<h4 class="hasAnchor">
<a href="#a1a2mafn" class="anchor"></a><code>A1</code>,<code>A2</code>,<code>MAF</code>,<code>N</code>
</h4>
</div>
</div>
<div id="susie" class="section level3">
<h3 class="hasAnchor">
<a href="#susie" class="anchor"></a><a href="https://github.com/stephenslab/susieR">SuSiE</a>
</h3>
<div id="n" class="section level4">
<h4 class="hasAnchor">
<a href="#n" class="anchor"></a><code>N</code>
</h4>
</div>
</div>
<div id="polyfun" class="section level3">
<h3 class="hasAnchor">
<a href="#polyfun" class="anchor"></a><a href="https://github.com/omerwe/polyfun">PolyFun</a>
</h3>
<div id="a1a2pn" class="section level4">
<h4 class="hasAnchor">
<a href="#a1a2pn" class="anchor"></a><code>A1</code>,<code>A2</code>,<code>P</code>,<code>N</code>
</h4>
</div>
</div>
<div id="paintor" class="section level3">
<h3 class="hasAnchor">
<a href="#paintor" class="anchor"></a><a href="https://github.com/gkichaev/PAINTOR_V3.0">PAINTOR</a>
</h3>
<div id="a1a2t-stat" class="section level4">
<h4 class="hasAnchor">
<a href="#a1a2t-stat" class="anchor"></a><code>A1</code>,<code>A2</code>,<code>t-stat</code>
</h4>
</div>
</div>
<div id="gcta-cojo" class="section level3">
<h3 class="hasAnchor">
<a href="#gcta-cojo" class="anchor"></a><a href="https://cnsgenomics.com/software/gcta/#COJO">GCTA-COJO</a>
</h3>
<div id="a1a2freqpn" class="section level4">
<h4 class="hasAnchor">
<a href="#a1a2freqpn" class="anchor"></a><code>A1</code>,<code>A2</code>,<code>Freq</code>,<code>P</code>,<code>N</code>
</h4>
</div>
</div>
<div id="coloc" class="section level3">
<h3 class="hasAnchor">
<a href="#coloc" class="anchor"></a><a href="https://cran.r-project.org/web/packages/coloc/vignettes/vignette.html">coloc</a>
</h3>
<div id="nmaf" class="section level4">
<h4 class="hasAnchor">
<a href="#nmaf" class="anchor"></a><code>N</code>,<code>MAF</code>
</h4>
<p><br></p>
</div>
</div>
</div>
<div id="multi-finemap-results-files" class="section level2">
<h2 class="hasAnchor">
<a href="#multi-finemap-results-files" class="anchor"></a>Multi-finemap results files</h2>
<p>The main output of <strong>echolocatoR</strong> are the multi-finemap files (for example, <code><a href="https://rdrr.io/r/utils/data.html">data("BST1")</a></code>). They are stored in the locus-specific <em>Multi-finemap</em> subfolders.</p>
<div id="column-descriptions" class="section level3">
<h3 class="hasAnchor">
<a href="#column-descriptions" class="anchor"></a>Column descriptions</h3>
<ul>
<li>
<strong>Standardized GWAS/QTL summary statistics</strong>: e.g.Â <code>SNP</code>,<code>CHR</code>,<code>POS</code>,<code>Effect</code>,<code>StdErr</code>. See <code>?finemap_loci()</code> for descriptions of each.<br>
</li>
<li>
<strong>leadSNP</strong>: The designated proxy SNP per locus, which is the SNP with the smallest p-value by default.</li>
<li>
<strong>&lt;tool&gt;.CS</strong>: The 95% probability Credible Set (CS) to which a SNP belongs within a given fine-mapping toolâ€™s results. If a SNP is not in any of the toolâ€™s CS, it is assigned <code>NA</code> (or <code>0</code> for the purposes of plotting).<br>
</li>
<li>
<strong>&lt;tool&gt;.PP</strong>: The posterior probability that a SNP is causal for a given GWAS/QTL trait.<br>
</li>
<li>
<strong>Support</strong>: The total number of fine-mapping tools that include the SNP in its CS.</li>
<li>
<strong>Consensus_SNP</strong>: By default, defined as a SNP that is included in the CS of more than <code>N</code> fine-mapping tool(s), i.e.Â <code>Support&gt;1</code> (default: <code>N=1</code>).<br>
</li>
<li>
<strong>mean.PP</strong>: The mean SNP-wise PP across all fine-mapping tools used.</li>
<li>
<strong>mean.CS</strong>: If mean PP is greater than the 95% probability threshold (<code>mean.PP&gt;0.95</code>) then <code>mean.CS</code> is 1, else 0. This tends to be a very stringent threshold as it requires a high degree of agreement between fine-mapping tools.</li>
</ul>
</div>
<div id="notes" class="section level3">
<h3 class="hasAnchor">
<a href="#notes" class="anchor"></a>Notes</h3>
<ul>
<li><p>Separate multi-finemap files are generated for each LD reference panel used, which is included in the file name (e.g.Â <em>UKB_LD.Multi-finemap.tsv.gz</em>).</p></li>
<li><p>Each fine-mapping tool defines its CS and PP slightly differently, so please refer to the associated original publications for the exact details of how these are calculated (links provided above).</p></li>
</ul>
<p><br></p>
</div>
</div>
<div id="datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#datasets" class="anchor"></a>Datasets</h2>
<p>For more detailed information about each dataset, use <code><a href="https://rdrr.io/r/utils/Question.html">?</a></code>:<br><code>R   library(echolocatoR)   ?NOTT_2019.interactome # example dataset</code></p>
<div id="epigenomic--genome-wide-annotations" class="section level3">
<h3 class="hasAnchor">
<a href="#epigenomic--genome-wide-annotations" class="anchor"></a>Epigenomic &amp; genome-wide annotations</h3>
<div id="nott-et-al-2019" class="section level4">
<h4 class="hasAnchor">
<a href="#nott-et-al-2019" class="anchor"></a><a href="https://science.sciencemag.org/content/366/6469/1134.abstract">Nott et al.Â (2019)</a>
</h4>
<ul>
<li><p>Data from this publication contains results from cell type-specific (neurons, oligodendrocytes, astrocytes, microglia, &amp; peripheral myeloid cells) epigenomic assays (H3K27ac, ATAC, H3K4me3) from human brain tissue.</p></li>
<li>
<p>For detailed metadata, see:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"NOTT_2019.bigwig_metadata"</span><span class="op">)</span></code></pre></div>
</li>
<li>
<p>Built-in datasets:</p>
<ul>
<li>Enhancer/promoter coordinates (as <em>GenomicRanges</em>)</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"NOTT_2019.interactome"</span><span class="op">)</span>
<span class="co"># Examples of the data nested in "NOTT_2019.interactome" object:</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Neuronal promoters`</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Neuronal enhancers`</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Microglia promoters`</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Microglia enhancers`</span>
<span class="va">...</span>
<span class="va">...</span></code></pre></div>
<ul>
<li>PLAC-seq enhancer-promoter interactome coordinates</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">H3K4me3_around_TSS_annotated_pe</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Microglia interactome`</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Neuronal interactome`</span>
<span class="va">NOTT_2019.interactome</span><span class="op">$</span><span class="va">`Oligo interactome`</span>
<span class="va">...</span>
<span class="va">...</span></code></pre></div>
</li>
<li>
<p>API access to full bigWig files on UCSC Genome Browser, which includes</p>
<ul>
<li>Epigenomic reads (as <em>GenomicRanges</em>)<br>
</li>
<li>Aggregate epigenomic <em>score</em> for each cell type - assay combination</li>
</ul>
</li>
</ul>
</div>
<div id="corces-et-al-2020" class="section level4">
<h4 class="hasAnchor">
<a href="#corces-et-al-2020" class="anchor"></a><a href="https://www.biorxiv.org/content/10.1101/2020.01.06.896159v1">Corces et al.Â (2020)</a>
</h4>
<ul>
<li>
<p>Data from this preprint contains results from bulk and single-cell chromatin accessibility epigenomic assays in 39 human brains.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"CORCES_2020.bulkATACseq_peaks"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"CORCES_2020.cicero_coaccessibility"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"CORCES_2020.HiChIP_FitHiChIP_loop_calls"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"CORCES_2020.scATACseq_celltype_peaks"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"CORCES_2020.scATACseq_peaks"</span><span class="op">)</span></code></pre></div>
</li>
</ul>
</div>
<div id="xgr" class="section level4">
<h4 class="hasAnchor">
<a href="#xgr" class="anchor"></a><a href="http://xgr.r-forge.r-project.org">XGR</a>
</h4>
<ul>
<li>API access to a diverse library of cell type/line-specific epigenomic (e.g.Â ENCODE) and other genome-wide annotations.</li>
</ul>
</div>
<div id="roadmap" class="section level4">
<h4 class="hasAnchor">
<a href="#roadmap" class="anchor"></a><a href="http://www.roadmapepigenomics.org">Roadmap</a>
</h4>
<ul>
<li>API access to cell type-specific epigenomic data.</li>
</ul>
</div>
<div id="biomart" class="section level4">
<h4 class="hasAnchor">
<a href="#biomart" class="anchor"></a><a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html">biomaRt</a>
</h4>
<ul>
<li>API access to various genome-wide SNP annotations (e.g.Â missense, nonsynonmous, intronic, enhancer).</li>
</ul>
</div>
<div id="haplor" class="section level4">
<h4 class="hasAnchor">
<a href="#haplor" class="anchor"></a><a href="https://cran.r-project.org/web/packages/haploR/vignettes/haplor-vignette.html">HaploR</a>
</h4>
<ul>
<li>API access to known per-SNP QTL and epigenomic data hits.</li>
</ul>
</div>
</div>
<div id="qtls" class="section level3">
<h3 class="hasAnchor">
<a href="#qtls" class="anchor"></a>QTLs</h3>
<div id="eqtl-catalogue" class="section level4">
<h4 class="hasAnchor">
<a href="#eqtl-catalogue" class="anchor"></a><a href="https://www.ebi.ac.uk/eqtl/">eQTL Catalogue</a>
</h4>
<ul>
<li>API access to full summary statistics from many standardized e/s/t-QTL datasets.<br>
</li>
<li>Data access and colocalization tests facilitated through the <a href="https://github.com/RajLabMSSM/catalogueR">catalogueR</a> R package.</li>
</ul>
<p><br></p>
</div>
</div>
</div>
<div id="enrichment-tools" class="section level2">
<h2 class="hasAnchor">
<a href="#enrichment-tools" class="anchor"></a>Enrichment tools</h2>
<div id="xgr-1" class="section level3">
<h3 class="hasAnchor">
<a href="#xgr-1" class="anchor"></a><a href="http://xgr.r-forge.r-project.org">XGR</a>
</h3>
<ul>
<li>Binomial enrichment tests between customisable foreground and background SNPs.</li>
</ul>
</div>
<div id="goshifter" class="section level3">
<h3 class="hasAnchor">
<a href="#goshifter" class="anchor"></a><a href="https://github.com/immunogenomics/goshifter">GoShifter</a>
</h3>
<ul>
<li>LD-informed iterative enrichment analysis.</li>
</ul>
</div>
<div id="s-ldsc" class="section level3">
<h3 class="hasAnchor">
<a href="#s-ldsc" class="anchor"></a><a href="https://www.nature.com/articles/ng.3954">S-LDSC</a>
</h3>
<ul>
<li>Genome-wide stratified LD score regression.</li>
<li>Inlccles 187-annotation baseline model from <a href="https://www.nature.com/articles/s41588-018-0231-8">Gazal et al. 2018</a>.<br>
</li>
<li>You can alternatively supply a custom annotations matrix.</li>
</ul>
</div>
<div id="motifbreakr" class="section level3">
<h3 class="hasAnchor">
<a href="#motifbreakr" class="anchor"></a><a href="https://github.com/Simon-Coetzee/motifBreakR">motifbreakR</a>
</h3>
<ul>
<li>Identification of transcript factor binding motifs (TFBM) and prediction of SNP disruption to said motifs.</li>
<li>Includes a comprehensive list of TFBM databases via <a href="https://bioconductor.org/packages/release/bioc/html/MotifDb.html">MotifDB</a> (9,900+ annotated position frequency matrices from 14 public sources, for multiple organisms).</li>
</ul>
</div>
<div id="garfield-under-construction" class="section level3">
<h3 class="hasAnchor">
<a href="#garfield-under-construction" class="anchor"></a><a href="https://www.bioconductor.org/packages/release/bioc/html/garfield.html">GARFIELD</a> (<strong>under construction</strong>)</h3>
<ul>
<li>Genomic enrichment with LD-informed heuristics.</li>
</ul>
<p><br></p>
</div>
</div>
<div id="ld-reference-panels" class="section level2">
<h2 class="hasAnchor">
<a href="#ld-reference-panels" class="anchor"></a>LD reference panels</h2>
<div id="uk-biobank" class="section level3">
<h3 class="hasAnchor">
<a href="#uk-biobank" class="anchor"></a><a href="https://www.ukbiobank.ac.uk">UK Biobank</a>
</h3>
</div>
<div id="1000-genomes-phase-1" class="section level3">
<h3 class="hasAnchor">
<a href="#1000-genomes-phase-1" class="anchor"></a><a href="https://www.internationalgenome.org">1000 Genomes Phase 1</a>
</h3>
</div>
<div id="1000-genomes-phase-3" class="section level3">
<h3 class="hasAnchor">
<a href="#1000-genomes-phase-3" class="anchor"></a><a href="https://www.internationalgenome.org">1000 Genomes Phase 3</a>
</h3>
<hr>
<hr>
</div>
</div>
<div id="creator" class="section level2">
<h2 class="hasAnchor">
<a href="#creator" class="anchor"></a>Creator</h2>
<p><a href="https://bschilder.github.io/BMSchilder/" target="_blank">Brian M. Schilder, Bioinformatician II</a><br><a href="https://rajlab.org" target="_blank">Raj Lab</a><br><a href="https://icahn.mssm.edu/about/departments/neuroscience" target="_blank">Department of Neuroscience, Icahn School of Medicine at Mount Sinai</a><br><img src="./images/sinai.png" alt="Sinai"></p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/RajLabMSSM/echolocatoR/">https://â€‹github.com/â€‹RajLabMSSM/â€‹echolocatoR/â€‹</a>
</li>
<li>Report a bug at <br><a href="https://github.com/RajLabMSSM/echolocatoR/issues">https://â€‹github.com/â€‹RajLabMSSM/â€‹echolocatoR/â€‹issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Brian Schilder <br><small class="roles"> Author, maintainer </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
