<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Brian M. Schilder" />


<title>echolocatoR: plotting vignette</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">echolocatoR: plotting vignette</h1>
<h4 class="author">Brian M. Schilder</h4>
<h4 class="date">Most Recent Update:<br> 2020-06-17</h4>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(echolocatoR) </a></code></pre></div>
<div id="plotting-only" class="section level1">
<h1>Plotting only</h1>
<p><em>echolocatoR</em> contains various functions that can be used separately<br />
from the comprehensive <code>finemap_loci()</code> pipeline.</p>
<p>Generate a multiview plot using <code>GGBIO.plot()</code>. * You can mix and match different tracks and annotations using the different arguments (see <code>?GGBIO.plot</code> for details).<br />
* Zoom in/out by adjusting the <code>plot.window</code> size (in units of basepairs).<br />
The plot is centered on the lead/index SNP.<br />
* <code>GGBIO.plot()</code> returns a <code>ggbio</code> track object. + Further modify this object using <code>ggplot2</code> functions like <code>+ theme()</code>. + The modifications will be applied to all tracks at once.<br />
* Save a high-resolution <em>.png</em> of the plot by setting <code>save_plot=T</code><br />
+ Further increase resolution by adjusting the <code>dpi</code> argument (<em>default=400</em>). + Adjust the <code>height</code> and <code>width</code> of the saved plot using these respective arguments. + The plot will be automatically saved in the locus-specific directory as:<br />
*multiview_<locus>_&lt;plot.window&gt;.png*.</p>
<div id="load-example-data" class="section level2">
<h2>Load example data</h2>
<p>Load example dataset of the results from fine-mapping the BST1 locus with <code>finemap_loci()</code>. Original data comes from the recent Nalls et al. (2019) Parkinsonâ€™s disease GWAS (see <code>?BST1</code> for details).</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">data</span>(<span class="st">&quot;BST1&quot;</span>); <span class="kw">data</span>(<span class="st">&quot;LD_matrix&quot;</span>); <span class="kw">data</span>(<span class="st">&quot;locus_dir&quot;</span>);</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">locus_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="st">&quot;~/Desktop&quot;</span>,locus_dir)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">finemap_DT &lt;-<span class="st"> </span>BST1 </a></code></pre></div>
</div>
<div id="using-no-annotations" class="section level2">
<h2>Using NO annotations</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">trk_plot &lt;-<span class="st"> </span><span class="kw">GGBIO.plot</span>(<span class="dt">finemap_dat=</span>finemap_DT, </a>
<a class="sourceLine" id="cb3-2" data-line-number="2">                       <span class="dt">LD_matrix=</span>LD_matrix, </a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                       <span class="dt">locus_dir=</span>locus_dir, </a>
<a class="sourceLine" id="cb3-4" data-line-number="4">                       <span class="dt">XGR_libnames=</span><span class="ot">NULL</span>, </a>
<a class="sourceLine" id="cb3-5" data-line-number="5">                       <span class="dt">save_plot=</span>F,</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">                       <span class="dt">show_plot=</span>F)</a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt; [1] &quot;+ Identifying Consensus SNPs...&quot;</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt; [1] &quot;++ support_thresh = 2&quot;</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt; [1] &quot;+ Calculating mean Posterior Probability (mean.PP)...&quot;</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; [1] &quot;++ 4 fine-mapping methods used.&quot;</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt; [1] &quot;++ 7 Credible Set SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt; [1] &quot;++ 3 Consensus SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt; [1] &quot;LD Matrix dimensions: 95 x 95&quot;</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt; [1] &quot;Extracting LD subset for lead SNP: rs4698412&quot;</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; [1] &quot;++ GGBIO:: ABF Track&quot;</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; [1] &quot;++ GGBIO:: FINEMAP Track&quot;</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt; [1] &quot;++ GGBIO:: SUSIE Track&quot;</span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt; [1] &quot;++ GGBIO:: POLYFUN_SUSIE Track&quot;</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#&gt; [1] &quot;++ GGBIO:: mean Track&quot;</span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="co">#&gt; [1] &quot;++ GGBIO:: Gene Model Track&quot;</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="co">#&gt; [1] &quot;+ Annotating at transcript-level.&quot;</span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-34" data-line-number="34"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-36" data-line-number="36"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-38" data-line-number="38"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb3-39" data-line-number="39"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb3-40" data-line-number="40">trk_plot &lt;-<span class="st"> </span>trk_plot <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">strip.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill=</span><span class="st">&quot;gray&quot;</span>),</a>
<a class="sourceLine" id="cb3-41" data-line-number="41">                             <span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;blue&quot;</span>))</a></code></pre></div>
</div>
<div id="using-only-xgr-annotations" class="section level2">
<h2>Using only XGR annotations</h2>
<p>Whenever you use annotation arguments (e.g. <code>XGR_libnames</code>,<code>Roadmap</code>,<code>Nott_epigenome</code>) the annotations that overlap with your locus will automatically be saved as <code>GRanges</code> objects<br />
in a locus-specific subdirectory:<br />
<em>results/<dataset_type>/<dataset_name>/<locus>/annotation</em></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">trk_plot.xgr &lt;-<span class="st"> </span><span class="kw">GGBIO.plot</span>(<span class="dt">finemap_dat=</span>finemap_DT, </a>
<a class="sourceLine" id="cb4-2" data-line-number="2">                           <span class="dt">LD_matrix=</span>LD_matrix, </a>
<a class="sourceLine" id="cb4-3" data-line-number="3">                           <span class="dt">locus_dir=</span>locus_dir, </a>
<a class="sourceLine" id="cb4-4" data-line-number="4">                           <span class="dt">XGR_libnames=</span><span class="kw">c</span>(<span class="st">&quot;ENCODE_TFBS_ClusteredV3_CellTypes&quot;</span>), </a>
<a class="sourceLine" id="cb4-5" data-line-number="5">                           <span class="dt">save_plot=</span>F,</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">                           <span class="dt">show_plot=</span>F)</a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt; [1] &quot;+ Identifying Consensus SNPs...&quot;</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; [1] &quot;++ support_thresh = 2&quot;</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; [1] &quot;+ Calculating mean Posterior Probability (mean.PP)...&quot;</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt; [1] &quot;++ 4 fine-mapping methods used.&quot;</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt; [1] &quot;++ 7 Credible Set SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; [1] &quot;++ 3 Consensus SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt; [1] &quot;LD Matrix dimensions: 95 x 95&quot;</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt; [1] &quot;Extracting LD subset for lead SNP: rs4698412&quot;</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; [1] &quot;++ GGBIO:: ABF Track&quot;</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt; [1] &quot;++ GGBIO:: FINEMAP Track&quot;</span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="co">#&gt; [1] &quot;++ GGBIO:: SUSIE Track&quot;</span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#&gt; [1] &quot;++ GGBIO:: POLYFUN_SUSIE Track&quot;</span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="co">#&gt; [1] &quot;++ GGBIO:: mean Track&quot;</span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="co">#&gt; [1] &quot;++ GGBIO:: Gene Model Track&quot;</span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co">#&gt; [1] &quot;+ Annotating at transcript-level.&quot;</span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="co">#&gt; [1] &quot;++ GGBIO:: XGR Tracks&quot;</span></a>
<a class="sourceLine" id="cb4-29" data-line-number="29"><span class="co">#&gt; Start at 2020-06-17 17:33:24</span></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-31" data-line-number="31"><span class="co">#&gt; 'ENCODE_TFBS_ClusteredV3_CellTypes' (from http://galahad.well.ox.ac.uk/bigdata/ENCODE_TFBS_ClusteredV3_CellTypes.RData) has been loaded into the working environment (at 2020-06-17 17:33:34)</span></a>
<a class="sourceLine" id="cb4-32" data-line-number="32"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="co">#&gt; End at 2020-06-17 17:33:34</span></a>
<a class="sourceLine" id="cb4-34" data-line-number="34"><span class="co">#&gt; Runtime in total is: 10 secs</span></a>
<a class="sourceLine" id="cb4-35" data-line-number="35"><span class="co">#&gt; 8702 query SNP(s) detected with reference overlap.</span></a>
<a class="sourceLine" id="cb4-36" data-line-number="36"><span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span></a>
<a class="sourceLine" id="cb4-37" data-line-number="37"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb4-38" data-line-number="38"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb4-39" data-line-number="39"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-40" data-line-number="40"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-41" data-line-number="41"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-42" data-line-number="42"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-43" data-line-number="43"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-44" data-line-number="44"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-45" data-line-number="45"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-46" data-line-number="46"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-47" data-line-number="47"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-48" data-line-number="48"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-49" data-line-number="49"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-50" data-line-number="50"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-51" data-line-number="51"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-52" data-line-number="52"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-53" data-line-number="53"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-54" data-line-number="54"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-55" data-line-number="55"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-56" data-line-number="56"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-57" data-line-number="57"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-58" data-line-number="58"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-59" data-line-number="59"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-60" data-line-number="60"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-61" data-line-number="61"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-62" data-line-number="62"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-63" data-line-number="63"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-64" data-line-number="64"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-65" data-line-number="65"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-66" data-line-number="66"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-67" data-line-number="67"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-68" data-line-number="68"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-69" data-line-number="69"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-70" data-line-number="70"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-71" data-line-number="71"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-72" data-line-number="72"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-73" data-line-number="73"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb4-74" data-line-number="74"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb4-75" data-line-number="75"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb4-76" data-line-number="76"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb4-77" data-line-number="77"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb4-78" data-line-number="78"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb4-79" data-line-number="79"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb4-80" data-line-number="80"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb4-81" data-line-number="81"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb4-82" data-line-number="82"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb4-83" data-line-number="83"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb4-84" data-line-number="84"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb4-85" data-line-number="85"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb4-86" data-line-number="86"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb4-87" data-line-number="87"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb4-88" data-line-number="88"><span class="co">#&gt; replace the existing scale.</span></a></code></pre></div>
</div>
<div id="using-only-roadmap-annotations" class="section level2">
<h2>Using only Roadmap annotations</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">trk_plot.roadmap &lt;-<span class="st"> </span><span class="kw">GGBIO.plot</span>(<span class="dt">finemap_dat=</span>finemap_DT, </a>
<a class="sourceLine" id="cb5-2" data-line-number="2">                               <span class="dt">LD_matrix=</span>LD_matrix, </a>
<a class="sourceLine" id="cb5-3" data-line-number="3">                               <span class="dt">locus_dir=</span>locus_dir, </a>
<a class="sourceLine" id="cb5-4" data-line-number="4">                               <span class="dt">XGR_libnames=</span><span class="ot">NULL</span>, </a>
<a class="sourceLine" id="cb5-5" data-line-number="5">                               <span class="dt">Roadmap=</span>T, </a>
<a class="sourceLine" id="cb5-6" data-line-number="6">                               <span class="dt">Roadmap_query=</span><span class="st">&quot;monocyte&quot;</span>, </a>
<a class="sourceLine" id="cb5-7" data-line-number="7">                               <span class="dt">save_plot=</span>F, </a>
<a class="sourceLine" id="cb5-8" data-line-number="8">                               <span class="dt">show_plot=</span>F)</a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt; [1] &quot;+ Identifying Consensus SNPs...&quot;</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt; [1] &quot;++ support_thresh = 2&quot;</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt; [1] &quot;+ Calculating mean Posterior Probability (mean.PP)...&quot;</span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt; [1] &quot;++ 4 fine-mapping methods used.&quot;</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt; [1] &quot;++ 7 Credible Set SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt; [1] &quot;++ 3 Consensus SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt; [1] &quot;LD Matrix dimensions: 95 x 95&quot;</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; [1] &quot;Extracting LD subset for lead SNP: rs4698412&quot;</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt; [1] &quot;++ GGBIO:: ABF Track&quot;</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt; [1] &quot;++ GGBIO:: FINEMAP Track&quot;</span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb5-22" data-line-number="22"><span class="co">#&gt; [1] &quot;++ GGBIO:: SUSIE Track&quot;</span></a>
<a class="sourceLine" id="cb5-23" data-line-number="23"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb5-24" data-line-number="24"><span class="co">#&gt; [1] &quot;++ GGBIO:: POLYFUN_SUSIE Track&quot;</span></a>
<a class="sourceLine" id="cb5-25" data-line-number="25"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb5-26" data-line-number="26"><span class="co">#&gt; [1] &quot;++ GGBIO:: mean Track&quot;</span></a>
<a class="sourceLine" id="cb5-27" data-line-number="27"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb5-28" data-line-number="28"><span class="co">#&gt; [1] &quot;++ GGBIO:: Gene Model Track&quot;</span></a>
<a class="sourceLine" id="cb5-29" data-line-number="29"><span class="co">#&gt; [1] &quot;+ Annotating at transcript-level.&quot;</span></a>
<a class="sourceLine" id="cb5-30" data-line-number="30"><span class="co">#&gt; [1] &quot;+ GGBIO:: Creating Roadmap track&quot;</span></a>
<a class="sourceLine" id="cb5-31" data-line-number="31"><span class="co">#&gt; [1] &quot;+ Saved annotation file detected. Loading...&quot;</span></a>
<a class="sourceLine" id="cb5-32" data-line-number="32"><span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span></a>
<a class="sourceLine" id="cb5-33" data-line-number="33"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-34" data-line-number="34"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-35" data-line-number="35"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-36" data-line-number="36"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-37" data-line-number="37"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-38" data-line-number="38"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-39" data-line-number="39"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-40" data-line-number="40"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-41" data-line-number="41"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-42" data-line-number="42"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-43" data-line-number="43"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-44" data-line-number="44"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-45" data-line-number="45"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-46" data-line-number="46"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-47" data-line-number="47"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-48" data-line-number="48"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-49" data-line-number="49"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-50" data-line-number="50"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-51" data-line-number="51"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-52" data-line-number="52"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-53" data-line-number="53"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-54" data-line-number="54"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-55" data-line-number="55"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-56" data-line-number="56"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-57" data-line-number="57"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-58" data-line-number="58"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-59" data-line-number="59"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-60" data-line-number="60"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-61" data-line-number="61"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-62" data-line-number="62"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-63" data-line-number="63"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-64" data-line-number="64"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-65" data-line-number="65"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-66" data-line-number="66"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-67" data-line-number="67"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb5-68" data-line-number="68"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb5-69" data-line-number="69"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb5-70" data-line-number="70"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb5-71" data-line-number="71"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb5-72" data-line-number="72"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb5-73" data-line-number="73"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb5-74" data-line-number="74"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb5-75" data-line-number="75"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb5-76" data-line-number="76"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb5-77" data-line-number="77"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb5-78" data-line-number="78"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb5-79" data-line-number="79"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb5-80" data-line-number="80"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb5-81" data-line-number="81"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb5-82" data-line-number="82"><span class="co">#&gt; replace the existing scale.</span></a></code></pre></div>
</div>
<div id="using-only-nott_2019-annotations" class="section level2">
<h2>Using only Nott_2019 annotations</h2>
<ul>
<li>Query and plot brain cell type-specific epigenomic assays from Nott et al. (2019)<br />
(see <code>?NOTT_2019.bigwig_metadata</code> for details).</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">trk_plot.nott_<span class="dv">2019</span> &lt;-<span class="st"> </span><span class="kw">GGBIO.plot</span>(<span class="dt">finemap_dat=</span>finemap_DT, </a>
<a class="sourceLine" id="cb6-2" data-line-number="2">                                 <span class="dt">LD_matrix=</span>LD_matrix, </a>
<a class="sourceLine" id="cb6-3" data-line-number="3">                                 <span class="dt">locus_dir=</span>locus_dir, </a>
<a class="sourceLine" id="cb6-4" data-line-number="4">                                 <span class="dt">XGR_libnames=</span><span class="ot">NULL</span>, </a>
<a class="sourceLine" id="cb6-5" data-line-number="5">                                 <span class="dt">Nott_epigenome=</span>T, </a>
<a class="sourceLine" id="cb6-6" data-line-number="6">                                 <span class="dt">Nott_binwidth =</span> <span class="dv">100</span>,</a>
<a class="sourceLine" id="cb6-7" data-line-number="7">                                 <span class="dt">Nott_regulatory_rects =</span> T, </a>
<a class="sourceLine" id="cb6-8" data-line-number="8">                                 <span class="dt">Nott_show_placseq =</span> T,</a>
<a class="sourceLine" id="cb6-9" data-line-number="9">                                 <span class="dt">save_plot=</span>F,</a>
<a class="sourceLine" id="cb6-10" data-line-number="10">                                 <span class="dt">show_plot=</span>F)</a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt; [1] &quot;+ Identifying Consensus SNPs...&quot;</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt; [1] &quot;++ support_thresh = 2&quot;</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt; [1] &quot;+ Calculating mean Posterior Probability (mean.PP)...&quot;</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt; [1] &quot;++ 4 fine-mapping methods used.&quot;</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt; [1] &quot;++ 7 Credible Set SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt; [1] &quot;++ 3 Consensus SNPs identified.&quot;</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt; [1] &quot;LD Matrix dimensions: 95 x 95&quot;</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt; [1] &quot;Extracting LD subset for lead SNP: rs4698412&quot;</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt; [1] &quot;++ GGBIO:: ABF Track&quot;</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt; [1] &quot;++ GGBIO:: FINEMAP Track&quot;</span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb6-24" data-line-number="24"><span class="co">#&gt; [1] &quot;++ GGBIO:: SUSIE Track&quot;</span></a>
<a class="sourceLine" id="cb6-25" data-line-number="25"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb6-26" data-line-number="26"><span class="co">#&gt; [1] &quot;++ GGBIO:: POLYFUN_SUSIE Track&quot;</span></a>
<a class="sourceLine" id="cb6-27" data-line-number="27"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb6-28" data-line-number="28"><span class="co">#&gt; [1] &quot;++ GGBIO:: mean Track&quot;</span></a>
<a class="sourceLine" id="cb6-29" data-line-number="29"><span class="co">#&gt; using coord:genome to parse x scale</span></a>
<a class="sourceLine" id="cb6-30" data-line-number="30"><span class="co">#&gt; [1] &quot;++ GGBIO:: Gene Model Track&quot;</span></a>
<a class="sourceLine" id="cb6-31" data-line-number="31"><span class="co">#&gt; [1] &quot;+ Annotating at transcript-level.&quot;</span></a>
<a class="sourceLine" id="cb6-32" data-line-number="32"><span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span></a>
<a class="sourceLine" id="cb6-33" data-line-number="33"><span class="co">#&gt; [1] &quot;Importing Astrocyte enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb6-34" data-line-number="34"><span class="co">#&gt; [1] &quot;Importing Astrocyte promoters ...&quot;</span></a>
<a class="sourceLine" id="cb6-35" data-line-number="35"><span class="co">#&gt; [1] &quot;Importing Neuronal enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb6-36" data-line-number="36"><span class="co">#&gt; [1] &quot;Importing Neuronal promoters ...&quot;</span></a>
<a class="sourceLine" id="cb6-37" data-line-number="37"><span class="co">#&gt; [1] &quot;Importing Oligo enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb6-38" data-line-number="38"><span class="co">#&gt; [1] &quot;Importing Oligo promoters ...&quot;</span></a>
<a class="sourceLine" id="cb6-39" data-line-number="39"><span class="co">#&gt; [1] &quot;Importing Microglia enhancers ...&quot;</span></a>
<a class="sourceLine" id="cb6-40" data-line-number="40"><span class="co">#&gt; [1] &quot;Importing Microglia promoters ...&quot;</span></a>
<a class="sourceLine" id="cb6-41" data-line-number="41"><span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span></a>
<a class="sourceLine" id="cb6-42" data-line-number="42"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb6-43" data-line-number="43"><span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span></a>
<a class="sourceLine" id="cb6-44" data-line-number="44"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb6-45" data-line-number="45"><span class="co">#&gt; [1] &quot;++ Nott sn-epigenomics:: Returning PLAC-seq track.&quot;</span></a>
<a class="sourceLine" id="cb6-46" data-line-number="46"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-47" data-line-number="47"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-48" data-line-number="48"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-49" data-line-number="49"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-50" data-line-number="50"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-51" data-line-number="51"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-52" data-line-number="52"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-53" data-line-number="53"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-54" data-line-number="54"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-55" data-line-number="55"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-56" data-line-number="56"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-57" data-line-number="57"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-58" data-line-number="58"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-59" data-line-number="59"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-60" data-line-number="60"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-61" data-line-number="61"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-62" data-line-number="62"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-63" data-line-number="63"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-64" data-line-number="64"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-65" data-line-number="65"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-66" data-line-number="66"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-67" data-line-number="67"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-68" data-line-number="68"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-69" data-line-number="69"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-70" data-line-number="70"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-71" data-line-number="71"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-72" data-line-number="72"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-73" data-line-number="73"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-74" data-line-number="74"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-75" data-line-number="75"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-76" data-line-number="76"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-77" data-line-number="77"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-78" data-line-number="78"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-79" data-line-number="79"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-80" data-line-number="80"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-81" data-line-number="81"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-82" data-line-number="82"><span class="co">#&gt; Found more than one class &quot;unit&quot; in cache; using the first, from namespace 'ggbio'</span></a>
<a class="sourceLine" id="cb6-83" data-line-number="83"><span class="co">#&gt; Also defined by 'hexbin'</span></a>
<a class="sourceLine" id="cb6-84" data-line-number="84"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb6-85" data-line-number="85"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb6-86" data-line-number="86"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb6-87" data-line-number="87"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb6-88" data-line-number="88"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb6-89" data-line-number="89"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb6-90" data-line-number="90"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb6-91" data-line-number="91"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb6-92" data-line-number="92"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb6-93" data-line-number="93"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb6-94" data-line-number="94"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb6-95" data-line-number="95"><span class="co">#&gt; replace the existing scale.</span></a>
<a class="sourceLine" id="cb6-96" data-line-number="96"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span></a>
<a class="sourceLine" id="cb6-97" data-line-number="97"><span class="co">#&gt; replace the existing scale.</span></a></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
