<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started • echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started">
<meta property="og:description" content="echolocatoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/echolocatoR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Started</h1>
                        <h4 data-toc-skip class="author"><h4>
Author: <i>Brian M. Schilder</i>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Updated: <i>Mar-04-2022</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/HEAD/vignettes/echolocatoR.Rmd" class="external-link"><code>vignettes/echolocatoR.Rmd</code></a></small>
      <div class="hidden name"><code>echolocatoR.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echolocatoR" class="external-link">echolocatoR</a></span><span class="op">)</span> 
<span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span>
<span class="co">#&gt;   method from   </span>
<span class="co">#&gt;   +.gg   ggplot2</span>

<span class="va">root.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="full-pipeline">Full pipeline<a class="anchor" aria-label="anchor" href="#full-pipeline"></a>
</h2>
<p>All examples below use data from the Parkinson’s disease GWAS by Nalls et al. (2019).</p>
<div class="section level3">
<h3 id="prepare-top_snps-data-frame">Prepare <code>top_SNPs</code> data.frame<a class="anchor" aria-label="anchor" href="#prepare-top_snps-data-frame"></a>
</h3>
<ul>
<li>To enable rapid fine-mapping of many loci, you can create a <code>top_SNPs</code> data.frame<br>
which contains the position of the lead/index SNP within each locus.</li>
<li>
<code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> (see next step) will then use this info to extract subsets of the<br>
full GWAS/QTL summary statistics using windows centered on each lead/index SNP.</li>
<li>The <code>topSS</code> argument can either be a data.frame, or a path to a topSS file saved somehwere. Most common tabular data formats (e.g. .tsv, .csv, .xlsx) are accepted.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Nalls_top_SNPs"</span><span class="op">)</span>;
<span class="va">top_SNPs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_topSNPs.html">import_topSNPs</a></span><span class="op">(</span>
  <span class="co"># topSS = "~/Desktop/Fine_Mapping/Data/GWAS/Nalls23andMe_2019/Nalls2019_TableS2.xlsx",</span>
  topSS <span class="op">=</span> <span class="va">Nalls_top_SNPs</span>,
  munge <span class="op">=</span> <span class="cn">TRUE</span>,
  ref_genome <span class="op">=</span> <span class="st">"GRCH37"</span>, 
  pval_col<span class="op">=</span><span class="st">"P, all studies"</span>, 
  effect_col<span class="op">=</span><span class="st">"Beta, all studies"</span>, 
  gene_col<span class="op">=</span><span class="st">"Nearest Gene"</span>, 
  locus_col <span class="op">=</span> <span class="st">"Nearest Gene"</span>,
  grouping_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Locus"</span><span class="op">)</span>,
  remove_variants <span class="op">=</span> <span class="st">"rs34637584"</span><span class="op">)</span> 
<span class="co">#&gt; [1] "+ Assigning gene_col and locus_col independently"</span>
<span class="co">#&gt; [1] "+ Munging top_SNPs"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">top_SNPs</span><span class="op">)</span>
<span class="co">#&gt;      Locus    Gene CHR       POS        SNP        P  Effect</span>
<span class="co">#&gt; 1:   ASXL3   ASXL3  18  31304318  rs1941685 1.69e-08  0.0531</span>
<span class="co">#&gt; 2:    BAG3    BAG3  10 121415685 rs72840788 1.57e-11  0.0763</span>
<span class="co">#&gt; 3:    BIN3    BIN3   8  22525980  rs2280104 1.16e-08  0.0556</span>
<span class="co">#&gt; 4:   BRIP1   BRIP1  17  59917366 rs61169879 9.28e-10  0.0820</span>
<span class="co">#&gt; 5:    BST1    BST1   4  15737348  rs4698412 2.06e-28  0.1035</span>
<span class="co">#&gt; 6: C5orf24 C5orf24   5 134199105 rs11950533 7.16e-09 -0.0916</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="path-to-full-summary-stats-file">Path to full summary stats file<a class="anchor" aria-label="anchor" href="#path-to-full-summary-stats-file"></a>
</h3>
<ul>
<li><p>Since a full GWAS summary stats file would be too large to include within <em>echolocatoR</em>,<br>
we instead provide an example subset of the full summary stats.</p></li>
<li><p>To simulate how you’d actually use your own full summary stats file, we will save our example dataset to your computer (you can change the path to wherever you like).</p></li>
<li><p>We highly recommend munging your full summary stats using the Bioconductor package <a href="https://github.com/neurogenomics/MungeSumstats" class="external-link"><code>MungeSumstats</code></a> first. It’s easy to use and very robust. It also means you don’t have to provide most column mapping arguments in <code>finemap_loci</code> when <code>munged=TRUE</code>.</p></li>
</ul>
<p>Here’s an example of how to munge your full summary stats file:</p>
<pre><code><span class="va">fullSS_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/example_fullSS.html">example_fullSS</a></span><span class="op">(</span>munged <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">fullSS_path</span> <span class="op">&lt;-</span> <span class="fu">MungeSumstats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MungeSumstats/man/format_sumstats.html" class="external-link">format_sumstats</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">fullSS_path</span>, ref_genome <span class="op">=</span> <span class="st">"GRCH37"</span><span class="op">)</span></code></pre>
<p>We have already munged the following example summary stats for you.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fullSS_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/example_fullSS.html">example_fullSS</a></span><span class="op">(</span>munged <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; [1] "Writing file to ==&gt; /tmp/RtmpdfLtJX/Nalls23andMe_2019.fullSS_subset.tsv"</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-fine-mapping-pipeline">Run fine-mapping pipeline<a class="anchor" aria-label="anchor" href="#run-fine-mapping-pipeline"></a>
</h3>
<p>For a full description of all arguments, see <code><a href="../reference/finemap_loci.html">?finemap_loci</a></code>.</p>
<p>Here are some key arguments:</p>
<ul>
<li>
<em>results_dir</em>: Where you want to store all of your results.<br>
</li>
<li>
<em>finemap_methods</em>: Which fine-mapping methods you want to run (currently includes ABF, FINEMAP, SUSIE, POLYFUN_SUSIE, and COJO).<br>
</li>
<li>
<em>bp_distance</em>: Controls window size. Specifically, <code>bp_distance</code> is the number of basepairs upstream/downstream you want to extract for each locus. For example, if you want a 2Mb window (+/- 1Mb from the lead/index SNP in <code>top_SNPs</code>), set <code>bp_distance=1e+06</code>.<br>
</li>
<li>
<em>plot.zoom</em>: Zoom in/out from the center of each locus when producing the multiview plot.<br>
You can adjust this separately from <code>bp_distance</code> so that you don’t have rerun the whole pipeline each time (locus subsets, LD matrices, and fine-mapping results are all automatically saved in locus-specific folders).</li>
</ul>
<p><strong>NOTE</strong>: This example assumes you have already installed tabix (via htslib) and bcftoools.</p>
<p><strong>WARNING</strong>: Please use the full absolute paths (instead of relative paths) wherever possible (e.g. <code>results_dir</code>). This is especially important for the tool <em>FINEMAP</em>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">Nalls23andMe_2019.results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/finemap_loci.html">finemap_loci</a></span><span class="op">(</span><span class="co"># GENERAL ARGUMENTS </span>
                                          top_SNPs <span class="op">=</span> <span class="va">top_SNPs</span>,  
                                          <span class="co">#  It's best to give absolute paths</span>
                                          results_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">root.dir</span>,<span class="st">"results"</span><span class="op">)</span>,
                                          loci <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BST1"</span>,<span class="st">"MEX3C"</span><span class="op">)</span>,<span class="co"># top_SNPs$Locus,</span>
                                          dataset_name <span class="op">=</span> <span class="st">"Nalls23andMe_2019"</span>,
                                          dataset_type <span class="op">=</span> <span class="st">"GWAS"</span>,  
                                          force_new_subset <span class="op">=</span> <span class="cn">TRUE</span>,
                                          force_new_LD <span class="op">=</span> <span class="cn">FALSE</span>,
                                          force_new_finemap <span class="op">=</span> <span class="cn">TRUE</span>,
                                          remove_tmps <span class="op">=</span> <span class="cn">FALSE</span>,
                                          
                                          <span class="co"># Munge full sumstats first</span>
                                          munged <span class="op">=</span> <span class="cn">TRUE</span>,
                                          
                                          <span class="co"># SUMMARY STATS ARGUMENTS</span>
                                          fullSS_path <span class="op">=</span> <span class="va">fullSS_path</span>,
                                          fullSS_genome_build <span class="op">=</span> <span class="st">"hg19"</span>,
                                          query_by <span class="op">=</span><span class="st">"tabix"</span>,  
                                          MAF_col <span class="op">=</span> <span class="st">"calculate"</span>,   
                                         
                                          <span class="co"># FILTERING ARGUMENTS</span>
                                          <span class="co">## It's often desirable to use a larger window size </span>
                                          <span class="co">## (e.g. 2Mb which is bp_distance=500000*2), </span>
                                          <span class="co">## but we use a small window here to speed up the process. </span>
                                          bp_distance <span class="op">=</span> <span class="fl">10000</span>,<span class="co">#500000*2,</span>
                                          min_MAF <span class="op">=</span> <span class="fl">0.001</span>,  
                                          trim_gene_limits <span class="op">=</span> <span class="cn">FALSE</span>,
                                         
                                          <span class="co"># FINE-MAPPING ARGUMENTS</span>
                                          <span class="co">## General</span>
                                          finemap_methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ABF"</span>,<span class="st">"FINEMAP"</span>,<span class="st">"SUSIE"</span>,<span class="st">"POLYFUN_SUSIE"</span><span class="op">)</span>, 
                                          n_causal <span class="op">=</span> <span class="fl">5</span>,
                                          PP_threshold <span class="op">=</span> <span class="fl">.95</span>, 
                                         
                                          <span class="co"># LD ARGUMENTS </span>
                                          LD_reference <span class="op">=</span> <span class="st">"1KGphase1"</span>,<span class="co">#"UKB",</span>
                                          superpopulation <span class="op">=</span> <span class="st">"EUR"</span>,
                                          download_method <span class="op">=</span> <span class="st">"axel"</span>,
                                         
                                          <span class="co"># PLOT ARGUMENTS </span>
                                          <span class="co">## general   </span>
                                          plot.types <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fancy"</span><span class="op">)</span>,
                                          <span class="co">## Generate multiple plots of different window sizes; </span>
                                          <span class="co">### all SNPs, 4x zoomed-in, and a 50000bp window</span>
                                          plot.zoom <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all"</span>,<span class="st">"4x"</span>,<span class="st">"10x"</span><span class="op">)</span>,
                                          <span class="co">## XGR</span>
                                          <span class="co"># plot.XGR_libnames=c("ENCODE_TFBS_ClusteredV3_CellTypes"), </span>
                                          <span class="co">## Roadmap</span>
                                          plot.Roadmap <span class="op">=</span> <span class="cn">FALSE</span>,
                                          plot.Roadmap_query <span class="op">=</span> <span class="cn">NULL</span>,
                                          <span class="co"># Nott et al. (2019)</span>
                                          plot.Nott_epigenome <span class="op">=</span> <span class="cn">TRUE</span>, 
                                          plot.Nott_show_placseq <span class="op">=</span> <span class="cn">TRUE</span>, 
                                         
                                          verbose <span class="op">=</span> <span class="cn">TRUE</span>
                                          <span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dataset</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">root.dir</span>,<span class="st">"results/GWAS/Nalls23andMe_2019/"</span><span class="op">)</span>
<span class="va">merged_dat</span> <span class="op">&lt;-</span> <span class="fu">echolocatoR</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_finemapping_results.html">merge_finemapping_results</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="va">dataset</span>,
                                                     minimum_support <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] echolocatoR_0.2.2 BiocStyle_2.22.0 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] backports_1.4.1             Hmisc_4.6-0                </span>
<span class="co">#&gt;   [3] BiocFileCache_2.2.1         systemfonts_1.0.4          </span>
<span class="co">#&gt;   [5] googleAuthR_2.0.0           plyr_1.8.6                 </span>
<span class="co">#&gt;   [7] lazyeval_0.2.2              splines_4.1.2              </span>
<span class="co">#&gt;   [9] crosstalk_1.2.0             BiocParallel_1.28.3        </span>
<span class="co">#&gt;  [11] GenomeInfoDb_1.30.1         ggplot2_3.3.5              </span>
<span class="co">#&gt;  [13] digest_0.6.29               ensembldb_2.18.3           </span>
<span class="co">#&gt;  [15] htmltools_0.5.2             fansi_1.0.2                </span>
<span class="co">#&gt;  [17] magrittr_2.0.2              checkmate_2.0.0            </span>
<span class="co">#&gt;  [19] memoise_2.0.1               BSgenome_1.62.0            </span>
<span class="co">#&gt;  [21] cluster_2.1.2               Biostrings_2.62.0          </span>
<span class="co">#&gt;  [23] matrixStats_0.61.0          R.utils_2.11.0             </span>
<span class="co">#&gt;  [25] ggbio_1.42.0                pkgdown_2.0.2              </span>
<span class="co">#&gt;  [27] prettyunits_1.1.1           jpeg_0.1-9                 </span>
<span class="co">#&gt;  [29] colorspace_2.0-3            rappdirs_0.3.3             </span>
<span class="co">#&gt;  [31] blob_1.2.2                  textshaping_0.3.6          </span>
<span class="co">#&gt;  [33] xfun_0.30                   dplyr_1.0.8                </span>
<span class="co">#&gt;  [35] crayon_1.5.0                RCurl_1.98-1.6             </span>
<span class="co">#&gt;  [37] jsonlite_1.8.0              graph_1.72.0               </span>
<span class="co">#&gt;  [39] Exact_3.1                   VariantAnnotation_1.40.0   </span>
<span class="co">#&gt;  [41] survival_3.2-13             glue_1.6.2                 </span>
<span class="co">#&gt;  [43] gargle_1.2.0                gtable_0.3.0               </span>
<span class="co">#&gt;  [45] zlibbioc_1.40.0             XVector_0.34.0             </span>
<span class="co">#&gt;  [47] DelayedArray_0.20.0         BiocGenerics_0.40.0        </span>
<span class="co">#&gt;  [49] scales_1.1.1                mvtnorm_1.1-3              </span>
<span class="co">#&gt;  [51] DBI_1.1.2                   GGally_2.1.2               </span>
<span class="co">#&gt;  [53] Rcpp_1.0.8                  progress_1.2.2             </span>
<span class="co">#&gt;  [55] htmlTable_2.4.0             foreign_0.8-81             </span>
<span class="co">#&gt;  [57] bit_4.0.4                   proxy_0.4-26               </span>
<span class="co">#&gt;  [59] OrganismDbi_1.36.0          Formula_1.2-4              </span>
<span class="co">#&gt;  [61] DT_0.21                     stats4_4.1.2               </span>
<span class="co">#&gt;  [63] htmlwidgets_1.5.4           httr_1.4.2                 </span>
<span class="co">#&gt;  [65] RColorBrewer_1.1-2          ellipsis_0.3.2             </span>
<span class="co">#&gt;  [67] R.methodsS3_1.8.1           pkgconfig_2.0.3            </span>
<span class="co">#&gt;  [69] reshape_0.8.8               XML_3.99-0.9               </span>
<span class="co">#&gt;  [71] dbplyr_2.1.1                nnet_7.3-16                </span>
<span class="co">#&gt;  [73] sass_0.4.0                  utf8_1.2.2                 </span>
<span class="co">#&gt;  [75] tidyselect_1.1.2            rlang_1.0.1                </span>
<span class="co">#&gt;  [77] reshape2_1.4.4              AnnotationDbi_1.56.2       </span>
<span class="co">#&gt;  [79] munsell_0.5.0               tools_4.1.2                </span>
<span class="co">#&gt;  [81] cachem_1.0.6                cli_3.2.0                  </span>
<span class="co">#&gt;  [83] generics_0.1.2              RSQLite_2.2.10             </span>
<span class="co">#&gt;  [85] MungeSumstats_1.2.2         evaluate_0.15              </span>
<span class="co">#&gt;  [87] stringr_1.4.0               fastmap_1.1.0              </span>
<span class="co">#&gt;  [89] yaml_2.3.5                  ragg_1.2.2                 </span>
<span class="co">#&gt;  [91] knitr_1.37                  bit64_4.0.5                </span>
<span class="co">#&gt;  [93] fs_1.5.2                    purrr_0.3.4                </span>
<span class="co">#&gt;  [95] KEGGREST_1.34.0             AnnotationFilter_1.18.0    </span>
<span class="co">#&gt;  [97] rootSolve_1.8.2.3           RBGL_1.70.0                </span>
<span class="co">#&gt;  [99] R.oo_1.24.0                 xml2_1.3.3                 </span>
<span class="co">#&gt; [101] biomaRt_2.50.3              compiler_4.1.2             </span>
<span class="co">#&gt; [103] rstudioapi_0.13             filelock_1.0.2             </span>
<span class="co">#&gt; [105] curl_4.3.2                  png_0.1-7                  </span>
<span class="co">#&gt; [107] e1071_1.7-9                 tibble_3.1.6               </span>
<span class="co">#&gt; [109] bslib_0.3.1                 DescTools_0.99.44          </span>
<span class="co">#&gt; [111] stringi_1.7.6               GenomicFeatures_1.46.5     </span>
<span class="co">#&gt; [113] desc_1.4.0                  lattice_0.20-45            </span>
<span class="co">#&gt; [115] ProtGenerics_1.26.0         Matrix_1.3-4               </span>
<span class="co">#&gt; [117] vctrs_0.3.8                 pillar_1.7.0               </span>
<span class="co">#&gt; [119] lifecycle_1.0.1             BiocManager_1.30.16        </span>
<span class="co">#&gt; [121] jquerylib_0.1.4             data.table_1.14.2          </span>
<span class="co">#&gt; [123] bitops_1.0-7                lmom_2.8                   </span>
<span class="co">#&gt; [125] rtracklayer_1.54.0          GenomicRanges_1.46.1       </span>
<span class="co">#&gt; [127] R6_2.5.1                    BiocIO_1.4.0               </span>
<span class="co">#&gt; [129] latticeExtra_0.6-29         bookdown_0.24              </span>
<span class="co">#&gt; [131] gridExtra_2.3               IRanges_2.28.0             </span>
<span class="co">#&gt; [133] gld_2.6.4                   dichromat_2.0-0            </span>
<span class="co">#&gt; [135] boot_1.3-28                 MASS_7.3-54                </span>
<span class="co">#&gt; [137] assertthat_0.2.1            SummarizedExperiment_1.24.0</span>
<span class="co">#&gt; [139] rprojroot_2.0.2             rjson_0.2.21               </span>
<span class="co">#&gt; [141] GenomicAlignments_1.30.0    Rsamtools_2.10.0           </span>
<span class="co">#&gt; [143] S4Vectors_0.32.3            GenomeInfoDbData_1.2.7     </span>
<span class="co">#&gt; [145] expm_0.999-6                parallel_4.1.2             </span>
<span class="co">#&gt; [147] hms_1.1.1                   grid_4.1.2                 </span>
<span class="co">#&gt; [149] rpart_4.1-15                class_7.3-19               </span>
<span class="co">#&gt; [151] rmarkdown_2.12              MatrixGenerics_1.6.0       </span>
<span class="co">#&gt; [153] biovizBase_1.42.0           Biobase_2.54.0             </span>
<span class="co">#&gt; [155] base64enc_0.1-3             restfulr_0.0.13</span></code></pre></div>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
