<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>echolocatoR: full pipeline vignette • echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="echolocatoR: full pipeline vignette">
<meta property="og:description" content="echolocatoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/full_pipeline_vignette.html">echolocatoR: full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">echolocatoR: plotting vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>echolocatoR: full pipeline vignette</h1>
                        <h4 class="author">Brian M. Schilder</h4>
            
            <h4 class="date">Most recent update:<br> 2020-06-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/vignettes/full_pipeline_vignette.Rmd"><code>vignettes/full_pipeline_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>full_pipeline_vignette.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">echolocatoR</span>)
<span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span>
<span class="co">#&gt;   method from   </span>
<span class="co">#&gt;   +.gg   ggplot2</span></pre></body></html></div>
<div id="full-pipeline" class="section level1">
<h1 class="hasAnchor">
<a href="#full-pipeline" class="anchor"></a>Full pipeline</h1>
<p>All examples below use data from the Parkinson’s disease GWAS by Nalls et al. (2019).</p>
<div id="prepare-top_snps-data-frame" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-top_snps-data-frame" class="anchor"></a>Prepare <code>top_SNPs</code> data.frame</h2>
<ul>
<li>To enable rapid fine-mapping of many loci, you can create a <code>top_SNPs</code> data.frame<br>
which contains the position of the lead/index SNP within each locus.</li>
<li>
<code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> (see next step) will then use this info to extract subsets of the<br>
full GWAS/QTL summary statistics using windows centered on each lead/index SNP.</li>
<li>The <code>topSS</code> argument can either be a data.frame, or a path to a topSS file saved somehwere. Most common tabular data formats (e.g. .tsv, .csv, .xlsx) are accepted.</li>
</ul>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"Nalls_top_SNPs"</span>);
<span class="no">top_SNPs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_topSNPs.html">import_topSNPs</a></span>(
  <span class="co"># topSS = "~/Desktop/Fine_Mapping/Data/GWAS/Nalls23andMe_2019/Nalls2019_TableS2.xlsx",</span>
  <span class="kw">topSS</span> <span class="kw">=</span> <span class="no">Nalls_top_SNPs</span>,
  <span class="kw">chrom_col</span> <span class="kw">=</span> <span class="st">"CHR"</span>, <span class="kw">position_col</span> <span class="kw">=</span> <span class="st">"BP"</span>, <span class="kw">snp_col</span><span class="kw">=</span><span class="st">"SNP"</span>,
  <span class="kw">pval_col</span><span class="kw">=</span><span class="st">"P, all studies"</span>, <span class="kw">effect_col</span><span class="kw">=</span><span class="st">"Beta, all studies"</span>, <span class="kw">gene_col</span><span class="kw">=</span><span class="st">"Nearest Gene"</span>,
  <span class="kw">group_by_locus</span> <span class="kw">=</span> <span class="no">T</span>,
  <span class="kw">locus_col</span> <span class="kw">=</span> <span class="st">"Nearest Gene"</span>,
  <span class="kw">remove_variants</span> <span class="kw">=</span> <span class="st">"rs34637584"</span>)</pre></body></html></div>
</div>
<div id="find-path-to-full-summary-stats-file" class="section level2">
<h2 class="hasAnchor">
<a href="#find-path-to-full-summary-stats-file" class="anchor"></a>Find path to full summary stats file</h2>
<ul>
<li>Since a full GWAS summary stats file would be too large to include within <em>echolocatoR</em>,<br>
we instead provide a example subset of the full summary stats.<br>
</li>
<li>To simulate how you’d actually use your own full summary stats file, we will save our example dataset to your computer (you can change the path to wherever you like).</li>
</ul>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="co"># fullSS_path &lt;- "./nallsEtAl2019_allSamples_allVariants.mod.txt.gz"</span>
<span class="no">fullSS_path</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/example_fullSS.html">example_fullSS</a></span>(<span class="kw">fullSS_path</span><span class="kw">=</span><span class="st">"~/Desktop/Nalls23andMe_2019.fullSS_subset.tsv"</span>)</pre></body></html></div>
</div>
<div id="run-fine-mapping-pipeline" class="section level2">
<h2 class="hasAnchor">
<a href="#run-fine-mapping-pipeline" class="anchor"></a>Run fine-mapping pipeline</h2>
<p>For a full description of all arguments, see <code><a href="../reference/finemap_loci.html">?finemap_loci</a></code>.</p>
<p>Here are some key arguments:</p>
<ul>
<li>
<em>results_dir</em>: Where you want to store all of your results.<br>
</li>
<li>
<em>finemap_methods</em>: Which fine-mapping methods you want to run (currently includes ABF, FINEMAP, SUSIE, POLYFUN_SUSIE, and COJO).<br>
</li>
<li>
<em>bp_distance</em>: Controls window size. Specifically, <code>bp_distance</code> is the number of basepairs upstream/downstream you want to extract for each locus. For example, if you want a 2Mb window (+/- 1Mb from the lead/index SNP in <code>top_SNPs</code>), set <code>bp_distance=1e+06</code>.<br>
</li>
<li>
<em>plot.window</em>: Zoom in/out from the center of each locus when producing the multiview plot.<br>
You can adjust this separately from <code>bp_distance</code> so that you don’t have rerun the whole pipeline each time (locus subsets, LD matrices, and fine-mapping results are all automatically saved in locus-specific folders).</li>
</ul>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">Nalls23andMe_2019.results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/finemap_loci.html">finemap_loci</a></span>(<span class="co"># GENERAL ARGUMENTS </span>
                                          <span class="kw">top_SNPs</span> <span class="kw">=</span> <span class="no">top_SNPs</span>,
                                          <span class="kw">results_dir</span> <span class="kw">=</span> <span class="st">"./echolocatoR_results"</span>,
                                          <span class="kw">loci</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"BST1"</span>,<span class="st">"MEX3C"</span>),<span class="co"># top_SNPs$Locus,</span>
                                          <span class="kw">dataset_name</span> <span class="kw">=</span> <span class="st">"Nalls23andMe_2019"</span>,
                                          <span class="kw">dataset_type</span> <span class="kw">=</span> <span class="st">"GWAS"</span>,
                                          <span class="kw">force_new_subset</span> <span class="kw">=</span> <span class="no">F</span>,
                                          <span class="kw">force_new_LD</span> <span class="kw">=</span> <span class="no">F</span>,
                                          <span class="kw">force_new_finemap</span> <span class="kw">=</span> <span class="no">T</span>,
                                          <span class="kw">remove_tmps</span> <span class="kw">=</span> <span class="no">T</span>,

                 <span class="co"># SUMMARY STATS ARGUMENTS</span>
                 <span class="kw">fullSS_path</span> <span class="kw">=</span> <span class="no">fullSS_path</span>,
                 <span class="kw">query_by</span> <span class="kw">=</span><span class="st">"tabix"</span>,
                 <span class="kw">chrom_col</span> <span class="kw">=</span> <span class="st">"CHR"</span>, <span class="kw">position_col</span> <span class="kw">=</span> <span class="st">"POS"</span>, <span class="kw">snp_col</span> <span class="kw">=</span> <span class="st">"RSID"</span>,
                 <span class="kw">pval_col</span> <span class="kw">=</span> <span class="st">"p"</span>, <span class="kw">effect_col</span> <span class="kw">=</span> <span class="st">"beta"</span>, <span class="kw">stderr_col</span> <span class="kw">=</span> <span class="st">"se"</span>,
                 <span class="kw">freq_col</span> <span class="kw">=</span> <span class="st">"freq"</span>, <span class="kw">MAF_col</span> <span class="kw">=</span> <span class="st">"calculate"</span>,
                 <span class="kw">A1_col</span> <span class="kw">=</span> <span class="st">"A1"</span>,
                 <span class="kw">A2_col</span> <span class="kw">=</span> <span class="st">"A2"</span>,

                 <span class="co"># FILTERING ARGUMENTS</span>
                 <span class="kw">bp_distance</span> <span class="kw">=</span> <span class="fl">500000</span>*<span class="fl">2</span>,
                 <span class="kw">min_MAF</span> <span class="kw">=</span> <span class="fl">0.001</span>,
                 <span class="kw">trim_gene_limits</span> <span class="kw">=</span> <span class="no">F</span>,

                 <span class="co"># FINE-MAPPING ARGUMENTS</span>
                 <span class="kw">finemap_methods</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ABF"</span>,<span class="st">"FINEMAP"</span>,<span class="st">"SUSIE"</span>,<span class="st">"POLYFUN_SUSIE"</span>),
                 <span class="kw">n_causal</span> <span class="kw">=</span> <span class="fl">5</span>,
                 <span class="kw">PP_threshold</span> <span class="kw">=</span> <span class="fl">.95</span>,

                 <span class="co"># LD ARGUMENTS </span>
                 <span class="kw">LD_reference</span> <span class="kw">=</span> <span class="st">"UKB"</span>,<span class="co">#"1KGphase1",</span>
                 <span class="kw">superpopulation</span> <span class="kw">=</span> <span class="st">"EUR"</span>,
                 <span class="kw">LD_download_method</span> <span class="kw">=</span> <span class="st">"axel"</span>,

                 <span class="co"># PLOT ARGUMENTS </span>
                 <span class="co">## general   </span>
                 <span class="kw">plot.types</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"fancy"</span>),
                 <span class="kw">plot.window</span> <span class="kw">=</span> <span class="fl">100000</span>,
                 <span class="co">## XGR</span>
                 <span class="co"># plot.XGR_libnames=c("ENCODE_TFBS_ClusteredV3_CellTypes"), </span>
                 <span class="co">## Roadmap</span>
                 <span class="kw">plot.Roadmap</span> <span class="kw">=</span> <span class="no">F</span>,
                 <span class="kw">plot.Roadmap_query</span> <span class="kw">=</span> <span class="kw">NULL</span>,
                 <span class="co"># Nott et al. (2019)</span>
                 <span class="kw">plot.Nott_epigenome</span> <span class="kw">=</span> <span class="no">T</span>,
                 <span class="kw">plot.Nott_binwidth</span> <span class="kw">=</span> <span class="fl">100</span>
                 )
<span class="co">#&gt; 🦇 🦇 🦇 BST1 🦇 🦇 🦇</span>
<span class="co">#&gt; ------------------ Step 1: Query ---------------</span>
<span class="co">#&gt; [1] "+ Extracting relevant variants from fullSS..."</span>
<span class="co">#&gt; [1] "++ Query Method: 'tabix"</span>
<span class="co">#&gt; [1] "---Min snp position: 14737348 ---"</span>
<span class="co">#&gt; [1] "---Max snp position: 16737348 ---"</span>
<span class="co">#&gt; [1] "TABIX:: Existing indexed tabix file detected"</span>
<span class="co">#&gt; [1] "TABIX:: Extracting subset of sum stats"</span>
<span class="co">#&gt; [1] "++ Returning 6282 x 11 data.table"</span>
<span class="co">#&gt; [1] "++ Saving query ==&gt; ./echolocatoR_results/GWAS/Nalls23andMe_2019/BST1/BST1_Nalls23andMe_2019_subset.tsv.gz"</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; ---------------- Step 1.5: Standardize ----------</span>
<span class="co">#&gt; [1] "+Inferring MAF from frequency column..."</span>
<span class="co">#&gt; [1] "+ Standardize:: Calculating proportion of cases."</span>
<span class="co">#&gt; [1] "+ Calculating t-statistic from Effect and StdErr..."</span>
<span class="co">#&gt; [1] "+ Extraction completed in 2.45 seconds"</span>
<span class="co">#&gt; [1] "+ 6273 SNPs x  15 columns"</span>
<span class="co">#&gt; [1] "+ Removing subset tmp..."</span>
<span class="co">#&gt; --- Step 2: Calculate Linkage Disequilibrium ---</span>
<span class="co">#&gt; [1] "LD:: Using UK Biobank LD reference panel..."</span>
<span class="co">#&gt; [1] "+ UKB LD file name: chr4_14000001_17000001"</span>
<span class="co">#&gt; [1] "+ LD:: Downloading full .gz/.npz UKB files and saving to disk."</span>
<span class="co">#&gt; [1] "+ LD:: ...this could take some time..."</span>
<span class="co">#&gt; Processed URL: ./echolocatoR_results/GWAS/Nalls23andMe_2019/BST1/LD/chr4_14000001_17000001</span>
<span class="co">#&gt; Some other message at the end</span>
<span class="co">#&gt; [1] "LD matrix dimensions 6254 x 6254"</span>
<span class="co">#&gt; [1] "+ LD:: Saving LD =&gt; ./echolocatoR_results/GWAS/Nalls23andMe_2019/BST1/LD/BST1.UKB_LD.RDS"</span>
<span class="co">#&gt; [1] "+ Removing .gz/.npz files."</span>
<span class="co">#&gt; -------------- Step 3: Filter SNPs -------------</span>
<span class="co">#&gt; [1] "++ Post-filtered data: 6273 x 15"</span>
<span class="co">#&gt; -------- Step 4: Statistically Fine-map --------</span>
<span class="co">#&gt; [1] "++ Fine-mapping using multiple tools: ABF, FINEMAP, SUSIE, POLYFUN_SUSIE"</span>
<span class="co">#&gt; Multi-finemap:: ABF</span>
<span class="co">#&gt; [1] "++ Credible Set SNPS identified = 0"</span>
<span class="co">#&gt; [1] "++ Merging ABF results with multi-finemap data."</span>
<span class="co">#&gt; Multi-finemap:: FINEMAP</span>
<span class="co">#&gt; [1] "++ Constructing FINEMAP master file."</span>
<span class="co">#&gt; [1] "++ Formatting data.z file for FINEMAP"</span>
<span class="co">#&gt; [1] "++ Formatting LD Matrix for FINEMAP"</span>
<span class="co">#&gt; [1] "+++ Writing FINEMAP z and ld files..."</span>
<span class="co">#&gt; [1] "cd ./echolocatoR_results/GWAS/Nalls23andMe_2019/BST1 &amp;&amp; /private/var/folders/sw/km9819713937hp640lj59ygw0000gn/T/RtmpnuHoUP/temp_libpathdf6530fe2c52/echolocatoR/tools/FINEMAP/finemap_v1.3_MacOSX --cond --in-files ./echolocatoR_results/GWAS/Nalls23andMe_2019/BST1/FINEMAP/master --log --n-causal-snps 5"</span>
<span class="co">#&gt; Error in data.table::fread(file.path(locus_dir, "FINEMAP/data.config")) : </span>
<span class="co">#&gt;   File './echolocatoR_results/GWAS/Nalls23andMe_2019/BST1/FINEMAP/data.config' does not exist or is non-readable. getwd()=='/Users/schilder/Desktop'</span>
<span class="co">#&gt; [1] "++ Credible Set SNPS identified = 0"</span>
<span class="co">#&gt; [1] "++ Merging FINEMAP results with multi-finemap data."</span>
<span class="co">#&gt; Multi-finemap:: SUSIE</span>
<span class="co">#&gt; [1] "++ Computing phenotype variance..."</span>
<span class="co">#&gt; [1] "+ SUSIE:: n_causal = 5"</span>
<span class="co">#&gt; Warning in susie_ss(XtX = XtX, Xty = Xty, yty = var_y * (n - 1), n = n, : IBSS</span>
<span class="co">#&gt; algorithm did not converge in 100 iterations!</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; [1] "++ Extracting Credible Sets..."</span>
<span class="co">#&gt; [1] "++ Credible Set SNPS identified = 3"</span>
<span class="co">#&gt; [1] "++ Merging SUSIE results with multi-finemap data."</span>
<span class="co">#&gt; Multi-finemap:: POLYFUN_SUSIE</span>
<span class="co">#&gt; [1] "PolyFun:: Preparing SNP input file..."</span>
<span class="co">#&gt; [1] "+ PolyFun:: 6216 SNPs identified."</span>
<span class="co">#&gt; [1] "+ PolyFun:: Writing SNP file ==&gt; ./echolocatoR_results/GWAS/Nalls23andMe_2019/BST1/PolyFun/snps_to_finemap.txt.gz"</span>
<span class="co">#&gt; [1] "/opt/anaconda3/envs/echoR/bin/python /private/var/folders/sw/km9819713937hp640lj59ygw0000gn/T/RtmpnuHoUP/temp_libpathdf6530fe2c52/echolocatoR/tools/polyfun//extract_snpvar.py --snps ./echolocatoR_results/GWAS/Nalls23andMe_2019/BST1/PolyFun/snps_to_finemap.txt.gz --out ./echolocatoR_results/GWAS/Nalls23andMe_2019/BST1/PolyFun/snps_with_priors.snpvar.tsv.gz"</span>
<span class="co">#&gt; [1] "++ Remove tmp file."</span>
<span class="co">#&gt; [1] "++ Computing phenotype variance..."</span>
<span class="co">#&gt; [1] "+ SUSIE:: n_causal = 5"</span>
<span class="co">#&gt; [1] "Utilizing prior_weights for 6216 SNPs."</span>
<span class="co">#&gt; Warning in susie_ss(XtX = XtX, Xty = Xty, yty = var_y * (n - 1), n = n, : IBSS</span>
<span class="co">#&gt; algorithm did not converge in 100 iterations!</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; [1] "++ Extracting Credible Sets..."</span>
<span class="co">#&gt; [1] "++ Credible Set SNPS identified = 3"</span>
<span class="co">#&gt; [1] "++ Merging POLYFUN_SUSIE results with multi-finemap data."</span>
<span class="co">#&gt; [1] "++ Fine-mapping with ' ABF, FINEMAP, SUSIE, POLYFUN_SUSIE ' completed in  5.91"</span>
<span class="co">#&gt; [1] "+ Identifying Consensus SNPs..."</span>
<span class="co">#&gt; [1] "++ support_thresh = 2"</span>
<span class="co">#&gt; [1] "+ Calculating mean Posterior Probability (mean.PP)..."</span>
<span class="co">#&gt; [1] "++ 4 fine-mapping methods used."</span>
<span class="co">#&gt; [1] "++ 3 Credible Set SNPs identified."</span>
<span class="co">#&gt; [1] "++ 3 Consensus SNPs identified."</span>
<span class="co">#&gt; --------------- Step 7: Visualize --------------</span>
<span class="co">#&gt; [1] "+ Identifying Consensus SNPs..."</span>
<span class="co">#&gt; [1] "++ support_thresh = 2"</span>
<span class="co">#&gt; [1] "+ Calculating mean Posterior Probability (mean.PP)..."</span>
<span class="co">#&gt; [1] "++ 4 fine-mapping methods used."</span>
<span class="co">#&gt; [1] "++ 3 Credible Set SNPs identified."</span>
<span class="co">#&gt; [1] "++ 3 Consensus SNPs identified."</span>
<span class="co">#&gt; [1] "GGBIO:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "LD Matrix dimensions: 6216 x 6216"</span>
<span class="co">#&gt; [1] "Extracting LD subset for lead SNP: rs4698412"</span>
<span class="co">#&gt; [1] "++ GGBIO:: GWAS track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: ABF track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: FINEMAP track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: POLYFUN_SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: mean track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: Gene Model Track"</span>
<span class="co">#&gt; [1] "+ Annotating at transcript-level."</span>
<span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span>
<span class="co">#&gt; [1] "Importing Astrocyte enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Astrocyte promoters ..."</span>
<span class="co">#&gt; [1] "Importing Neuronal enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Neuronal promoters ..."</span>
<span class="co">#&gt; [1] "Importing Oligo enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Oligo promoters ..."</span>
<span class="co">#&gt; [1] "Importing Microglia enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Microglia promoters ..."</span>
<span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span>
<span class="co">#&gt; replace the existing scale.</span>
<span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span>
<span class="co">#&gt; replace the existing scale.</span>
<span class="co">#&gt; [1] "++ Nott sn-epigenomics:: Returning PLAC-seq track."</span>
<span class="co">#&gt; [1] "+ GGBIO:: Saving plot ==&gt; ./echolocatoR_results/GWAS/Nalls23andMe_2019/BST1/multiview_BST1_100kb.png"</span>
<span class="co">#&gt; Warning: Removed 100 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 6212 rows containing missing values (geom_point).</span>
<span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_point).</span>
<span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_label_repel).</span>

<span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_label_repel).</span>
<span class="co">#&gt; Error : Discrete value supplied to continuous scale</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Fine-mappping complete:BST1</span>
<span class="co">#&gt; Time difference of 19.7 mins</span>
<span class="co">#&gt; 🦇 🦇 🦇 MEX3C 🦇 🦇 🦇</span>
<span class="co">#&gt; ------------------ Step 1: Query ---------------</span>
<span class="co">#&gt; [1] "+ Extracting relevant variants from fullSS..."</span>
<span class="co">#&gt; [1] "++ Query Method: 'tabix"</span>
<span class="co">#&gt; [1] "---Min snp position: 47683589 ---"</span>
<span class="co">#&gt; [1] "---Max snp position: 49683589 ---"</span>
<span class="co">#&gt; [1] "TABIX:: Existing indexed tabix file detected"</span>
<span class="co">#&gt; [1] "TABIX:: Extracting subset of sum stats"</span>
<span class="co">#&gt; [1] "++ Returning 5405 x 11 data.table"</span>
<span class="co">#&gt; [1] "++ Saving query ==&gt; ./echolocatoR_results/GWAS/Nalls23andMe_2019/MEX3C/MEX3C_Nalls23andMe_2019_subset.tsv.gz"</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; ---------------- Step 1.5: Standardize ----------</span>
<span class="co">#&gt; [1] "+Inferring MAF from frequency column..."</span>
<span class="co">#&gt; [1] "+ Standardize:: Calculating proportion of cases."</span>
<span class="co">#&gt; [1] "+ Calculating t-statistic from Effect and StdErr..."</span>
<span class="co">#&gt; [1] "+ Extraction completed in 1.08 seconds"</span>
<span class="co">#&gt; [1] "+ 5399 SNPs x  15 columns"</span>
<span class="co">#&gt; [1] "+ Removing subset tmp..."</span>
<span class="co">#&gt; --- Step 2: Calculate Linkage Disequilibrium ---</span>
<span class="co">#&gt; [1] "LD:: Using UK Biobank LD reference panel..."</span>
<span class="co">#&gt; [1] "+ UKB LD file name: chr18_47000001_50000001"</span>
<span class="co">#&gt; [1] "+ LD:: Downloading full .gz/.npz UKB files and saving to disk."</span>
<span class="co">#&gt; [1] "+ LD:: ...this could take some time..."</span>
<span class="co">#&gt; Processed URL: ./echolocatoR_results/GWAS/Nalls23andMe_2019/MEX3C/LD/chr18_47000001_50000001</span>
<span class="co">#&gt; Some other message at the end</span>
<span class="co">#&gt; [1] "LD matrix dimensions 5388 x 5388"</span>
<span class="co">#&gt; [1] "+ LD:: Saving LD =&gt; ./echolocatoR_results/GWAS/Nalls23andMe_2019/MEX3C/LD/MEX3C.UKB_LD.RDS"</span>
<span class="co">#&gt; [1] "+ Removing .gz/.npz files."</span>
<span class="co">#&gt; -------------- Step 3: Filter SNPs -------------</span>
<span class="co">#&gt; [1] "++ Post-filtered data: 5399 x 15"</span>
<span class="co">#&gt; -------- Step 4: Statistically Fine-map --------</span>
<span class="co">#&gt; [1] "++ Fine-mapping using multiple tools: ABF, FINEMAP, SUSIE, POLYFUN_SUSIE"</span>
<span class="co">#&gt; Multi-finemap:: ABF</span>
<span class="co">#&gt; [1] "++ Credible Set SNPS identified = 0"</span>
<span class="co">#&gt; [1] "++ Merging ABF results with multi-finemap data."</span>
<span class="co">#&gt; Multi-finemap:: FINEMAP</span>
<span class="co">#&gt; [1] "++ Constructing FINEMAP master file."</span>
<span class="co">#&gt; [1] "++ Formatting data.z file for FINEMAP"</span>
<span class="co">#&gt; [1] "++ Formatting LD Matrix for FINEMAP"</span>
<span class="co">#&gt; [1] "+++ Writing FINEMAP z and ld files..."</span>
<span class="co">#&gt; [1] "cd ./echolocatoR_results/GWAS/Nalls23andMe_2019/MEX3C &amp;&amp; /private/var/folders/sw/km9819713937hp640lj59ygw0000gn/T/RtmpnuHoUP/temp_libpathdf6530fe2c52/echolocatoR/tools/FINEMAP/finemap_v1.3_MacOSX --cond --in-files ./echolocatoR_results/GWAS/Nalls23andMe_2019/MEX3C/FINEMAP/master --log --n-causal-snps 5"</span>
<span class="co">#&gt; Error in data.table::fread(file.path(locus_dir, "FINEMAP/data.config")) : </span>
<span class="co">#&gt;   File './echolocatoR_results/GWAS/Nalls23andMe_2019/MEX3C/FINEMAP/data.config' does not exist or is non-readable. getwd()=='/Users/schilder/Desktop'</span>
<span class="co">#&gt; [1] "++ Credible Set SNPS identified = 0"</span>
<span class="co">#&gt; [1] "++ Merging FINEMAP results with multi-finemap data."</span>
<span class="co">#&gt; Multi-finemap:: SUSIE</span>
<span class="co">#&gt; [1] "++ Computing phenotype variance..."</span>
<span class="co">#&gt; [1] "+ SUSIE:: n_causal = 5"</span>
<span class="co">#&gt; Warning in susie_ss(XtX = XtX, Xty = Xty, yty = var_y * (n - 1), n = n, : IBSS</span>
<span class="co">#&gt; algorithm did not converge in 100 iterations!</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; [1] "++ Extracting Credible Sets..."</span>
<span class="co">#&gt; [1] "++ Credible Set SNPS identified = 1"</span>
<span class="co">#&gt; [1] "++ Merging SUSIE results with multi-finemap data."</span>
<span class="co">#&gt; Multi-finemap:: POLYFUN_SUSIE</span>
<span class="co">#&gt; [1] "PolyFun:: Preparing SNP input file..."</span>
<span class="co">#&gt; [1] "+ PolyFun:: 5368 SNPs identified."</span>
<span class="co">#&gt; [1] "+ PolyFun:: Writing SNP file ==&gt; ./echolocatoR_results/GWAS/Nalls23andMe_2019/MEX3C/PolyFun/snps_to_finemap.txt.gz"</span>
<span class="co">#&gt; [1] "/opt/anaconda3/envs/echoR/bin/python /private/var/folders/sw/km9819713937hp640lj59ygw0000gn/T/RtmpnuHoUP/temp_libpathdf6530fe2c52/echolocatoR/tools/polyfun//extract_snpvar.py --snps ./echolocatoR_results/GWAS/Nalls23andMe_2019/MEX3C/PolyFun/snps_to_finemap.txt.gz --out ./echolocatoR_results/GWAS/Nalls23andMe_2019/MEX3C/PolyFun/snps_with_priors.snpvar.tsv.gz"</span>
<span class="co">#&gt; [1] "++ Remove tmp file."</span>
<span class="co">#&gt; [1] "++ Computing phenotype variance..."</span>
<span class="co">#&gt; [1] "+ SUSIE:: n_causal = 5"</span>
<span class="co">#&gt; [1] "Utilizing prior_weights for 5368 SNPs."</span>
<span class="co">#&gt; Warning in susie_ss(XtX = XtX, Xty = Xty, yty = var_y * (n - 1), n = n, : IBSS</span>
<span class="co">#&gt; algorithm did not converge in 100 iterations!</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; [1] "++ Extracting Credible Sets..."</span>
<span class="co">#&gt; [1] "++ Credible Set SNPS identified = 3"</span>
<span class="co">#&gt; [1] "++ Merging POLYFUN_SUSIE results with multi-finemap data."</span>
<span class="co">#&gt; [1] "++ Fine-mapping with ' ABF, FINEMAP, SUSIE, POLYFUN_SUSIE ' completed in  4.17"</span>
<span class="co">#&gt; [1] "+ Identifying Consensus SNPs..."</span>
<span class="co">#&gt; [1] "++ support_thresh = 2"</span>
<span class="co">#&gt; [1] "+ Calculating mean Posterior Probability (mean.PP)..."</span>
<span class="co">#&gt; [1] "++ 4 fine-mapping methods used."</span>
<span class="co">#&gt; [1] "++ 3 Credible Set SNPs identified."</span>
<span class="co">#&gt; [1] "++ 1 Consensus SNPs identified."</span>
<span class="co">#&gt; --------------- Step 7: Visualize --------------</span>
<span class="co">#&gt; [1] "+ Identifying Consensus SNPs..."</span>
<span class="co">#&gt; [1] "++ support_thresh = 2"</span>
<span class="co">#&gt; [1] "+ Calculating mean Posterior Probability (mean.PP)..."</span>
<span class="co">#&gt; [1] "++ 4 fine-mapping methods used."</span>
<span class="co">#&gt; [1] "++ 3 Credible Set SNPs identified."</span>
<span class="co">#&gt; [1] "++ 1 Consensus SNPs identified."</span>
<span class="co">#&gt; [1] "GGBIO:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "LD Matrix dimensions: 5368 x 5368"</span>
<span class="co">#&gt; [1] "Extracting LD subset for lead SNP: rs8087969"</span>
<span class="co">#&gt; [1] "++ GGBIO:: GWAS track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: ABF track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: FINEMAP track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: POLYFUN_SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: mean track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: Gene Model Track"</span>
<span class="co">#&gt; [1] "+ Annotating at transcript-level."</span>
<span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span>
<span class="co">#&gt; [1] "Importing Astrocyte enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Astrocyte promoters ..."</span>
<span class="co">#&gt; [1] "Importing Neuronal enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Neuronal promoters ..."</span>
<span class="co">#&gt; [1] "Importing Oligo enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Oligo promoters ..."</span>
<span class="co">#&gt; [1] "Importing Microglia enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Microglia promoters ..."</span>
<span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span>
<span class="co">#&gt; replace the existing scale.</span>
<span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span>
<span class="co">#&gt; replace the existing scale.</span>
<span class="co">#&gt; [1] "++ Nott sn-epigenomics:: Returning PLAC-seq track."</span>
<span class="co">#&gt; [1] "+ GGBIO:: Saving plot ==&gt; ./echolocatoR_results/GWAS/Nalls23andMe_2019/MEX3C/multiview_MEX3C_100kb.png"</span>
<span class="co">#&gt; Warning: Removed 304 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 5364 rows containing missing values (geom_point).</span>
<span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_point).</span>
<span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_label_repel).</span>

<span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_label_repel).</span>
<span class="co">#&gt; Error : Discrete value supplied to continuous scale</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Fine-mappping complete:MEX3C</span>
<span class="co">#&gt; Time difference of 14.8 mins</span>
<span class="co">#&gt; &lt;div id="htmlwidget-f843a680a1fcd62a0dd1" style="width:960px;height:500px;" class="datatables html-widget"&gt;&lt;/div&gt;</span>
<span class="co">#&gt; &lt;script type="application/json" data-for="htmlwidget-f843a680a1fcd62a0dd1"&gt;{"x":{"filter":"none","extensions":["Buttons"],"caption":"&lt;caption&gt;&lt;\/caption&gt;","data":[["1","2","3","4","5","6"],["BST1","BST1","BST1","MEX3C","MEX3C","MEX3C"],["rs34559912","rs4389574","rs4541502","rs1893379","rs11082860","rs4939656"],[4,4,4,18,18,18],[15730146,15730398,15712787,48718536,48714154,48714576],[1.213e-27,3.141e-17,8.309e-22,2.847e-08,3.007e-08,3.19e-08],[0.103,-0.0977,-0.0897,0.0558,0.0557,-0.0556],[0.0095,0.0116,0.0093,0.01,0.01,0.01],["T","A","T","A","T","T"],["G","G","G","G","G","C"],[0.5526,0.4443,0.4749,0.4542,0.4546,0.5454],[0.4474,0.4443,0.4749,0.4542,0.4546,0.4546],[56306,42598,56306,49053,49053,49053],[1417791,1322509,1417791,1411006,1411006,1411006],[0.0381969436203995,0.0312048799105125,0.0381969436203995,0.0335965875351613,0.0335965875351613,0.0335965875351613],[10.8421052631579,-8.42241379310345,-9.64516129032258,5.58,5.57,-5.56],[false,false,false,false,false,false],[null,null,null,null,null,null],[null,null,null,null,null,null],[null,null,null,null,null,null],[null,null,null,null,null,null],[3,1,2,1,0,0],[1,1,1,1,0,1.04650871348699e-05],[3,1,2,1,3,2],[1,1,1,1,1,1],[2,2,2,2,1,1],[true,true,true,true,false,false],[0.5,0.5,0.5,0.5,0.25,0.250002616271784],[0,0,0,0,0,0]],"container":"&lt;table class=\"display\"&gt;\n  &lt;thead&gt;\n    &lt;tr&gt;\n      &lt;th&gt; &lt;\/th&gt;\n      &lt;th&gt;Locus&lt;\/th&gt;\n      &lt;th&gt;SNP&lt;\/th&gt;\n      &lt;th&gt;CHR&lt;\/th&gt;\n      &lt;th&gt;POS&lt;\/th&gt;\n      &lt;th&gt;P&lt;\/th&gt;\n      &lt;th&gt;Effect&lt;\/th&gt;\n      &lt;th&gt;StdErr&lt;\/th&gt;\n      &lt;th&gt;A1&lt;\/th&gt;\n      &lt;th&gt;A2&lt;\/th&gt;\n      &lt;th&gt;Freq&lt;\/th&gt;\n      &lt;th&gt;MAF&lt;\/th&gt;\n      &lt;th&gt;N_cases&lt;\/th&gt;\n      &lt;th&gt;N_controls&lt;\/th&gt;\n      &lt;th&gt;proportion_cases&lt;\/th&gt;\n      &lt;th&gt;t_stat&lt;\/th&gt;\n      &lt;th&gt;leadSNP&lt;\/th&gt;\n      &lt;th&gt;ABF.CS&lt;\/th&gt;\n      &lt;th&gt;ABF.PP&lt;\/th&gt;\n      &lt;th&gt;FINEMAP.CS&lt;\/th&gt;\n      &lt;th&gt;FINEMAP.PP&lt;\/th&gt;\n      &lt;th&gt;SUSIE.CS&lt;\/th&gt;\n      &lt;th&gt;SUSIE.PP&lt;\/th&gt;\n      &lt;th&gt;POLYFUN_SUSIE.CS&lt;\/th&gt;\n      &lt;th&gt;POLYFUN_SUSIE.PP&lt;\/th&gt;\n      &lt;th&gt;Support&lt;\/th&gt;\n      &lt;th&gt;Consensus_SNP&lt;\/th&gt;\n      &lt;th&gt;mean.PP&lt;\/th&gt;\n      &lt;th&gt;mean.CS&lt;\/th&gt;\n    &lt;\/tr&gt;\n  &lt;\/thead&gt;\n&lt;\/table&gt;","options":{"dom":"Bfrtip","buttons":["copy","csv","excel","pdf","print"],"scrollY":400,"scrollX":true,"scrollCollapse":true,"paging":false,"columnDefs":[{"className":"dt-center","targets":"_all"},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}&lt;/script&gt;</span></pre></body></html></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
