<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Plotting vignette • echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Plotting vignette">
<meta property="og:description" content="echolocatoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/echolocatoR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Plotting vignette</h1>
                        <h4 data-toc-skip class="author"><h4>
Author: <i>Brian M. Schilder</i>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Updated: <i>Mar-04-2022</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/HEAD/vignettes/plotting_vignette.Rmd" class="external-link"><code>vignettes/plotting_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>plotting_vignette.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">root.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>
  collapse <span class="op">=</span> <span class="cn">TRUE</span>,
  comment <span class="op">=</span> <span class="st">"#&gt;"</span>,
  root.dir <span class="op">=</span> <span class="va">root.dir</span>, 
  fig.height <span class="op">=</span> <span class="fl">12</span>,
  fig.width <span class="op">=</span> <span class="fl">10</span>
<span class="op">)</span>  
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>
  root.dir <span class="op">=</span> <span class="va">root.dir</span>
  <span class="op">)</span>
<span class="co"># knitr::opts_chunk$get("root.dir")</span>
<span class="co"># devtools::build_vignettes(quiet = F)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echolocatoR" class="external-link">echolocatoR</a></span><span class="op">)</span> 
<span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span>
<span class="co">#&gt;   method from   </span>
<span class="co">#&gt;   +.gg   ggplot2</span></code></pre></div>
<div class="section level2">
<h2 id="plotting-loci-with-echolocator">Plotting loci with <em>echolocatoR</em><a class="anchor" aria-label="anchor" href="#plotting-loci-with-echolocator"></a>
</h2>
<p><em>echolocatoR</em> contains various functions that can be used separately<br>
from the comprehensive <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> pipeline.</p>
<p>Generate a multi-view plot of a given locus using <code><a href="../reference/PLOT.locus.html">PLOT.locus()</a></code>.</p>
<ul>
<li>You can mix and match different tracks and annotations using the different arguments (see <code><a href="../reference/PLOT.locus.html">?PLOT.locus</a></code> for details).</li>
</ul>
<p>The plot is centered on the lead/index SNP. If a list is supplied to plot.xoom * <code><a href="../reference/PLOT.locus.html">PLOT.locus()</a></code> returns a series of <code>ggplot</code> objects bound together with <a href="https://patchwork.data-imaginist.com" class="external-link"><code>patchwork</code></a>. One can further modify this object using <code>ggplot2</code> functions like <code>+ theme()</code>. + The modifications will be applied to all tracks at once.</p>
<ul>
<li>Save a high-resolution versions the plot by setting <code>save_plot=T</code>.
<ul>
<li>Further increase resolution by adjusting the <code>dpi</code> argument (<em>default=300</em>).</li>
<li>Files are saved in <em>jpg</em> format by default, but users can specify their preferred file format (e.g. <code>file_format="png"</code>)</li>
<li>Adjust the <code>height</code> and <code>width</code> of the saved plot using these respective arguments.</li>
<li>The plot will be automatically saved in the locus-specific directory as:<br>
*multiview_<locus>_&lt;plot.zoom&gt;.jpg*.</locus>
</li>
</ul>
</li>
</ul>
<div class="section level3">
<h3 id="load-example-data">Load example data<a class="anchor" aria-label="anchor" href="#load-example-data"></a>
</h3>
<p>Load example dataset of the results from fine-mapping the BST1 locus with <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code>. Original data comes from the recent Nalls et al. (2019) Parkinson’s disease GWAS (see <code><a href="../reference/BST1.html">?BST1</a></code> for details).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"BST1"</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"LD_matrix"</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"locus_dir"</span><span class="op">)</span>;
<span class="va">locus_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">root.dir</span>,<span class="va">locus_dir</span><span class="op">)</span>
<span class="va">finemap_DT</span> <span class="op">&lt;-</span> <span class="va">BST1</span>  
<span class="va">LD_matrix</span> <span class="op">&lt;-</span> <span class="va">BST1_LD_matrix</span>
<span class="va">LD_reference</span> <span class="op">&lt;-</span> <span class="st">"UKB"</span> <span class="co"># Used for naming saved plots</span>
<span class="va">plot.zoom</span> <span class="op">=</span> <span class="st">"10x"</span>
<span class="va">show_plot</span> <span class="op">&lt;-</span> <span class="cn">F</span>


<span class="co"># locus &lt;- "DNAH17"</span>
<span class="co"># locus_dir &lt;- file.path("~/Desktop/Fine_Mapping/Data/GWAS/Nalls23andMe_2019",locus)</span>
<span class="co"># LD_matrix &lt;- readRDS(file.path(locus_dir,"plink/UKB_LD.RDS"))</span>
<span class="co"># finemap_DT &lt;- data.table::fread(file.path(locus_dir,"Multi-finemap/Multi-finemap_results.txt"))</span>
<span class="co"># finemap_DT &lt;- find_consensus_SNPs(finemap_DT) </span>
<span class="co"># finemap_DT &lt;- update_CS_cols(finemap_DT)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="full-window">Full window<a class="anchor" aria-label="anchor" href="#full-window"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                       LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                       LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                       locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                       save_plot<span class="op">=</span><span class="cn">F</span>,
                       show_plot<span class="op">=</span><span class="va">show_plot</span>,
                       plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">trk_plot</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="at-multiple-zooms">At multiple zooms<a class="anchor" aria-label="anchor" href="#at-multiple-zooms"></a>
</h3>
<ul>
<li>You can easily generate the same locus plot at multiple zoomed in views by supplying a list to <code>plot.zoom</code>.<br>
</li>
<li>This list can be composed of zoom multipliers (e.g. <code>c("1x", "2x")</code>), window widths in units of basepairs (e.g. <code>c(5000, 1500)</code>), or a mixture of both (e.g. <code>c("1x","4x", 5000, 2000)</code>).</li>
<li>Each zoom view will be saved individually with its respective scale as the suffix (e.g. <code>multiview.BST1.UKB.4x.jpg</code>).<br>
</li>
<li>Each zoom view is stored as a named item within the returned list.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot_zooms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                             LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                             LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                             locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                             save_plot<span class="op">=</span><span class="cn">F</span>,
                             show_plot<span class="op">=</span><span class="va">show_plot</span>,
                             plot.zoom <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1x"</span>,<span class="st">"5x"</span>,<span class="st">"10x"</span><span class="op">)</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">trk_plot_zooms</span><span class="op">)</span> <span class="co"># Get zoom view names</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">trk_plot_zooms</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="return-as-list">Return as list<a class="anchor" aria-label="anchor" href="#return-as-list"></a>
</h3>
<ul>
<li>For even further control over each track of the multi-view plot, specify <code>PLOT.locus(..., return_list=T)</code> to instead return a named list (nested within each zoom view list item) of <code>ggplot</code> objects which can each be modified individually.</li>
<li>Once you’ve made your modifications, you can then bind this list of plots back together with <code>patchwork::wrap_plots(tracks_list, ncol = 1)</code>.</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                             LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                             LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                             locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                             save_plot<span class="op">=</span><span class="cn">F</span>,
                             show_plot<span class="op">=</span><span class="va">show_plot</span>,
                             plot.zoom<span class="op">=</span><span class="va">plot.zoom</span>,
                             return_list<span class="op">=</span><span class="cn">T</span><span class="op">)</span>  </code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">view1_list</span> <span class="op">&lt;-</span> <span class="va">trk_plot_list</span><span class="op">[[</span><span class="va">plot.zoom</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">view1_list</span><span class="op">)</span> <span class="co"># Get track names from a particular zoom view</span></code></pre></div>
<p>Modify a specific tracks within a view.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Modify your selected track</span>
<span class="va">modified_track</span> <span class="op">&lt;-</span> <span class="va">view1_list</span><span class="op">$</span><span class="va">GWAS</span> <span class="op">+</span> 
                      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Modified GWAS"</span><span class="op">)</span> <span class="op">+</span> 
                      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_dark</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
                      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Put it back into your track list</span>
<span class="va">view1_list</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">modified_track</span>
<span class="co"># Remove a plot you don't want</span>
<span class="va">view1_list</span><span class="op">[[</span><span class="st">"Genes"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="co"># Specify the relative heights of each track (make sure it matches your new # of plots!)</span>
<span class="va">track_heights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">.3</span>,<span class="fl">.1</span>,<span class="fl">.3</span>,<span class="fl">1</span><span class="op">)</span>

<span class="co"># Bind them together and plot</span>
<span class="va">fused_plot</span> <span class="op">&lt;-</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">view1_list</span>, 
                                    heights <span class="op">=</span> <span class="va">track_heights</span>,
                                    ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">fused_plot</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="using-xgr-annotations">Using XGR annotations<a class="anchor" aria-label="anchor" href="#using-xgr-annotations"></a>
</h3>
<ul>
<li>Whenever you use annotation arguments (e.g. <code>XGR_libnames</code>,<code>Roadmap</code>,<code>Nott_epigenome</code>) the annotations that overlap with your locus will automatically be saved as <code>GRanges</code> objects in a locus-specific subdirectory:<br><em>results/<dataset_type>/<dataset_name>/<locus>/annotation</locus></dataset_name></dataset_type></em>
</li>
<li>If a selected annotation has previously been downloaded and stored for that locus, <code><a href="../reference/PLOT.locus.html">PLOT.locus()</a></code> will automatically detect and import it to save time.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot.xgr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                           LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                           LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                           locus_dir<span class="op">=</span><span class="va">locus_dir</span>, 
                           
                           XGR_libnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENCODE_TFBS_ClusteredV3_CellTypes"</span><span class="op">)</span>, 
                           save_plot<span class="op">=</span><span class="cn">F</span>,
                           show_plot<span class="op">=</span><span class="va">show_plot</span>,
                           plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">trk_plot.xgr</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="using-roadmap-annotations">Using <a href="http://www.roadmapepigenomics.org/data/tables/brain" class="external-link">Roadmap</a> annotations<a class="anchor" aria-label="anchor" href="#using-roadmap-annotations"></a>
</h3>
<ul>
<li>Using the <code>Roadmap=T</code> and <code>Roadmap_query="&lt;query&gt;"</code> arguments searches the Roadmap for chromatin mark data across various cell-types, cell-lines and tissues.<br>
</li>
<li>Note that Roadmap queries requires <code>tabix</code> to be installed on your machine, or within a conda environment (<code>conda_env = "echoR"</code>).</li>
<li>Parallelizing these queries across multiple thredas speeds up this process (<code>nThread=&lt;n_cores_available&gt;</code>), as does reusing previously stored data which is automatically saved to the locus-specific subfolder (<code>&lt;dataset_type&gt;/&lt;dataset_name&gt;/&lt;locus&gt;/annotations/Roadmap.ChromatinMarks_CellTypes.RDS</code>) .</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot.roadmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                               LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                               LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                               locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                               
                               Roadmap<span class="op">=</span><span class="cn">T</span>, 
                               Roadmap_query<span class="op">=</span><span class="st">"monocyte"</span>, 
                               
                               save_plot<span class="op">=</span><span class="cn">F</span>, 
                               show_plot<span class="op">=</span><span class="va">show_plot</span>,
                               plot.zoom<span class="op">=</span><span class="st">"5x"</span>, 
                               nThread <span class="op">=</span> <span class="fu">parallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/parallel/detectCores.html" class="external-link">detectCores</a></span><span class="op">(</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span>,
                               conda_env <span class="op">=</span> <span class="st">"echoR"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">trk_plot.roadmap</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="using-nott_2019-annotations">Using <a href="https://science.sciencemag.org/content/366/6469/1134.abstract" class="external-link">Nott_2019</a> annotations<a class="anchor" aria-label="anchor" href="#using-nott_2019-annotations"></a>
</h3>
<ul>
<li>Query and plot brain cell type-specific epigenomic assays from <a href="https://science.sciencemag.org/content/366/6469/1134.abstract" class="external-link">Nott et al. (Science, 2019)</a><br>
(see <code><a href="../reference/NOTT_2019.bigwig_metadata.html">?NOTT_2019.bigwig_metadata</a></code> for details).</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot.nott_2019</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>,  
                                 LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                                 LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                                 locus_dir<span class="op">=</span><span class="va">locus_dir</span>, 
                                 
                                 Nott_epigenome<span class="op">=</span><span class="cn">T</span>,  
                                 Nott_binwidth <span class="op">=</span> <span class="fl">200</span>,
                                 Nott_regulatory_rects <span class="op">=</span> <span class="cn">T</span>, 
                                 Nott_show_placseq <span class="op">=</span> <span class="cn">T</span>,
                                 
                                 save_plot<span class="op">=</span><span class="cn">F</span>,
                                 show_plot<span class="op">=</span><span class="va">show_plot</span>,
                                 plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">trk_plot.nott_2019</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="using-qtl-datasets">Using QTL datasets<a class="anchor" aria-label="anchor" href="#using-qtl-datasets"></a>
</h3>
<ul>
<li>Plot multiple QTL p-value columns (or really P-value columns from any kind of datset).<br>
</li>
<li>Each QTL dataset will be plotted as a new track.</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Make fake QTL P-values for the sake a demonstration</span>
<span class="va">finemap_DT</span><span class="op">$</span><span class="va">fake_eQTL.P</span> <span class="op">&lt;-</span> <span class="va">finemap_DT</span><span class="op">$</span><span class="va">P</span>  <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">.9</span>,<span class="fl">.7</span><span class="op">)</span>
<span class="va">finemap_DT</span><span class="op">$</span><span class="va">fake_sQTL.P</span> <span class="op">&lt;-</span> <span class="va">finemap_DT</span><span class="op">$</span><span class="va">P</span>  <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">.8</span>,<span class="fl">.5</span><span class="op">)</span>

<span class="va">trk_plot.qtl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                           LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                           LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                           locus_dir<span class="op">=</span><span class="va">locus_dir</span>,
                           
                           QTL_prefixes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fake_eQTL."</span>,<span class="st">"fake_sQTL."</span><span class="op">)</span>, 
                           
                           save_plot<span class="op">=</span><span class="cn">F</span>,
                           show_plot<span class="op">=</span><span class="va">show_plot</span>,
                           plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">trk_plot.qtl</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] echolocatoR_0.2.2 BiocStyle_2.22.0 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] backports_1.4.1             Hmisc_4.6-0                </span>
<span class="co">#&gt;   [3] BiocFileCache_2.2.1         systemfonts_1.0.4          </span>
<span class="co">#&gt;   [5] plyr_1.8.6                  lazyeval_0.2.2             </span>
<span class="co">#&gt;   [7] splines_4.1.2               BiocParallel_1.28.3        </span>
<span class="co">#&gt;   [9] GenomeInfoDb_1.30.1         ggplot2_3.3.5              </span>
<span class="co">#&gt;  [11] digest_0.6.29               ensembldb_2.18.3           </span>
<span class="co">#&gt;  [13] htmltools_0.5.2             fansi_1.0.2                </span>
<span class="co">#&gt;  [15] magrittr_2.0.2              checkmate_2.0.0            </span>
<span class="co">#&gt;  [17] memoise_2.0.1               BSgenome_1.62.0            </span>
<span class="co">#&gt;  [19] cluster_2.1.2               Biostrings_2.62.0          </span>
<span class="co">#&gt;  [21] matrixStats_0.61.0          ggbio_1.42.0               </span>
<span class="co">#&gt;  [23] pkgdown_2.0.2               prettyunits_1.1.1          </span>
<span class="co">#&gt;  [25] jpeg_0.1-9                  colorspace_2.0-3           </span>
<span class="co">#&gt;  [27] rappdirs_0.3.3              blob_1.2.2                 </span>
<span class="co">#&gt;  [29] textshaping_0.3.6           xfun_0.30                  </span>
<span class="co">#&gt;  [31] dplyr_1.0.8                 crayon_1.5.0               </span>
<span class="co">#&gt;  [33] RCurl_1.98-1.6              jsonlite_1.8.0             </span>
<span class="co">#&gt;  [35] graph_1.72.0                Exact_3.1                  </span>
<span class="co">#&gt;  [37] VariantAnnotation_1.40.0    survival_3.2-13            </span>
<span class="co">#&gt;  [39] glue_1.6.2                  gtable_0.3.0               </span>
<span class="co">#&gt;  [41] zlibbioc_1.40.0             XVector_0.34.0             </span>
<span class="co">#&gt;  [43] DelayedArray_0.20.0         BiocGenerics_0.40.0        </span>
<span class="co">#&gt;  [45] scales_1.1.1                mvtnorm_1.1-3              </span>
<span class="co">#&gt;  [47] DBI_1.1.2                   GGally_2.1.2               </span>
<span class="co">#&gt;  [49] Rcpp_1.0.8                  progress_1.2.2             </span>
<span class="co">#&gt;  [51] htmlTable_2.4.0             foreign_0.8-81             </span>
<span class="co">#&gt;  [53] bit_4.0.4                   proxy_0.4-26               </span>
<span class="co">#&gt;  [55] OrganismDbi_1.36.0          Formula_1.2-4              </span>
<span class="co">#&gt;  [57] stats4_4.1.2                htmlwidgets_1.5.4          </span>
<span class="co">#&gt;  [59] httr_1.4.2                  RColorBrewer_1.1-2         </span>
<span class="co">#&gt;  [61] ellipsis_0.3.2              pkgconfig_2.0.3            </span>
<span class="co">#&gt;  [63] reshape_0.8.8               XML_3.99-0.9               </span>
<span class="co">#&gt;  [65] dbplyr_2.1.1                nnet_7.3-16                </span>
<span class="co">#&gt;  [67] sass_0.4.0                  utf8_1.2.2                 </span>
<span class="co">#&gt;  [69] tidyselect_1.1.2            rlang_1.0.1                </span>
<span class="co">#&gt;  [71] reshape2_1.4.4              AnnotationDbi_1.56.2       </span>
<span class="co">#&gt;  [73] munsell_0.5.0               tools_4.1.2                </span>
<span class="co">#&gt;  [75] cachem_1.0.6                cli_3.2.0                  </span>
<span class="co">#&gt;  [77] generics_0.1.2              RSQLite_2.2.10             </span>
<span class="co">#&gt;  [79] evaluate_0.15               stringr_1.4.0              </span>
<span class="co">#&gt;  [81] fastmap_1.1.0               yaml_2.3.5                 </span>
<span class="co">#&gt;  [83] ragg_1.2.2                  knitr_1.37                 </span>
<span class="co">#&gt;  [85] bit64_4.0.5                 fs_1.5.2                   </span>
<span class="co">#&gt;  [87] purrr_0.3.4                 KEGGREST_1.34.0            </span>
<span class="co">#&gt;  [89] AnnotationFilter_1.18.0     rootSolve_1.8.2.3          </span>
<span class="co">#&gt;  [91] RBGL_1.70.0                 xml2_1.3.3                 </span>
<span class="co">#&gt;  [93] biomaRt_2.50.3              compiler_4.1.2             </span>
<span class="co">#&gt;  [95] rstudioapi_0.13             filelock_1.0.2             </span>
<span class="co">#&gt;  [97] curl_4.3.2                  png_0.1-7                  </span>
<span class="co">#&gt;  [99] e1071_1.7-9                 tibble_3.1.6               </span>
<span class="co">#&gt; [101] bslib_0.3.1                 DescTools_0.99.44          </span>
<span class="co">#&gt; [103] stringi_1.7.6               GenomicFeatures_1.46.5     </span>
<span class="co">#&gt; [105] desc_1.4.0                  lattice_0.20-45            </span>
<span class="co">#&gt; [107] ProtGenerics_1.26.0         Matrix_1.3-4               </span>
<span class="co">#&gt; [109] vctrs_0.3.8                 pillar_1.7.0               </span>
<span class="co">#&gt; [111] lifecycle_1.0.1             BiocManager_1.30.16        </span>
<span class="co">#&gt; [113] jquerylib_0.1.4             data.table_1.14.2          </span>
<span class="co">#&gt; [115] bitops_1.0-7                lmom_2.8                   </span>
<span class="co">#&gt; [117] rtracklayer_1.54.0          GenomicRanges_1.46.1       </span>
<span class="co">#&gt; [119] R6_2.5.1                    BiocIO_1.4.0               </span>
<span class="co">#&gt; [121] latticeExtra_0.6-29         bookdown_0.24              </span>
<span class="co">#&gt; [123] gridExtra_2.3               IRanges_2.28.0             </span>
<span class="co">#&gt; [125] gld_2.6.4                   dichromat_2.0-0            </span>
<span class="co">#&gt; [127] boot_1.3-28                 MASS_7.3-54                </span>
<span class="co">#&gt; [129] assertthat_0.2.1            SummarizedExperiment_1.24.0</span>
<span class="co">#&gt; [131] rprojroot_2.0.2             rjson_0.2.21               </span>
<span class="co">#&gt; [133] GenomicAlignments_1.30.0    Rsamtools_2.10.0           </span>
<span class="co">#&gt; [135] S4Vectors_0.32.3            GenomeInfoDbData_1.2.7     </span>
<span class="co">#&gt; [137] expm_0.999-6                parallel_4.1.2             </span>
<span class="co">#&gt; [139] hms_1.1.1                   grid_4.1.2                 </span>
<span class="co">#&gt; [141] rpart_4.1-15                class_7.3-19               </span>
<span class="co">#&gt; [143] rmarkdown_2.12              MatrixGenerics_1.6.0       </span>
<span class="co">#&gt; [145] biovizBase_1.42.0           Biobase_2.54.0             </span>
<span class="co">#&gt; [147] base64enc_0.1-3             restfulr_0.0.13</span></code></pre></div>
<details></details></details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
