<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Plotting vignette • echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Plotting vignette">
<meta property="og:description" content="echolocatoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/full_pipeline_vignette.html">Full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="plotting_vignette_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Plotting vignette</h1>
                        <h4 class="author">Brian M. Schilder</h4>
            
            <h4 class="date">Most recent update:<br> 2021-01-19</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/vignettes/plotting_vignette.Rmd"><code>vignettes/plotting_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>plotting_vignette.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echolocatoR">echolocatoR</a></span><span class="op">)</span> 
<span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span>
<span class="co">#&gt;   method from   </span>
<span class="co">#&gt;   +.gg   ggplot2</span></code></pre></div>
<div id="plotting-loci-with-echolocator" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-loci-with-echolocator" class="anchor"></a>Plotting loci with <em>echolocatoR</em>
</h1>
<p><em>echolocatoR</em> contains various functions that can be used separately<br>
from the comprehensive <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> pipeline.</p>
<p>Generate a multi-view plot of a given locus using <code><a href="../reference/PLOT.locus.html">PLOT.locus()</a></code>.</p>
<ul>
<li>You can mix and match different tracks and annotations using the different arguments (see <code><a href="../reference/PLOT.locus.html">?PLOT.locus</a></code> for details).</li>
</ul>
<p>The plot is centered on the lead/index SNP. If a list is supplied to plot.xoom * <code><a href="../reference/PLOT.locus.html">PLOT.locus()</a></code> returns a series of <code>ggplot</code> objects bound together with <a href="https://patchwork.data-imaginist.com"><code>patchwork</code></a>. One can further modify this object using <code>ggplot2</code> functions like <code>+ theme()</code>. + The modifications will be applied to all tracks at once.</p>
<ul>
<li>Save a high-resolution versions the plot by setting <code>save_plot=T</code>.
<ul>
<li>Further increase resolution by adjusting the <code>dpi</code> argument (<em>default=300</em>).</li>
<li>Files are saved in <em>jpg</em> format by default, but users can specify their preferred file format (e.g. <code>file_format="png"</code>)</li>
<li>Adjust the <code>height</code> and <code>width</code> of the saved plot using these respective arguments.</li>
<li>The plot will be automatically saved in the locus-specific directory as:<br>
*multiview_<locus>_&lt;plot.zoom&gt;.jpg*.</locus>
</li>
</ul>
</li>
</ul>
<div id="load-example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-example-data" class="anchor"></a>Load example data</h2>
<p>Load example dataset of the results from fine-mapping the BST1 locus with <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code>. Original data comes from the recent Nalls et al. (2019) Parkinson’s disease GWAS (see <code><a href="../reference/BST1.html">?BST1</a></code> for details).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"BST1"</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"LD_matrix"</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"locus_dir"</span><span class="op">)</span>;
<span class="va">locus_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"~/Desktop"</span>,<span class="va">locus_dir</span><span class="op">)</span>
<span class="va">finemap_DT</span> <span class="op">&lt;-</span> <span class="va">BST1</span>  
<span class="va">LD_matrix</span> <span class="op">&lt;-</span> <span class="va">BST1_LD_matrix</span>
<span class="va">LD_reference</span> <span class="op">&lt;-</span> <span class="st">"UKB"</span> <span class="co"># Used for naming saved plots</span>
<span class="va">plot.zoom</span> <span class="op">=</span> <span class="st">"10x"</span>
<span class="va">show_plot</span> <span class="op">&lt;-</span> <span class="cn">F</span>


<span class="co"># locus &lt;- "DNAH17"</span>
<span class="co"># locus_dir &lt;- file.path("~/Desktop/Fine_Mapping/Data/GWAS/Nalls23andMe_2019",locus)</span>
<span class="co"># LD_matrix &lt;- readRDS(file.path(locus_dir,"plink/UKB_LD.RDS"))</span>
<span class="co"># finemap_DT &lt;- data.table::fread(file.path(locus_dir,"Multi-finemap/Multi-finemap_results.txt"))</span>
<span class="co"># finemap_DT &lt;- find_consensus_SNPs(finemap_DT) </span>
<span class="co"># finemap_DT &lt;- update_CS_cols(finemap_DT)</span></code></pre></div>
</div>
<div id="full-window" class="section level2">
<h2 class="hasAnchor">
<a href="#full-window" class="anchor"></a>Full window</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                       LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                       LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                       locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                       save_plot<span class="op">=</span><span class="cn">F</span>,
                       show_plot<span class="op">=</span><span class="va">show_plot</span>,
                       plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span> 
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span></code></pre></div>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Filling NAs in CS cols with 0"
#&gt; [1] "+ Filling NAs in PP cols with 0"
#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."
#&gt; [1] "++ PLOT:: GWAS full window track"
#&gt; [1] "++ PLOT:: GWAS track"
#&gt; [1] "++ PLOT:: Merged fine-mapping track"
#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"
#&gt; Warning: The `.dots` argument of `group_by()` is deprecated as of dplyr 1.0.0.
#&gt; This warning is displayed once every 8 hours.
#&gt; Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "++ PLOT:: Adding Gene model track."
#&gt; Loading required package: ensembldb
#&gt; Loading required package: BiocGenerics
#&gt; Loading required package: parallel
#&gt; 
#&gt; Attaching package: 'BiocGenerics'
#&gt; The following objects are masked from 'package:parallel':
#&gt; 
#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,
#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB
#&gt; The following objects are masked from 'package:stats':
#&gt; 
#&gt;     IQR, mad, sd, var, xtabs
#&gt; The following objects are masked from 'package:base':
#&gt; 
#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
#&gt;     union, unique, unsplit, which, which.max, which.min
#&gt; Loading required package: GenomicRanges
#&gt; Loading required package: stats4
#&gt; Loading required package: S4Vectors
#&gt; 
#&gt; Attaching package: 'S4Vectors'
#&gt; The following object is masked from 'package:base':
#&gt; 
#&gt;     expand.grid
#&gt; Loading required package: IRanges
#&gt; Loading required package: GenomeInfoDb
#&gt; Loading required package: GenomicFeatures
#&gt; Loading required package: AnnotationDbi
#&gt; Loading required package: Biobase
#&gt; Welcome to Bioconductor
#&gt; 
#&gt;     Vignettes contain introductory material; view with
#&gt;     'browseVignettes()'. To cite Bioconductor, see
#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.
#&gt; Loading required package: AnnotationFilter
#&gt; 
#&gt; Attaching package: 'ensembldb'
#&gt; The following object is masked from 'package:stats':
#&gt; 
#&gt;     filter
#&gt; max_transcripts=1.
#&gt; 15 transcripts from 15 genes returned.
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...
#&gt; Warning: `quo_expr()` is deprecated as of rlang 0.2.0.
#&gt; Please use `quo_squash()` instead.
#&gt; This warning is displayed once per session.</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."
#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ PLOT:: Get window suffix..."
#&gt; [1] "+ Constructing zoom polygon..."
#&gt; [1] "+ Highlighting zoom origin..."
#&gt; [1] "+ Removing subplot margins..."
#&gt; [1] "+ Reordering tracks..."
#&gt; [1] "+ Ensuring last track shows genomic units..."
#&gt; [1] "+ Aligning xlimits for each subplot..."
#&gt; [1] "+ Checking track heights..."
#&gt; [1] "+ PLOT:: Saving ggplots tracks list ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/plot_tracks/multiview.BST1.UKB.10x.RDS"</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot</span><span class="op">)</span>
<span class="co">#&gt; $`10x`</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-2-1.png" width="960"></p>
</div>
<div id="at-multiple-zooms" class="section level2">
<h2 class="hasAnchor">
<a href="#at-multiple-zooms" class="anchor"></a>At multiple zooms</h2>
<ul>
<li>You can easily generate the same locus plot at multiple zoomed in views by supplying a list to <code>plot.zoom</code>.<br>
</li>
<li>This list can be composed of zoom multipliers (e.g. <code><a href="https://rdrr.io/r/base/c.html">c("1x", "2x")</a></code>), window widths in units of basepairs (e.g. <code><a href="https://rdrr.io/r/base/c.html">c(5000, 1500)</a></code>), or a mixture of both (e.g. <code><a href="https://rdrr.io/r/base/c.html">c("1x","4x", 5000, 2000)</a></code>).</li>
<li>Each zoom view will be saved individually with its respective scale as the suffix (e.g. <code>multiview.BST1.UKB.4x.jpg</code>).<br>
</li>
<li>Each zoom view is stored as a named item within the returned list.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot_zooms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                             LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                             LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                             locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                             save_plot<span class="op">=</span><span class="cn">F</span>,
                             show_plot<span class="op">=</span><span class="va">show_plot</span>,
                             plot.zoom <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1x"</span>,<span class="st">"5x"</span>,<span class="st">"10x"</span><span class="op">)</span><span class="op">)</span> 
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span></code></pre></div>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Filling NAs in CS cols with 0"
#&gt; [1] "+ Filling NAs in PP cols with 0"
#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."
#&gt; [1] "++ PLOT:: GWAS full window track"
#&gt; [1] "++ PLOT:: GWAS track"
#&gt; [1] "++ PLOT:: Merged fine-mapping track"
#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"
#&gt; [1] "++ PLOT:: Adding Gene model track."
#&gt; max_transcripts=1.
#&gt; 15 transcripts from 15 genes returned.
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."
#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 1x +&lt;+&lt;+&lt;+&lt;+</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ PLOT:: Get window suffix..."
#&gt; [1] "+ PLOT:: Removing GWAS full window track @ zoom=1x"
#&gt; [1] "+ Removing subplot margins..."
#&gt; [1] "+ Reordering tracks..."
#&gt; [1] "+ Ensuring last track shows genomic units..."
#&gt; [1] "+ Aligning xlimits for each subplot..."
#&gt; [1] "+ Checking track heights..."
#&gt; [1] "+ PLOT:: Saving ggplots tracks list ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/plot_tracks/multiview.BST1.UKB.1x.RDS"
#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 5x +&lt;+&lt;+&lt;+&lt;+</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ PLOT:: Get window suffix..."
#&gt; [1] "+ Constructing zoom polygon..."
#&gt; [1] "+ Highlighting zoom origin..."
#&gt; [1] "+ Removing subplot margins..."
#&gt; [1] "+ Reordering tracks..."
#&gt; [1] "+ Ensuring last track shows genomic units..."
#&gt; [1] "+ Aligning xlimits for each subplot..."
#&gt; [1] "+ Checking track heights..."
#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</code></pre>
<pre style="max-height: 200px;"><code><a href="https://rdrr.io/r/base/names.html">#&gt; [1] "+ PLOT:: Get window suffix..."
#&gt; [1] "+ Constructing zoom polygon..."
#&gt; [1] "+ Highlighting zoom origin..."
#&gt; [1] "+ Removing subplot margins..."
#&gt; [1] "+ Reordering tracks..."
#&gt; [1] "+ Ensuring last track shows genomic units..."
#&gt; [1] "+ Aligning xlimits for each subplot..."
#&gt; [1] "+ Checking track heights..."
names(trk_plot_zooms) # Get zoom view names</a></code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "1x"  "5x"  "10x"</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot_zooms</span><span class="op">)</span>
<span class="co">#&gt; $`1x`</span>
<span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span>
<span class="co">#&gt; increasing max.overlaps</span>

<span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span>
<span class="co">#&gt; increasing max.overlaps</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-3-1.png" width="960"></p>
<pre><code>#&gt; 
#&gt; $`5x`
#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider
#&gt; increasing max.overlaps
#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider
#&gt; increasing max.overlaps</code></pre>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-3-2.png" width="960"></p>
<pre><code>#&gt; 
#&gt; $`10x`</code></pre>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-3-3.png" width="960"></p>
</div>
<div id="return-as-list" class="section level2">
<h2 class="hasAnchor">
<a href="#return-as-list" class="anchor"></a>Return as list</h2>
<ul>
<li>For even further control over each track of the multi-view plot, specify <code><a href="../reference/PLOT.locus.html">PLOT.locus(..., return_list=T)</a></code> to instead return a named list (nested within each zoom view list item) of <code>ggplot</code> objects which can each be modified individually.</li>
<li>Once you’ve made your modifications, you can then bind this list of plots back together with <code><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">patchwork::wrap_plots(tracks_list, ncol = 1)</a></code>.</li>
</ul>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                             LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                             LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                             locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                             save_plot<span class="op">=</span><span class="cn">F</span>,
                             show_plot<span class="op">=</span><span class="va">show_plot</span>,
                             plot.zoom<span class="op">=</span><span class="va">plot.zoom</span>,
                             return_list<span class="op">=</span><span class="cn">T</span><span class="op">)</span>  
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span></code></pre></div>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Filling NAs in CS cols with 0"
#&gt; [1] "+ Filling NAs in PP cols with 0"
#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."
#&gt; [1] "++ PLOT:: GWAS full window track"
#&gt; [1] "++ PLOT:: GWAS track"
#&gt; [1] "++ PLOT:: Merged fine-mapping track"
#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"
#&gt; [1] "++ PLOT:: Adding Gene model track."
#&gt; max_transcripts=1.
#&gt; 15 transcripts from 15 genes returned.
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."
#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ PLOT:: Get window suffix..."
#&gt; [1] "+ Constructing zoom polygon..."
#&gt; [1] "+ Highlighting zoom origin..."
#&gt; [1] "+ Removing subplot margins..."
#&gt; [1] "+ Reordering tracks..."
#&gt; [1] "+ Ensuring last track shows genomic units..."
#&gt; [1] "+ Aligning xlimits for each subplot..."
#&gt; [1] "+ Checking track heights..."
#&gt; [1] "+ PLOT:: Saving ggplots tracks list ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/plot_tracks/multiview.BST1.UKB.10x.RDS"</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">view1_list</span> <span class="op">&lt;-</span> <span class="va">trk_plot_list</span><span class="op">[[</span><span class="va">plot.zoom</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">view1_list</span><span class="op">)</span> <span class="co"># Get track names from a particular zoom view</span>
<span class="co">#&gt; [1] "GWAS full window" "zoom_polygon"     "Genes"            "GWAS"            </span>
<span class="co">#&gt; [5] "Fine-mapping"</span></code></pre></div>
<p>Modify a specific tracks within a view.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Modify your selected track</span>
<span class="va">modified_track</span> <span class="op">&lt;-</span> <span class="va">view1_list</span><span class="op">$</span><span class="va">GWAS</span> <span class="op">+</span> 
                      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Modified GWAS"</span><span class="op">)</span> <span class="op">+</span> 
                      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_dark</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
                      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Put it back into your track list</span>
<span class="va">view1_list</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">modified_track</span>
<span class="co"># Remove a plot you don't want</span>
<span class="va">view1_list</span><span class="op">[[</span><span class="st">"Genes"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="co"># Specify the relative heights of each track (make sure it matches your new # of plots!)</span>
<span class="va">track_heights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.3</span>,<span class="fl">.1</span>,<span class="fl">.3</span>,<span class="fl">1</span><span class="op">)</span>

<span class="co"># Bind them together and plot</span>
<span class="va">fused_plot</span> <span class="op">&lt;-</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">wrap_plots</a></span><span class="op">(</span><span class="va">view1_list</span>, 
                                    heights <span class="op">=</span> <span class="va">track_heights</span>,
                                    ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">fused_plot</span><span class="op">)</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/modify%20track-1.png" width="960"></p>
</div>
<div id="using-xgr-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#using-xgr-annotations" class="anchor"></a>Using XGR annotations</h2>
<ul>
<li>Whenever you use annotation arguments (e.g. <code>XGR_libnames</code>,<code>Roadmap</code>,<code>Nott_epigenome</code>) the annotations that overlap with your locus will automatically be saved as <code>GRanges</code> objects in a locus-specific subdirectory:<br><em>results/<dataset_type>/<dataset_name>/<locus>/annotation</locus></dataset_name></dataset_type></em>
</li>
<li>If a selected annotation has previously been downloaded and stored for that locus, <code><a href="../reference/PLOT.locus.html">PLOT.locus()</a></code> will automatically detect and import it to save time.</li>
</ul>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot.xgr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                           LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                           LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                           locus_dir<span class="op">=</span><span class="va">locus_dir</span>, 
                           
                           XGR_libnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ENCODE_TFBS_ClusteredV3_CellTypes"</span><span class="op">)</span>, 
                           save_plot<span class="op">=</span><span class="cn">F</span>,
                           show_plot<span class="op">=</span><span class="va">show_plot</span>,
                           plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span>
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span></code></pre></div>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Filling NAs in CS cols with 0"
#&gt; [1] "+ Filling NAs in PP cols with 0"
#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."
#&gt; [1] "++ PLOT:: GWAS full window track"
#&gt; [1] "++ PLOT:: GWAS track"
#&gt; [1] "++ PLOT:: Merged fine-mapping track"
#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"
#&gt; [1] "++ PLOT:: Adding Gene model track."
#&gt; max_transcripts=1.
#&gt; 15 transcripts from 15 genes returned.
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "++ PLOT:: XGR Tracks"
#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."
#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ PLOT:: Get window suffix..."
#&gt; [1] "+ Constructing zoom polygon..."
#&gt; [1] "+ Highlighting zoom origin..."
#&gt; [1] "+ Removing subplot margins..."
#&gt; [1] "+ Reordering tracks..."
#&gt; [1] "+ Ensuring last track shows genomic units..."
#&gt; [1] "+ Aligning xlimits for each subplot..."
#&gt; [1] "+ Checking track heights..."
#&gt; [1] "+ PLOT:: Saving ggplots tracks list ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/plot_tracks/multiview.BST1.UKB.10x.RDS"</code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot.xgr</span><span class="op">)</span>
<span class="co">#&gt; $`10x`</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-4-1.png" width="960"></p>
</div>
<div id="using-roadmap-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#using-roadmap-annotations" class="anchor"></a>Using <a href="http://www.roadmapepigenomics.org/data/tables/brain">Roadmap</a> annotations</h2>
<ul>
<li>Using the <code>Roadmap=T</code> and <code>Roadmap_query="&lt;query&gt;"</code> arguments searches the Roadmap for chromatin mark data across various cell-types, cell-lines and tissues.<br>
</li>
<li>Note that Roadmap queries requires <code>tabix</code> to be installed on your machine, or within a conda environment (<code>conda_env = "echoR"</code>).</li>
<li>Parallelizing these queries across multiple thredas speeds up this process (<code>nThread=&lt;n_cores_available&gt;</code>), as does reusing previously stored data which is automatically saved to the locus-specific subfolder (<code>&lt;dataset_type&gt;/&lt;dataset_name&gt;/&lt;locus&gt;/annotations/Roadmap.ChromatinMarks_CellTypes.RDS</code>) .</li>
</ul>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot.roadmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                               LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                               LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                               locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                               
                               Roadmap<span class="op">=</span><span class="cn">T</span>, 
                               Roadmap_query<span class="op">=</span><span class="st">"monocyte"</span>, 
                               
                               save_plot<span class="op">=</span><span class="cn">F</span>, 
                               show_plot<span class="op">=</span><span class="va">show_plot</span>,
                               plot.zoom<span class="op">=</span><span class="va">plot.zoom</span>, 
                               nThread <span class="op">=</span> <span class="fu">parallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/parallel/detectCores.html">detectCores</a></span><span class="op">(</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span>,
                               conda_env <span class="op">=</span> <span class="st">"echoR"</span><span class="op">)</span>
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span></code></pre></div>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Filling NAs in CS cols with 0"
#&gt; [1] "+ Filling NAs in PP cols with 0"
#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."
#&gt; [1] "++ PLOT:: GWAS full window track"
#&gt; [1] "++ PLOT:: GWAS track"
#&gt; [1] "++ PLOT:: Merged fine-mapping track"
#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"
#&gt; [1] "++ PLOT:: Adding Gene model track."
#&gt; max_transcripts=1.
#&gt; 15 transcripts from 15 genes returned.
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ PLOT:: Creating Roadmap track"
#&gt; [1] "+ Saved annotation file detected. Loading..."
#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."
#&gt; Warning: Groups with fewer than two data points have been dropped.
#&gt; Warning: Groups with fewer than two data points have been dropped.

#&gt; Warning: Groups with fewer than two data points have been dropped.

#&gt; Warning: Groups with fewer than two data points have been dropped.
#&gt; Warning: Removed 4 rows containing missing values (position_stack).
#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ PLOT:: Get window suffix..."
#&gt; [1] "+ Constructing zoom polygon..."
#&gt; [1] "+ Highlighting zoom origin..."
#&gt; [1] "+ Removing subplot margins..."
#&gt; [1] "+ Reordering tracks..."
#&gt; [1] "+ Ensuring last track shows genomic units..."
#&gt; Warning: Groups with fewer than two data points have been dropped.
#&gt; Warning: Groups with fewer than two data points have been dropped.

#&gt; Warning: Groups with fewer than two data points have been dropped.

#&gt; Warning: Groups with fewer than two data points have been dropped.
#&gt; Warning: Removed 4 rows containing missing values (position_stack).</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Aligning xlimits for each subplot..."
#&gt; Warning: Groups with fewer than two data points have been dropped.
#&gt; Warning: Groups with fewer than two data points have been dropped.

#&gt; Warning: Groups with fewer than two data points have been dropped.

#&gt; Warning: Groups with fewer than two data points have been dropped.
#&gt; Warning: Removed 4 rows containing missing values (position_stack).</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Checking track heights..."
#&gt; [1] "+ PLOT:: Saving ggplots tracks list ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/plot_tracks/multiview.BST1.UKB.10x.RDS"</code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot.roadmap</span><span class="op">)</span>
<span class="co">#&gt; $`10x`</span>
<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>
<span class="co">#&gt; Warning: Removed 4 rows containing missing values (position_stack).</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-5-1.png" width="960"></p>
</div>
<div id="using-nott_2019-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#using-nott_2019-annotations" class="anchor"></a>Using <a href="https://science.sciencemag.org/content/366/6469/1134.abstract">Nott_2019</a> annotations</h2>
<ul>
<li>Query and plot brain cell type-specific epigenomic assays from <a href="https://science.sciencemag.org/content/366/6469/1134.abstract">Nott et al. (Science, 2019)</a><br>
(see <code><a href="../reference/NOTT_2019.bigwig_metadata.html">?NOTT_2019.bigwig_metadata</a></code> for details).</li>
</ul>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot.nott_2019</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>,  
                                 LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                                 LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                                 locus_dir<span class="op">=</span><span class="va">locus_dir</span>, 
                                 
                                 Nott_epigenome<span class="op">=</span><span class="cn">T</span>,  
                                 Nott_binwidth <span class="op">=</span> <span class="fl">200</span>,
                                 Nott_regulatory_rects <span class="op">=</span> <span class="cn">T</span>, 
                                 Nott_show_placseq <span class="op">=</span> <span class="cn">T</span>,
                                 
                                 save_plot<span class="op">=</span><span class="cn">F</span>,
                                 show_plot<span class="op">=</span><span class="va">show_plot</span>,
                                 plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span> 
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span></code></pre></div>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Filling NAs in CS cols with 0"
#&gt; [1] "+ Filling NAs in PP cols with 0"
#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."
#&gt; [1] "++ PLOT:: GWAS full window track"
#&gt; [1] "++ PLOT:: GWAS track"
#&gt; [1] "++ PLOT:: Merged fine-mapping track"
#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"
#&gt; [1] "++ PLOT:: Adding Gene model track."
#&gt; max_transcripts=1.
#&gt; 15 transcripts from 15 genes returned.
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "NOTT_2019:: Creating epigenomic histograms plot"
#&gt; [1] "NOTT_2019:: Importing bigWig subsets from UCSC..."
#&gt; [1] "+ Inferring genomic limits for window: 1x"
#&gt; [1] "++ NOTT_2019:: Downloading and merging 12 peaks BED files."
#&gt; [1] "++ NOTT_2019:: Converting merged BED files to GRanges."
#&gt; [1] "++ NOTT_2019:: 634540 ranges retrieved."
#&gt; 1860925 query SNP(s) detected with reference overlap.</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ PLOT:: Calculating max histogram height"
#&gt; [1] "++ Converting label units to Mb..."
#&gt; [1] "NOTT_2019:: Creating PLAC-seq interactome plot"
#&gt; [1] "+ NOTT_2019:: Getting interactome data."
#&gt; [1] "++ Converting to GRanges."
#&gt; [1] "Importing Microglia interactome ..."
#&gt; [1] "Importing Neuronal interactome ..."
#&gt; [1] "Importing Oligo interactome ..."
#&gt; Warning in .replace_seqlevels_style(x_seqlevels, value): found more than one
#&gt; best sequence renaming map compatible with seqname style "NCBI" for this object,
#&gt; using the first one
#&gt; 1403 query SNP(s) detected with reference overlap.
#&gt; Warning in .replace_seqlevels_style(x_seqlevels, value): found more than one
#&gt; best sequence renaming map compatible with seqname style "NCBI" for this object,
#&gt; using the first one
#&gt; 1770 query SNP(s) detected with reference overlap.</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "++ NOTT_2019:: Adding enhancer/promoter rectangles"
#&gt; [1] "++ Converting genomic units to Mb..."
#&gt; [1] "++ NOTT_2019:: Returning PLAC-seq track."
#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."
#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ PLOT:: Get window suffix..."
#&gt; [1] "+ Constructing zoom polygon..."
#&gt; [1] "+ Highlighting zoom origin..."
#&gt; [1] "+ Removing subplot margins..."
#&gt; [1] "+ Reordering tracks..."
#&gt; [1] "+ Ensuring last track shows genomic units..."
#&gt; [1] "+ Aligning xlimits for each subplot..."
#&gt; [1] "+ Checking track heights..."
#&gt; [1] "+ PLOT:: Saving ggplots tracks list ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/plot_tracks/multiview.BST1.UKB.10x.RDS"</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot.nott_2019</span><span class="op">)</span>
<span class="co">#&gt; $`10x`</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-6-1.png" width="960"></p>
</div>
<div id="using-qtl-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#using-qtl-datasets" class="anchor"></a>Using QTL datasets</h2>
<ul>
<li>Plot multiple QTL p-value columns (or really P-value columns from any kind of datset).<br>
</li>
<li>Each QTL dataset will be plotted as a new track.</li>
</ul>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Make fake QTL P-values for the sake a demonstration</span>
<span class="va">finemap_DT</span><span class="op">$</span><span class="va">fake_eQTL.P</span> <span class="op">&lt;-</span> <span class="va">finemap_DT</span><span class="op">$</span><span class="va">P</span>  <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">.9</span>,<span class="fl">.7</span><span class="op">)</span>
<span class="va">finemap_DT</span><span class="op">$</span><span class="va">fake_sQTL.P</span> <span class="op">&lt;-</span> <span class="va">finemap_DT</span><span class="op">$</span><span class="va">P</span>  <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">.8</span>,<span class="fl">.5</span><span class="op">)</span>

<span class="va">trk_plot.qtl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                           LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                           LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                           locus_dir<span class="op">=</span><span class="va">locus_dir</span>,
                           
                           QTL_prefixes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"fake_eQTL."</span>,<span class="st">"fake_sQTL."</span><span class="op">)</span>, 
                           
                           save_plot<span class="op">=</span><span class="cn">F</span>,
                           show_plot<span class="op">=</span><span class="va">show_plot</span>,
                           plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span>
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span></code></pre></div>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Filling NAs in CS cols with 0"
#&gt; [1] "+ Filling NAs in PP cols with 0"
#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."
#&gt; [1] "++ PLOT:: GWAS full window track"
#&gt; [1] "++ PLOT:: GWAS track"
#&gt; [1] "++ PLOT:: fake_eQTL. track"
#&gt; [1] "++ PLOT:: fake_sQTL. track"
#&gt; [1] "++ PLOT:: Merged fine-mapping track"
#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"
#&gt; [1] "++ PLOT:: Adding Gene model track."
#&gt; max_transcripts=1.
#&gt; 15 transcripts from 15 genes returned.
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."
#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</code></pre>
<pre style="max-height: 200px;"><code>#&gt; [1] "+ PLOT:: Get window suffix..."
#&gt; [1] "+ Constructing zoom polygon..."
#&gt; [1] "+ Highlighting zoom origin..."
#&gt; [1] "+ Removing subplot margins..."
#&gt; [1] "+ Reordering tracks..."
#&gt; [1] "+ Ensuring last track shows genomic units..."
#&gt; [1] "+ Aligning xlimits for each subplot..."
#&gt; [1] "+ Checking track heights..."
#&gt; [1] "+ PLOT:: Saving ggplots tracks list ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/plot_tracks/multiview.BST1.UKB.10x.RDS"</code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot.qtl</span><span class="op">)</span>
<span class="co">#&gt; $`10x`</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-7-1.png" width="960"></p>
<details><summary><h1>
Session Info
</h1>
</summary><div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sessioninfo/man/session_info.html">session_info</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre style="max-height: 200px;"><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 3.6.3 (2020-02-29)
#&gt;  os       macOS  10.16                
#&gt;  system   x86_64, darwin15.6.0        
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_GB.UTF-8                 
#&gt;  ctype    en_GB.UTF-8                 
#&gt;  tz       Europe/London               
#&gt;  date     2021-01-19                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────
#&gt;  package              * version  date       lib source               
#&gt;  AnnotationDbi        * 1.48.0   2019-10-29 [1] Bioconductor         
#&gt;  AnnotationFilter     * 1.10.0   2019-10-29 [1] Bioconductor         
#&gt;  askpass                1.1      2019-01-13 [1] CRAN (R 3.6.0)       
#&gt;  assertthat             0.2.1    2019-03-21 [1] CRAN (R 3.6.0)       
#&gt;  backports              1.2.1    2020-12-09 [1] CRAN (R 3.6.2)       
#&gt;  base64enc              0.1-3    2015-07-28 [1] CRAN (R 3.6.0)       
#&gt;  Biobase              * 2.46.0   2019-10-29 [1] Bioconductor         
#&gt;  BiocFileCache          1.10.2   2019-11-08 [1] Bioconductor         
#&gt;  BiocGenerics         * 0.32.0   2019-10-29 [1] Bioconductor         
#&gt;  BiocManager            1.30.10  2019-11-16 [1] CRAN (R 3.6.0)       
#&gt;  BiocParallel           1.20.1   2019-12-21 [1] Bioconductor         
#&gt;  biomaRt                2.42.1   2020-03-26 [1] Bioconductor         
#&gt;  Biostrings             2.54.0   2019-10-29 [1] Bioconductor         
#&gt;  biovizBase             1.34.1   2019-12-04 [1] Bioconductor         
#&gt;  bit                    4.0.4    2020-08-04 [1] CRAN (R 3.6.2)       
#&gt;  bit64                  4.0.5    2020-08-30 [1] CRAN (R 3.6.2)       
#&gt;  bitops                 1.0-6    2013-08-17 [1] CRAN (R 3.6.0)       
#&gt;  blob                   1.2.1    2020-01-20 [1] CRAN (R 3.6.0)       
#&gt;  boot                   1.3-25   2020-04-26 [1] CRAN (R 3.6.2)       
#&gt;  BSgenome               1.54.0   2019-10-29 [1] Bioconductor         
#&gt;  checkmate              2.0.0    2020-02-06 [1] CRAN (R 3.6.0)       
#&gt;  class                  7.3-17   2020-04-26 [1] CRAN (R 3.6.2)       
#&gt;  cli                    2.2.0    2020-11-20 [1] CRAN (R 3.6.2)       
#&gt;  cluster                2.1.0    2019-06-19 [1] CRAN (R 3.6.3)       
#&gt;  colorspace             2.0-0    2020-11-11 [1] CRAN (R 3.6.2)       
#&gt;  crayon                 1.3.4    2017-09-16 [1] CRAN (R 3.6.0)       
#&gt;  curl                   4.3      2019-12-02 [1] CRAN (R 3.6.0)       
#&gt;  data.table             1.13.6   2020-12-30 [1] CRAN (R 3.6.2)       
#&gt;  DBI                    1.1.1    2021-01-15 [1] CRAN (R 3.6.2)       
#&gt;  dbplyr                 2.0.0    2020-11-03 [1] CRAN (R 3.6.2)       
#&gt;  DelayedArray           0.12.3   2020-04-09 [1] Bioconductor         
#&gt;  desc                   1.2.0    2018-05-01 [1] CRAN (R 3.6.0)       
#&gt;  DescTools              0.99.39  2020-12-07 [1] CRAN (R 3.6.2)       
#&gt;  dichromat              2.0-0    2013-01-24 [1] CRAN (R 3.6.0)       
#&gt;  digest                 0.6.27   2020-10-24 [1] CRAN (R 3.6.2)       
#&gt;  dplyr                  1.0.3    2021-01-15 [1] CRAN (R 3.6.2)       
#&gt;  e1071                  1.7-4    2020-10-14 [1] CRAN (R 3.6.2)       
#&gt;  echolocatoR          * 0.1.0    2021-01-18 [1] local                
#&gt;  ellipsis               0.3.1    2020-05-15 [1] CRAN (R 3.6.2)       
#&gt;  EnsDb.Hsapiens.v75   * 2.99.0   2020-11-23 [1] Bioconductor         
#&gt;  ensembldb            * 2.10.2   2019-11-20 [1] Bioconductor         
#&gt;  evaluate               0.14     2019-05-28 [1] CRAN (R 3.6.0)       
#&gt;  Exact                  2.1      2020-10-02 [1] CRAN (R 3.6.2)       
#&gt;  expm                   0.999-6  2021-01-13 [1] CRAN (R 3.6.2)       
#&gt;  fansi                  0.4.2    2021-01-15 [1] CRAN (R 3.6.2)       
#&gt;  farver                 2.0.3    2020-01-16 [1] CRAN (R 3.6.0)       
#&gt;  foreign                0.8-75   2020-01-20 [1] CRAN (R 3.6.3)       
#&gt;  Formula                1.2-4    2020-10-16 [1] CRAN (R 3.6.2)       
#&gt;  fs                     1.5.0    2020-07-31 [1] CRAN (R 3.6.2)       
#&gt;  generics               0.1.0    2020-10-31 [1] CRAN (R 3.6.2)       
#&gt;  GenomeInfoDb         * 1.22.1   2020-03-27 [1] Bioconductor         
#&gt;  GenomeInfoDbData       1.2.2    2020-11-16 [1] Bioconductor         
#&gt;  GenomicAlignments      1.22.1   2019-11-12 [1] Bioconductor         
#&gt;  GenomicFeatures      * 1.38.2   2020-02-15 [1] Bioconductor         
#&gt;  GenomicRanges        * 1.38.0   2019-10-29 [1] Bioconductor         
#&gt;  GGally                 2.1.0    2021-01-06 [1] CRAN (R 3.6.2)       
#&gt;  ggbio                  1.34.0   2020-11-23 [1] bioc_git2r (@15346db)
#&gt;  ggplot2              * 3.3.3    2020-12-30 [1] CRAN (R 3.6.2)       
#&gt;  ggrepel                0.9.1    2021-01-15 [1] CRAN (R 3.6.2)       
#&gt;  gld                    2.6.2    2020-01-08 [1] CRAN (R 3.6.0)       
#&gt;  glue                   1.4.2    2020-08-27 [1] CRAN (R 3.6.2)       
#&gt;  graph                  1.64.0   2019-10-29 [1] Bioconductor         
#&gt;  gridExtra              2.3      2017-09-09 [1] CRAN (R 3.6.0)       
#&gt;  gtable                 0.3.0    2019-03-25 [1] CRAN (R 3.6.0)       
#&gt;  Hmisc                  4.4-2    2020-11-29 [1] CRAN (R 3.6.2)       
#&gt;  hms                    1.0.0    2021-01-13 [1] CRAN (R 3.6.3)       
#&gt;  htmlTable              2.1.0    2020-09-16 [1] CRAN (R 3.6.2)       
#&gt;  htmltools              0.5.1    2021-01-12 [1] CRAN (R 3.6.2)       
#&gt;  htmlwidgets            1.5.3    2020-12-10 [1] CRAN (R 3.6.2)       
#&gt;  httr                   1.4.2    2020-07-20 [1] CRAN (R 3.6.2)       
#&gt;  IRanges              * 2.20.2   2020-01-13 [1] Bioconductor         
#&gt;  jpeg                   0.1-8.1  2019-10-24 [1] CRAN (R 3.6.0)       
#&gt;  knitr                  1.30     2020-09-22 [1] CRAN (R 3.6.2)       
#&gt;  labeling               0.4.2    2020-10-20 [1] CRAN (R 3.6.2)       
#&gt;  lattice                0.20-41  2020-04-02 [1] CRAN (R 3.6.2)       
#&gt;  latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 3.6.0)       
#&gt;  lazyeval               0.2.2    2019-03-15 [1] CRAN (R 3.6.0)       
#&gt;  lifecycle              0.2.0    2020-03-06 [1] CRAN (R 3.6.0)       
#&gt;  lmom                   2.8      2019-03-12 [1] CRAN (R 3.6.0)       
#&gt;  magrittr               2.0.1    2020-11-17 [1] CRAN (R 3.6.2)       
#&gt;  mapproj                1.2.7    2020-02-03 [1] CRAN (R 3.6.0)       
#&gt;  maps                   3.3.0    2018-04-03 [1] CRAN (R 3.6.0)       
#&gt;  MASS                   7.3-53   2020-09-09 [1] CRAN (R 3.6.2)       
#&gt;  Matrix                 1.3-2    2021-01-06 [1] CRAN (R 3.6.2)       
#&gt;  matrixStats            0.57.0   2020-09-25 [1] CRAN (R 3.6.2)       
#&gt;  memoise                1.1.0    2017-04-21 [1] CRAN (R 3.6.0)       
#&gt;  munsell                0.5.0    2018-06-12 [1] CRAN (R 3.6.0)       
#&gt;  mvtnorm                1.1-1    2020-06-09 [1] CRAN (R 3.6.2)       
#&gt;  nnet                   7.3-14   2020-04-26 [1] CRAN (R 3.6.2)       
#&gt;  openssl                1.4.3    2020-09-18 [1] CRAN (R 3.6.2)       
#&gt;  OrganismDbi            1.28.0   2019-10-29 [1] Bioconductor         
#&gt;  pals                   1.6      2019-12-04 [1] CRAN (R 3.6.0)       
#&gt;  patchwork              1.1.1    2020-12-17 [1] CRAN (R 3.6.2)       
#&gt;  pillar                 1.4.7    2020-11-20 [1] CRAN (R 3.6.2)       
#&gt;  pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 3.6.0)       
#&gt;  pkgdown                1.6.1    2020-09-12 [1] CRAN (R 3.6.2)       
#&gt;  plyr                   1.8.6    2020-03-03 [1] CRAN (R 3.6.0)       
#&gt;  png                    0.1-7    2013-12-03 [1] CRAN (R 3.6.0)       
#&gt;  prettyunits            1.1.1    2020-01-24 [1] CRAN (R 3.6.0)       
#&gt;  progress               1.2.2    2019-05-16 [1] CRAN (R 3.6.0)       
#&gt;  ProtGenerics           1.18.0   2019-10-29 [1] Bioconductor         
#&gt;  purrr                  0.3.4    2020-04-17 [1] CRAN (R 3.6.2)       
#&gt;  R6                     2.5.0    2020-10-28 [1] CRAN (R 3.6.2)       
#&gt;  ragg                   0.4.1    2021-01-11 [1] CRAN (R 3.6.2)       
#&gt;  rappdirs               0.3.1    2016-03-28 [1] CRAN (R 3.6.0)       
#&gt;  RBGL                   1.62.1   2019-10-30 [1] Bioconductor         
#&gt;  RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 3.6.0)       
#&gt;  Rcpp                   1.0.6    2021-01-15 [1] CRAN (R 3.6.2)       
#&gt;  RCurl                  1.98-1.2 2020-04-18 [1] CRAN (R 3.6.2)       
#&gt;  reshape                0.8.8    2018-10-23 [1] CRAN (R 3.6.0)       
#&gt;  reshape2               1.4.4    2020-04-09 [1] CRAN (R 3.6.2)       
#&gt;  rlang                  0.4.10   2020-12-30 [1] CRAN (R 3.6.2)       
#&gt;  rmarkdown              2.6      2020-12-14 [1] CRAN (R 3.6.2)       
#&gt;  rootSolve              1.8.2.1  2020-04-27 [1] CRAN (R 3.6.2)       
#&gt;  rpart                  4.1-15   2019-04-12 [1] CRAN (R 3.6.3)       
#&gt;  rprojroot              2.0.2    2020-11-15 [1] CRAN (R 3.6.3)       
#&gt;  Rsamtools              2.2.3    2020-02-23 [1] Bioconductor         
#&gt;  RSQLite                2.2.2    2021-01-08 [1] CRAN (R 3.6.2)       
#&gt;  rstudioapi             0.13     2020-11-12 [1] CRAN (R 3.6.2)       
#&gt;  rtracklayer            1.46.0   2019-10-29 [1] Bioconductor         
#&gt;  S4Vectors            * 0.24.4   2020-04-09 [1] Bioconductor         
#&gt;  scales                 1.1.1    2020-05-11 [1] CRAN (R 3.6.2)       
#&gt;  sessioninfo            1.1.1    2018-11-05 [1] CRAN (R 3.6.0)       
#&gt;  stringi                1.5.3    2020-09-09 [1] CRAN (R 3.6.2)       
#&gt;  stringr                1.4.0    2019-02-10 [1] CRAN (R 3.6.0)       
#&gt;  SummarizedExperiment   1.16.1   2019-12-19 [1] Bioconductor         
#&gt;  survival               3.2-7    2020-09-28 [1] CRAN (R 3.6.2)       
#&gt;  systemfonts            0.3.2    2020-09-29 [1] CRAN (R 3.6.2)       
#&gt;  textshaping            0.2.1    2020-11-13 [1] CRAN (R 3.6.2)       
#&gt;  tibble                 3.0.5    2021-01-15 [1] CRAN (R 3.6.2)       
#&gt;  tidyr                  1.1.2    2020-08-27 [1] CRAN (R 3.6.2)       
#&gt;  tidyselect             1.1.0    2020-05-11 [1] CRAN (R 3.6.2)       
#&gt;  VariantAnnotation      1.32.0   2019-10-29 [1] Bioconductor         
#&gt;  vctrs                  0.3.6    2020-12-17 [1] CRAN (R 3.6.2)       
#&gt;  withr                  2.4.0    2021-01-16 [1] CRAN (R 3.6.2)       
#&gt;  xfun                   0.20     2021-01-06 [1] CRAN (R 3.6.2)       
#&gt;  XML                    3.99-0.3 2020-01-20 [1] CRAN (R 3.6.0)       
#&gt;  XVector                0.26.0   2019-10-29 [1] Bioconductor         
#&gt;  yaml                   2.2.1    2020-02-01 [1] CRAN (R 3.6.0)       
#&gt;  zlibbioc               1.32.0   2019-10-29 [1] Bioconductor         
#&gt; 
#&gt; [1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library</code></pre>
<details></details></details>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
