<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>echolocatoR: plotting vignette • echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="echolocatoR: plotting vignette">
<meta property="og:description" content="echolocatoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QTL_pipeline_vignette.html">echolocatoR: QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/full_pipeline_vignette.html">echolocatoR: full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">echolocatoR: plotting vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>echolocatoR: plotting vignette</h1>
                        <h4 class="author">Brian M. Schilder</h4>
            
            <h4 class="date">Most recent update:<br> 2020-07-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/vignettes/plotting_vignette.Rmd"><code>vignettes/plotting_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>plotting_vignette.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">echolocatoR</span>)
<span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span>
<span class="co">#&gt;   method from   </span>
<span class="co">#&gt;   +.gg   ggplot2</span></pre></body></html></div>
<div id="plotting-only" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-only" class="anchor"></a>Plotting only</h1>
<p><em>echolocatoR</em> contains various functions that can be used separately<br>
from the comprehensive <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> pipeline.</p>
<p>Generate a multiview plot using <code><a href="../reference/GGBIO.plot.html">GGBIO.plot()</a></code>.</p>
<ul>
<li>You can mix and match different tracks and annotations using the different arguments (see <code><a href="../reference/GGBIO.plot.html">?GGBIO.plot</a></code> for details).<br>
</li>
<li>Zoom in/out by adjusting the <code>plot.window</code> size (in units of basepairs).<br>
The plot is centered on the lead/index SNP.<br>
</li>
<li>
<code><a href="../reference/GGBIO.plot.html">GGBIO.plot()</a></code> returns a <code>ggbio</code> track object.
<ul>
<li>Further modify this object using <code>ggplot2</code> functions like <code>+ theme()</code>.</li>
<li>The modifications will be applied to all tracks at once.<br>
</li>
</ul>
</li>
<li>Save a high-resolution <em>.png</em> of the plot by setting <code>save_plot=T</code>
<ul>
<li>Further increase resolution by adjusting the <code>dpi</code> argument (<em>default=300</em>).</li>
<li>Adjust the <code>height</code> and <code>width</code> of the saved plot using these respective arguments.</li>
<li>The plot will be automatically saved in the locus-specific directory as:<br>
*multiview_<locus>_&lt;plot.window&gt;.png*.</locus>
</li>
</ul>
</li>
</ul>
<div id="load-example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-example-data" class="anchor"></a>Load example data</h2>
<p>Load example dataset of the results from fine-mapping the BST1 locus with <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code>. Original data comes from the recent Nalls et al. (2019) Parkinson’s disease GWAS (see <code><a href="../reference/BST1.html">?BST1</a></code> for details).</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"BST1"</span>); <span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"LD_matrix"</span>); <span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"locus_dir"</span>);
<span class="no">locus_dir</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"~/Desktop"</span>,<span class="no">locus_dir</span>)
<span class="no">finemap_DT</span> <span class="kw">&lt;-</span> <span class="no">BST1</span>
<span class="no">show_plot</span> <span class="kw">&lt;-</span> <span class="no">T</span></pre></body></html></div>
</div>
<div id="using-no-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#using-no-annotations" class="anchor"></a>Using NO annotations</h2>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">trk_plot</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/GGBIO.plot.html">GGBIO.plot</a></span>(<span class="kw">finemap_dat</span><span class="kw">=</span><span class="no">finemap_DT</span>,
                       <span class="kw">LD_matrix</span><span class="kw">=</span><span class="no">LD_matrix</span>,
                       <span class="kw">locus_dir</span><span class="kw">=</span><span class="no">locus_dir</span>,
                       <span class="kw">XGR_libnames</span><span class="kw">=</span><span class="kw">NULL</span>,
                       <span class="kw">save_plot</span><span class="kw">=</span><span class="no">F</span>,
                       <span class="kw">show_plot</span><span class="kw">=</span><span class="no">show_plot</span>)
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "GGBIO:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "LD Matrix dimensions: 95 x 95"</span>
<span class="co">#&gt; [1] "Extracting LD subset for lead SNP: rs4698412"</span>
<span class="co">#&gt; [1] "++ GGBIO:: GWAS track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: ABF track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: FINEMAP track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: POLYFUN_SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: mean track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: Gene Model Track"</span>
<span class="co">#&gt; [1] "+ Annotating at transcript-level."</span>
<span class="co">#&gt; Fetching data...</span>
<span class="co">#&gt; OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; Warning: Removed 6131 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span></pre></body></html></div>
<p><img src="plotting_vignette_files/figure-html/trk_plot-1.png" width="960"></p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">trk_plot</span> <span class="kw">&lt;-</span> <span class="no">trk_plot</span> + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">strip.background</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span>(<span class="kw">fill</span><span class="kw">=</span><span class="st">"gray"</span>),
                             <span class="kw">text</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">size</span><span class="kw">=</span><span class="fl">3</span>, <span class="kw">color</span><span class="kw">=</span><span class="st">"blue"</span>))</pre></body></html></div>
</div>
<div id="using-only-xgr-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#using-only-xgr-annotations" class="anchor"></a>Using only XGR annotations</h2>
<p>Whenever you use annotation arguments (e.g. <code>XGR_libnames</code>,<code>Roadmap</code>,<code>Nott_epigenome</code>) the annotations that overlap with your locus will automatically be saved as <code>GRanges</code> objects<br>
in a locus-specific subdirectory:<br><em>results/<dataset_type>/<dataset_name>/<locus>/annotation</locus></dataset_name></dataset_type></em></p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">trk_plot.xgr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/GGBIO.plot.html">GGBIO.plot</a></span>(<span class="kw">finemap_dat</span><span class="kw">=</span><span class="no">finemap_DT</span>,
                           <span class="kw">LD_matrix</span><span class="kw">=</span><span class="no">LD_matrix</span>,
                           <span class="kw">locus_dir</span><span class="kw">=</span><span class="no">locus_dir</span>,
                           <span class="kw">XGR_libnames</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ENCODE_TFBS_ClusteredV3_CellTypes"</span>),
                           <span class="kw">save_plot</span><span class="kw">=</span><span class="no">F</span>,
                           <span class="kw">show_plot</span><span class="kw">=</span><span class="no">show_plot</span>)
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "GGBIO:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "LD Matrix dimensions: 95 x 95"</span>
<span class="co">#&gt; [1] "Extracting LD subset for lead SNP: rs4698412"</span>
<span class="co">#&gt; [1] "++ GGBIO:: GWAS track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: ABF track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: FINEMAP track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: POLYFUN_SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: mean track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: Gene Model Track"</span>
<span class="co">#&gt; [1] "+ Annotating at transcript-level."</span>
<span class="co">#&gt; [1] "++ GGBIO:: XGR Tracks"</span>
<span class="co">#&gt; Start at 2020-07-09 18:29:56</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'ENCODE_TFBS_ClusteredV3_CellTypes' (from http://galahad.well.ox.ac.uk/bigdata/ENCODE_TFBS_ClusteredV3_CellTypes.RData) has been loaded into the working environment (at 2020-07-09 18:30:15)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; End at 2020-07-09 18:30:15</span>
<span class="co">#&gt; Runtime in total is: 19 secs</span>
<span class="co">#&gt; [1] "+ dat2 already in GRanges format"</span>
<span class="co">#&gt; Warning in .replace_seqlevels_style(x_seqlevels, value): found more than one</span>
<span class="co">#&gt; best sequence renaming map compatible with seqname style "NCBI" for this object,</span>
<span class="co">#&gt; using the first one</span>
<span class="co">#&gt; 8702 query SNP(s) detected with reference overlap.</span>
<span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span>
<span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which will</span>
<span class="co">#&gt; replace the existing scale.</span>
<span class="co">#&gt; Warning: Removed 6131 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span></pre></body></html></div>
<p><img src="plotting_vignette_files/figure-html/trk_plot.xgr-1.png" width="960"></p>
</div>
<div id="using-only-roadmap-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#using-only-roadmap-annotations" class="anchor"></a>Using only Roadmap annotations</h2>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">trk_plot.roadmap</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/GGBIO.plot.html">GGBIO.plot</a></span>(<span class="kw">finemap_dat</span><span class="kw">=</span><span class="no">finemap_DT</span>,
                               <span class="kw">LD_matrix</span><span class="kw">=</span><span class="no">LD_matrix</span>,
                               <span class="kw">locus_dir</span><span class="kw">=</span><span class="no">locus_dir</span>,
                               <span class="kw">XGR_libnames</span><span class="kw">=</span><span class="kw">NULL</span>,
                               <span class="kw">Roadmap</span><span class="kw">=</span><span class="no">T</span>,
                               <span class="kw">Roadmap_query</span><span class="kw">=</span><span class="st">"monocyte"</span>,
                               <span class="kw">save_plot</span><span class="kw">=</span><span class="no">F</span>,
                               <span class="kw">show_plot</span><span class="kw">=</span><span class="no">show_plot</span>)
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "GGBIO:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "LD Matrix dimensions: 95 x 95"</span>
<span class="co">#&gt; [1] "Extracting LD subset for lead SNP: rs4698412"</span>
<span class="co">#&gt; [1] "++ GGBIO:: GWAS track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: ABF track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: FINEMAP track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: POLYFUN_SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: mean track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: Gene Model Track"</span>
<span class="co">#&gt; [1] "+ Annotating at transcript-level."</span>
<span class="co">#&gt; [1] "+ GGBIO:: Creating Roadmap track"</span>
<span class="co">#&gt; [1] "+ Saved annotation file detected. Loading..."</span>
<span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span>
<span class="co">#&gt; Warning: Removed 6131 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span>
<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>
<span class="co">#&gt; Warning: Removed 2 rows containing missing values (position_stack).</span>
<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>
<span class="co">#&gt; Warning: Removed 2 rows containing missing values (position_stack).</span></pre></body></html></div>
<p><img src="plotting_vignette_files/figure-html/trk_plot.roadmap-1.png" width="960"></p>
</div>
<div id="using-only-nott_2019-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#using-only-nott_2019-annotations" class="anchor"></a>Using only Nott_2019 annotations</h2>
<ul>
<li>Query and plot brain cell type-specific epigenomic assays from Nott et al. (2019)<br>
(see <code><a href="../reference/NOTT_2019.bigwig_metadata.html">?NOTT_2019.bigwig_metadata</a></code> for details).</li>
</ul>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">trk_plot.nott_2019</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/GGBIO.plot.html">GGBIO.plot</a></span>(<span class="kw">finemap_dat</span><span class="kw">=</span><span class="no">finemap_DT</span>,
                                 <span class="kw">LD_matrix</span><span class="kw">=</span><span class="no">LD_matrix</span>,
                                 <span class="kw">locus_dir</span><span class="kw">=</span><span class="no">locus_dir</span>,
                                 <span class="kw">XGR_libnames</span><span class="kw">=</span><span class="kw">NULL</span>,
                                 <span class="kw">Nott_epigenome</span><span class="kw">=</span><span class="no">T</span>,
                                 <span class="kw">Nott_binwidth</span> <span class="kw">=</span> <span class="fl">100</span>,
                                 <span class="kw">Nott_regulatory_rects</span> <span class="kw">=</span> <span class="no">T</span>,
                                 <span class="kw">Nott_show_placseq</span> <span class="kw">=</span> <span class="no">T</span>,
                                 <span class="kw">save_plot</span><span class="kw">=</span><span class="no">F</span>,
                                 <span class="kw">show_plot</span><span class="kw">=</span><span class="no">show_plot</span>)
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "GGBIO:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "LD Matrix dimensions: 95 x 95"</span>
<span class="co">#&gt; [1] "Extracting LD subset for lead SNP: rs4698412"</span>
<span class="co">#&gt; [1] "++ GGBIO:: GWAS track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: ABF track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: FINEMAP track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: POLYFUN_SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: mean track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: Gene Model Track"</span>
<span class="co">#&gt; [1] "+ Annotating at transcript-level."</span>
<span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span>
<span class="co">#&gt; [1] "Importing Astrocyte enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Astrocyte promoters ..."</span>
<span class="co">#&gt; [1] "Importing Neuronal enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Neuronal promoters ..."</span>
<span class="co">#&gt; [1] "Importing Oligo enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Oligo promoters ..."</span>
<span class="co">#&gt; [1] "Importing Microglia enhancers ..."</span>
<span class="co">#&gt; [1] "Importing Microglia promoters ..."</span>
<span class="co">#&gt; [1] "+ Converting to GRanges."</span>
<span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span>
<span class="co">#&gt; replace the existing scale.</span>
<span class="co">#&gt; Scale for 'y' is already present. Adding another scale for 'y', which will</span>
<span class="co">#&gt; replace the existing scale.</span>
<span class="co">#&gt; [1] "++ Nott sn-epigenomics:: Returning PLAC-seq track."</span>
<span class="co">#&gt; Warning: Removed 6131 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span></pre></body></html></div>
<p><img src="plotting_vignette_files/figure-html/trk_plot.nott_2019-1.png" width="960"></p>
</div>
<div id="using-qtl-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#using-qtl-datasets" class="anchor"></a>Using QTL datasets</h2>
<ul>
<li>Plot multiple QTL p-value columns (or really P-value columns from any kind of datset).<br>
</li>
<li>Each QTL dataset will be plotted as a new track.</li>
</ul>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="co"># Make fake QTL P-values for the sake a demonstration</span>
<span class="no">finemap_DT</span>$<span class="no">fake_eQTL.P</span> <span class="kw">&lt;-</span> <span class="no">finemap_DT</span>$<span class="no">P</span>  * <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">.9</span>,<span class="fl">.7</span>)
<span class="no">finemap_DT</span>$<span class="no">fake_sQTL.P</span> <span class="kw">&lt;-</span> <span class="no">finemap_DT</span>$<span class="no">P</span>  * <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">.8</span>,<span class="fl">.5</span>)

<span class="no">trk_plot.qtl</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/GGBIO.plot.html">GGBIO.plot</a></span>(<span class="kw">finemap_dat</span><span class="kw">=</span><span class="no">finemap_DT</span>,
                           <span class="kw">LD_matrix</span><span class="kw">=</span><span class="no">LD_matrix</span>,
                           <span class="kw">locus_dir</span><span class="kw">=</span><span class="no">locus_dir</span>,
                           <span class="kw">QTL_prefixes</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"fake_eQTL."</span>,<span class="st">"fake_sQTL."</span>),
                           <span class="kw">XGR_libnames</span><span class="kw">=</span><span class="kw">NULL</span>,
                           <span class="kw">save_plot</span><span class="kw">=</span><span class="no">F</span>,
                           <span class="kw">show_plot</span><span class="kw">=</span><span class="no">show_plot</span>)
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "GGBIO:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "LD Matrix dimensions: 95 x 95"</span>
<span class="co">#&gt; [1] "Extracting LD subset for lead SNP: rs4698412"</span>
<span class="co">#&gt; [1] "++ GGBIO:: GWAS track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: fake_eQTL. track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: fake_sQTL. track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: ABF track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: FINEMAP track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: POLYFUN_SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: mean track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: Gene Model Track"</span>
<span class="co">#&gt; [1] "+ Annotating at transcript-level."</span>
<span class="co">#&gt; Fetching data...</span>
<span class="co">#&gt; OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; Warning: Removed 6131 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Removed 6131 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Removed 6131 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span></pre></body></html></div>
<p><img src="plotting_vignette_files/figure-html/trk_plot.QTL-1.png" width="960"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
