<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Plotting vignette • echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Plotting vignette">
<meta property="og:description" content="echolocatoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/full_pipeline_vignette.html">Full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="plotting_vignette_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Plotting vignette</h1>
                        <h4 class="author">Brian M. Schilder</h4>
            
            <h4 class="date">Most recent update:<br> 2021-06-22</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/vignettes/plotting_vignette.Rmd"><code>vignettes/plotting_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>plotting_vignette.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echolocatoR">echolocatoR</a></span><span class="op">)</span> 
<span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span>
<span class="co">#&gt;   method from   </span>
<span class="co">#&gt;   +.gg   ggplot2</span></code></pre></div>
<div id="plotting-loci-with-echolocator" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-loci-with-echolocator" class="anchor"></a>Plotting loci with <em>echolocatoR</em>
</h1>
<p><em>echolocatoR</em> contains various functions that can be used separately<br>
from the comprehensive <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> pipeline.</p>
<p>Generate a multi-view plot of a given locus using <code><a href="../reference/PLOT.locus.html">PLOT.locus()</a></code>.</p>
<ul>
<li>You can mix and match different tracks and annotations using the different arguments (see <code><a href="../reference/PLOT.locus.html">?PLOT.locus</a></code> for details).</li>
</ul>
<p>The plot is centered on the lead/index SNP. If a list is supplied to plot.xoom * <code><a href="../reference/PLOT.locus.html">PLOT.locus()</a></code> returns a series of <code>ggplot</code> objects bound together with <a href="https://patchwork.data-imaginist.com"><code>patchwork</code></a>. One can further modify this object using <code>ggplot2</code> functions like <code>+ theme()</code>. + The modifications will be applied to all tracks at once.</p>
<ul>
<li>Save a high-resolution versions the plot by setting <code>save_plot=T</code>.
<ul>
<li>Further increase resolution by adjusting the <code>dpi</code> argument (<em>default=300</em>).</li>
<li>Files are saved in <em>jpg</em> format by default, but users can specify their preferred file format (e.g. <code>file_format="png"</code>)</li>
<li>Adjust the <code>height</code> and <code>width</code> of the saved plot using these respective arguments.</li>
<li>The plot will be automatically saved in the locus-specific directory as:<br>
*multiview_<locus>_&lt;plot.zoom&gt;.jpg*.</locus>
</li>
</ul>
</li>
</ul>
<div id="load-example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-example-data" class="anchor"></a>Load example data</h2>
<p>Load example dataset of the results from fine-mapping the BST1 locus with <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code>. Original data comes from the recent Nalls et al. (2019) Parkinson’s disease GWAS (see <code><a href="../reference/BST1.html">?BST1</a></code> for details).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"BST1"</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"LD_matrix"</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"locus_dir"</span><span class="op">)</span>;
<span class="va">locus_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"~/Desktop"</span>,<span class="va">locus_dir</span><span class="op">)</span>
<span class="va">finemap_DT</span> <span class="op">&lt;-</span> <span class="va">BST1</span>  
<span class="va">LD_matrix</span> <span class="op">&lt;-</span> <span class="va">BST1_LD_matrix</span>
<span class="va">LD_reference</span> <span class="op">&lt;-</span> <span class="st">"UKB"</span> <span class="co"># Used for naming saved plots</span>
<span class="va">plot.zoom</span> <span class="op">=</span> <span class="st">"10x"</span>
<span class="va">show_plot</span> <span class="op">&lt;-</span> <span class="cn">F</span>


<span class="co"># locus &lt;- "DNAH17"</span>
<span class="co"># locus_dir &lt;- file.path("~/Desktop/Fine_Mapping/Data/GWAS/Nalls23andMe_2019",locus)</span>
<span class="co"># LD_matrix &lt;- readRDS(file.path(locus_dir,"plink/UKB_LD.RDS"))</span>
<span class="co"># finemap_DT &lt;- data.table::fread(file.path(locus_dir,"Multi-finemap/Multi-finemap_results.txt"))</span>
<span class="co"># finemap_DT &lt;- find_consensus_SNPs(finemap_DT) </span>
<span class="co"># finemap_DT &lt;- update_CS_cols(finemap_DT)</span></code></pre></div>
</div>
<div id="full-window" class="section level2">
<h2 class="hasAnchor">
<a href="#full-window" class="anchor"></a>Full window</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                       LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                       LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                       locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                       save_plot<span class="op">=</span><span class="cn">F</span>,
                       show_plot<span class="op">=</span><span class="va">show_plot</span>,
                       plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span> 
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span>
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS full window track"</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS track"</span>
<span class="co">#&gt; [1] "++ PLOT:: Merged fine-mapping track"</span>
<span class="co">#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"</span>
<span class="co">#&gt; Warning: The `.dots` argument of `group_by()` is deprecated as of dplyr 1.0.0.</span>
<span class="co">#&gt; [1] "++ PLOT:: Adding Gene model track."</span>
<span class="co">#&gt; max_transcripts=1.</span>
<span class="co">#&gt; 15 transcripts from 15 genes returned.</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."</span>
<span class="co">#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</span>
<span class="co">#&gt; [1] "+ PLOT:: Get window suffix..."</span>
<span class="co">#&gt; [1] "+ Constructing zoom polygon..."</span>
<span class="co">#&gt; [1] "+ Highlighting zoom origin..."</span>
<span class="co">#&gt; [1] "+ Removing subplot margins..."</span>
<span class="co">#&gt; [1] "+ Reordering tracks..."</span>
<span class="co">#&gt; [1] "+ Ensuring last track shows genomic units..."</span>
<span class="co">#&gt; [1] "+ Aligning xlimits for each subplot..."</span>
<span class="co">#&gt; [1] "+ Checking track heights..."</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot</span><span class="op">)</span>
<span class="co">#&gt; $`10x`</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-2-1.png" width="960"></p>
</div>
<div id="at-multiple-zooms" class="section level2">
<h2 class="hasAnchor">
<a href="#at-multiple-zooms" class="anchor"></a>At multiple zooms</h2>
<ul>
<li>You can easily generate the same locus plot at multiple zoomed in views by supplying a list to <code>plot.zoom</code>.<br>
</li>
<li>This list can be composed of zoom multipliers (e.g. <code><a href="https://rdrr.io/r/base/c.html">c("1x", "2x")</a></code>), window widths in units of basepairs (e.g. <code><a href="https://rdrr.io/r/base/c.html">c(5000, 1500)</a></code>), or a mixture of both (e.g. <code><a href="https://rdrr.io/r/base/c.html">c("1x","4x", 5000, 2000)</a></code>).</li>
<li>Each zoom view will be saved individually with its respective scale as the suffix (e.g. <code>multiview.BST1.UKB.4x.jpg</code>).<br>
</li>
<li>Each zoom view is stored as a named item within the returned list.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot_zooms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                             LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                             LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                             locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                             save_plot<span class="op">=</span><span class="cn">F</span>,
                             show_plot<span class="op">=</span><span class="va">show_plot</span>,
                             plot.zoom <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1x"</span>,<span class="st">"5x"</span>,<span class="st">"10x"</span><span class="op">)</span><span class="op">)</span> 
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span>
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS full window track"</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS track"</span>
<span class="co">#&gt; [1] "++ PLOT:: Merged fine-mapping track"</span>
<span class="co">#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"</span>
<span class="co">#&gt; [1] "++ PLOT:: Adding Gene model track."</span>
<span class="co">#&gt; max_transcripts=1.</span>
<span class="co">#&gt; 15 transcripts from 15 genes returned.</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."</span>
<span class="co">#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 1x +&lt;+&lt;+&lt;+&lt;+</span>
<span class="co">#&gt; [1] "+ PLOT:: Get window suffix..."</span>
<span class="co">#&gt; [1] "+ PLOT:: Removing GWAS full window track @ zoom=1x"</span>
<span class="co">#&gt; [1] "+ Removing subplot margins..."</span>
<span class="co">#&gt; [1] "+ Reordering tracks..."</span>
<span class="co">#&gt; [1] "+ Ensuring last track shows genomic units..."</span>
<span class="co">#&gt; [1] "+ Aligning xlimits for each subplot..."</span>
<span class="co">#&gt; [1] "+ Checking track heights..."</span>
<span class="co">#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 5x +&lt;+&lt;+&lt;+&lt;+</span>
<span class="co">#&gt; [1] "+ PLOT:: Get window suffix..."</span>
<span class="co">#&gt; [1] "+ Constructing zoom polygon..."</span>
<span class="co">#&gt; [1] "+ Highlighting zoom origin..."</span>
<span class="co">#&gt; [1] "+ Removing subplot margins..."</span>
<span class="co">#&gt; [1] "+ Reordering tracks..."</span>
<span class="co">#&gt; [1] "+ Ensuring last track shows genomic units..."</span>
<span class="co">#&gt; [1] "+ Aligning xlimits for each subplot..."</span>
<span class="co">#&gt; [1] "+ Checking track heights..."</span>
<span class="co">#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</span>
<span class="co">#&gt; [1] "+ PLOT:: Get window suffix..."</span>
<span class="co">#&gt; [1] "+ Constructing zoom polygon..."</span>
<span class="co">#&gt; [1] "+ Highlighting zoom origin..."</span>
<span class="co">#&gt; [1] "+ Removing subplot margins..."</span>
<span class="co">#&gt; [1] "+ Reordering tracks..."</span>
<span class="co">#&gt; [1] "+ Ensuring last track shows genomic units..."</span>
<span class="co">#&gt; [1] "+ Aligning xlimits for each subplot..."</span>
<span class="co">#&gt; [1] "+ Checking track heights..."</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">trk_plot_zooms</span><span class="op">)</span> <span class="co"># Get zoom view names</span>
<span class="co">#&gt; [1] "1x"  "5x"  "10x"</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot_zooms</span><span class="op">)</span>
<span class="co">#&gt; $`1x`</span>
<span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span>
<span class="co">#&gt; increasing max.overlaps</span>

<span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span>
<span class="co">#&gt; increasing max.overlaps</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-3-1.png" width="960"></p>
<pre><code>#&gt; 
#&gt; $`5x`
#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider
#&gt; increasing max.overlaps
#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider
#&gt; increasing max.overlaps</code></pre>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-3-2.png" width="960"></p>
<pre><code>#&gt; 
#&gt; $`10x`</code></pre>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-3-3.png" width="960"></p>
</div>
<div id="return-as-list" class="section level2">
<h2 class="hasAnchor">
<a href="#return-as-list" class="anchor"></a>Return as list</h2>
<ul>
<li>For even further control over each track of the multi-view plot, specify <code><a href="../reference/PLOT.locus.html">PLOT.locus(..., return_list=T)</a></code> to instead return a named list (nested within each zoom view list item) of <code>ggplot</code> objects which can each be modified individually.</li>
<li>Once you’ve made your modifications, you can then bind this list of plots back together with <code><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">patchwork::wrap_plots(tracks_list, ncol = 1)</a></code>.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                             LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                             LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                             locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                             save_plot<span class="op">=</span><span class="cn">F</span>,
                             show_plot<span class="op">=</span><span class="va">show_plot</span>,
                             plot.zoom<span class="op">=</span><span class="va">plot.zoom</span>,
                             return_list<span class="op">=</span><span class="cn">T</span><span class="op">)</span>  
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span>
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS full window track"</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS track"</span>
<span class="co">#&gt; [1] "++ PLOT:: Merged fine-mapping track"</span>
<span class="co">#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"</span>
<span class="co">#&gt; [1] "++ PLOT:: Adding Gene model track."</span>
<span class="co">#&gt; max_transcripts=1.</span>
<span class="co">#&gt; 15 transcripts from 15 genes returned.</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."</span>
<span class="co">#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</span>
<span class="co">#&gt; [1] "+ PLOT:: Get window suffix..."</span>
<span class="co">#&gt; [1] "+ Constructing zoom polygon..."</span>
<span class="co">#&gt; [1] "+ Highlighting zoom origin..."</span>
<span class="co">#&gt; [1] "+ Removing subplot margins..."</span>
<span class="co">#&gt; [1] "+ Reordering tracks..."</span>
<span class="co">#&gt; [1] "+ Ensuring last track shows genomic units..."</span>
<span class="co">#&gt; [1] "+ Aligning xlimits for each subplot..."</span>
<span class="co">#&gt; [1] "+ Checking track heights..."</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">view1_list</span> <span class="op">&lt;-</span> <span class="va">trk_plot_list</span><span class="op">[[</span><span class="va">plot.zoom</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">view1_list</span><span class="op">)</span> <span class="co"># Get track names from a particular zoom view</span>
<span class="co">#&gt; [1] "GWAS full window" "zoom_polygon"     "Genes"            "GWAS"            </span>
<span class="co">#&gt; [5] "Fine-mapping"</span></code></pre></div>
<p>Modify a specific tracks within a view.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Modify your selected track</span>
<span class="va">modified_track</span> <span class="op">&lt;-</span> <span class="va">view1_list</span><span class="op">$</span><span class="va">GWAS</span> <span class="op">+</span> 
                      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Modified GWAS"</span><span class="op">)</span> <span class="op">+</span> 
                      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_dark</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
                      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Put it back into your track list</span>
<span class="va">view1_list</span><span class="op">[[</span><span class="st">"GWAS"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">modified_track</span>
<span class="co"># Remove a plot you don't want</span>
<span class="va">view1_list</span><span class="op">[[</span><span class="st">"Genes"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="co"># Specify the relative heights of each track (make sure it matches your new # of plots!)</span>
<span class="va">track_heights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.3</span>,<span class="fl">.1</span>,<span class="fl">.3</span>,<span class="fl">1</span><span class="op">)</span>

<span class="co"># Bind them together and plot</span>
<span class="va">fused_plot</span> <span class="op">&lt;-</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">wrap_plots</a></span><span class="op">(</span><span class="va">view1_list</span>, 
                                    heights <span class="op">=</span> <span class="va">track_heights</span>,
                                    ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">fused_plot</span><span class="op">)</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/modify%20track-1.png" width="960"></p>
</div>
<div id="using-xgr-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#using-xgr-annotations" class="anchor"></a>Using XGR annotations</h2>
<ul>
<li>Whenever you use annotation arguments (e.g. <code>XGR_libnames</code>,<code>Roadmap</code>,<code>Nott_epigenome</code>) the annotations that overlap with your locus will automatically be saved as <code>GRanges</code> objects in a locus-specific subdirectory:<br><em>results/<dataset_type>/<dataset_name>/<locus>/annotation</locus></dataset_name></dataset_type></em>
</li>
<li>If a selected annotation has previously been downloaded and stored for that locus, <code><a href="../reference/PLOT.locus.html">PLOT.locus()</a></code> will automatically detect and import it to save time.</li>
</ul>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot.xgr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                           LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                           LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                           locus_dir<span class="op">=</span><span class="va">locus_dir</span>, 
                           
                           XGR_libnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ENCODE_TFBS_ClusteredV3_CellTypes"</span><span class="op">)</span>, 
                           save_plot<span class="op">=</span><span class="cn">F</span>,
                           show_plot<span class="op">=</span><span class="va">show_plot</span>,
                           plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span>
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span>
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS full window track"</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS track"</span>
<span class="co">#&gt; [1] "++ PLOT:: Merged fine-mapping track"</span>
<span class="co">#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"</span>
<span class="co">#&gt; [1] "++ PLOT:: Adding Gene model track."</span>
<span class="co">#&gt; max_transcripts=1.</span>
<span class="co">#&gt; 15 transcripts from 15 genes returned.</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; [1] "++ PLOT:: XGR Tracks"</span>
<span class="co">#&gt; Start at 2021-06-22 17:11:17</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'ENCODE_TFBS_ClusteredV3_CellTypes' (from http://galahad.well.ox.ac.uk/bigdata/ENCODE_TFBS_ClusteredV3_CellTypes.RData) has been loaded into the working environment (at 2021-06-22 17:11:24)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; End at 2021-06-22 17:11:24</span>
<span class="co">#&gt; Runtime in total is: 7 secs</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 8702 query SNP(s) detected with reference overlap.</span>
<span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span>
<span class="co">#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."</span>
<span class="co">#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</span>
<span class="co">#&gt; [1] "+ PLOT:: Get window suffix..."</span>
<span class="co">#&gt; [1] "+ Constructing zoom polygon..."</span>
<span class="co">#&gt; [1] "+ Highlighting zoom origin..."</span>
<span class="co">#&gt; [1] "+ Removing subplot margins..."</span>
<span class="co">#&gt; [1] "+ Reordering tracks..."</span>
<span class="co">#&gt; [1] "+ Ensuring last track shows genomic units..."</span>
<span class="co">#&gt; [1] "+ Aligning xlimits for each subplot..."</span>
<span class="co">#&gt; [1] "+ Checking track heights..."</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot.xgr</span><span class="op">)</span>
<span class="co">#&gt; $`10x`</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-4-1.png" width="960"></p>
</div>
<div id="using-roadmap-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#using-roadmap-annotations" class="anchor"></a>Using <a href="http://www.roadmapepigenomics.org/data/tables/brain">Roadmap</a> annotations</h2>
<ul>
<li>Using the <code>Roadmap=T</code> and <code>Roadmap_query="&lt;query&gt;"</code> arguments searches the Roadmap for chromatin mark data across various cell-types, cell-lines and tissues.<br>
</li>
<li>Note that Roadmap queries requires <code>tabix</code> to be installed on your machine, or within a conda environment (<code>conda_env = "echoR"</code>).</li>
<li>Parallelizing these queries across multiple thredas speeds up this process (<code>nThread=&lt;n_cores_available&gt;</code>), as does reusing previously stored data which is automatically saved to the locus-specific subfolder (<code>&lt;dataset_type&gt;/&lt;dataset_name&gt;/&lt;locus&gt;/annotations/Roadmap.ChromatinMarks_CellTypes.RDS</code>) .</li>
</ul>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot.roadmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                               LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                               LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                               locus_dir<span class="op">=</span><span class="va">locus_dir</span>,  
                               
                               Roadmap<span class="op">=</span><span class="cn">T</span>, 
                               Roadmap_query<span class="op">=</span><span class="st">"monocyte"</span>, 
                               
                               save_plot<span class="op">=</span><span class="cn">F</span>, 
                               show_plot<span class="op">=</span><span class="va">show_plot</span>,
                               plot.zoom<span class="op">=</span><span class="st">"5x"</span>, 
                               nThread <span class="op">=</span> <span class="fu">parallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/parallel/detectCores.html">detectCores</a></span><span class="op">(</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span>,
                               conda_env <span class="op">=</span> <span class="st">"echoR"</span><span class="op">)</span>
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span>
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS full window track"</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS track"</span>
<span class="co">#&gt; [1] "++ PLOT:: Merged fine-mapping track"</span>
<span class="co">#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"</span>
<span class="co">#&gt; [1] "++ PLOT:: Adding Gene model track."</span>
<span class="co">#&gt; max_transcripts=1.</span>
<span class="co">#&gt; 15 transcripts from 15 genes returned.</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; [1] "+ PLOT:: Creating Roadmap track"</span>
<span class="co">#&gt; [1] "+ ROADMAP:: 2 annotation(s) identified that match `keyword_query`."</span>
<span class="co">#&gt; [1] "ROADMAP:: All downloads complete"</span>
<span class="co">#&gt; Time difference of 15.2 secs</span>
<span class="co">#&gt; Warning in max(xlim): no non-missing arguments to max; returning -Inf</span>
<span class="co">#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."</span>
<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>
<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>
<span class="co">#&gt; Warning: Removed 4 rows containing missing values (position_stack).</span>
<span class="co">#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 5x +&lt;+&lt;+&lt;+&lt;+</span>
<span class="co">#&gt; [1] "+ PLOT:: Get window suffix..."</span>
<span class="co">#&gt; [1] "+ Constructing zoom polygon..."</span>
<span class="co">#&gt; [1] "+ Highlighting zoom origin..."</span>
<span class="co">#&gt; [1] "+ Removing subplot margins..."</span>
<span class="co">#&gt; [1] "+ Reordering tracks..."</span>
<span class="co">#&gt; [1] "+ Ensuring last track shows genomic units..."</span>
<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>
<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>
<span class="co">#&gt; Warning: Removed 4 rows containing missing values (position_stack).</span>
<span class="co">#&gt; [1] "+ Aligning xlimits for each subplot..."</span>
<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>
<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>
<span class="co">#&gt; Warning: Removed 4 rows containing missing values (position_stack).</span>
<span class="co">#&gt; [1] "+ Checking track heights..."</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot.roadmap</span><span class="op">)</span>
<span class="co">#&gt; $`5x`</span>
<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>

<span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span>
<span class="co">#&gt; Warning: Removed 4 rows containing missing values (position_stack).</span>
<span class="co">#&gt; Warning: ggrepel: 5 unlabeled data points (too many overlaps). Consider</span>
<span class="co">#&gt; increasing max.overlaps</span>

<span class="co">#&gt; Warning: ggrepel: 5 unlabeled data points (too many overlaps). Consider</span>
<span class="co">#&gt; increasing max.overlaps</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-5-1.png" width="960"></p>
</div>
<div id="using-nott_2019-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#using-nott_2019-annotations" class="anchor"></a>Using <a href="https://science.sciencemag.org/content/366/6469/1134.abstract">Nott_2019</a> annotations</h2>
<ul>
<li>Query and plot brain cell type-specific epigenomic assays from <a href="https://science.sciencemag.org/content/366/6469/1134.abstract">Nott et al. (Science, 2019)</a><br>
(see <code><a href="../reference/NOTT_2019.bigwig_metadata.html">?NOTT_2019.bigwig_metadata</a></code> for details).</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trk_plot.nott_2019</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>,  
                                 LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                                 LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                                 locus_dir<span class="op">=</span><span class="va">locus_dir</span>, 
                                 
                                 Nott_epigenome<span class="op">=</span><span class="cn">T</span>,  
                                 Nott_binwidth <span class="op">=</span> <span class="fl">200</span>,
                                 Nott_regulatory_rects <span class="op">=</span> <span class="cn">T</span>, 
                                 Nott_show_placseq <span class="op">=</span> <span class="cn">T</span>,
                                 
                                 save_plot<span class="op">=</span><span class="cn">F</span>,
                                 show_plot<span class="op">=</span><span class="va">show_plot</span>,
                                 plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span> 
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span>
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS full window track"</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS track"</span>
<span class="co">#&gt; [1] "++ PLOT:: Merged fine-mapping track"</span>
<span class="co">#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"</span>
<span class="co">#&gt; [1] "++ PLOT:: Adding Gene model track."</span>
<span class="co">#&gt; max_transcripts=1.</span>
<span class="co">#&gt; 15 transcripts from 15 genes returned.</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; [1] "NOTT_2019:: Creating epigenomic histograms plot"</span>
<span class="co">#&gt; [1] "NOTT_2019:: Importing bigWig subsets from UCSC..."</span>
<span class="co">#&gt; [1] "+ Inferring genomic limits for window: 1x"</span>
<span class="co">#&gt; [1] "++ NOTT_2019:: Downloading and merging 12 peaks BED files."</span>
<span class="co">#&gt; [1] "++ NOTT_2019:: Converting merged BED files to GRanges."</span>
<span class="co">#&gt; [1] "++ NOTT_2019:: 634540 ranges retrieved."</span>
<span class="co">#&gt; 1860925 query SNP(s) detected with reference overlap.</span>
<span class="co">#&gt; [1] "+ PLOT:: Calculating max histogram height"</span>
<span class="co">#&gt; [1] "++ Converting label units to Mb..."</span>
<span class="co">#&gt; [1] "NOTT_2019:: Creating PLAC-seq interactome plot"</span>
<span class="co">#&gt; [1] "+ NOTT_2019:: Getting interactome data."</span>
<span class="co">#&gt; [1] "++ Converting to GRanges."</span>
<span class="co">#&gt; [1] "Importing Microglia interactome ..."</span>
<span class="co">#&gt; [1] "Importing Neuronal interactome ..."</span>
<span class="co">#&gt; [1] "Importing Oligo interactome ..."</span>
<span class="co">#&gt; 1403 query SNP(s) detected with reference overlap.</span>
<span class="co">#&gt; 1770 query SNP(s) detected with reference overlap.</span>
<span class="co">#&gt; Error in h(simpleError(msg, call)) : </span>
<span class="co">#&gt;   error in evaluating the argument 'args' in selecting a method for function 'do.call': GRanges objects don't support [[, as.list(), lapply(), or unlist() at</span>
<span class="co">#&gt;   the moment</span>
<span class="co">#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."</span>
<span class="co">#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</span>
<span class="co">#&gt; [1] "+ PLOT:: Get window suffix..."</span>
<span class="co">#&gt; [1] "+ Constructing zoom polygon..."</span>
<span class="co">#&gt; [1] "+ Highlighting zoom origin..."</span>
<span class="co">#&gt; [1] "+ Removing subplot margins..."</span>
<span class="co">#&gt; [1] "+ Reordering tracks..."</span>
<span class="co">#&gt; [1] "+ Ensuring last track shows genomic units..."</span>
<span class="co">#&gt; [1] "+ Aligning xlimits for each subplot..."</span>
<span class="co">#&gt; [1] "+ Checking track heights..."</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot.nott_2019</span><span class="op">)</span>
<span class="co">#&gt; $`10x`</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-6-1.png" width="960"></p>
</div>
<div id="using-qtl-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#using-qtl-datasets" class="anchor"></a>Using QTL datasets</h2>
<ul>
<li>Plot multiple QTL p-value columns (or really P-value columns from any kind of datset).<br>
</li>
<li>Each QTL dataset will be plotted as a new track.</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Make fake QTL P-values for the sake a demonstration</span>
<span class="va">finemap_DT</span><span class="op">$</span><span class="va">fake_eQTL.P</span> <span class="op">&lt;-</span> <span class="va">finemap_DT</span><span class="op">$</span><span class="va">P</span>  <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">.9</span>,<span class="fl">.7</span><span class="op">)</span>
<span class="va">finemap_DT</span><span class="op">$</span><span class="va">fake_sQTL.P</span> <span class="op">&lt;-</span> <span class="va">finemap_DT</span><span class="op">$</span><span class="va">P</span>  <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">.8</span>,<span class="fl">.5</span><span class="op">)</span>

<span class="va">trk_plot.qtl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PLOT.locus.html">PLOT.locus</a></span><span class="op">(</span>finemap_dat<span class="op">=</span><span class="va">finemap_DT</span>, 
                           LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, 
                           LD_reference<span class="op">=</span><span class="va">LD_reference</span>,
                           locus_dir<span class="op">=</span><span class="va">locus_dir</span>,
                           
                           QTL_prefixes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"fake_eQTL."</span>,<span class="st">"fake_sQTL."</span><span class="op">)</span>, 
                           
                           save_plot<span class="op">=</span><span class="cn">F</span>,
                           show_plot<span class="op">=</span><span class="va">show_plot</span>,
                           plot.zoom<span class="op">=</span><span class="va">plot.zoom</span><span class="op">)</span>
<span class="co">#&gt; +-------- Locus Plot: BST1--------+</span>
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS full window track"</span>
<span class="co">#&gt; [1] "++ PLOT:: GWAS track"</span>
<span class="co">#&gt; [1] "++ PLOT:: fake_eQTL. track"</span>
<span class="co">#&gt; [1] "++ PLOT:: fake_sQTL. track"</span>
<span class="co">#&gt; [1] "++ PLOT:: Merged fine-mapping track"</span>
<span class="co">#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"</span>
<span class="co">#&gt; [1] "++ PLOT:: Adding Gene model track."</span>
<span class="co">#&gt; max_transcripts=1.</span>
<span class="co">#&gt; 15 transcripts from 15 genes returned.</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."</span>
<span class="co">#&gt; +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 10x +&lt;+&lt;+&lt;+&lt;+</span>
<span class="co">#&gt; [1] "+ PLOT:: Get window suffix..."</span>
<span class="co">#&gt; [1] "+ Constructing zoom polygon..."</span>
<span class="co">#&gt; [1] "+ Highlighting zoom origin..."</span>
<span class="co">#&gt; [1] "+ Removing subplot margins..."</span>
<span class="co">#&gt; [1] "+ Reordering tracks..."</span>
<span class="co">#&gt; [1] "+ Ensuring last track shows genomic units..."</span>
<span class="co">#&gt; [1] "+ Aligning xlimits for each subplot..."</span>
<span class="co">#&gt; [1] "+ Checking track heights..."</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">trk_plot.qtl</span><span class="op">)</span>
<span class="co">#&gt; $`10x`</span></code></pre></div>
<p><img src="plotting_vignette_files/figure-html/unnamed-chunk-7-1.png" width="960"></p>
<details><summary><h1>
Session Info
</h1>
</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sessioninfo/man/session_info.html">session_info</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────</span>
<span class="co">#&gt;  setting  value                       </span>
<span class="co">#&gt;  version  R version 4.1.0 (2021-05-18)</span>
<span class="co">#&gt;  os       macOS Big Sur 10.16         </span>
<span class="co">#&gt;  system   x86_64, darwin17.0          </span>
<span class="co">#&gt;  ui       X11                         </span>
<span class="co">#&gt;  language (EN)                        </span>
<span class="co">#&gt;  collate  en_GB.UTF-8                 </span>
<span class="co">#&gt;  ctype    en_GB.UTF-8                 </span>
<span class="co">#&gt;  tz       Europe/London               </span>
<span class="co">#&gt;  date     2021-06-22                  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt;  package              * version  date       lib</span>
<span class="co">#&gt;  AnnotationDbi        * 1.54.1   2021-06-08 [3]</span>
<span class="co">#&gt;  AnnotationFilter     * 1.16.0   2021-05-19 [3]</span>
<span class="co">#&gt;  ape                    5.5      2021-04-25 [3]</span>
<span class="co">#&gt;  assertthat             0.2.1    2019-03-21 [3]</span>
<span class="co">#&gt;  backports              1.2.1    2020-12-09 [3]</span>
<span class="co">#&gt;  base64enc              0.1-3    2015-07-28 [3]</span>
<span class="co">#&gt;  Biobase              * 2.52.0   2021-05-19 [3]</span>
<span class="co">#&gt;  BiocFileCache          2.0.0    2021-05-19 [3]</span>
<span class="co">#&gt;  BiocGenerics         * 0.38.0   2021-05-19 [3]</span>
<span class="co">#&gt;  BiocIO                 1.2.0    2021-05-19 [3]</span>
<span class="co">#&gt;  BiocManager            1.30.16  2021-06-15 [3]</span>
<span class="co">#&gt;  BiocParallel           1.26.0   2021-05-19 [3]</span>
<span class="co">#&gt;  biomaRt                2.48.1   2021-06-08 [3]</span>
<span class="co">#&gt;  Biostrings             2.60.1   2021-06-06 [3]</span>
<span class="co">#&gt;  biovizBase             1.40.0   2021-05-19 [3]</span>
<span class="co">#&gt;  bit                    4.0.4    2020-08-04 [3]</span>
<span class="co">#&gt;  bit64                  4.0.5    2020-08-30 [3]</span>
<span class="co">#&gt;  bitops                 1.0-7    2021-04-24 [3]</span>
<span class="co">#&gt;  blob                   1.2.1    2020-01-20 [3]</span>
<span class="co">#&gt;  boot                   1.3-28   2021-05-03 [3]</span>
<span class="co">#&gt;  BSgenome               1.60.0   2021-05-19 [3]</span>
<span class="co">#&gt;  bslib                  0.2.5.1  2021-05-18 [3]</span>
<span class="co">#&gt;  cachem                 1.0.5    2021-05-15 [3]</span>
<span class="co">#&gt;  checkmate              2.0.0    2020-02-06 [3]</span>
<span class="co">#&gt;  class                  7.3-19   2021-05-03 [3]</span>
<span class="co">#&gt;  cli                    2.5.0    2021-04-26 [3]</span>
<span class="co">#&gt;  cluster                2.1.2    2021-04-17 [3]</span>
<span class="co">#&gt;  colorspace             2.0-1    2021-05-04 [3]</span>
<span class="co">#&gt;  crayon                 1.4.1    2021-02-08 [3]</span>
<span class="co">#&gt;  crul                   1.1.0    2021-02-15 [3]</span>
<span class="co">#&gt;  curl                   4.3.1    2021-04-30 [3]</span>
<span class="co">#&gt;  data.table             1.14.0   2021-02-21 [3]</span>
<span class="co">#&gt;  DBI                    1.1.1    2021-01-15 [3]</span>
<span class="co">#&gt;  dbplyr                 2.1.1    2021-04-06 [3]</span>
<span class="co">#&gt;  DelayedArray           0.18.0   2021-05-19 [3]</span>
<span class="co">#&gt;  desc                   1.3.0    2021-03-05 [3]</span>
<span class="co">#&gt;  DescTools              0.99.42  2021-06-17 [3]</span>
<span class="co">#&gt;  dichromat              2.0-0    2013-01-24 [3]</span>
<span class="co">#&gt;  digest                 0.6.27   2020-10-24 [3]</span>
<span class="co">#&gt;  dnet                   1.1.7    2020-02-20 [3]</span>
<span class="co">#&gt;  dplyr                  1.0.7    2021-06-18 [3]</span>
<span class="co">#&gt;  e1071                  1.7-7    2021-05-23 [3]</span>
<span class="co">#&gt;  echolocatoR          * 0.1.2    2021-06-22 [1]</span>
<span class="co">#&gt;  ellipsis               0.3.2    2021-04-29 [3]</span>
<span class="co">#&gt;  EnsDb.Hsapiens.v75   * 2.99.0   2021-06-06 [3]</span>
<span class="co">#&gt;  ensembldb            * 2.16.0   2021-05-19 [3]</span>
<span class="co">#&gt;  evaluate               0.14     2019-05-28 [3]</span>
<span class="co">#&gt;  Exact                  2.1      2020-10-02 [3]</span>
<span class="co">#&gt;  expm                   0.999-6  2021-01-13 [3]</span>
<span class="co">#&gt;  fansi                  0.5.0    2021-05-25 [3]</span>
<span class="co">#&gt;  farver                 2.1.0    2021-02-28 [3]</span>
<span class="co">#&gt;  fastmap                1.1.0    2021-01-25 [3]</span>
<span class="co">#&gt;  filelock               1.0.2    2018-10-05 [3]</span>
<span class="co">#&gt;  foreign                0.8-81   2020-12-22 [3]</span>
<span class="co">#&gt;  Formula                1.2-4    2020-10-16 [3]</span>
<span class="co">#&gt;  fs                     1.5.0    2020-07-31 [3]</span>
<span class="co">#&gt;  generics               0.1.0    2020-10-31 [3]</span>
<span class="co">#&gt;  GenomeInfoDb         * 1.28.0   2021-05-19 [3]</span>
<span class="co">#&gt;  GenomeInfoDbData       1.2.6    2021-05-28 [3]</span>
<span class="co">#&gt;  GenomicAlignments      1.28.0   2021-05-19 [3]</span>
<span class="co">#&gt;  GenomicFeatures      * 1.44.0   2021-05-19 [3]</span>
<span class="co">#&gt;  GenomicRanges        * 1.44.0   2021-05-19 [3]</span>
<span class="co">#&gt;  GGally                 2.1.2    2021-06-21 [3]</span>
<span class="co">#&gt;  ggbio                  1.40.0   2021-05-19 [3]</span>
<span class="co">#&gt;  ggnetwork              0.5.9    2021-06-04 [3]</span>
<span class="co">#&gt;  ggplot2              * 3.3.4    2021-06-16 [3]</span>
<span class="co">#&gt;  ggrepel                0.9.1    2021-01-15 [3]</span>
<span class="co">#&gt;  gld                    2.6.2    2020-01-08 [3]</span>
<span class="co">#&gt;  glue                   1.4.2    2020-08-27 [3]</span>
<span class="co">#&gt;  graph                  1.70.0   2021-05-19 [3]</span>
<span class="co">#&gt;  gridExtra              2.3      2017-09-09 [3]</span>
<span class="co">#&gt;  gtable                 0.3.0    2019-03-25 [3]</span>
<span class="co">#&gt;  hexbin                 1.28.2   2021-01-08 [3]</span>
<span class="co">#&gt;  highr                  0.9      2021-04-16 [3]</span>
<span class="co">#&gt;  Hmisc                  4.5-0    2021-02-28 [3]</span>
<span class="co">#&gt;  hms                    1.1.0    2021-05-17 [3]</span>
<span class="co">#&gt;  htmlTable              2.2.1    2021-05-18 [3]</span>
<span class="co">#&gt;  htmltools              0.5.1.1  2021-01-22 [3]</span>
<span class="co">#&gt;  htmlwidgets            1.5.3    2020-12-10 [3]</span>
<span class="co">#&gt;  httpcode               0.3.0    2020-04-10 [3]</span>
<span class="co">#&gt;  httr                   1.4.2    2020-07-20 [3]</span>
<span class="co">#&gt;  igraph                 1.2.6    2020-10-06 [3]</span>
<span class="co">#&gt;  IRanges              * 2.26.0   2021-05-19 [3]</span>
<span class="co">#&gt;  jpeg                   0.1-8.1  2019-10-24 [3]</span>
<span class="co">#&gt;  jquerylib              0.1.4    2021-04-26 [3]</span>
<span class="co">#&gt;  jsonlite               1.7.2    2020-12-09 [3]</span>
<span class="co">#&gt;  KEGGREST               1.32.0   2021-05-19 [3]</span>
<span class="co">#&gt;  knitr                  1.33     2021-04-24 [3]</span>
<span class="co">#&gt;  labeling               0.4.2    2020-10-20 [3]</span>
<span class="co">#&gt;  lattice                0.20-44  2021-05-02 [3]</span>
<span class="co">#&gt;  latticeExtra           0.6-29   2019-12-19 [3]</span>
<span class="co">#&gt;  lazyeval               0.2.2    2019-03-15 [3]</span>
<span class="co">#&gt;  lifecycle              1.0.0    2021-02-15 [3]</span>
<span class="co">#&gt;  lmom                   2.8      2019-03-12 [3]</span>
<span class="co">#&gt;  magrittr               2.0.1    2020-11-17 [3]</span>
<span class="co">#&gt;  mapproj                1.2.7    2020-02-03 [3]</span>
<span class="co">#&gt;  maps                   3.3.0    2018-04-03 [3]</span>
<span class="co">#&gt;  MASS                   7.3-54   2021-05-03 [3]</span>
<span class="co">#&gt;  Matrix                 1.3-4    2021-06-01 [3]</span>
<span class="co">#&gt;  MatrixGenerics         1.4.0    2021-05-19 [3]</span>
<span class="co">#&gt;  matrixStats            0.59.0   2021-06-01 [3]</span>
<span class="co">#&gt;  memoise                2.0.0    2021-01-26 [3]</span>
<span class="co">#&gt;  munsell                0.5.0    2018-06-12 [3]</span>
<span class="co">#&gt;  mvtnorm                1.1-2    2021-06-07 [3]</span>
<span class="co">#&gt;  nlme                   3.1-152  2021-02-04 [3]</span>
<span class="co">#&gt;  nnet                   7.3-16   2021-05-03 [3]</span>
<span class="co">#&gt;  OrganismDbi            1.34.0   2021-05-19 [3]</span>
<span class="co">#&gt;  osfr                   0.2.8    2020-02-17 [3]</span>
<span class="co">#&gt;  pals                   1.7      2021-04-17 [3]</span>
<span class="co">#&gt;  patchwork              1.1.1    2020-12-17 [3]</span>
<span class="co">#&gt;  pillar                 1.6.1    2021-05-16 [3]</span>
<span class="co">#&gt;  pkgconfig              2.0.3    2019-09-22 [3]</span>
<span class="co">#&gt;  pkgdown                1.6.1    2020-09-12 [3]</span>
<span class="co">#&gt;  plyr                   1.8.6    2020-03-03 [3]</span>
<span class="co">#&gt;  png                    0.1-7    2013-12-03 [3]</span>
<span class="co">#&gt;  prettyunits            1.1.1    2020-01-24 [3]</span>
<span class="co">#&gt;  progress               1.2.2    2019-05-16 [3]</span>
<span class="co">#&gt;  ProtGenerics           1.24.0   2021-05-19 [3]</span>
<span class="co">#&gt;  proxy                  0.4-26   2021-06-07 [3]</span>
<span class="co">#&gt;  purrr                  0.3.4    2020-04-17 [3]</span>
<span class="co">#&gt;  R6                     2.5.0    2020-10-28 [3]</span>
<span class="co">#&gt;  ragg                   1.1.3    2021-06-09 [3]</span>
<span class="co">#&gt;  rappdirs               0.3.3    2021-01-31 [3]</span>
<span class="co">#&gt;  RBGL                   1.68.0   2021-05-19 [3]</span>
<span class="co">#&gt;  RCircos                1.2.1    2019-03-12 [3]</span>
<span class="co">#&gt;  RColorBrewer           1.1-2    2014-12-07 [3]</span>
<span class="co">#&gt;  Rcpp                   1.0.6    2021-01-15 [3]</span>
<span class="co">#&gt;  RCurl                  1.98-1.3 2021-03-16 [3]</span>
<span class="co">#&gt;  readr                  1.4.0    2020-10-05 [3]</span>
<span class="co">#&gt;  reshape                0.8.8    2018-10-23 [3]</span>
<span class="co">#&gt;  reshape2               1.4.4    2020-04-09 [3]</span>
<span class="co">#&gt;  restfulr               0.0.13   2017-08-06 [3]</span>
<span class="co">#&gt;  Rgraphviz              2.36.0   2021-05-19 [3]</span>
<span class="co">#&gt;  rjson                  0.2.20   2018-06-08 [3]</span>
<span class="co">#&gt;  rlang                  0.4.11   2021-04-30 [3]</span>
<span class="co">#&gt;  rmarkdown              2.9      2021-06-15 [3]</span>
<span class="co">#&gt;  rootSolve              1.8.2.1  2020-04-27 [3]</span>
<span class="co">#&gt;  rpart                  4.1-15   2019-04-12 [3]</span>
<span class="co">#&gt;  rprojroot              2.0.2    2020-11-15 [3]</span>
<span class="co">#&gt;  Rsamtools              2.8.0    2021-05-19 [3]</span>
<span class="co">#&gt;  RSQLite                2.2.7    2021-04-22 [3]</span>
<span class="co">#&gt;  rstudioapi             0.13     2020-11-12 [3]</span>
<span class="co">#&gt;  rtracklayer            1.52.0   2021-05-19 [3]</span>
<span class="co">#&gt;  S4Vectors            * 0.30.0   2021-05-19 [3]</span>
<span class="co">#&gt;  sass                   0.4.0    2021-05-12 [3]</span>
<span class="co">#&gt;  scales                 1.1.1    2020-05-11 [3]</span>
<span class="co">#&gt;  sessioninfo            1.1.1    2018-11-05 [3]</span>
<span class="co">#&gt;  stringi                1.6.2    2021-05-17 [3]</span>
<span class="co">#&gt;  stringr                1.4.0    2019-02-10 [3]</span>
<span class="co">#&gt;  SummarizedExperiment   1.22.0   2021-05-19 [3]</span>
<span class="co">#&gt;  supraHex               1.30.0   2021-05-19 [3]</span>
<span class="co">#&gt;  survival               3.2-11   2021-04-26 [3]</span>
<span class="co">#&gt;  systemfonts            1.0.2    2021-05-11 [3]</span>
<span class="co">#&gt;  textshaping            0.3.5    2021-06-09 [3]</span>
<span class="co">#&gt;  tibble                 3.1.2    2021-05-16 [3]</span>
<span class="co">#&gt;  tidyr                  1.1.3    2021-03-03 [3]</span>
<span class="co">#&gt;  tidyselect             1.1.1    2021-04-30 [3]</span>
<span class="co">#&gt;  utf8                   1.2.1    2021-03-12 [3]</span>
<span class="co">#&gt;  VariantAnnotation      1.38.0   2021-05-19 [3]</span>
<span class="co">#&gt;  vctrs                  0.3.8    2021-04-29 [3]</span>
<span class="co">#&gt;  withr                  2.4.2    2021-04-18 [3]</span>
<span class="co">#&gt;  xfun                   0.24     2021-06-15 [3]</span>
<span class="co">#&gt;  XGR                    1.1.8    2021-06-22 [3]</span>
<span class="co">#&gt;  XML                    3.99-0.6 2021-03-16 [3]</span>
<span class="co">#&gt;  xml2                   1.3.2    2020-04-23 [3]</span>
<span class="co">#&gt;  XVector                0.32.0   2021-05-19 [3]</span>
<span class="co">#&gt;  yaml                   2.2.1    2020-02-01 [3]</span>
<span class="co">#&gt;  zlibbioc               1.38.0   2021-05-19 [3]</span>
<span class="co">#&gt;  source                            </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  local                             </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Github (hfang-bristol/XGR@4b86c9f)</span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt;  CRAN (R 4.1.0)                    </span>
<span class="co">#&gt;  Bioconductor                      </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [1] /private/var/folders/zq/h7mtybc533b1qzkys_ttgpth0000gn/T/RtmplIUggK/temp_libpath50a46ff44e71</span>
<span class="co">#&gt; [2] /private/var/folders/zq/h7mtybc533b1qzkys_ttgpth0000gn/T/RtmplIUggK/temp_libpath50a42a4792c9</span>
<span class="co">#&gt; [3] /Library/Frameworks/R.framework/Versions/4.1/Resources/library</span></code></pre></div>
<details></details></details>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
