<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>echolocatoR: QTL pipeline vignette • echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="echolocatoR: QTL pipeline vignette">
<meta property="og:description" content="echolocatoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QTL_pipeline_vignette.html">echolocatoR: QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/full_pipeline_vignette.html">echolocatoR: full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">echolocatoR: plotting vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>echolocatoR: QTL pipeline vignette</h1>
                        <h4 class="author">Brian M. Schilder</h4>
            
            <h4 class="date">Most recent update:<br> 2020-07-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/master/vignettes/QTL_pipeline_vignette.Rmd"><code>vignettes/QTL_pipeline_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>QTL_pipeline_vignette.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">echolocatoR</span>)
<span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span>
<span class="co">#&gt;   method from   </span>
<span class="co">#&gt;   +.gg   ggplot2</span></pre></body></html></div>
<div id="qtl-pipeline" class="section level1">
<h1 class="hasAnchor">
<a href="#qtl-pipeline" class="anchor"></a>QTL pipeline</h1>
<div id="prepare-top_snps-data-frame" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-top_snps-data-frame" class="anchor"></a>Prepare <code>top_SNPs</code> data.frame</h2>
<ul>
<li>In this case, we don’t have a top SNPs file ready. So we’re just going to make one directly from the full summary stats file itself (<em>NOTE</em>: You can only use this approach if you can fit the entire file in memory).<br>
</li>
<li>In this case, you’ll want to make sure to set <code>grouping_vars=c("Locus","Gene")</code> so that you get top SNPs for each eGene-locus pair (not just one SNP per locus).</li>
</ul>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">fullSS_path</span> <span class="kw">&lt;-</span> <span class="st">"~/Desktop/merged_coloc_results.snp.tsv.gz"</span>

<span class="no">top_SNPs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/import_topSNPs.html">import_topSNPs</a></span>(
  <span class="kw">topSS</span> <span class="kw">=</span> <span class="no">fullSS_path</span>,
  <span class="kw">chrom_col</span> <span class="kw">=</span> <span class="st">"chr"</span>,
  <span class="kw">position_col</span> <span class="kw">=</span> <span class="st">"pos"</span>,
  <span class="kw">snp_col</span><span class="kw">=</span><span class="st">"snp"</span>,
  <span class="kw">pval_col</span><span class="kw">=</span><span class="st">"gwas.pvalues"</span>,
  <span class="kw">effect_col</span><span class="kw">=</span><span class="st">"gwas.beta"</span>,
  <span class="kw">gene_col</span><span class="kw">=</span><span class="st">"gene"</span>,
  <span class="kw">locus_col</span> <span class="kw">=</span> <span class="st">"Locus"</span>,
  <span class="kw">grouping_vars</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Locus"</span>,<span class="st">"Gene"</span>))
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">top_SNPs</span>)
<span class="co">#&gt;          Gene Locus CHR     POS        SNP         P  Effect</span>
<span class="co">#&gt; 1:      ABCA7 ABCA7  19 1050874 rs12151021 2.562e-10  0.1071</span>
<span class="co">#&gt; 2:    ABHD17A ABCA7  19 1050874 rs12151021 2.562e-10  0.1071</span>
<span class="co">#&gt; 3: AC004623.1 ABCA7  19 1050874 rs12151021 2.562e-10  0.1071</span>
<span class="co">#&gt; 4: AC006277.1 ABCA7  19 1959773 rs56076117 2.345e-02 -0.0423</span>
<span class="co">#&gt; 5: AC006538.3 ABCA7  19 1923909 rs12610391 2.449e-03 -0.0693</span>
<span class="co">#&gt; 6: AC008993.1 ABCA7  19 1050874 rs12151021 2.562e-10  0.1071</span></pre></body></html></div>
</div>
<div id="egene-locus-list" class="section level2">
<h2 class="hasAnchor">
<a href="#egene-locus-list" class="anchor"></a>eGene-locus list</h2>
<ul>
<li>In <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> we usually supply <code>loci=</code> with a list of locus names. However,when fine-mapping GWAS-QTL jointly, you’ll need to specify which QTL eGene-GWAS locus pairs you want to run.</li>
<li>You can easy get this named list of all eGene-locus pairs with <code><a href="../reference/gene_locus_list.html">gene_locus_list()</a></code>. The returned list’s values are the GWAS loci, while the names are the QTL eGenes.<br>
</li>
<li>If you don’t want to test all pairs, you can filter the <code>top_SNPs</code> object first. In this example, I’m going to remove any pairs that don’t have matching eGene-locus names.<br>
We’ll also limit the anaylsis to just 3 loci.</li>
</ul>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">loci</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/gene_locus_list.html">gene_locus_list</a></span>(<span class="kw">top_SNPs</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="no">top_SNPs</span>, <span class="no">Gene</span><span class="kw">==</span><span class="no">Locus</span>))
<span class="co"># We'll just fine-map 2 loci to start</span>
<span class="no">loci</span> <span class="kw">&lt;-</span> <span class="no">loci</span>[<span class="fl">1</span>:<span class="fl">3</span>]
<span class="no">loci</span>
<span class="co">#&gt;    ABCA7      ACE   ADAM10 </span>
<span class="co">#&gt;  "ABCA7"    "ACE" "ADAM10"</span></pre></body></html></div>
</div>
<div id="run-fine-mapping-pipeline" class="section level2">
<h2 class="hasAnchor">
<a href="#run-fine-mapping-pipeline" class="anchor"></a>Run fine-mapping pipeline</h2>
<p><strong>NOTE</strong>: Currently this <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> will only fine-map the GWAS results. However you can still plot the fine-mapped GWAS results using the <code>QTL_prefixes</code> argument.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">Kunkle_2019.microgliaQTL</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/finemap_loci.html">finemap_loci</a></span>(<span class="co"># GENERAL ARGUMENTS </span>
                                        <span class="kw">top_SNPs</span> <span class="kw">=</span> <span class="no">top_SNPs</span>,
                                        <span class="kw">results_dir</span> <span class="kw">=</span> <span class="st">"~/Desktop/results"</span>,

                                        <span class="kw">loci</span> <span class="kw">=</span> <span class="no">loci</span>,
                                        <span class="kw">dataset_name</span> <span class="kw">=</span> <span class="st">"Kunkle_2019.microgliaQTL"</span>,
                                        <span class="kw">dataset_type</span> <span class="kw">=</span> <span class="st">"GWAS"</span>,
                                        <span class="kw">sample_size</span> <span class="kw">=</span> <span class="fl">21982</span> + <span class="fl">41944</span>,
                                        <span class="kw">proportion_cases</span> <span class="kw">=</span> <span class="fl">21982</span> / <span class="fl">41944</span>,
                                        <span class="kw">force_new_subset</span> <span class="kw">=</span> <span class="no">F</span>,
                                        <span class="kw">force_new_LD</span> <span class="kw">=</span> <span class="no">F</span>,
                                        <span class="kw">force_new_finemap</span> <span class="kw">=</span> <span class="no">F</span>,
                                        <span class="kw">remove_tmps</span> <span class="kw">=</span> <span class="no">T</span>,

                 <span class="co"># SUMMARY STATS ARGUMENTS</span>
                 <span class="kw">fullSS_path</span> <span class="kw">=</span> <span class="no">fullSS_path</span>,
                 <span class="kw">query_by</span> <span class="kw">=</span><span class="st">"coordinates"</span>,
                 <span class="kw">chrom_col</span> <span class="kw">=</span> <span class="st">"chr"</span>,
                 <span class="kw">position_col</span> <span class="kw">=</span> <span class="st">"pos"</span>,
                 <span class="kw">snp_col</span> <span class="kw">=</span> <span class="st">"snp"</span>,
                 <span class="kw">pval_col</span> <span class="kw">=</span> <span class="st">"gwas.pvalues"</span>,
                 <span class="kw">effect_col</span> <span class="kw">=</span> <span class="st">"gwas.beta"</span>,
                 <span class="kw">stderr_col</span> <span class="kw">=</span> <span class="st">"gwas.varbeta"</span>,
                 <span class="kw">MAF_col</span> <span class="kw">=</span> <span class="st">"gwas.MAF"</span>,
                 <span class="kw">gene_col</span> <span class="kw">=</span> <span class="st">"gene"</span>,

                 <span class="co"># QTL prefixes</span>
                 <span class="kw">QTL_prefixes</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"qtl."</span>),

                 <span class="co"># FILTERING ARGUMENTS</span>
                 <span class="kw">bp_distance</span> <span class="kw">=</span> <span class="fl">500000</span>, <span class="co">#100000,</span>
                 <span class="kw">min_MAF</span> <span class="kw">=</span> <span class="fl">0.001</span>,

                 <span class="co"># FINE-MAPPING ARGUMENTS</span>
                 <span class="kw">finemap_methods</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ABF"</span>,<span class="st">"FINEMAP"</span>,<span class="st">"SUSIE"</span>,<span class="st">"POLYFUN_SUSIE"</span>),
                 <span class="kw">n_causal</span> <span class="kw">=</span> <span class="fl">5</span>,
                 <span class="kw">PP_threshold</span> <span class="kw">=</span> <span class="fl">.95</span>,

                 <span class="co"># LD ARGUMENTS </span>
                 <span class="kw">LD_reference</span> <span class="kw">=</span> <span class="st">"1KGphase1"</span>,
                 <span class="kw">superpopulation</span> <span class="kw">=</span> <span class="st">"EUR"</span>,
                 <span class="kw">download_method</span> <span class="kw">=</span> <span class="st">"axel"</span>,

                 <span class="co"># PLOT ARGUMENTS </span>
                 <span class="kw">plot.types</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"simple"</span>),
                 <span class="kw">plot.window</span> <span class="kw">=</span> <span class="fl">500000</span>/<span class="fl">2</span>,
                 )
<span class="co">#&gt; [1] "Fine-mapping 3 gene:Locus pairs."</span>
<span class="co">#&gt; 🦇 🦇 🦇 ABCA7 🦇 🦇 🦇</span>
<span class="co">#&gt; ------------------ Step 1: Query ---------------</span>
<span class="co">#&gt; [1] "+ Importing  pre-existing file: /Users/schilder/Desktop/results/GWAS/Kunkle_2019.microgliaQTL/ABCA7/Multi-finemap/ABCA7_Kunkle_2019.microgliaQTL_Multi-finemap.tsv.gz"</span>
<span class="co">#&gt; [1] "+ Subset file looks good! :)"</span>
<span class="co">#&gt; [1] "+ Removing subset tmp..."</span>
<span class="co">#&gt; --- Step 2: Calculate Linkage Disequilibrium ---</span>
<span class="co">#&gt; [1] "+ Previously computed LD matrix detected. Importing... /Users/schilder/Desktop/results/GWAS/Kunkle_2019.microgliaQTL/ABCA7/LD/ABCA7.1KGphase1_LD.RDS"</span>
<span class="co">#&gt; -------------- Step 3: Filter SNPs -------------</span>
<span class="co">#&gt; [1] "++ Post-filtered data: 724 x 22"</span>
<span class="co">#&gt; -------- Step 4: Statistically Fine-map --------</span>
<span class="co">#&gt; [1] "++ Previously multi-fine-mapped results identified. Importing..."</span>
<span class="co">#&gt; [1] "++ Fine-mapping with ' ABF, FINEMAP, SUSIE, POLYFUN_SUSIE ' completed in  0.04"</span>
<span class="co">#&gt; [1] "+ Identifying Consensus SNPs..."</span>
<span class="co">#&gt; [1] "++ support_thresh = 2"</span>
<span class="co">#&gt; [1] "+ Calculating mean Posterior Probability (mean.PP)..."</span>
<span class="co">#&gt; [1] "++ 2 fine-mapping methods used."</span>
<span class="co">#&gt; [1] "++ 0 Credible Set SNPs identified."</span>
<span class="co">#&gt; [1] "++ 0 Consensus SNPs identified."</span>
<span class="co">#&gt; --------------- Step 7: Visualize --------------</span>
<span class="co">#&gt; [1] "+ Filling NAs in CS cols with 0"</span>
<span class="co">#&gt; [1] "+ Filling NAs in PP cols with 0"</span>
<span class="co">#&gt; [1] "GGBIO:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="co">#&gt; [1] "LD Matrix dimensions: 724 x 724"</span>
<span class="co">#&gt; [1] "Extracting LD subset for lead SNP: rs12151021"</span>
<span class="co">#&gt; [1] "++ GGBIO:: GWAS track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: qtl. track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: ABF track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: SUSIE track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: mean track"</span>
<span class="co">#&gt; using coord:genome to parse x scale</span>
<span class="co">#&gt; [1] "++ GGBIO:: Gene Model Track"</span>
<span class="co">#&gt; [1] "+ Annotating at transcript-level."</span>
<span class="co">#&gt; [1] "+ GGBIO:: Saving plot ==&gt; /Users/schilder/Desktop/results/GWAS/Kunkle_2019.microgliaQTL/ABCA7/multiview_ABCA7_250kb.png"</span>
<span class="co">#&gt; Fetching data...</span>
<span class="co">#&gt; OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; Warning: Removed 128 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Removed 128 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Fetching data...OK</span>
<span class="co">#&gt; Parsing exons...OK</span>
<span class="co">#&gt; Defining introns...OK</span>
<span class="co">#&gt; Defining UTRs...OK</span>
<span class="co">#&gt; Defining CDS...OK</span>
<span class="co">#&gt; aggregating...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; Constructing graphics...</span>
<span class="co">#&gt; Warning: Removed 128 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Removed 128 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 row(s) containing missing values (geom_path).</span>
<span class="co">#&gt; Warning: Duplicated override.aes is ignored.</span></pre></body></html></div>
<pre><code>#&gt; Fine-mappping complete:ABCA7
#&gt; Time difference of 23.3 secs
#&gt; 🦇 🦇 🦇 ACE 🦇 🦇 🦇 
#&gt; ------------------ Step 1: Query ---------------
#&gt; [1] "+ Importing  pre-existing file: /Users/schilder/Desktop/results/GWAS/Kunkle_2019.microgliaQTL/ACE/Multi-finemap/ACE_Kunkle_2019.microgliaQTL_Multi-finemap.tsv.gz"
#&gt; [1] "+ Subset file looks good! :)"
#&gt; [1] "+ Removing subset tmp..."
#&gt; --- Step 2: Calculate Linkage Disequilibrium ---
#&gt; [1] "+ Previously computed LD matrix detected. Importing... /Users/schilder/Desktop/results/GWAS/Kunkle_2019.microgliaQTL/ACE/LD/ACE.1KGphase1_LD.RDS"
#&gt; -------------- Step 3: Filter SNPs -------------
#&gt; [1] "++ Post-filtered data: 186 x 22"
#&gt; -------- Step 4: Statistically Fine-map --------
#&gt; [1] "++ Previously multi-fine-mapped results identified. Importing..."
#&gt; [1] "++ Fine-mapping with ' ABF, FINEMAP, SUSIE, POLYFUN_SUSIE ' completed in  0.04"
#&gt; [1] "+ Identifying Consensus SNPs..."
#&gt; [1] "++ support_thresh = 2"
#&gt; [1] "+ Calculating mean Posterior Probability (mean.PP)..."
#&gt; [1] "++ 2 fine-mapping methods used."
#&gt; [1] "++ 0 Credible Set SNPs identified."
#&gt; [1] "++ 0 Consensus SNPs identified."
#&gt; --------------- Step 7: Visualize --------------
#&gt; [1] "+ Filling NAs in CS cols with 0"
#&gt; [1] "+ Filling NAs in PP cols with 0"
#&gt; [1] "GGBIO:: LD_matrix detected. Coloring SNPs by LD with lead SNP."
#&gt; [1] "LD Matrix dimensions: 186 x 186"
#&gt; [1] "Extracting LD subset for lead SNP: rs4323"
#&gt; [1] "++ GGBIO:: GWAS track"
#&gt; using coord:genome to parse x scale
#&gt; [1] "++ GGBIO:: qtl. track"
#&gt; using coord:genome to parse x scale
#&gt; [1] "++ GGBIO:: ABF track"
#&gt; using coord:genome to parse x scale
#&gt; [1] "++ GGBIO:: SUSIE track"
#&gt; using coord:genome to parse x scale
#&gt; [1] "++ GGBIO:: mean track"
#&gt; using coord:genome to parse x scale
#&gt; [1] "++ GGBIO:: Gene Model Track"
#&gt; [1] "+ Annotating at transcript-level."
#&gt; [1] "+ GGBIO:: Saving plot ==&gt; /Users/schilder/Desktop/results/GWAS/Kunkle_2019.microgliaQTL/ACE/multiview_ACE_250kb.png"
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...
#&gt; Warning: Removed 148 rows containing non-finite values (stat_smooth).
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : span too small. fewer data values than degrees of freedom.
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : pseudoinverse used at 6.147e+07
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : neighborhood radius 1045.8
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : reciprocal condition number 0
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : There are other near singularities as well. 2.8556e+07
#&gt; Warning: Removed 1 rows containing missing values (geom_hline).
#&gt; Warning: Removed 186 rows containing missing values (geom_text).
#&gt; Warning: Removed 148 rows containing non-finite values (stat_smooth).
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : span too small. fewer data values than degrees of freedom.
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : pseudoinverse used at 6.147e+07
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : neighborhood radius 1045.8
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : reciprocal condition number 0
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : There are other near singularities as well. 2.8556e+07
#&gt; Warning: Removed 1 rows containing missing values (geom_hline).
#&gt; Warning: Removed 186 rows containing missing values (geom_text).
#&gt; Warning: Duplicated override.aes is ignored.</code></pre>
<p><img src="QTL_pipeline_vignette_files/figure-html/unnamed-chunk-3-1.png" width="960"></p>
<pre><code>#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; Defining CDS...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...
#&gt; Warning: Removed 148 rows containing non-finite values (stat_smooth).
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : span too small. fewer data values than degrees of freedom.
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : pseudoinverse used at 6.147e+07
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : neighborhood radius 1045.8
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : reciprocal condition number 0
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : There are other near singularities as well. 2.8556e+07
#&gt; Warning: Removed 1 rows containing missing values (geom_hline).
#&gt; Warning: Removed 186 rows containing missing values (geom_text).
#&gt; Warning: Removed 148 rows containing non-finite values (stat_smooth).
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : span too small. fewer data values than degrees of freedom.
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : pseudoinverse used at 6.147e+07
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : neighborhood radius 1045.8
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : reciprocal condition number 0
#&gt; Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
#&gt; parametric, : There are other near singularities as well. 2.8556e+07
#&gt; Warning: Removed 1 rows containing missing values (geom_hline).
#&gt; Warning: Removed 186 rows containing missing values (geom_text).
#&gt; Warning: Duplicated override.aes is ignored.</code></pre>
<pre><code>#&gt; Fine-mappping complete:ACE
#&gt; Time difference of 14.5 secs
#&gt; 🦇 🦇 🦇 ADAM10 🦇 🦇 🦇 
#&gt; ------------------ Step 1: Query ---------------
#&gt; [1] "+ Importing  pre-existing file: /Users/schilder/Desktop/results/GWAS/Kunkle_2019.microgliaQTL/ADAM10/Multi-finemap/ADAM10_Kunkle_2019.microgliaQTL_Multi-finemap.tsv.gz"
#&gt; [1] "+ Subset file looks good! :)"
#&gt; [1] "+ Removing subset tmp..."
#&gt; --- Step 2: Calculate Linkage Disequilibrium ---
#&gt; [1] "+ Previously computed LD matrix detected. Importing... /Users/schilder/Desktop/results/GWAS/Kunkle_2019.microgliaQTL/ADAM10/LD/ADAM10.1KGphase1_LD.RDS"
#&gt; -------------- Step 3: Filter SNPs -------------
#&gt; [1] "++ Post-filtered data: 270 x 22"
#&gt; -------- Step 4: Statistically Fine-map --------
#&gt; [1] "++ Previously multi-fine-mapped results identified. Importing..."
#&gt; [1] "++ Fine-mapping with ' ABF, FINEMAP, SUSIE, POLYFUN_SUSIE ' completed in  0.04"
#&gt; [1] "+ Identifying Consensus SNPs..."
#&gt; [1] "++ support_thresh = 2"
#&gt; [1] "+ Calculating mean Posterior Probability (mean.PP)..."
#&gt; [1] "++ 2 fine-mapping methods used."
#&gt; [1] "++ 0 Credible Set SNPs identified."
#&gt; [1] "++ 0 Consensus SNPs identified."
#&gt; --------------- Step 7: Visualize --------------
#&gt; [1] "+ Filling NAs in CS cols with 0"
#&gt; [1] "+ Filling NAs in PP cols with 0"
#&gt; [1] "GGBIO:: LD_matrix detected. Coloring SNPs by LD with lead SNP."
#&gt; [1] "LD Matrix dimensions: 270 x 270"
#&gt; [1] "Extracting LD subset for lead SNP: rs383902"
#&gt; [1] "++ GGBIO:: GWAS track"
#&gt; using coord:genome to parse x scale
#&gt; [1] "++ GGBIO:: qtl. track"
#&gt; using coord:genome to parse x scale
#&gt; [1] "++ GGBIO:: ABF track"
#&gt; using coord:genome to parse x scale
#&gt; [1] "++ GGBIO:: SUSIE track"
#&gt; using coord:genome to parse x scale
#&gt; [1] "++ GGBIO:: mean track"
#&gt; using coord:genome to parse x scale
#&gt; [1] "++ GGBIO:: Gene Model Track"
#&gt; [1] "+ Annotating at transcript-level."
#&gt; [1] "+ GGBIO:: Saving plot ==&gt; /Users/schilder/Desktop/results/GWAS/Kunkle_2019.microgliaQTL/ADAM10/multiview_ADAM10_250kb.png"
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; aggregating...
#&gt; Done
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...
#&gt; Warning: Removed 263 rows containing non-finite values (stat_smooth).
#&gt; Warning: Computation failed in `stat_smooth()`:
#&gt; span is too small
#&gt; Warning: Removed 1 rows containing missing values (geom_hline).
#&gt; Warning: Removed 270 rows containing missing values (geom_text).
#&gt; Warning: Removed 263 rows containing non-finite values (stat_smooth).
#&gt; Warning: Computation failed in `stat_smooth()`:
#&gt; span is too small
#&gt; Warning: Removed 1 rows containing missing values (geom_hline).
#&gt; Warning: Removed 270 rows containing missing values (geom_text).
#&gt; Warning: Duplicated override.aes is ignored.</code></pre>
<p><img src="QTL_pipeline_vignette_files/figure-html/unnamed-chunk-3-2.png" width="960"></p>
<pre><code>#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; aggregating...
#&gt; Done
#&gt; Fetching data...OK
#&gt; Parsing exons...OK
#&gt; Defining introns...OK
#&gt; Defining UTRs...OK
#&gt; aggregating...
#&gt; Done
#&gt; Constructing graphics...
#&gt; Warning: Removed 263 rows containing non-finite values (stat_smooth).
#&gt; Warning: Computation failed in `stat_smooth()`:
#&gt; span is too small
#&gt; Warning: Removed 1 rows containing missing values (geom_hline).
#&gt; Warning: Removed 270 rows containing missing values (geom_text).
#&gt; Warning: Removed 263 rows containing non-finite values (stat_smooth).
#&gt; Warning: Computation failed in `stat_smooth()`:
#&gt; span is too small
#&gt; Warning: Removed 1 rows containing missing values (geom_hline).
#&gt; Warning: Removed 270 rows containing missing values (geom_text).
#&gt; Warning: Duplicated override.aes is ignored.</code></pre>
<pre><code>#&gt; Fine-mappping complete:ADAM10</code></pre>
<p><img src="QTL_pipeline_vignette_files/figure-html/unnamed-chunk-3-3.png" width="960"></p>
<pre><code>#&gt; Time difference of 12.7 secs
#&gt; &lt;div id="htmlwidget-a680a1fcd62a0dd1ba32" style="width:960px;height:500px;" class="datatables html-widget"&gt;&lt;/div&gt;
#&gt; &lt;script type="application/json" data-for="htmlwidget-a680a1fcd62a0dd1ba32"&gt;{"x":{"filter":"none","extensions":["Buttons"],"caption":"&lt;caption&gt;&lt;\/caption&gt;","data":[[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[]],"container":"&lt;table class=\"display\"&gt;\n  &lt;thead&gt;\n    &lt;tr&gt;\n      &lt;th&gt; &lt;\/th&gt;\n      &lt;th&gt;Locus&lt;\/th&gt;\n      &lt;th&gt;SNP&lt;\/th&gt;\n      &lt;th&gt;CHR&lt;\/th&gt;\n      &lt;th&gt;POS&lt;\/th&gt;\n      &lt;th&gt;P&lt;\/th&gt;\n      &lt;th&gt;Effect&lt;\/th&gt;\n      &lt;th&gt;StdErr&lt;\/th&gt;\n      &lt;th&gt;Gene&lt;\/th&gt;\n      &lt;th&gt;MAF&lt;\/th&gt;\n      &lt;th&gt;proportion_cases&lt;\/th&gt;\n      &lt;th&gt;N&lt;\/th&gt;\n      &lt;th&gt;t_stat&lt;\/th&gt;\n      &lt;th&gt;leadSNP&lt;\/th&gt;\n      &lt;th&gt;qtl.pvalues&lt;\/th&gt;\n      &lt;th&gt;qtl.beta&lt;\/th&gt;\n      &lt;th&gt;qtl.varbeta&lt;\/th&gt;\n      &lt;th&gt;qtl.MAF&lt;\/th&gt;\n      &lt;th&gt;qtl.N&lt;\/th&gt;\n      &lt;th&gt;qtl.type&lt;\/th&gt;\n      &lt;th&gt;ABF.CS&lt;\/th&gt;\n      &lt;th&gt;ABF.PP&lt;\/th&gt;\n      &lt;th&gt;SUSIE.CS&lt;\/th&gt;\n      &lt;th&gt;SUSIE.PP&lt;\/th&gt;\n      &lt;th&gt;Support&lt;\/th&gt;\n      &lt;th&gt;Consensus_SNP&lt;\/th&gt;\n      &lt;th&gt;mean.PP&lt;\/th&gt;\n      &lt;th&gt;mean.CS&lt;\/th&gt;\n    &lt;\/tr&gt;\n  &lt;\/thead&gt;\n&lt;\/table&gt;","options":{"dom":"Bfrtip","buttons":["copy","csv","excel","pdf","print"],"scrollY":400,"scrollX":true,"scrollCollapse":true,"paging":false,"columnDefs":[{"className":"dt-center","targets":"_all"},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}&lt;/script&gt;</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
