<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>QTL pipeline vignette • echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="QTL pipeline vignette">
<meta property="og:description" content="echolocatoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/echolocatoR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>QTL pipeline vignette</h1>
                        <h4 data-toc-skip class="author"><h4>
Author: <i>Brian M. Schilder</i>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Updated: <i>Mar-04-2022</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/HEAD/vignettes/QTL_pipeline_vignette.Rmd" class="external-link"><code>vignettes/QTL_pipeline_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>QTL_pipeline_vignette.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">root.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>
  collapse <span class="op">=</span> <span class="cn">TRUE</span>,
  comment <span class="op">=</span> <span class="st">"#&gt;"</span>,
  root.dir <span class="op">=</span> <span class="va">root.dir</span>,
  fig.height <span class="op">=</span> <span class="fl">12</span>,
  fig.width <span class="op">=</span> <span class="fl">10</span>
<span class="op">)</span>  
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>root.dir <span class="op">=</span> <span class="va">root.dir</span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echolocatoR" class="external-link">echolocatoR</a></span><span class="op">)</span> 
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="va">results_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">root.dir</span>,<span class="st">"results"</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">results_dir</span>, showWarnings <span class="op">=</span> <span class="cn">F</span>, recursive <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="qtl-pipeline">QTL pipeline<a class="anchor" aria-label="anchor" href="#qtl-pipeline"></a>
</h2>
<ul>
<li>Here, we will use GWAS-eQTL colocalization results provided via the <a href="https://github.com/RajLabMSSM/Fine_Mapping_Shiny" class="external-link"><em>echolocatoR Fine-mapping Portal</em></a>, from the preprint: &gt; K de Paiva Lopes, GJL Snijders, J Humphrey, A Allan, M Sneeboer, E Navarro, BM Schilder…T Raj (2020) Atlas of Genetic Effects in Human Microglia Transcriptome across Brain Regions, Aging and Disease Pathologies. bioRxiv; <a href="https://doi.org/10.1101/2020.10.27.356113" class="external-link">https://doi.org/10.1101/2020.10.27.356113</a>.</li>
</ul>
<div class="section level3">
<h3 id="prepare-top_snps-data-frame">Prepare <code>top_SNPs</code> data.frame<a class="anchor" aria-label="anchor" href="#prepare-top_snps-data-frame"></a>
</h3>
<ul>
<li>In this case, we don’t have a top SNPs file ready. So we’re just going to make one directly from the full summary stats file itself (<em>NOTE</em>: You can only use this approach if you can fit the entire file in memory).<br>
</li>
<li>In this case, you’ll want to make sure to set <code>grouping_vars=c("Locus","Gene")</code> so that you get top SNPs for each eGene-locus pair (not just one SNP per locus).</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fullSS_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">root.dir</span>,<span class="st">"Microglia_all_regions_Kunkle_2019_COLOC.snp-level_select.tsv.gz"</span><span class="op">)</span>
<span class="co">## Download example data </span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">fullSS_path</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://github.com/RajLabMSSM/Fine_Mapping_Shiny/raw/master/www/Microglia_all_regions_Kunkle_2019_COLOC.snp-level_select.tsv.gz"</span>, <span class="va">fullSS_path</span><span class="op">)</span>
<span class="op">}</span>


<span class="va">top_SNPs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_topSNPs.html">import_topSNPs</a></span><span class="op">(</span>  
  topSS <span class="op">=</span> <span class="va">fullSS_path</span>,
  chrom_col <span class="op">=</span> <span class="st">"chr"</span>, 
  position_col <span class="op">=</span> <span class="st">"pos"</span>,
  snp_col<span class="op">=</span><span class="st">"snp"</span>,
  pval_col<span class="op">=</span><span class="st">"gwas.pvalues"</span>, 
  effect_col<span class="op">=</span><span class="st">"gwas.beta"</span>, 
  gene_col<span class="op">=</span><span class="st">"gene"</span>,
  locus_col <span class="op">=</span> <span class="st">"Locus"</span>,
  grouping_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Locus"</span>,<span class="st">"Gene"</span><span class="op">)</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">top_SNPs</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="egene-locus-list">eGene-locus list<a class="anchor" aria-label="anchor" href="#egene-locus-list"></a>
</h3>
<ul>
<li>In <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> we usually supply <code>loci=</code> with a list of locus names. However,when fine-mapping GWAS-QTL jointly, you’ll need to specify which QTL eGene-GWAS locus pairs you want to run.</li>
<li>You can easy get this named list of all eGene-locus pairs with <code><a href="../reference/gene_locus_list.html">gene_locus_list()</a></code>. The returned list’s values are the GWAS loci, while the names are the QTL eGenes.<br>
</li>
<li>If you don’t want to test all pairs, you can filter the <code>top_SNPs</code> object first. In this example, I’m going to remove any pairs that don’t have matching eGene-locus names.<br>
We’ll also limit the anaylsis to just 3 loci.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">loci</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_locus_list.html">gene_locus_list</a></span><span class="op">(</span><span class="va">top_SNPs</span><span class="op">)</span>
<span class="co"># We'll just fine-map 2 loci to start</span>
<span class="va">loci</span> <span class="op">&lt;-</span> <span class="va">loci</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> 
<span class="va">loci</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-fine-mapping-pipeline">Run fine-mapping pipeline<a class="anchor" aria-label="anchor" href="#run-fine-mapping-pipeline"></a>
</h3>
<p><strong>NOTE</strong>: Currently this <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> will only fine-map the GWAS results. However you can still plot the fine-mapped GWAS results using the <code>QTL_prefixes</code> argument.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Kunkle_2019.microgliaQTL</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/finemap_loci.html">finemap_loci</a></span><span class="op">(</span><span class="co"># GENERAL ARGUMENTS </span>
                                        top_SNPs <span class="op">=</span> <span class="va">top_SNPs</span>,  
                                        results_dir <span class="op">=</span> <span class="va">results_dir</span>,
                                        
                                        loci <span class="op">=</span> <span class="va">loci</span>,
                                        dataset_name <span class="op">=</span> <span class="st">"Kunkle_2019.microgliaQTL"</span>,
                                        dataset_type <span class="op">=</span> <span class="st">"GWAS"</span>,  
                                        sample_size <span class="op">=</span> <span class="fl">21982</span> <span class="op">+</span> <span class="fl">41944</span>,
                                        proportion_cases <span class="op">=</span> <span class="fl">21982</span> <span class="op">/</span> <span class="fl">41944</span>,
                                        force_new_subset <span class="op">=</span> <span class="cn">F</span>,
                                        force_new_LD <span class="op">=</span> <span class="cn">F</span>,
                                        force_new_finemap <span class="op">=</span> <span class="cn">F</span>,
                                        remove_tmps <span class="op">=</span> <span class="cn">T</span>,
                                          
                 <span class="co"># SUMMARY STATS ARGUMENTS</span>
                 fullSS_path <span class="op">=</span> <span class="va">fullSS_path</span>,
                 query_by <span class="op">=</span><span class="st">"fullSS"</span>,
                 chrom_col <span class="op">=</span> <span class="st">"chr"</span>, 
                 position_col <span class="op">=</span> <span class="st">"pos"</span>, 
                 snp_col <span class="op">=</span> <span class="st">"snp"</span>,
                 pval_col <span class="op">=</span> <span class="st">"gwas.pvalues"</span>, 
                 effect_col <span class="op">=</span> <span class="st">"gwas.beta"</span>, 
                 stderr_col <span class="op">=</span> <span class="st">"gwas.varbeta"</span>, 
                 MAF_col <span class="op">=</span> <span class="st">"gwas.MAF"</span>, 
                 gene_col <span class="op">=</span> <span class="st">"gene"</span>,
                 
                 <span class="co"># QTL prefixes</span>
                 QTL_prefixes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"qtl."</span><span class="op">)</span>,
                 
                 <span class="co"># FILTERING ARGUMENTS</span>
                 bp_distance <span class="op">=</span> <span class="fl">500000</span>, <span class="co">#100000,</span>
                 min_MAF <span class="op">=</span> <span class="fl">0.001</span>, 
                 
                 <span class="co"># FINE-MAPPING ARGUMENTS</span>
                 finemap_methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ABF"</span>,<span class="st">"FINEMAP"</span>,<span class="st">"SUSIE"</span>,<span class="st">"POLYFUN_SUSIE"</span><span class="op">)</span>,
                 n_causal <span class="op">=</span> <span class="fl">5</span>,
                 PP_threshold <span class="op">=</span> <span class="fl">.95</span>,
                 
                 <span class="co"># LD ARGUMENTS </span>
                 LD_reference <span class="op">=</span> <span class="st">"1KGphase1"</span>,
                 superpopulation <span class="op">=</span> <span class="st">"EUR"</span>,
                 download_method <span class="op">=</span> <span class="st">"axel"</span>,
                 
                 <span class="co"># PLOT ARGUMENTS </span>
                 plot.types<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"simple"</span><span class="op">)</span>,
                 plot.zoom <span class="op">=</span> <span class="st">"1x"</span> 
                 <span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] dplyr_1.0.8       echolocatoR_0.2.2 BiocStyle_2.22.0 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] backports_1.4.1             Hmisc_4.6-0                </span>
<span class="co">#&gt;   [3] BiocFileCache_2.2.1         systemfonts_1.0.4          </span>
<span class="co">#&gt;   [5] plyr_1.8.6                  lazyeval_0.2.2             </span>
<span class="co">#&gt;   [7] splines_4.1.2               BiocParallel_1.28.3        </span>
<span class="co">#&gt;   [9] GenomeInfoDb_1.30.1         ggplot2_3.3.5              </span>
<span class="co">#&gt;  [11] digest_0.6.29               ensembldb_2.18.3           </span>
<span class="co">#&gt;  [13] htmltools_0.5.2             fansi_1.0.2                </span>
<span class="co">#&gt;  [15] magrittr_2.0.2              checkmate_2.0.0            </span>
<span class="co">#&gt;  [17] memoise_2.0.1               BSgenome_1.62.0            </span>
<span class="co">#&gt;  [19] cluster_2.1.2               Biostrings_2.62.0          </span>
<span class="co">#&gt;  [21] matrixStats_0.61.0          ggbio_1.42.0               </span>
<span class="co">#&gt;  [23] pkgdown_2.0.2               prettyunits_1.1.1          </span>
<span class="co">#&gt;  [25] jpeg_0.1-9                  colorspace_2.0-3           </span>
<span class="co">#&gt;  [27] rappdirs_0.3.3              blob_1.2.2                 </span>
<span class="co">#&gt;  [29] textshaping_0.3.6           xfun_0.30                  </span>
<span class="co">#&gt;  [31] crayon_1.5.0                RCurl_1.98-1.6             </span>
<span class="co">#&gt;  [33] jsonlite_1.8.0              graph_1.72.0               </span>
<span class="co">#&gt;  [35] Exact_3.1                   VariantAnnotation_1.40.0   </span>
<span class="co">#&gt;  [37] survival_3.2-13             glue_1.6.2                 </span>
<span class="co">#&gt;  [39] gtable_0.3.0                zlibbioc_1.40.0            </span>
<span class="co">#&gt;  [41] XVector_0.34.0              DelayedArray_0.20.0        </span>
<span class="co">#&gt;  [43] BiocGenerics_0.40.0         scales_1.1.1               </span>
<span class="co">#&gt;  [45] mvtnorm_1.1-3               DBI_1.1.2                  </span>
<span class="co">#&gt;  [47] GGally_2.1.2                Rcpp_1.0.8                 </span>
<span class="co">#&gt;  [49] progress_1.2.2              htmlTable_2.4.0            </span>
<span class="co">#&gt;  [51] foreign_0.8-81              bit_4.0.4                  </span>
<span class="co">#&gt;  [53] proxy_0.4-26                OrganismDbi_1.36.0         </span>
<span class="co">#&gt;  [55] Formula_1.2-4               stats4_4.1.2               </span>
<span class="co">#&gt;  [57] htmlwidgets_1.5.4           httr_1.4.2                 </span>
<span class="co">#&gt;  [59] RColorBrewer_1.1-2          ellipsis_0.3.2             </span>
<span class="co">#&gt;  [61] pkgconfig_2.0.3             reshape_0.8.8              </span>
<span class="co">#&gt;  [63] XML_3.99-0.9                dbplyr_2.1.1               </span>
<span class="co">#&gt;  [65] nnet_7.3-16                 sass_0.4.0                 </span>
<span class="co">#&gt;  [67] utf8_1.2.2                  tidyselect_1.1.2           </span>
<span class="co">#&gt;  [69] rlang_1.0.1                 reshape2_1.4.4             </span>
<span class="co">#&gt;  [71] AnnotationDbi_1.56.2        munsell_0.5.0              </span>
<span class="co">#&gt;  [73] tools_4.1.2                 cachem_1.0.6               </span>
<span class="co">#&gt;  [75] cli_3.2.0                   generics_0.1.2             </span>
<span class="co">#&gt;  [77] RSQLite_2.2.10              evaluate_0.15              </span>
<span class="co">#&gt;  [79] stringr_1.4.0               fastmap_1.1.0              </span>
<span class="co">#&gt;  [81] yaml_2.3.5                  ragg_1.2.2                 </span>
<span class="co">#&gt;  [83] knitr_1.37                  bit64_4.0.5                </span>
<span class="co">#&gt;  [85] fs_1.5.2                    purrr_0.3.4                </span>
<span class="co">#&gt;  [87] KEGGREST_1.34.0             AnnotationFilter_1.18.0    </span>
<span class="co">#&gt;  [89] rootSolve_1.8.2.3           RBGL_1.70.0                </span>
<span class="co">#&gt;  [91] xml2_1.3.3                  biomaRt_2.50.3             </span>
<span class="co">#&gt;  [93] compiler_4.1.2              rstudioapi_0.13            </span>
<span class="co">#&gt;  [95] filelock_1.0.2              curl_4.3.2                 </span>
<span class="co">#&gt;  [97] png_0.1-7                   e1071_1.7-9                </span>
<span class="co">#&gt;  [99] tibble_3.1.6                bslib_0.3.1                </span>
<span class="co">#&gt; [101] DescTools_0.99.44           stringi_1.7.6              </span>
<span class="co">#&gt; [103] GenomicFeatures_1.46.5      desc_1.4.0                 </span>
<span class="co">#&gt; [105] lattice_0.20-45             ProtGenerics_1.26.0        </span>
<span class="co">#&gt; [107] Matrix_1.3-4                vctrs_0.3.8                </span>
<span class="co">#&gt; [109] pillar_1.7.0                lifecycle_1.0.1            </span>
<span class="co">#&gt; [111] BiocManager_1.30.16         jquerylib_0.1.4            </span>
<span class="co">#&gt; [113] data.table_1.14.2           bitops_1.0-7               </span>
<span class="co">#&gt; [115] lmom_2.8                    rtracklayer_1.54.0         </span>
<span class="co">#&gt; [117] GenomicRanges_1.46.1        R6_2.5.1                   </span>
<span class="co">#&gt; [119] BiocIO_1.4.0                latticeExtra_0.6-29        </span>
<span class="co">#&gt; [121] bookdown_0.24               gridExtra_2.3              </span>
<span class="co">#&gt; [123] IRanges_2.28.0              gld_2.6.4                  </span>
<span class="co">#&gt; [125] dichromat_2.0-0             boot_1.3-28                </span>
<span class="co">#&gt; [127] MASS_7.3-54                 assertthat_0.2.1           </span>
<span class="co">#&gt; [129] SummarizedExperiment_1.24.0 rprojroot_2.0.2            </span>
<span class="co">#&gt; [131] rjson_0.2.21                GenomicAlignments_1.30.0   </span>
<span class="co">#&gt; [133] Rsamtools_2.10.0            S4Vectors_0.32.3           </span>
<span class="co">#&gt; [135] GenomeInfoDbData_1.2.7      expm_0.999-6               </span>
<span class="co">#&gt; [137] parallel_4.1.2              hms_1.1.1                  </span>
<span class="co">#&gt; [139] grid_4.1.2                  rpart_4.1-15               </span>
<span class="co">#&gt; [141] class_7.3-19                rmarkdown_2.12             </span>
<span class="co">#&gt; [143] MatrixGenerics_1.6.0        biovizBase_1.42.0          </span>
<span class="co">#&gt; [145] Biobase_2.54.0              base64enc_0.1-3            </span>
<span class="co">#&gt; [147] restfulr_0.0.13</span></code></pre></div>
</details><p><br></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
