<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Summarise vignette â€¢ echolocatoR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Summarise vignette">
<meta property="og:description" content="echolocatoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/echolocatoR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Summarise vignette</h1>
                        <h4 data-toc-skip class="author"><h4>
Author: <i>Brian M. Schilder</i>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Updated: <i>Mar-04-2022</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/HEAD/vignettes/summarise_vignette.Rmd" class="external-link"><code>vignettes/summarise_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>summarise_vignette.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">root.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>
  collapse <span class="op">=</span> <span class="cn">TRUE</span>,
  comment <span class="op">=</span> <span class="st">"#&gt;"</span>,
  root.dir <span class="op">=</span> <span class="va">root.dir</span>, 
  fig.height <span class="op">=</span> <span class="fl">12</span>,
  fig.width <span class="op">=</span> <span class="fl">10</span>
<span class="op">)</span>  
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>
  root.dir <span class="op">=</span> <span class="va">root.dir</span>
  <span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echolocatoR" class="external-link">echolocatoR</a></span><span class="op">)</span> 
<span class="va">results_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">root.dir</span>,<span class="st">"results"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echolocatoR" class="external-link">echolocatoR</a></span><span class="op">)</span>
<span class="co"># if(!require("XGR")) install.packages("XGR")  </span></code></pre></div>
<div class="section level2">
<h2 id="download-data">Download data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h2>
<ul>
<li>Many fine-mapping results can be found on the <a href="https://rajlab.shinyapps.io/Fine_Mapping_Shiny" class="external-link">echolocatoR Fine-mapping Portal</a>.<br>
</li>
<li>We will download some of those files here using some built-in APIs.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">local_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GITHUB.portal_query.html">GITHUB.portal_query</a></span><span class="op">(</span>results_dir <span class="op">=</span> <span class="va">results_dir</span>, 
                                   dataset_types<span class="op">=</span><span class="st">"GWAS"</span>,
                                   phenotypes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"parkinson"</span><span class="op">)</span>,
                                   file_types <span class="op">=</span> <span class="st">"multi_finemap"</span>,
                                   <span class="co"># loci = c("BST1","CHRNB1","LRRK2"),</span>
                                   LD_panels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UKB"</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
</div>
<div class="section level2">
<h2 id="merge-echolocator-results">Merge <code>echolocatoR</code> results<a class="anchor" aria-label="anchor" href="#merge-echolocator-results"></a>
</h2>
<p>Gather all of the fine-mapping results generated by <code><a href="../reference/finemap_loci.html">finemap_loci()</a></code> previously.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">merged_DT</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_finemapping_results.html">merge_finemapping_results</a></span><span class="op">(</span>minimum_support<span class="op">=</span><span class="fl">0</span>,
                                       include_leadSNPs<span class="op">=</span><span class="cn">T</span>,
                                       dataset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">results_dir</span>,
                                                           <span class="st">"GWAS/Nalls23andMe_2019"</span><span class="op">)</span>,  
                                       from_storage<span class="op">=</span><span class="cn">T</span>,
                                       consensus_threshold <span class="op">=</span> <span class="fl">2</span>,
                                       top_CS_only <span class="op">=</span> <span class="cn">T</span>,
                                       haploreg_annotation<span class="op">=</span><span class="cn">F</span>,
                                       biomart_annotation<span class="op">=</span><span class="cn">F</span>, 
                                       verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>    
<span class="fu"><a href="../reference/results_report.html">results_report</a></span><span class="op">(</span><span class="va">merged_DT</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="summarise">Summarise<a class="anchor" aria-label="anchor" href="#summarise"></a>
</h2>
<div class="section level3">
<h3 id="summarise-get_snpgroup_counts">
<code>SUMMARISE.get_SNPgroup_counts()</code><a class="anchor" aria-label="anchor" href="#summarise-get_snpgroup_counts"></a>
</h3>
<p>Get the number of SNPs for each SNP group per locus.<br>
It also prints the mean number of SNPs for each SNP group across all loci.<br><strong>NOTE</strong>: You will need to make sure to set <code>merge_finemapping_results(minimum_support=1)</code> in the above step to get accurate counts for all SNP groups.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snp_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SUMMARISE.get_SNPgroup_counts.html">SUMMARISE.get_SNPgroup_counts</a></span><span class="op">(</span>merged_DT<span class="op">=</span><span class="va">merged_DT</span><span class="op">)</span>   
<span class="fu"><a href="../reference/createDT.html">createDT</a></span><span class="op">(</span><span class="va">snp_groups</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summarise-get_cs_counts">
<code>SUMMARISE.get_CS_counts()</code><a class="anchor" aria-label="anchor" href="#summarise-get_cs_counts"></a>
</h3>
<p>County the number of tool-specific and UCS Credible Set SNPs per locus.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">UCS_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SUMMARISE.get_CS_counts.html">SUMMARISE.get_CS_counts</a></span><span class="op">(</span>merged_DT <span class="op">=</span> <span class="va">merged_DT</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">UCS_counts</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="plot">Plot<a class="anchor" aria-label="anchor" href="#plot"></a>
</h2>
<ul>
<li>The following functions each return a list containing both the <code>...$plot</code> and the <code>...$data</code> used to make the plot.<br>
</li>
<li>Where available, <code>snp_filter</code> allows user to use any filtering argument (supplied as a string) to subset the data they want to use in the plot/data.</li>
</ul>
<div class="section level3">
<h3 id="summarise-cs_bin_plot">
<code>SUMMARISE.CS_bin_plot()</code><a class="anchor" aria-label="anchor" href="#summarise-cs_bin_plot"></a>
</h3>
<p>Plot the number of number that had Credible Sets of a certain size (per tool).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bin_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SUMMARISE.CS_bin_plot.html">SUMMARISE.CS_bin_plot</a></span><span class="op">(</span>merged_DT <span class="op">=</span> <span class="va">merged_DT</span>, 
                                  show_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> 
<span class="fu"><a href="../reference/createDT.html">createDT</a></span><span class="op">(</span><span class="va">bin_plot</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summarise-coloc_nominated_egenes">
<code>SUMMARISE.coloc_nominated_eGenes()</code><a class="anchor" aria-label="anchor" href="#summarise-coloc_nominated_egenes"></a>
</h3>
<p>If you ran colocalization tests with <code>echolocatoR</code> (via <code>catalogueR</code>) you can use those results to come up with a top QTL nominated gene for each locus (potentially implicating that gene in your phenotype).</p>
<div class="section level4">
<h4 id="download-coloc-results-from-echolocator-fine-mapping-portal-">Download COLOC results from <a href="https://rajlab.shinyapps.io/Fine_Mapping_Shiny" class="external-link">echolocatoR Fine-mapping Portal</a>.<a class="anchor" aria-label="anchor" href="#download-coloc-results-from-echolocator-fine-mapping-portal-"></a>
</h4>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">remote_coloc</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/RajLabMSSM/Fine_Mapping_Shiny/raw/master/www/data/GWAS/Nalls23andMe_2019/_genome_wide/COLOC/coloc.eQTL_Catalogue_ALL.csv.gz"</span>
<span class="va">local_coloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"https://github.com/RajLabMSSM/Fine_Mapping_Shiny/raw/master/www/data"</span>,
                    <span class="va">results_dir</span>,
                    <span class="va">remote_coloc</span><span class="op">)</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">local_coloc</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">dirname</a></span><span class="op">(</span><span class="va">local_coloc</span><span class="op">)</span>, showWarnings <span class="op">=</span> <span class="cn">F</span>, recursive <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">remote_coloc</span>, destfile <span class="op">=</span> <span class="va">local_coloc</span><span class="op">)</span>
<span class="op">}</span> </code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">gg_egene</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SUMMARISE.coloc_nominated_eGenes.html">SUMMARISE.coloc_nominated_eGenes</a></span><span class="op">(</span>coloc_results <span class="op">=</span> <span class="va">local_coloc</span>,
                                             merged_DT <span class="op">=</span> <span class="va">merged_DT</span>,
                                             PP_threshold <span class="op">=</span> <span class="fl">.8</span>,
                                             fill_var <span class="op">=</span> <span class="cn">NULL</span>,
                                             text_size <span class="op">=</span> <span class="fl">2.5</span>,
                                             y_lab <span class="op">=</span> <span class="st">"Locus"</span>,
                                             x_lab <span class="op">=</span> <span class="cn">NULL</span>,
                                             label_yaxis <span class="op">=</span> <span class="cn">T</span>, 
                                             show_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="../reference/createDT.html">createDT</a></span><span class="op">(</span><span class="va">gg_egene</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="summarise-cs_counts_plot">
<code>SUMMARISE.CS_counts_plot()</code><a class="anchor" aria-label="anchor" href="#summarise-cs_counts_plot"></a>
</h3>
<p>An extension of <code><a href="../reference/SUMMARISE.get_CS_counts.html">SUMMARISE.get_CS_counts()</a></code> that plots these results.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gg_CS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SUMMARISE.CS_counts_plot.html">SUMMARISE.CS_counts_plot</a></span><span class="op">(</span>merged_DT <span class="op">=</span> <span class="va">merged_DT</span>, 
                                  show_numbers<span class="op">=</span><span class="cn">T</span>,
                                  label_yaxis<span class="op">=</span><span class="cn">T</span>, 
                                  ylabel<span class="op">=</span><span class="cn">NULL</span>, 
                                  show_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="../reference/createDT.html">createDT</a></span><span class="op">(</span><span class="va">gg_CS</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="annotate-plot_missense">
<code>ANNOTATE.plot_missense()</code><a class="anchor" aria-label="anchor" href="#annotate-plot_missense"></a>
</h3>
<p>Query <em>biomart</em> for SNP-level annotations, and then return only those that contain missense mutations.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Run this in case you don't want to wait to ANNOTATE.plot_missense() to finish querying.</span>
<span class="va">gg_missense</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>plot<span class="op">=</span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_spacer.html" class="external-link">plot_spacer</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gg_missense</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ANNOTATE.plot_missense.html">ANNOTATE.plot_missense</a></span><span class="op">(</span>merged_DT <span class="op">=</span> <span class="va">merged_DT</span>, 
                                      snp_filter<span class="op">=</span><span class="st">"Support&gt;0"</span>, 
                                      label_yaxis <span class="op">=</span> <span class="cn">T</span>,
                                      show_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>  
<span class="fu"><a href="../reference/createDT.html">createDT</a></span><span class="op">(</span><span class="va">gg_missense</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summarise-peak_overlap_plot">
<code>SUMMARISE.peak_overlap_plot()</code><a class="anchor" aria-label="anchor" href="#summarise-peak_overlap_plot"></a>
</h3>
<p>Plot the overlap between some SNP group and cell-type-specific epigenomic peaks / interactome anchors.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gg_peaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SUMMARISE.peak_overlap_plot.html">SUMMARISE.peak_overlap_plot</a></span><span class="op">(</span><span class="va">merged_DT</span>, 
                                        snp_filter<span class="op">=</span><span class="st">"Consensus_SNP==T"</span>,
                                        include.NOTT_2019_peaks<span class="op">=</span><span class="cn">T</span>,
                                        include.NOTT_2019_enhancers_promoters<span class="op">=</span><span class="cn">T</span>,
                                        include.NOTT_2019_PLACseq<span class="op">=</span><span class="cn">T</span>,
                                        include.CORCES_2020_scATACpeaks<span class="op">=</span><span class="cn">T</span>,
                                        include.CORCES_2020_Cicero_coaccess<span class="op">=</span><span class="cn">T</span>,
                                        include.CORCES_2020_bulkATACpeaks<span class="op">=</span><span class="cn">T</span>,
                                        include.CORCES_2020_HiChIP_FitHiChIP_coaccess<span class="op">=</span><span class="cn">T</span>,
                                        include.CORCES_2020_gene_annotations<span class="op">=</span><span class="cn">T</span>,
                                        plot_celltype_specificity<span class="op">=</span><span class="cn">F</span>,
                                        facets_formula<span class="op">=</span><span class="st">". ~ Cell_type"</span>,
                                        show_plot<span class="op">=</span><span class="cn">T</span>,  
                                        label_yaxis<span class="op">=</span><span class="cn">T</span>, 
                                        x_strip_angle <span class="op">=</span> <span class="fl">90</span>,
                                        drop_empty_cols <span class="op">=</span> <span class="cn">T</span>,
                                        fill_title<span class="op">=</span><span class="st">"Consensus SNPs\nin epigenomic peaks"</span>,
                                        save_path<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">results_dir</span>,<span class="st">"GWAS/Nalls23andMe_2019/_genome_wide"</span>,
                                                            <span class="st">"peak_plot.consensus.png"</span><span class="op">)</span>,
                                        verbose<span class="op">=</span><span class="cn">T</span><span class="op">)</span>  
<span class="fu"><a href="../reference/createDT.html">createDT</a></span><span class="op">(</span><span class="va">gg_peaks</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="multi-plot">Multi-plot<a class="anchor" aria-label="anchor" href="#multi-plot"></a>
</h3>
<p>Now for the grand finale, put them all together by calling a single function,<code><a href="../reference/super_summary_plot.html">super_summary_plot()</a></code>!</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">super_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/super_summary_plot.html">super_summary_plot</a></span><span class="op">(</span>merged_DT <span class="op">=</span> <span class="va">merged_DT</span>,
                                  snp_filter<span class="op">=</span><span class="st">"Consensus_SNP==T"</span>,
                                  coloc_results<span class="op">=</span><span class="va">local_coloc</span>,
                                  plot_missense<span class="op">=</span><span class="cn">F</span>,
                                  show_plot<span class="op">=</span><span class="cn">T</span>, 
                                  
                                  save_plot<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">results_dir</span>,
                                                      <span class="st">"GWAS/Nalls23andMe_2019/_genome_wide"</span>,
                                                      <span class="st">"PD_summary.studies.png"</span><span class="op">)</span>,
                                  height<span class="op">=</span><span class="fl">15</span>,
                                  width<span class="op">=</span><span class="fl">12</span>,
                                  dpi<span class="op">=</span><span class="fl">500</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="manual-multi-plot">Manual multi-plot<a class="anchor" aria-label="anchor" href="#manual-multi-plot"></a>
</h3>
<p>Alternatively, you can gain more customizability by merging your plots together manually with <code>patchwork</code>.</p>
<p>Make sure to set <code>label_yaxis=F</code> for some of the plots to avoid redundancy and save space.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span>

<span class="co"># Merge</span>
<span class="va">row1</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_spacer.html" class="external-link">plot_spacer</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
           <span class="va">bin_plot</span><span class="op">$</span><span class="va">plot</span> <span class="op">)</span> <span class="op">+</span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">.4</span>,<span class="fl">.6</span><span class="op">)</span><span class="op">)</span>
<span class="va">row2</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">gg_egene</span><span class="op">$</span><span class="va">plot</span> <span class="op">+</span> 
           <span class="va">gg_CS</span><span class="op">$</span><span class="va">plot</span> <span class="op">+</span> 
           <span class="va">gg_missense</span><span class="op">$</span><span class="va">plot</span> <span class="op">+</span>
           <span class="va">gg_peaks</span><span class="op">$</span><span class="va">plot</span>   
           <span class="op">)</span> <span class="op">+</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">.125</span>,<span class="fl">.3</span>,<span class="fl">.05</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span>

<span class="va">gg_merged</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">row1</span>  <span class="op">/</span> <span class="va">row2</span><span class="op">)</span>  <span class="op">+</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">.15</span>,<span class="fl">1</span><span class="op">)</span>,ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">gg_merged</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] echolocatoR_0.2.2 BiocStyle_2.22.0 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] backports_1.4.1             Hmisc_4.6-0                </span>
<span class="co">#&gt;   [3] BiocFileCache_2.2.1         systemfonts_1.0.4          </span>
<span class="co">#&gt;   [5] plyr_1.8.6                  lazyeval_0.2.2             </span>
<span class="co">#&gt;   [7] splines_4.1.2               BiocParallel_1.28.3        </span>
<span class="co">#&gt;   [9] GenomeInfoDb_1.30.1         ggplot2_3.3.5              </span>
<span class="co">#&gt;  [11] digest_0.6.29               ensembldb_2.18.3           </span>
<span class="co">#&gt;  [13] htmltools_0.5.2             fansi_1.0.2                </span>
<span class="co">#&gt;  [15] magrittr_2.0.2              checkmate_2.0.0            </span>
<span class="co">#&gt;  [17] memoise_2.0.1               BSgenome_1.62.0            </span>
<span class="co">#&gt;  [19] cluster_2.1.2               Biostrings_2.62.0          </span>
<span class="co">#&gt;  [21] matrixStats_0.61.0          ggbio_1.42.0               </span>
<span class="co">#&gt;  [23] pkgdown_2.0.2               prettyunits_1.1.1          </span>
<span class="co">#&gt;  [25] jpeg_0.1-9                  colorspace_2.0-3           </span>
<span class="co">#&gt;  [27] rappdirs_0.3.3              blob_1.2.2                 </span>
<span class="co">#&gt;  [29] textshaping_0.3.6           xfun_0.30                  </span>
<span class="co">#&gt;  [31] dplyr_1.0.8                 crayon_1.5.0               </span>
<span class="co">#&gt;  [33] RCurl_1.98-1.6              jsonlite_1.8.0             </span>
<span class="co">#&gt;  [35] graph_1.72.0                Exact_3.1                  </span>
<span class="co">#&gt;  [37] VariantAnnotation_1.40.0    survival_3.2-13            </span>
<span class="co">#&gt;  [39] glue_1.6.2                  gtable_0.3.0               </span>
<span class="co">#&gt;  [41] zlibbioc_1.40.0             XVector_0.34.0             </span>
<span class="co">#&gt;  [43] DelayedArray_0.20.0         BiocGenerics_0.40.0        </span>
<span class="co">#&gt;  [45] scales_1.1.1                mvtnorm_1.1-3              </span>
<span class="co">#&gt;  [47] DBI_1.1.2                   GGally_2.1.2               </span>
<span class="co">#&gt;  [49] Rcpp_1.0.8                  progress_1.2.2             </span>
<span class="co">#&gt;  [51] htmlTable_2.4.0             foreign_0.8-81             </span>
<span class="co">#&gt;  [53] bit_4.0.4                   proxy_0.4-26               </span>
<span class="co">#&gt;  [55] OrganismDbi_1.36.0          Formula_1.2-4              </span>
<span class="co">#&gt;  [57] stats4_4.1.2                htmlwidgets_1.5.4          </span>
<span class="co">#&gt;  [59] httr_1.4.2                  RColorBrewer_1.1-2         </span>
<span class="co">#&gt;  [61] ellipsis_0.3.2              pkgconfig_2.0.3            </span>
<span class="co">#&gt;  [63] reshape_0.8.8               XML_3.99-0.9               </span>
<span class="co">#&gt;  [65] dbplyr_2.1.1                nnet_7.3-16                </span>
<span class="co">#&gt;  [67] sass_0.4.0                  utf8_1.2.2                 </span>
<span class="co">#&gt;  [69] tidyselect_1.1.2            rlang_1.0.1                </span>
<span class="co">#&gt;  [71] reshape2_1.4.4              AnnotationDbi_1.56.2       </span>
<span class="co">#&gt;  [73] munsell_0.5.0               tools_4.1.2                </span>
<span class="co">#&gt;  [75] cachem_1.0.6                cli_3.2.0                  </span>
<span class="co">#&gt;  [77] generics_0.1.2              RSQLite_2.2.10             </span>
<span class="co">#&gt;  [79] evaluate_0.15               stringr_1.4.0              </span>
<span class="co">#&gt;  [81] fastmap_1.1.0               yaml_2.3.5                 </span>
<span class="co">#&gt;  [83] ragg_1.2.2                  knitr_1.37                 </span>
<span class="co">#&gt;  [85] bit64_4.0.5                 fs_1.5.2                   </span>
<span class="co">#&gt;  [87] purrr_0.3.4                 KEGGREST_1.34.0            </span>
<span class="co">#&gt;  [89] AnnotationFilter_1.18.0     rootSolve_1.8.2.3          </span>
<span class="co">#&gt;  [91] RBGL_1.70.0                 xml2_1.3.3                 </span>
<span class="co">#&gt;  [93] biomaRt_2.50.3              compiler_4.1.2             </span>
<span class="co">#&gt;  [95] rstudioapi_0.13             filelock_1.0.2             </span>
<span class="co">#&gt;  [97] curl_4.3.2                  png_0.1-7                  </span>
<span class="co">#&gt;  [99] e1071_1.7-9                 tibble_3.1.6               </span>
<span class="co">#&gt; [101] bslib_0.3.1                 DescTools_0.99.44          </span>
<span class="co">#&gt; [103] stringi_1.7.6               GenomicFeatures_1.46.5     </span>
<span class="co">#&gt; [105] desc_1.4.0                  lattice_0.20-45            </span>
<span class="co">#&gt; [107] ProtGenerics_1.26.0         Matrix_1.3-4               </span>
<span class="co">#&gt; [109] vctrs_0.3.8                 pillar_1.7.0               </span>
<span class="co">#&gt; [111] lifecycle_1.0.1             BiocManager_1.30.16        </span>
<span class="co">#&gt; [113] jquerylib_0.1.4             data.table_1.14.2          </span>
<span class="co">#&gt; [115] bitops_1.0-7                lmom_2.8                   </span>
<span class="co">#&gt; [117] rtracklayer_1.54.0          GenomicRanges_1.46.1       </span>
<span class="co">#&gt; [119] R6_2.5.1                    BiocIO_1.4.0               </span>
<span class="co">#&gt; [121] latticeExtra_0.6-29         bookdown_0.24              </span>
<span class="co">#&gt; [123] gridExtra_2.3               IRanges_2.28.0             </span>
<span class="co">#&gt; [125] gld_2.6.4                   dichromat_2.0-0            </span>
<span class="co">#&gt; [127] boot_1.3-28                 MASS_7.3-54                </span>
<span class="co">#&gt; [129] assertthat_0.2.1            SummarizedExperiment_1.24.0</span>
<span class="co">#&gt; [131] rprojroot_2.0.2             rjson_0.2.21               </span>
<span class="co">#&gt; [133] GenomicAlignments_1.30.0    Rsamtools_2.10.0           </span>
<span class="co">#&gt; [135] S4Vectors_0.32.3            GenomeInfoDbData_1.2.7     </span>
<span class="co">#&gt; [137] expm_0.999-6                parallel_4.1.2             </span>
<span class="co">#&gt; [139] hms_1.1.1                   grid_4.1.2                 </span>
<span class="co">#&gt; [141] rpart_4.1-15                class_7.3-19               </span>
<span class="co">#&gt; [143] rmarkdown_2.12              MatrixGenerics_1.6.0       </span>
<span class="co">#&gt; [145] biovizBase_1.42.0           Biobase_2.54.0             </span>
<span class="co">#&gt; [147] base64enc_0.1-3             restfulr_0.0.13</span></code></pre></div>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
