<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Multi-track plot of fine-mapping results — GGBIO.plot • echolocatoR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Multi-track plot of fine-mapping results — GGBIO.plot" />
<meta property="og:description" content="Plots fine-mapping results from echolocatoR.
Create separate tracks for the original results (-log10(P))
and each fine-mapping method (Posterior Probability).
Uses ggbio." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/full_pipeline_vignette.html">Full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Multi-track plot of fine-mapping results</h1>
    <small class="dont-index">Source: <a href='https://github.com/RajLabMSSM/echolocatoR/blob/master/R/ggbio.R'><code>R/ggbio.R</code></a></small>
    <div class="hidden name"><code>GGBIO.plot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plots fine-mapping results from <em>echolocatoR</em>.
Create separate tracks for the original results (-log10(P))
and each fine-mapping method (Posterior Probability).
Uses <code>ggbio</code>.</p>
    </div>

    <pre class="usage"><span class='fu'>GGBIO.plot</span>(
  <span class='no'>finemap_dat</span>,
  <span class='no'>locus_dir</span>,
  <span class='kw'>LD_matrix</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>color_r2</span> <span class='kw'>=</span> <span class='no'>T</span>,
  <span class='kw'>method_list</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"ABF"</span>, <span class='st'>"FINEMAP"</span>, <span class='st'>"SUSIE"</span>, <span class='st'>"POLYFUN_SUSIE"</span>),
  <span class='kw'>label_snps</span> <span class='kw'>=</span> <span class='no'>T</span>,
  <span class='kw'>dot_summary</span> <span class='kw'>=</span> <span class='no'>F</span>,
  <span class='kw'>QTL_prefixes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>mean.PP</span> <span class='kw'>=</span> <span class='no'>T</span>,
  <span class='kw'>PP_threshold</span> <span class='kw'>=</span> <span class='fl'>0.95</span>,
  <span class='kw'>consensus_threshold</span> <span class='kw'>=</span> <span class='fl'>2</span>,
  <span class='kw'>sig_cutoff</span> <span class='kw'>=</span> <span class='fl'>5e-08</span>,
  <span class='kw'>XGR_libnames</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"ENCODE_TFBS_ClusteredV3_CellTypes"</span>,
    <span class='st'>"ENCODE_DNaseI_ClusteredV3_CellTypes"</span>, <span class='st'>"Broad_Histone"</span>),
  <span class='kw'>n_top_xgr</span> <span class='kw'>=</span> <span class='fl'>5</span>,
  <span class='kw'>Roadmap</span> <span class='kw'>=</span> <span class='no'>F</span>,
  <span class='kw'>Roadmap_query</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>n_top_roadmap</span> <span class='kw'>=</span> <span class='fl'>7</span>,
  <span class='kw'>annot_overlap_threshold</span> <span class='kw'>=</span> <span class='fl'>5</span>,
  <span class='kw'>Nott_epigenome</span> <span class='kw'>=</span> <span class='no'>F</span>,
  <span class='kw'>Nott_regulatory_rects</span> <span class='kw'>=</span> <span class='no'>T</span>,
  <span class='kw'>Nott_show_placseq</span> <span class='kw'>=</span> <span class='no'>T</span>,
  <span class='kw'>Nott_binwidth</span> <span class='kw'>=</span> <span class='fl'>200</span>,
  <span class='kw'>Nott_bigwig_dir</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>save_plot</span> <span class='kw'>=</span> <span class='no'>T</span>,
  <span class='kw'>show_plot</span> <span class='kw'>=</span> <span class='no'>T</span>,
  <span class='kw'>max_transcripts</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>plot.window</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>dpi</span> <span class='kw'>=</span> <span class='fl'>300</span>,
  <span class='kw'>height</span> <span class='kw'>=</span> <span class='fl'>12</span>,
  <span class='kw'>width</span> <span class='kw'>=</span> <span class='fl'>10</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='no'>T</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>color_r2</th>
      <td><p>If an LD_matrix is supplied, color each SNP by its LD with the lead SNP from the GWAS ()</p></td>
    </tr>
    <tr>
      <th>QTL_prefixes</th>
      <td><p>Prefixes to the columns that contain QTL data.
For exmaple, P-values for the <em>Fairfax_2014</em> QTL study would be
stored in the column <em>Fairfax_2014.P</em>.</p></td>
    </tr>
    <tr>
      <th>PP_threshold</th>
      <td><p>The minimum fine-mapped posterior probability for a SNP to be considered part of a Credible Set.
For example, <code>PP_threshold=.95</code> means that all Credible Set SNPs will be 95% Credible Set SNPs.</p></td>
    </tr>
    <tr>
      <th>consensus_threshold</th>
      <td><p>The minimum number of fine-mapping tools that include a SNP
in their 95% Credible Sets to consider that it a "Consensus SNP" (<em>default=2</em>).</p></td>
    </tr>
    <tr>
      <th>max_transcripts</th>
      <td><p>Max number of transcripts per gene to show.</p></td>
    </tr>
    <tr>
      <th>plot.window</th>
      <td><p>Zoom into the center of the locus when plotting (without editing the fine-mapping results file).
You can also pass a list of window sizes (e.g. <code><a href='https://rdrr.io/r/base/c.html'>c(50000,100000,500000)</a></code>) to automatically generate
multiple views of each locus.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Whether <span class="pkg">echolocatoR</span> should be verbose or silent.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p><a href='http://bioconductor.org/packages/release/bioc/vignettes/ggbio/inst/doc/ggbio.pdf'>http://bioconductor.org/packages/release/bioc/vignettes/ggbio/inst/doc/ggbio.pdf</a></p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other plot: 
<code><a href='GGBIO.QTL_track.html'>GGBIO.QTL_track</a>()</code>,
<code><a href='GGBIO.SNP_track.html'>GGBIO.SNP_track</a>()</code>,
<code><a href='GGBIO.add_lines.html'>GGBIO.add_lines</a>()</code>,
<code><a href='GGBIO.invisible_legend.html'>GGBIO.invisible_legend</a>()</code>,
<code><a href='GGBIO.track_heights_dict.html'>GGBIO.track_heights_dict</a>()</code>,
<code><a href='GGBIO.transcript_model_track.html'>GGBIO.transcript_model_track</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


