<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ echolocatoR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echolocatoR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="ANNOTATE.annotate_missense.html">ANNOTATE.annotate_missense()</a></code> </p>
        </td>
        <td><p>Annotate any missense variants</p></td>
      </tr><tr><td>
          <p><code><a href="ANNOTATE.plot_missense.html">ANNOTATE.plot_missense()</a></code> </p>
        </td>
        <td><p>Plot any missense variants</p></td>
      </tr><tr><td>
          <p><code><a href="BST1.html">BST1</a></code> </p>
        </td>
        <td><p><span class="pkg">echolocatoR</span> output example: BST1 locus</p></td>
      </tr><tr><td>
          <p><code><a href="BST1_LD_matrix.html">BST1_LD_matrix</a></code> </p>
        </td>
        <td><p>LD with the lead SNP: BST1 locus</p></td>
      </tr><tr><td>
          <p><code><a href="COJO.html">COJO()</a></code> </p>
        </td>
        <td><p>Run <em>GCTA-COJO</em></p></td>
      </tr><tr><td>
          <p><code><a href="COJO.conditional.html">COJO.conditional()</a></code> </p>
        </td>
        <td><p>Run <em>GCTA-COJO</em> conditional analysis</p></td>
      </tr><tr><td>
          <p><code><a href="COJO.stepwise.html">COJO.stepwise()</a></code> </p>
        </td>
        <td><p>Run <em>GCTA-COJO</em> conditional stepwise procedure</p></td>
      </tr><tr><td>
          <p><code><a href="CONDA.activate_env.html">CONDA.activate_env()</a></code> </p>
        </td>
        <td><p>Activate conda env</p></td>
      </tr><tr><td>
          <p><code><a href="CONDA.create_echoR_env.html">CONDA.create_echoR_env()</a></code> </p>
        </td>
        <td><p>Create conda env for <em>echolocatoR</em></p></td>
      </tr><tr><td>
          <p><code><a href="CONDA.find_env_Rlib.html">CONDA.find_env_Rlib()</a></code> </p>
        </td>
        <td><p>Find the R library for a specific env</p></td>
      </tr><tr><td>
          <p><code><a href="CONDA.find_python_path.html">CONDA.find_python_path()</a></code> </p>
        </td>
        <td><p>Find the python file for a specific env</p></td>
      </tr><tr><td>
          <p><code><a href="CONDA.install.html">CONDA.install()</a></code> </p>
        </td>
        <td><p>Install conda if it's missing</p></td>
      </tr><tr><td>
          <p><code><a href="CONDA.install_reticulate.html">CONDA.install_reticulate()</a></code> </p>
        </td>
        <td><p>Install reticulate</p></td>
      </tr><tr><td>
          <p><code><a href="CORCES_2020.HiChIP_FitHiChIP_loop_calls.html">CORCES_2020.HiChIP_FitHiChIP_loop_calls</a></code> </p>
        </td>
        <td><p>FitHiChIP loop calls from human brain tissue</p></td>
      </tr><tr><td>
          <p><code><a href="CORCES_2020.bulkATACseq_peaks.html">CORCES_2020.bulkATACseq_peaks</a></code> </p>
        </td>
        <td><p>bulkATACseq peaks from human brain tissue</p></td>
      </tr><tr><td>
          <p><code><a href="CORCES_2020.cicero_coaccessibility.html">CORCES_2020.cicero_coaccessibility</a></code> </p>
        </td>
        <td><p>Cicero_coaccessibility from human brain tissue</p></td>
      </tr><tr><td>
          <p><code><a href="CORCES_2020.get_ATAC_peak_overlap.html">CORCES_2020.get_ATAC_peak_overlap()</a></code> </p>
        </td>
        <td><p>Get overlap between datatable of SNPs and scATAC peaks</p></td>
      </tr><tr><td>
          <p><code><a href="CORCES_2020.get_HiChIP_FitHiChIP_overlap.html">CORCES_2020.get_HiChIP_FitHiChIP_overlap()</a></code> </p>
        </td>
        <td><p>Get overlap between data table of SNPs and HiChIP_FitHiChIP coaccessibility anchors</p></td>
      </tr><tr><td>
          <p><code><a href="CORCES_2020.prepare_bulkATAC_peak_overlap.html">CORCES_2020.prepare_bulkATAC_peak_overlap()</a></code> </p>
        </td>
        <td><p>Prepare data to plot overlap between datatable of SNPs and
cell-type-specific epigenomic peaks and coaccessibility data.</p></td>
      </tr><tr><td>
          <p><code><a href="CORCES_2020.prepare_scATAC_peak_overlap.html">CORCES_2020.prepare_scATAC_peak_overlap()</a></code> </p>
        </td>
        <td><p>Prepare data to plot overlap between datatable of SNPs and
cell-type-specific epigenomic peaks and coaccessibility data.</p></td>
      </tr><tr><td>
          <p><code><a href="CORCES_2020.scATACseq_celltype_peaks.html">CORCES_2020.scATACseq_celltype_peaks</a></code> </p>
        </td>
        <td><p>scATACseq cell type-specific peaks from human brain tissue</p></td>
      </tr><tr><td>
          <p><code><a href="CORCES_2020.scATACseq_peaks.html">CORCES_2020.scATACseq_peaks</a></code> </p>
        </td>
        <td><p>scATACseq peaks from human brain tissue</p></td>
      </tr><tr><td>
          <p><code><a href="DEEPLEARNING.melt.html">DEEPLEARNING.melt()</a></code> </p>
        </td>
        <td><p>Melt deep learning annotations into long-format</p></td>
      </tr><tr><td>
          <p><code><a href="DEEPLEARNING.plot.html">DEEPLEARNING.plot()</a></code> </p>
        </td>
        <td><p>Plot deep learning predictions</p></td>
      </tr><tr><td>
          <p><code><a href="DEEPLEARNING.query.html">DEEPLEARNING.query()</a></code> </p>
        </td>
        <td><p>Iteratively collect deep learning annotations</p></td>
      </tr><tr><td>
          <p><code><a href="DEEPLEARNING.query_multi_chr.html">DEEPLEARNING.query_multi_chr()</a></code> </p>
        </td>
        <td><p>Query deep learning annotations and LDscores (iterate)</p></td>
      </tr><tr><td>
          <p><code><a href="DEEPLEARNING.query_one_chr.html">DEEPLEARNING.query_one_chr()</a></code> </p>
        </td>
        <td><p>Query deep learning annotations and LDscores</p></td>
      </tr><tr><td>
          <p><code><a href="Directory_info.html">Directory_info()</a></code> </p>
        </td>
        <td><p>Retrieve the location of summary stats files</p></td>
      </tr><tr><td>
          <p><code><a href="GITHUB.portal_query.html">GITHUB.portal_query()</a></code> </p>
        </td>
        <td><p>Search and download fine-mapping files</p></td>
      </tr><tr><td>
          <p><code><a href="GOSHIFTER.html">GOSHIFTER()</a></code> </p>
        </td>
        <td><p>Run <code>GoShifter</code> enrichment pipeline</p></td>
      </tr><tr><td>
          <p><code><a href="GOSHIFTER.create_LD.html">GOSHIFTER.create_LD()</a></code> </p>
        </td>
        <td><p>Prepare LD input for <em>GoShifter</em></p></td>
      </tr><tr><td>
          <p><code><a href="GOSHIFTER.create_snpmap.html">GOSHIFTER.create_snpmap()</a></code> </p>
        </td>
        <td><p>Prepare SNP input for <em>GoShifter</em></p></td>
      </tr><tr><td>
          <p><code><a href="GOSHIFTER.run.html">GOSHIFTER.run()</a></code> </p>
        </td>
        <td><p>Run <code>GoShifter</code> enrichment test</p></td>
      </tr><tr><td>
          <p><code><a href="IMPACT_annotation_key.html">IMPACT_annotation_key</a></code> </p>
        </td>
        <td><p>IMPACT annotation key</p></td>
      </tr><tr><td>
          <p><code><a href="LD.fill_NA.html">LD.fill_NA()</a></code> </p>
        </td>
        <td><p>Fill NAs in an LD matrix</p></td>
      </tr><tr><td>
          <p><code><a href="LD.get_lead_r2.html">LD.get_lead_r2()</a></code> </p>
        </td>
        <td><p>Find correlates of the lead GWAS/QTL SNP</p></td>
      </tr><tr><td>
          <p><code><a href="LDlinkR.LDproxy_batch.html">LDlinkR.LDproxy_batch()</a></code> </p>
        </td>
        <td><p>Extract LD proxies from 1KGphase3</p></td>
      </tr><tr><td>
          <p><code><a href="LIFTOVER.html">LIFTOVER()</a></code> </p>
        </td>
        <td><p>Lift genome across builds</p></td>
      </tr><tr><td>
          <p><code><a href="LRRK2.html">LRRK2</a></code> </p>
        </td>
        <td><p><span class="pkg">echolocatoR</span> output example: LRRK2 locus</p></td>
      </tr><tr><td>
          <p><code><a href="MEX3C.html">MEX3C</a></code> </p>
        </td>
        <td><p><em>echolocatoR</em> output example: MEX3C locus</p></td>
      </tr><tr><td>
          <p><code><a href="MOTIFBREAKR.html">MOTIFBREAKR()</a></code> </p>
        </td>
        <td><p>Run <code>motifbreakR</code></p></td>
      </tr><tr><td>
          <p><code><a href="MOTIFBREAKR.filter.html">MOTIFBREAKR.filter()</a></code> </p>
        </td>
        <td><p>Summarise <code>motifbreakR</code> + <code>echolocatoR</code> results</p></td>
      </tr><tr><td>
          <p><code><a href="MOTIFBREAKR.plot.html">MOTIFBREAKR.plot()</a></code> </p>
        </td>
        <td><p>Plot <code>motifbreakR</code> results</p></td>
      </tr><tr><td>
          <p><code><a href="MUNGESUMSTATS.check_syn.html">MUNGESUMSTATS.check_syn()</a></code> </p>
        </td>
        <td><p>Check whether a table has headers that can be mapped to an attribute in <span class="pkg">MungeSumstats</span></p></td>
      </tr><tr><td>
          <p><code><a href="MUNGESUMSTATS.col_map.html">MUNGESUMSTATS.col_map()</a></code> </p>
        </td>
        <td><p>Column name mappings from <span class="pkg">MungeSumstats</span> to <span class="pkg">echolocatoR</span></p></td>
      </tr><tr><td>
          <p><code><a href="MUNGESUMSTATS.to_echolocatoR.html">MUNGESUMSTATS.to_echolocatoR()</a></code> </p>
        </td>
        <td><p>Convert from <span class="pkg">MungeSumstats</span> to <span class="pkg">echolocatoR</span> format</p></td>
      </tr><tr><td>
          <p><code><a href="NOTT_2019.bigwig_metadata.html">NOTT_2019.bigwig_metadata</a></code> </p>
        </td>
        <td><p>Metadata and links to data</p></td>
      </tr><tr><td>
          <p><code><a href="NOTT_2019.interactome.html">NOTT_2019.interactome</a></code> </p>
        </td>
        <td><p>Brain cell type-specific enhancers, promoters, and interactomes</p></td>
      </tr><tr><td>
          <p><code><a href="NOTT_2019.superenhancer_interactome.html">NOTT_2019.superenhancer_interactome</a></code> </p>
        </td>
        <td><p>Brain cell type-specific interactomes with superenhancers</p></td>
      </tr><tr><td>
          <p><code><a href="Nalls_top_SNPs.html">Nalls_top_SNPs</a></code> </p>
        </td>
        <td><p>TopSS example file</p></td>
      </tr><tr><td>
          <p><code><a href="PAINTOR.help_menu.html">PAINTOR.help_menu()</a></code> </p>
        </td>
        <td><p>Show PAINTOR help menu</p></td>
      </tr><tr><td>
          <p><code><a href="PAINTOR.import_QTL_DT.html">PAINTOR.import_QTL_DT()</a></code> </p>
        </td>
        <td><p>Import QTL data for PAINTOR</p></td>
      </tr><tr><td>
          <p><code><a href="PAINTOR.merge_results.html">PAINTOR.merge_results()</a></code> </p>
        </td>
        <td><p>Merge PAINTOR results</p></td>
      </tr><tr><td>
          <p><code><a href="PAINTOR.process_results.html">PAINTOR.process_results()</a></code> </p>
        </td>
        <td><p>Process PAINTOR results</p></td>
      </tr><tr><td>
          <p><code><a href="PAINTOR.run.html">PAINTOR.run()</a></code> </p>
        </td>
        <td><p>Run PAINTOR</p></td>
      </tr><tr><td>
          <p><code><a href="PAINTOR.survey_annotation.html">PAINTOR.survey_annotation()</a></code> </p>
        </td>
        <td><p>Report the number of annotations &gt; 0</p></td>
      </tr><tr><td>
          <p><code><a href="PLOT.dot_summary.html">PLOT.dot_summary()</a></code> </p>
        </td>
        <td><p>Multi-fine-map summary dot plot</p></td>
      </tr><tr><td>
          <p><code><a href="PLOT.locus.html">PLOT.locus()</a></code> </p>
        </td>
        <td><p>Generate a locus plot</p></td>
      </tr><tr><td>
          <p><code><a href="PLOT.transcript_model_track.html">PLOT.transcript_model_track()</a></code> </p>
        </td>
        <td><p>Plot gene/transcript models</p></td>
      </tr><tr><td>
          <p><code><a href="POLYFUN.get_ref_prefix.html">POLYFUN.get_ref_prefix()</a></code> </p>
        </td>
        <td><p>Get ref data path prefix</p></td>
      </tr><tr><td>
          <p><code><a href="POLYFUN.h2_enrichment_SNPgroups_plot.html">POLYFUN.h2_enrichment_SNPgroups_plot()</a></code> </p>
        </td>
        <td><p>Plot heritability (h2) enrichment</p></td>
      </tr><tr><td>
          <p><code><a href="ROADMAP.construct_reference.html">ROADMAP.construct_reference()</a></code> </p>
        </td>
        <td><p>Gather Roadmap annotation metadata</p></td>
      </tr><tr><td>
          <p><code><a href="ROADMAP.merge_and_process_grl.html">ROADMAP.merge_and_process_grl()</a></code> </p>
        </td>
        <td><p>Standardize Roadmap query</p></td>
      </tr><tr><td>
          <p><code><a href="ROADMAP.query.html">ROADMAP.query()</a></code> </p>
        </td>
        <td><p>Query Roadmap by genomic coordinates</p></td>
      </tr><tr><td>
          <p><code><a href="ROADMAP.query_and_plot.html">ROADMAP.query_and_plot()</a></code> </p>
        </td>
        <td><p>Query and plot Roadmap epigenomic annotations</p></td>
      </tr><tr><td>
          <p><code><a href="ROADMAP.tabix.html">ROADMAP.tabix()</a></code> </p>
        </td>
        <td><p>Query Roadmap API</p></td>
      </tr><tr><td>
          <p><code><a href="ROADMAP.track_plot.html">ROADMAP.track_plot()</a></code> </p>
        </td>
        <td><p>Plot Roadmap query</p></td>
      </tr><tr><td>
          <p><code><a href="SPLICEAI.plot.html">SPLICEAI.plot()</a></code> </p>
        </td>
        <td><p>Plot <em>SpliceAI</em> predictions</p></td>
      </tr><tr><td>
          <p><code><a href="SPLICEAI.run.html">SPLICEAI.run()</a></code> </p>
        </td>
        <td><p>Run pre-trained <em>SpliceAI</em> model to get predictions</p></td>
      </tr><tr><td>
          <p><code><a href="SPLICEAI.snp_probs.html">SPLICEAI.snp_probs()</a></code> </p>
        </td>
        <td><p>Postprocess <em>SpliceAI</em> results after querying</p></td>
      </tr><tr><td>
          <p><code><a href="SPLICEAI.subset_precomputed_tsv.html">SPLICEAI.subset_precomputed_tsv()</a></code> </p>
        </td>
        <td><p>Query genome-wide <em>SpliceAI</em> results file (tsv format)</p></td>
      </tr><tr><td>
          <p><code><a href="SPLICEAI.subset_precomputed_tsv_iterate.html">SPLICEAI.subset_precomputed_tsv_iterate()</a></code> </p>
        </td>
        <td><p>Make multiple queries to genome-wide <em>SpliceAI</em> results file (tsv format)</p></td>
      </tr><tr><td>
          <p><code><a href="SPLICEAI.subset_precomputed_vcf.html">SPLICEAI.subset_precomputed_vcf()</a></code> </p>
        </td>
        <td><p>Query genome-wide <em>SpliceAI</em> results file (vcf format)</p></td>
      </tr><tr><td>
          <p><code><a href="SUMMARISE.CS_bin_plot.html">SUMMARISE.CS_bin_plot()</a></code> </p>
        </td>
        <td><p>Plot CS bin counts</p></td>
      </tr><tr><td>
          <p><code><a href="SUMMARISE.CS_counts_plot.html">SUMMARISE.CS_counts_plot()</a></code> </p>
        </td>
        <td><p>Bar plot of tool-specific CS sizes</p></td>
      </tr><tr><td>
          <p><code><a href="SUMMARISE.cell_type_specificity.html">SUMMARISE.cell_type_specificity()</a></code> </p>
        </td>
        <td><p>Get cell-type-specifity score for each cell type</p></td>
      </tr><tr><td>
          <p><code><a href="SUMMARISE.coloc_nominated_eGenes.html">SUMMARISE.coloc_nominated_eGenes()</a></code> </p>
        </td>
        <td><p>Nominate target genes within each locus</p></td>
      </tr><tr><td>
          <p><code><a href="SUMMARISE.get_CS_bins.html">SUMMARISE.get_CS_bins()</a></code> </p>
        </td>
        <td><p>Count bins of tool-specific and union CS sizes</p></td>
      </tr><tr><td>
          <p><code><a href="SUMMARISE.get_CS_counts.html">SUMMARISE.get_CS_counts()</a></code> </p>
        </td>
        <td><p>Tally tool-specific and union CS sizes</p></td>
      </tr><tr><td>
          <p><code><a href="SUMMARISE.get_SNPgroup_counts.html">SUMMARISE.get_SNPgroup_counts()</a></code> </p>
        </td>
        <td><p>Tally locus-specific SNP group sizes</p></td>
      </tr><tr><td>
          <p><code><a href="SUMMARISE.plot_dataset_overlap.html">SUMMARISE.plot_dataset_overlap()</a></code> </p>
        </td>
        <td><p>Plot inter-study SNP overlap</p></td>
      </tr><tr><td>
          <p><code><a href="SURE.download_annotations.html">SURE.download_annotations()</a></code> </p>
        </td>
        <td><p>Download <em>SuRE</em> annotations</p></td>
      </tr><tr><td>
          <p><code><a href="SURE.melt_snp_groups.html">SURE.melt_snp_groups()</a></code> </p>
        </td>
        <td><p>Aggregate <em>SuRE</em> p-values for each SNP group</p></td>
      </tr><tr><td>
          <p><code><a href="SURE.merge.html">SURE.merge()</a></code> </p>
        </td>
        <td><p>Merge with fine-mapping and <em>SuRE</em> results</p></td>
      </tr><tr><td>
          <p><code><a href="SURE.plot.html">SURE.plot()</a></code> </p>
        </td>
        <td><p>Plot <em>SuRE</em> results across SNP groups</p></td>
      </tr><tr><td>
          <p><code><a href="SUSIE.html">SUSIE()</a></code> </p>
        </td>
        <td><p>Fine-map with SUSIE</p></td>
      </tr><tr><td>
          <p><code><a href="TABIX.html">TABIX()</a></code> </p>
        </td>
        <td><p>Covert and query</p></td>
      </tr><tr><td>
          <p><code><a href="TABIX.convert_file.html">TABIX.convert_file()</a></code> </p>
        </td>
        <td><p>Convert summary stats file to tabix format</p></td>
      </tr><tr><td>
          <p><code><a href="TABIX.query.html">TABIX.query()</a></code> </p>
        </td>
        <td><p>Query a tabix file</p></td>
      </tr><tr><td>
          <p><code><a href="VALIDATION.bootstrap.html">VALIDATION.bootstrap()</a></code> </p>
        </td>
        <td><p>Perform validation bootstrap procedure</p></td>
      </tr><tr><td>
          <p><code><a href="VALIDATION.bootstrap_multimetric.html">VALIDATION.bootstrap_multimetric()</a></code> </p>
        </td>
        <td><p>Conduct bootstrap procedure on multiple columns</p></td>
      </tr><tr><td>
          <p><code><a href="VALIDATION.bootstrap_plot.html">VALIDATION.bootstrap_plot()</a></code> </p>
        </td>
        <td><p>Plot permutation test results</p></td>
      </tr><tr><td>
          <p><code><a href="VALIDATION.compare_bootstrap_distributions.html">VALIDATION.compare_bootstrap_distributions()</a></code> </p>
        </td>
        <td><p>Test whether permutation p-value distributions are different</p></td>
      </tr><tr><td>
          <p><code><a href="VALIDATION.compare_support.html">VALIDATION.compare_support()</a></code> </p>
        </td>
        <td><p>Check whether Support level correlates with some variable</p></td>
      </tr><tr><td>
          <p><code><a href="VALIDATION.permute.html">VALIDATION.permute()</a></code> </p>
        </td>
        <td><p>Validation permutation tests</p></td>
      </tr><tr><td>
          <p><code><a href="VALIDATION.super_plot.html">VALIDATION.super_plot()</a></code> </p>
        </td>
        <td><p>Merge validation assays into a single plot</p></td>
      </tr><tr><td>
          <p><code><a href="XGR.enrichment.html">XGR.enrichment()</a></code> </p>
        </td>
        <td><p>XGR enrichment</p></td>
      </tr><tr><td>
          <p><code><a href="XGR.enrichment_bootstrap.html">XGR.enrichment_bootstrap()</a></code> </p>
        </td>
        <td><p>XGR enrichment (bootstrapped)</p></td>
      </tr><tr><td>
          <p><code><a href="XGR.enrichment_plot.html">XGR.enrichment_plot()</a></code> </p>
        </td>
        <td><p>Plot enrichment results</p></td>
      </tr><tr><td>
          <p><code><a href="auto_topSNPs_sub.html">auto_topSNPs_sub()</a></code> </p>
        </td>
        <td><p>Automatically identify top SNP per locus</p></td>
      </tr><tr><td>
          <p><code><a href="calculate_tstat.html">calculate_tstat()</a></code> </p>
        </td>
        <td><p>Compute t-stat</p></td>
      </tr><tr><td>
          <p><code><a href="check_necessary_cols.html">check_necessary_cols()</a></code> </p>
        </td>
        <td><p>Check for necessary columns</p></td>
      </tr><tr><td>
          <p><code><a href="column_map.html">column_map()</a></code> </p>
        </td>
        <td><p>Column name mappings for a given package</p></td>
      </tr><tr><td>
          <p><code><a href="dataframe_2_vcf.html">dataframe_2_vcf()</a></code> </p>
        </td>
        <td><p>Convert a dataframe to a vcf</p></td>
      </tr><tr><td>
          <p><code><a href="detect_genes.html">detect_genes()</a></code> </p>
        </td>
        <td><p>Detect QTL genes in full summary stats file</p></td>
      </tr><tr><td>
          <p><code><a href="echolocatoR-package.html">echolocatoR-package</a></code> </p>
        </td>
        <td><p>echolocatoR: Automated genomic fine-mapping</p></td>
      </tr><tr><td>
          <p><code><a href="extra_installs.html">extra_installs()</a></code> </p>
        </td>
        <td><p>Install tricky R packages</p></td>
      </tr><tr><td>
          <p><code><a href="finemap_loci.html">finemap_loci()</a></code> </p>
        </td>
        <td><p>Fine-map multiple loci</p></td>
      </tr><tr><td>
          <p><code><a href="finemap_pipeline.html">finemap_pipeline()</a></code> </p>
        </td>
        <td><p>Run <span class="pkg">echolocatoR</span> pipeline on a single locus</p></td>
      </tr><tr><td>
          <p><code><a href="fullSS_dat.html">fullSS_dat</a></code> </p>
        </td>
        <td><p>Example subset of full summary stats</p></td>
      </tr><tr><td>
          <p><code><a href="fullSS_munged.html">fullSS_munged</a></code> </p>
        </td>
        <td><p>Example subset of full summary stats: munged</p></td>
      </tr><tr><td>
          <p><code><a href="gene_locus_list.html">gene_locus_list()</a></code> </p>
        </td>
        <td><p>Generate a named list of [e]gene-locus pairs</p></td>
      </tr><tr><td>
          <p><code><a href="genome_wide_dir.html">genome_wide_dir</a></code> </p>
        </td>
        <td><p>Example results path for genome-wide results</p></td>
      </tr><tr><td>
          <p><code><a href="get_UKB_MAF.html">get_UKB_MAF()</a></code> </p>
        </td>
        <td><p>Get MAF from UK Biobank.</p></td>
      </tr><tr><td>
          <p><code><a href="get_stepwise_results.html">get_stepwise_results()</a></code> </p>
        </td>
        <td><p>Gather stepwise conditional results</p></td>
      </tr><tr><td>
          <p><code><a href="import.bw.filt.html">import.bw.filt()</a></code> </p>
        </td>
        <td><p>Import a subset of a bigwig file
based on the coordinates in a GRanges object (gr.dat).</p></td>
      </tr><tr><td>
          <p><code><a href="import_topSNPs.html">import_topSNPs()</a></code> </p>
        </td>
        <td><p>Import top GWAS/QTL summary statistics</p></td>
      </tr><tr><td>
          <p><code><a href="install_tricky_packages.html">install_tricky_packages()</a></code> </p>
        </td>
        <td><p>Install tricky packages</p></td>
      </tr><tr><td>
          <p><code><a href="locus_dir.html">locus_dir</a></code> </p>
        </td>
        <td><p>Example results path for BST1 locus</p></td>
      </tr><tr><td>
          <p><code><a href="merge_celltype_specific_epigenomics.html">merge_celltype_specific_epigenomics()</a></code> </p>
        </td>
        <td><p>Merge all cell-type-specific epigenomics</p></td>
      </tr><tr><td>
          <p><code><a href="merge_finemapping_results.html">merge_finemapping_results()</a></code> </p>
        </td>
        <td><p>Merge fine-mapping results from all loci</p></td>
      </tr><tr><td>
          <p><code><a href="merge_finemapping_results_each.html">merge_finemapping_results_each()</a></code> </p>
        </td>
        <td><p>Create full cross-locus merged files for each dataset,
then return a subset of those files as one super-merged table.</p></td>
      </tr><tr><td>
          <p><code><a href="merged_DT.html">merged_DT</a></code> </p>
        </td>
        <td><p><em>echolocatoR</em> output example: all loci</p></td>
      </tr><tr><td>
          <p><code><a href="order_loci.html">order_loci()</a></code> </p>
        </td>
        <td><p>Order loci by UCS size, or alphabetically</p></td>
      </tr><tr><td>
          <p><code><a href="popDat_1KGphase1.html">popDat_1KGphase1</a></code> </p>
        </td>
        <td><p>Population metadata: 1KGphase1</p></td>
      </tr><tr><td>
          <p><code><a href="popDat_1KGphase3.html">popDat_1KGphase3</a></code> </p>
        </td>
        <td><p>Population metadata: 1KGphase3</p></td>
      </tr><tr><td>
          <p><code><a href="process_COJO_results.html">process_COJO_results()</a></code> </p>
        </td>
        <td><p>Gather all <em>GCTA-COJO</em> results</p></td>
      </tr><tr><td>
          <p><code><a href="results_report.html">results_report()</a></code> </p>
        </td>
        <td><p>Give a quick summary report of the fine-mapping results</p></td>
      </tr><tr><td>
          <p><code><a href="snps_to_condition.html">snps_to_condition()</a></code> </p>
        </td>
        <td><p>Identify SNPs to condition on.</p></td>
      </tr><tr><td>
          <p><code><a href="standardize_subset.html">standardize_subset()</a></code> </p>
        </td>
        <td><p>Standardize the locus subset</p></td>
      </tr><tr><td>
          <p><code><a href="super_summary_plot.html">super_summary_plot()</a></code> </p>
        </td>
        <td><p>Merge all summary plots into one super plot</p></td>
      </tr><tr><td>
          <p><code><a href="top_SNPs.html">top_SNPs</a></code> </p>
        </td>
        <td><p>TopSS example file (processed)</p></td>
      </tr><tr><td>
          <p><code><a href="update_cols.html">update_cols()</a></code> </p>
        </td>
        <td><p>Update CS cols</p></td>
      </tr><tr><td>
          <p><code><a href="vcf_cleaning.html">vcf_cleaning()</a></code> </p>
        </td>
        <td><p>Remove old vcf files</p></td>
      </tr><tr><td>
          <p><code><a href="xLiftOver.html">xLiftOver()</a></code> </p>
        </td>
        <td><p>Genome build liftover</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

