<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ echolocatoR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/full_pipeline_vignette.html">echolocatoR: full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">echolocatoR: plotting vignette</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="BST1.html">BST1</a></code> </p>
        </td>
        <td><p><em>echolocatoR</em> output example (BST1 locus)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="BST1_LD_matrix.html">BST1_LD_matrix</a></code> </p>
        </td>
        <td><p>LD with the lead SNP (BST1 locus)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="COJO.html">COJO()</a></code> </p>
        </td>
        <td><p>Run GCTA-COJO</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="COJO.conditional.html">COJO.conditional()</a></code> </p>
        </td>
        <td><p>Run COJO conditional analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="COJO.stepwise.html">COJO.stepwise()</a></code> </p>
        </td>
        <td><p>Conditional stepwise procedure</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CONDA.activate_env.html">CONDA.activate_env()</a></code> </p>
        </td>
        <td><p>Activate conda env</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CONDA.create_env.html">CONDA.create_env()</a></code> </p>
        </td>
        <td><p>Install necessary command line and python tools</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CONDA.find_env_path.html">CONDA.find_env_path()</a></code> </p>
        </td>
        <td><p>Identify the right conda env to use</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CONDA.install.html">CONDA.install()</a></code> </p>
        </td>
        <td><p>Install conda if it's missing</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.HiChIP_FitHiChIP_loop_calls.html">CORCES_2020.HiChIP_FitHiChIP_loop_calls</a></code> </p>
        </td>
        <td><p>FitHiChIP loop calls from Alzheimer's disease brain tissue</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.bulkATACseq_peaks.html">CORCES_2020.bulkATACseq_peaks</a></code> </p>
        </td>
        <td><p>bulkATACseq peaks from Alzheimer's disease brain tissue</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.cicero_coaccessibility.html">CORCES_2020.cicero_coaccessibility</a></code> </p>
        </td>
        <td><p>Cicero_coaccessibility from Alzheimer's disease brain tissue</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.scATACseq_celltype_peaks.html">CORCES_2020.scATACseq_celltype_peaks</a></code> </p>
        </td>
        <td><p>scATACseq cell type-specific peaks from Alzheimer's disease brain tissue</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.scATACseq_peaks.html">CORCES_2020.scATACseq_peaks</a></code> </p>
        </td>
        <td><p>scATACseq peaks from Alzheimer's disease brain tissue</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="GGBIO.dot_summary.html">GGBIO.dot_summary()</a></code> </p>
        </td>
        <td><p>Multi-fine-map summary dot plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="IMPACT_annotation_key.html">IMPACT_annotation_key</a></code> </p>
        </td>
        <td><p>IMPACT annotation key</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="LRRK2.html">LRRK2</a></code> </p>
        </td>
        <td><p><em>echolocatoR</em> output example (LRRK2 locus)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MEX3C.html">MEX3C</a></code> </p>
        </td>
        <td><p><em>echolocatoR</em> output example (MEX3C locus)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MOTIFBREAKR.html">MOTIFBREAKR()</a></code> </p>
        </td>
        <td><p>Run <code>motifbreakR</code></p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MOTIFBREAKR.plot.html">MOTIFBREAKR.plot()</a></code> </p>
        </td>
        <td><p>Plot <code>motifbreakR</code> results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MOTIFBREAKR.summarize.html">MOTIFBREAKR.summarize()</a></code> </p>
        </td>
        <td><p>Summarise <code>motifbreakR</code> + <code>echolocatoR</code> results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="NOTT_2019.bigwig_metadata.html">NOTT_2019.bigwig_metadata</a></code> </p>
        </td>
        <td><p>Metadata and links to data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="NOTT_2019.interactome.html">NOTT_2019.interactome</a></code> </p>
        </td>
        <td><p>Brain cell type-specific enhancers, promoters, and interactomes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="NOTT_2019.superenhancer_interactome.html">NOTT_2019.superenhancer_interactome</a></code> </p>
        </td>
        <td><p>Brain cell type-specific interactomes with superenhancers</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Nalls_top_SNPs.html">Nalls_top_SNPs</a></code> </p>
        </td>
        <td><p>TopSS example file</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="PAINTOR.help_menu.html">PAINTOR.help_menu()</a></code> </p>
        </td>
        <td><p>Show PAINTOR help menu</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="PAINTOR.import_QTL_DT.html">PAINTOR.import_QTL_DT()</a></code> </p>
        </td>
        <td><p>Import QTL data for PAINTOR</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="PAINTOR.merge_results.html">PAINTOR.merge_results()</a></code> </p>
        </td>
        <td><p>Merge PAINTOR results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="PAINTOR.process_results.html">PAINTOR.process_results()</a></code> </p>
        </td>
        <td><p>Process PAINTOR results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="PAINTOR.run.html">PAINTOR.run()</a></code> </p>
        </td>
        <td><p>Run PAINTOR</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="PAINTOR.survey_annotation.html">PAINTOR.survey_annotation()</a></code> </p>
        </td>
        <td><p>Report the number of annotations &gt; 0</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="POLYFUN.get_ref_prefix.html">POLYFUN.get_ref_prefix()</a></code> </p>
        </td>
        <td><p>Get ref data path prefix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ROADMAP.construct_reference.html">ROADMAP.construct_reference()</a></code> </p>
        </td>
        <td><p>Gather Roadmap annotation metadata</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ROADMAP.merge_and_process_grl.html">ROADMAP.merge_and_process_grl()</a></code> </p>
        </td>
        <td><p>Standardize Roadmap query</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ROADMAP.query.html">ROADMAP.query()</a></code> </p>
        </td>
        <td><p>Query Roadmap by genomic coordinates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ROADMAP.query_and_plot.html">ROADMAP.query_and_plot()</a></code> </p>
        </td>
        <td><p>Query and plot Roadmap epigenomic annotations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ROADMAP.tabix.html">ROADMAP.tabix()</a></code> </p>
        </td>
        <td><p>Query Roadmap API</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ROADMAP.track_plot.html">ROADMAP.track_plot()</a></code> </p>
        </td>
        <td><p>Plot Roadmap query</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="TABIX.html">TABIX()</a></code> </p>
        </td>
        <td><p>Covert and query</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="TABIX.convert_file.html">TABIX.convert_file()</a></code> </p>
        </td>
        <td><p>Convert summary stats file to tabix format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="TABIX.query.html">TABIX.query()</a></code> </p>
        </td>
        <td><p>Query a tabix file</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="auto_topSNPs_sub.html">auto_topSNPs_sub()</a></code> </p>
        </td>
        <td><p>Automatically identify top SNP per locus</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculate.tstat.html">calculate.tstat()</a></code> </p>
        </td>
        <td><p>Compute t-stat</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="finemap_loci.html">finemap_loci()</a></code> </p>
        </td>
        <td><p>Fine-map multiple loci</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="finemap_pipeline.html">finemap_pipeline()</a></code> </p>
        </td>
        <td><p>Run <em>echolocatoR</em> pipeline on a single locus</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fullSS_dat.html">fullSS_dat</a></code> </p>
        </td>
        <td><p>"Example subset of full summary stats</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="genome_wide_dir.html">genome_wide_dir</a></code> </p>
        </td>
        <td><p>Example results path for genome-wide results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_UKB_MAF.html">get_UKB_MAF()</a></code> </p>
        </td>
        <td><p>Get MAF from UK Biobank.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_stepwise_results.html">get_stepwise_results()</a></code> </p>
        </td>
        <td><p>Gather stepwise conditional results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="import_topSNPs.html">import_topSNPs()</a></code> </p>
        </td>
        <td><p>Import top GWAS/QTL summary statistics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="locus_dir.html">locus_dir</a></code> </p>
        </td>
        <td><p>Example results path for BST1 locus</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="merge_finemapping_results.html">merge_finemapping_results()</a></code> </p>
        </td>
        <td><p>Merge fine-mapping results from all loci</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="merged_DT.html">merged_DT</a></code> </p>
        </td>
        <td><p><em>echolocatoR</em> output example (all loci)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="process_COJO_results.html">process_COJO_results()</a></code> </p>
        </td>
        <td><p>Gather all GCTA-COJO results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="snps_to_condition.html">snps_to_condition()</a></code> </p>
        </td>
        <td><p>Identify SNPs to condition on.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="standardize_subset.html">standardize_subset()</a></code> </p>
        </td>
        <td><p>Stanardize the locus subset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="top_SNPs.html">top_SNPs</a></code> </p>
        </td>
        <td><p>TopSS example file (processed)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="update_CS_cols.html">update_CS_cols()</a></code> </p>
        </td>
        <td><p>Update CS cols</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


