<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Procure an LD matrix for fine-mapping — LD.load_or_create • echolocatoR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Procure an LD matrix for fine-mapping — LD.load_or_create" />
<meta property="og:description" content="Calculate and/or query linkage disequilibrium (LD) from reference panels (UK Biobank, 1000 Genomes),
or user-supplied datasets." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/full_pipeline_vignette.html">Full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Procure an LD matrix for fine-mapping</h1>
    <small class="dont-index">Source: <a href='https://github.com/RajLabMSSM/echolocatoR/blob/master/R/LD.R'><code>R/LD.R</code></a></small>
    <div class="hidden name"><code>LD.load_or_create.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Calculate and/or query linkage disequilibrium (LD) from reference panels (UK Biobank, 1000 Genomes),
or user-supplied datasets.</p>
    </div>

    <pre class="usage"><span class='fu'>LD.load_or_create</span><span class='op'>(</span>
  <span class='va'>locus_dir</span>,
  <span class='va'>subset_DT</span>,
  force_new_LD <span class='op'>=</span> <span class='cn'>F</span>,
  LD_reference <span class='op'>=</span> <span class='st'>"1KGphase1"</span>,
  LD_genome_build <span class='op'>=</span> <span class='st'>"hg19"</span>,
  superpopulation <span class='op'>=</span> <span class='st'>"EUR"</span>,
  remote_LD <span class='op'>=</span> <span class='cn'>T</span>,
  download_method <span class='op'>=</span> <span class='st'>"direct"</span>,
  vcf_folder <span class='op'>=</span> <span class='cn'>NULL</span>,
  LD_block <span class='op'>=</span> <span class='cn'>F</span>,
  LD_block_size <span class='op'>=</span> <span class='fl'>0.7</span>,
  remove_correlates <span class='op'>=</span> <span class='cn'>F</span>,
  fillNA <span class='op'>=</span> <span class='fl'>0</span>,
  verbose <span class='op'>=</span> <span class='cn'>T</span>,
  server <span class='op'>=</span> <span class='cn'>F</span>,
  remove_tmps <span class='op'>=</span> <span class='cn'>T</span>,
  conda_env <span class='op'>=</span> <span class='st'>"echoR"</span>,
  nThread <span class='op'>=</span> <span class='fl'>4</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>subset_DT</th>
      <td><p>The locus subset of the full summary stats file.</p></td>
    </tr>
    <tr>
      <th>force_new_LD</th>
      <td><p>By default, if an LD matrix file for a given locus is already present,
then <span class="pkg">echolocatoR</span> will just use the preexisting file.
Set <code>force_new_LD=T</code> to override this and extract a new subset.</p></td>
    </tr>
    <tr>
      <th>LD_reference</th>
      <td><p>Which linkage disequilibrium reference panel do you want to use.
Options include:</p><dl>
<dt>"UKB"</dt><dd><p>A pre-caclulated LD reference matrix from a subset of caucasian British individuals from the UK Biobank. See <a href='https://www.biorxiv.org/content/10.1101/807792v2'>Wiessbrod et al. (2019)</a> for more details.</p></dd>
<dt>"1KGphase1"</dt><dd><p>Download a subset of the 1000 Genomes Project Phase 1 vcf and calculate LD on the fly with plink.</p></dd>
<dt>"1KGphase3"</dt><dd><p>Download a subset of the 1000 Genomes Project Phase 3 vcf and calculate LD on the fly with plink.</p></dd>
<dt>"&lt;path&gt;/*.vcf" or "&lt;path&gt;/*.vcf.gz"</dt><dd><p>Alternatively, users can provide their own custom panel by supplying a list of <em>.vcf</em> file path (one per locus) which <span class="pkg">echolocatoR</span> will use to compute LD (using <em>plink</em>).</p></dd>

</dl></td>
    </tr>
    <tr>
      <th>superpopulation</th>
      <td><p>Subset your LD reference panel by superopulation.
Setting the superpopulation is not currently possible when <code>LD_reference="UKB"</code>.
<a href='https://www.internationalgenome.org/faq/which-populations-are-part-your-study/'>1KGphase1 options</a> include:</p><dl>
<dt>"AFR"</dt><dd><p>African [descent]</p></dd>
<dt>"AMR"</dt><dd><p>Ad-mixed American</p></dd>
<dt>"EAS"</dt><dd><p>East Asian</p></dd>
<dt>"EUR"</dt><dd><p>European</p></dd>
<dt>"SAS"</dt><dd><p>South Asian</p></dd>

</dl></td>
    </tr>
    <tr>
      <th>remote_LD</th>
      <td><p>When acquiring LD matrixes,
the default is to delete the full vcf or npz files after <span class="pkg">echolocatoR</span> has extracted the necssary subset.
However, if you wish to keep these full files (which can be quite large) set <code>remote_LD=T</code>.</p></td>
    </tr>
    <tr>
      <th>LD_block</th>
      <td><p>Calculate LD blocks with <em>plink</em> and only include the block to which the lead SNP belongs.</p></td>
    </tr>
    <tr>
      <th>LD_block_size</th>
      <td><p>Adjust the granularity of block sizes when <code>LD_block=T</code>.</p></td>
    </tr>
    <tr>
      <th>remove_correlates</th>
      <td><p>A named list, where the names are the RSIDs of SNPs
whose LD correlates you wish to remove,
and the value is the absolute r2 threshold you wish to filter at for each RSID respectively
(e.g. <code> remove_correlates = c("rs76904798"=.2, "rs10000737"=.8)</code>).
This will also remove the SNPs in <code>remove_correlates</code> themselves.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Whether <span class="pkg">echolocatoR</span> should be verbose or silent.</p></td>
    </tr>
    <tr>
      <th>server</th>
      <td><p>Whether <span class="pkg">echolocatoR</span> is being run on a computing cluster/server or on a local machine.</p></td>
    </tr>
    <tr>
      <th>remove_tmps</th>
      <td><p>Whether to remove any temporary files (e.g. FINEMAP output files) after the pipeline is done running.</p></td>
    </tr>
    <tr>
      <th>conda_env</th>
      <td><p>The name of a conda environment to use.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A symmetric LD matrix of pairwise <em>r</em> values.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Options:</p><ul>
<li><p>Download pre-computed LD matrix from UK Biobank.</p></li>
<li><p>Download raw vcf file from 1KG and compute LD on the fly.</p></li>
<li><p>Compute LD on the fly from a user-supplied vcf file.</p></li>
<li><p>Use a user-supplied pre-computed LD-matrix.</p></li>
</ul>

    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other LD: 
<code><a href='LD.1KG_download_vcf.html'>LD.1KG_download_vcf</a>()</code>,
<code><a href='LD.1KG.html'>LD.1KG</a>()</code>,
<code><a href='LD.LD_blocks.html'>LD.LD_blocks</a>()</code>,
<code><a href='LD.UKBiobank.html'>LD.UKBiobank</a>()</code>,
<code><a href='LD.calculate_LD.html'>LD.calculate_LD</a>()</code>,
<code><a href='LD.construct_subset_vcf_name.html'>LD.construct_subset_vcf_name</a>()</code>,
<code><a href='LD.custom_panel.html'>LD.custom_panel</a>()</code>,
<code><a href='LD.dprime_table.html'>LD.dprime_table</a>()</code>,
<code><a href='LD.filter_LD.html'>LD.filter_LD</a>()</code>,
<code><a href='LD.filter_vcf_gaston.html'>LD.filter_vcf_gaston</a>()</code>,
<code><a href='LD.filter_vcf.html'>LD.filter_vcf</a>()</code>,
<code><a href='LD.get_locus_vcf_folder.html'>LD.get_locus_vcf_folder</a>()</code>,
<code><a href='LD.index_vcf.html'>LD.index_vcf</a>()</code>,
<code><a href='LD.leadSNP_block.html'>LD.leadSNP_block</a>()</code>,
<code><a href='LD.plink_LD.html'>LD.plink_LD</a>()</code>,
<code><a href='LD.plink_file.html'>LD.plink_file</a>()</code>,
<code><a href='LD.plot_LD.html'>LD.plot_LD</a>()</code>,
<code><a href='LD.query_vcf.html'>LD.query_vcf</a>()</code>,
<code><a href='LD.rds_to_npz.html'>LD.rds_to_npz</a>()</code>,
<code><a href='LD.read_bin.html'>LD.read_bin</a>()</code>,
<code><a href='LD.read_ld_table.html'>LD.read_ld_table</a>()</code>,
<code><a href='LD.run_plink_LD.html'>LD.run_plink_LD</a>()</code>,
<code><a href='LD.save_LD_matrix.html'>LD.save_LD_matrix</a>()</code>,
<code><a href='LD.snpstats_get_LD.html'>LD.snpstats_get_LD</a>()</code>,
<code><a href='LD.snpstats_get_MAF.html'>LD.snpstats_get_MAF</a>()</code>,
<code><a href='LD.translate_population.html'>LD.translate_population</a>()</code>,
<code><a href='LD.vcf_to_bed.html'>LD.vcf_to_bed</a>()</code>,
<code><a href='LDlinkR.LDproxy_batch.html'>LDlinkR.LDproxy_batch</a>()</code>,
<code><a href='popDat_1KGphase1.html'>popDat_1KGphase1</a></code>,
<code><a href='popDat_1KGphase3.html'>popDat_1KGphase3</a></code>,
<code><a href='saveSparse.html'>saveSparse</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>"BST1"</span><span class='op'>)</span>; <span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>"locus_dir"</span><span class='op'>)</span>;
<span class='va'>locus_dir</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='st'>"~/Desktop"</span>,<span class='va'>locus_dir</span><span class='op'>)</span>
 <span class='co'># BST1 &lt;- limit_SNPs(500, BST1)</span>

<span class='co'># UK Biobank LD</span>
<span class='va'>LD_matrix</span> <span class='op'>&lt;-</span> <span class='fu'>LD.load_or_create</span><span class='op'>(</span>locus_dir<span class='op'>=</span><span class='va'>locus_dir</span>, subset_DT<span class='op'>=</span><span class='va'>BST1</span>, LD_reference<span class='op'>=</span><span class='st'>"UKB"</span><span class='op'>)</span>

<span class='co'># 1000 Genomes</span>
<span class='va'>LD_matrix</span> <span class='op'>&lt;-</span> <span class='fu'>LD.load_or_create</span><span class='op'>(</span>locus_dir<span class='op'>=</span><span class='va'>locus_dir</span>, subset_DT<span class='op'>=</span><span class='va'>BST1</span>, LD_reference<span class='op'>=</span><span class='st'>"1KGphase1"</span>, force_new_LD<span class='op'>=</span><span class='cn'>T</span><span class='op'>)</span>

<span class='co'># Local vcf file</span>
<span class='va'>LD_reference</span><span class='op'>=</span><span class='st'>"~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/LD/BST1.1KGphase1.vcf.gz"</span>
<span class='va'>LD_matrix</span> <span class='op'>&lt;-</span> <span class='fu'>LD.load_or_create</span><span class='op'>(</span>locus_dir<span class='op'>=</span><span class='va'>locus_dir</span>, subset_DT<span class='op'>=</span><span class='va'>BST1</span>, LD_reference<span class='op'>=</span><span class='va'>LD_reference</span>, force_new_LD<span class='op'>=</span><span class='cn'>T</span><span class='op'>)</span>
<span class='op'>}</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


