<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generate a locus plot — PLOT.locus • echolocatoR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Generate a locus plot — PLOT.locus" />
<meta property="og:description" content="Generate a locus-specific plot with multiple selectable tracks.
Users can also generate multiple zoomed in views of the plot at multiple resolutions." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/full_pipeline_vignette.html">Full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate a locus plot</h1>
    <small class="dont-index">Source: <a href='https://github.com/RajLabMSSM/echolocatoR/blob/master/R/PLOT.locus.R'><code>R/PLOT.locus.R</code></a></small>
    <div class="hidden name"><code>PLOT.locus.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generate a locus-specific plot with multiple selectable tracks.
Users can also generate multiple zoomed in views of the plot at multiple resolutions.</p>
    </div>

    <pre class="usage"><span class='fu'>PLOT.locus</span><span class='op'>(</span>
  <span class='va'>finemap_dat</span>,
  <span class='va'>locus_dir</span>,
  LD_matrix <span class='op'>=</span> <span class='cn'>NULL</span>,
  LD_reference <span class='op'>=</span> <span class='cn'>NULL</span>,
  dataset_type <span class='op'>=</span> <span class='st'>"GWAS"</span>,
  color_r2 <span class='op'>=</span> <span class='cn'>T</span>,
  method_list <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ABF"</span>, <span class='st'>"FINEMAP"</span>, <span class='st'>"SUSIE"</span>, <span class='st'>"POLYFUN_SUSIE"</span><span class='op'>)</span>,
  track_order <span class='op'>=</span> <span class='cn'>NULL</span>,
  track_heights <span class='op'>=</span> <span class='cn'>NULL</span>,
  plot_full_window <span class='op'>=</span> <span class='cn'>T</span>,
  dot_summary <span class='op'>=</span> <span class='cn'>F</span>,
  QTL_prefixes <span class='op'>=</span> <span class='cn'>NULL</span>,
  mean.PP <span class='op'>=</span> <span class='cn'>T</span>,
  PP_threshold <span class='op'>=</span> <span class='fl'>0.95</span>,
  consensus_threshold <span class='op'>=</span> <span class='fl'>2</span>,
  sig_cutoff <span class='op'>=</span> <span class='fl'>5e-08</span>,
  gene_track <span class='op'>=</span> <span class='cn'>T</span>,
  point_size <span class='op'>=</span> <span class='fl'>1</span>,
  point_alpha <span class='op'>=</span> <span class='fl'>0.6</span>,
  snp_group_lines <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Lead"</span>, <span class='st'>"UCS"</span>, <span class='st'>"Consensus"</span><span class='op'>)</span>,
  xtext <span class='op'>=</span> <span class='cn'>F</span>,
  show.legend_genes <span class='op'>=</span> <span class='cn'>T</span>,
  XGR_libnames <span class='op'>=</span> <span class='cn'>NULL</span>,
  n_top_xgr <span class='op'>=</span> <span class='fl'>5</span>,
  Roadmap <span class='op'>=</span> <span class='cn'>F</span>,
  Roadmap_query <span class='op'>=</span> <span class='cn'>NULL</span>,
  n_top_roadmap <span class='op'>=</span> <span class='fl'>7</span>,
  annot_overlap_threshold <span class='op'>=</span> <span class='fl'>5</span>,
  zoom_exceptions_str <span class='op'>=</span> <span class='st'>"*full window$|zoom_polygon"</span>,
  Nott_epigenome <span class='op'>=</span> <span class='cn'>F</span>,
  Nott_regulatory_rects <span class='op'>=</span> <span class='cn'>T</span>,
  Nott_show_placseq <span class='op'>=</span> <span class='cn'>T</span>,
  Nott_binwidth <span class='op'>=</span> <span class='fl'>200</span>,
  Nott_bigwig_dir <span class='op'>=</span> <span class='cn'>NULL</span>,
  save_plot <span class='op'>=</span> <span class='cn'>T</span>,
  show_plot <span class='op'>=</span> <span class='cn'>T</span>,
  genomic_units <span class='op'>=</span> <span class='st'>"Mb"</span>,
  strip.text.y.angle <span class='op'>=</span> <span class='fl'>0</span>,
  max_transcripts <span class='op'>=</span> <span class='fl'>1</span>,
  plot.zoom <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"1x"</span><span class='op'>)</span>,
  dpi <span class='op'>=</span> <span class='fl'>300</span>,
  height <span class='op'>=</span> <span class='fl'>12</span>,
  width <span class='op'>=</span> <span class='fl'>10</span>,
  plot_format <span class='op'>=</span> <span class='st'>"jpg"</span>,
  save_RDS <span class='op'>=</span> <span class='cn'>T</span>,
  return_list <span class='op'>=</span> <span class='cn'>F</span>,
  conda_env <span class='op'>=</span> <span class='st'>"echoR"</span>,
  nThread <span class='op'>=</span> <span class='fl'>4</span>,
  verbose <span class='op'>=</span> <span class='cn'>T</span>
<span class='op'>)</span></pre>


    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>#### TO-do
Add anchors to plac-seq arches</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/RajLabMSSM/echolocatoR'>echolocatoR</a></span><span class='op'>)</span>
<span class='va'>finemap_dat</span><span class='op'>&lt;-</span> <span class='fu'>echolocatoR</span><span class='fu'>::</span><span class='va'><a href='BST1.html'>BST1</a></span>; <span class='va'>LD_matrix</span> <span class='op'>&lt;-</span> <span class='fu'>echolocatoR</span><span class='fu'>::</span><span class='va'><a href='BST1_LD_matrix.html'>BST1_LD_matrix</a></span>;
<span class='va'>locus_dir</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='st'>"~/Desktop"</span>,<span class='st'>"results/GWAS/Nalls23andMe_2019/BST1"</span><span class='op'>)</span>

<span class='va'>locus_plot</span> <span class='op'>&lt;-</span> <span class='fu'>PLOT.locus</span><span class='op'>(</span><span class='va'>finemap_dat</span>, locus_dir<span class='op'>=</span><span class='va'>locus_dir</span>, LD_matrix<span class='op'>=</span><span class='va'>LD_matrix</span>, Nott_epigenome<span class='op'>=</span><span class='cn'>T</span>, xtext<span class='op'>=</span><span class='cn'>F</span>, plot.zoom<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"5x"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>+-------- Locus Plot: BST1--------+</span></div><div class='output co'>#&gt; [1] "+ Filling NAs in CS cols with 0"
#&gt; [1] "+ Filling NAs in PP cols with 0"
#&gt; [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."
#&gt; [1] "++ PLOT:: GWAS full window track"
#&gt; [1] "++ PLOT:: GWAS track"
#&gt; [1] "++ PLOT:: Merged fine-mapping track"
#&gt; [1] "Melting PP and CS from 5 fine-mapping methods"</div><div class='output co'>#&gt; <span class='warning'>Warning: The `.dots` argument of `group_by()` is deprecated as of dplyr 1.0.0.</span>
#&gt; <span class='warning'><span style='color: #555555;'>This warning is displayed once every 8 hours.</span><span></span>
#&gt; <span class='warning'></span><span style='color: #555555;'>Call `lifecycle::last_warnings()` to see where this warning was generated.</span><span></span></div><div class='output co'>#&gt; [1] "++ PLOT:: Adding Gene model track."</div><div class='output co'>#&gt; <span class='message'>max_transcripts=1. </span></div><div class='output co'>#&gt; <span class='message'>15 transcripts from 15 genes returned.</span></div><div class='output co'>#&gt; <span class='message'>Fetching data...</span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Parsing exons...</span></div><div class='output co'>#&gt; <span class='message'>OK</span>
#&gt; <span class='message'>Defining introns...</span></div><div class='output co'>#&gt; <span class='message'>OK</span>
#&gt; <span class='message'>Defining UTRs...</span></div><div class='output co'>#&gt; <span class='message'>OK</span>
#&gt; <span class='message'>Defining CDS...</span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>aggregating...</span></div><div class='output co'>#&gt; <span class='message'>Done</span></div><div class='output co'>#&gt; <span class='message'>Constructing graphics...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `quo_expr()` is deprecated as of rlang 0.2.0.</span>
#&gt; <span class='warning'>Please use `quo_squash()` instead.</span>
#&gt; <span class='warning'></span><span style='color: #555555;'>This warning is displayed once per session.</span><span></span></div><div class='output co'>#&gt; [1] "NOTT_2019:: Creating epigenomic histograms plot"
#&gt; [1] "NOTT_2019:: Importing bigWig subsets from UCSC..."
#&gt; [1] "+ Inferring genomic limits for window: 1x"
#&gt; [1] "++ NOTT_2019:: Downloading and merging 12 peaks BED files."
#&gt; [1] "++ NOTT_2019:: Converting merged BED files to GRanges."
#&gt; [1] "++ NOTT_2019:: 634540 ranges retrieved."</div><div class='output co'>#&gt; <span class='message'>1860925 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; [1] "+ PLOT:: Calculating max histogram height"
#&gt; [1] "++ Converting label units to Mb..."
#&gt; [1] "NOTT_2019:: Creating PLAC-seq interactome plot"
#&gt; [1] "+ NOTT_2019:: Getting interactome data."
#&gt; [1] "++ Converting to GRanges."
#&gt; [1] "Importing Microglia interactome ..."
#&gt; [1] "Importing Neuronal interactome ..."
#&gt; [1] "Importing Oligo interactome ..."</div><div class='output co'>#&gt; <span class='warning'>Warning: found more than one best sequence renaming map compatible with seqname style "NCBI" for this object, using the first one</span></div><div class='output co'>#&gt; <span class='message'>1403 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: found more than one best sequence renaming map compatible with seqname style "NCBI" for this object, using the first one</span></div><div class='output co'>#&gt; <span class='message'>1770 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; [1] "++ NOTT_2019:: Adding enhancer/promoter rectangles"
#&gt; [1] "++ Converting genomic units to Mb..."
#&gt; [1] "++ NOTT_2019:: Returning PLAC-seq track."
#&gt; [1] "+ Adding vertical lines to highlight SNP groups..."</div><div class='output co'>#&gt; <span class='message'>+&gt;+&gt;+&gt;+&gt;+ plot.zoom = 5x +&lt;+&lt;+&lt;+&lt;+</span></div><div class='output co'>#&gt; [1] "+ PLOT:: Get window suffix..."
#&gt; [1] "+ Constructing zoom polygon..."
#&gt; [1] "+ Highlighting zoom origin..."
#&gt; [1] "+ Removing subplot margins..."
#&gt; [1] "+ Reordering tracks..."
#&gt; [1] "+ Ensuring last track shows genomic units..."
#&gt; [1] "+ Aligning xlimits for each subplot..."
#&gt; [1] "+ Checking track heights..."
#&gt; [1] "+ PLOT:: Saving plot ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/multiview.BST1..5x.jpg"</div><div class='output co'>#&gt; <span class='warning'>Warning: ggrepel: 4 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></div><div class='output co'>#&gt; <span class='warning'>Warning: ggrepel: 4 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></div><div class='output co'>#&gt; [1] "+ PLOT:: Saving ggplots tracks list ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/plot_tracks/multiview.BST1..5x.RDS"</div><div class='output co'>#&gt; <span class='warning'>Warning: ggrepel: 7 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></div><div class='output co'>#&gt; <span class='warning'>Warning: ggrepel: 7 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></div><div class='img'><img src='PLOT.locus-1.png' alt='' width='700' height='433' /></div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


