<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate a locus plot — PLOT.locus • echolocatoR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Generate a locus plot — PLOT.locus"><meta property="og:description" content="Generate a locus-specific plot with multiple selectable tracks.
Users can also generate multiple zoomed in views of the plot at multiple resolutions."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echolocatoR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate a locus plot</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/HEAD/R/PLOT.locus.R" class="external-link"><code>R/PLOT.locus.R</code></a></small>
    <div class="hidden name"><code>PLOT.locus.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generate a locus-specific plot with multiple selectable tracks.
Users can also generate multiple zoomed in views of the plot at multiple resolutions.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">PLOT.locus</span><span class="op">(</span>
  <span class="va">finemap_dat</span>,
  <span class="va">locus_dir</span>,
  LD_matrix <span class="op">=</span> <span class="cn">NULL</span>,
  LD_reference <span class="op">=</span> <span class="cn">NULL</span>,
  dataset_type <span class="op">=</span> <span class="st">"GWAS"</span>,
  color_r2 <span class="op">=</span> <span class="cn">T</span>,
  method_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ABF"</span>, <span class="st">"FINEMAP"</span>, <span class="st">"SUSIE"</span>, <span class="st">"POLYFUN_SUSIE"</span><span class="op">)</span>,
  track_order <span class="op">=</span> <span class="cn">NULL</span>,
  track_heights <span class="op">=</span> <span class="cn">NULL</span>,
  plot_full_window <span class="op">=</span> <span class="cn">T</span>,
  dot_summary <span class="op">=</span> <span class="cn">F</span>,
  QTL_prefixes <span class="op">=</span> <span class="cn">NULL</span>,
  mean.PP <span class="op">=</span> <span class="cn">T</span>,
  PP_threshold <span class="op">=</span> <span class="fl">0.95</span>,
  consensus_threshold <span class="op">=</span> <span class="fl">2</span>,
  sig_cutoff <span class="op">=</span> <span class="fl">5e-08</span>,
  gene_track <span class="op">=</span> <span class="cn">T</span>,
  point_size <span class="op">=</span> <span class="fl">1</span>,
  point_alpha <span class="op">=</span> <span class="fl">0.6</span>,
  snp_group_lines <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Lead"</span>, <span class="st">"UCS"</span>, <span class="st">"Consensus"</span><span class="op">)</span>,
  xtext <span class="op">=</span> <span class="cn">F</span>,
  show.legend_genes <span class="op">=</span> <span class="cn">T</span>,
  XGR_libnames <span class="op">=</span> <span class="cn">NULL</span>,
  n_top_xgr <span class="op">=</span> <span class="fl">5</span>,
  Roadmap <span class="op">=</span> <span class="cn">F</span>,
  Roadmap_query <span class="op">=</span> <span class="cn">NULL</span>,
  n_top_roadmap <span class="op">=</span> <span class="fl">7</span>,
  annot_overlap_threshold <span class="op">=</span> <span class="fl">5</span>,
  zoom_exceptions_str <span class="op">=</span> <span class="st">"*full window$|zoom_polygon"</span>,
  Nott_epigenome <span class="op">=</span> <span class="cn">F</span>,
  Nott_regulatory_rects <span class="op">=</span> <span class="cn">T</span>,
  Nott_show_placseq <span class="op">=</span> <span class="cn">T</span>,
  Nott_binwidth <span class="op">=</span> <span class="fl">200</span>,
  Nott_bigwig_dir <span class="op">=</span> <span class="cn">NULL</span>,
  save_plot <span class="op">=</span> <span class="cn">T</span>,
  show_plot <span class="op">=</span> <span class="cn">T</span>,
  genomic_units <span class="op">=</span> <span class="st">"Mb"</span>,
  strip.text.y.angle <span class="op">=</span> <span class="fl">0</span>,
  max_transcripts <span class="op">=</span> <span class="fl">1</span>,
  plot.zoom <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1x"</span><span class="op">)</span>,
  dpi <span class="op">=</span> <span class="fl">300</span>,
  height <span class="op">=</span> <span class="fl">12</span>,
  width <span class="op">=</span> <span class="fl">10</span>,
  plot_format <span class="op">=</span> <span class="st">"jpg"</span>,
  save_RDS <span class="op">=</span> <span class="cn">F</span>,
  return_list <span class="op">=</span> <span class="cn">F</span>,
  conda_env <span class="op">=</span> <span class="st">"echoR"</span>,
  nThread <span class="op">=</span> <span class="fl">4</span>,
  verbose <span class="op">=</span> <span class="cn">T</span>
<span class="op">)</span></code></pre></div>
    </div>


    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echolocatoR" class="external-link">echolocatoR</a></span><span class="op">)</span></span>
<span class="r-in"><span class="va">finemap_dat</span><span class="op">&lt;-</span> <span class="fu">echolocatoR</span><span class="fu">::</span><span class="va"><a href="BST1.html">BST1</a></span>; <span class="va">LD_matrix</span> <span class="op">&lt;-</span> <span class="fu">echolocatoR</span><span class="fu">::</span><span class="va"><a href="BST1_LD_matrix.html">BST1_LD_matrix</a></span>;</span>
<span class="r-in"><span class="va">locus_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"~/Desktop"</span>,<span class="st">"results/GWAS/Nalls23andMe_2019/BST1"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">locus_plot</span> <span class="op">&lt;-</span> <span class="fu">PLOT.locus</span><span class="op">(</span><span class="va">finemap_dat</span>, locus_dir<span class="op">=</span><span class="va">locus_dir</span>, LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, Nott_epigenome<span class="op">=</span><span class="cn">T</span>, xtext<span class="op">=</span><span class="cn">F</span>, plot.zoom<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"5x"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> +-------- Locus Plot: BST1--------+</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Filling NAs in CS cols with 0"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Filling NAs in PP cols with 0"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ LD:: LD_matrix detected. Coloring SNPs by LD with lead SNP."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "++ PLOT:: GWAS full window track"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "++ PLOT:: GWAS track"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "++ PLOT:: Merged fine-mapping track"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Melting PP and CS from 5 fine-mapping methods"</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The `.dots` argument of `group_by()` is deprecated as of dplyr 1.0.0.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "++ PLOT:: Adding Gene model track."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error in dplyr::slice_max(., order_by = c(symbol, width), n = max_transcripts,  : </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Problem while computing indices.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> The error occurred in group 1: symbol = "AC006427.2".</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Caused by error:</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> `order_by` must have size 1, not size 2.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "NOTT_2019:: Creating epigenomic histograms plot"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "NOTT_2019:: Importing bigWig subsets from UCSC..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Inferring genomic limits for window: 1x"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "++ NOTT_2019:: Downloading and merging 12 peaks BED files."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "++ NOTT_2019:: Converting merged BED files to GRanges."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "++ NOTT_2019:: 634540 ranges retrieved."</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1860925 query SNP(s) detected with reference overlap.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ PLOT:: Calculating max histogram height"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "++ Converting label units to Mb..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "NOTT_2019:: Creating PLAC-seq interactome plot"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ NOTT_2019:: Getting interactome data."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "++ Converting to GRanges."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Importing Microglia interactome ..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Importing Neuronal interactome ..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Importing Oligo interactome ..."</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1403 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1770 query SNP(s) detected with reference overlap.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error in h(simpleError(msg, call)) : </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   error in evaluating the argument 'args' in selecting a method for function 'do.call': GRanges objects don't support [[, as.list(), lapply(), or unlist() at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   the moment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Adding vertical lines to highlight SNP groups..."</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> +&gt;+&gt;+&gt;+&gt;+ plot.zoom = 5x +&lt;+&lt;+&lt;+&lt;+</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ PLOT:: Get window suffix..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Constructing zoom polygon..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Highlighting zoom origin..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Removing subplot margins..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Reordering tracks..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Ensuring last track shows genomic units..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Aligning xlimits for each subplot..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Checking track heights..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ PLOT:: Saving plot ==&gt; ~/Desktop/results/GWAS/Nalls23andMe_2019/BST1/multiview.BST1..5x.jpg"</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>ggrepel: 5 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>ggrepel: 5 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>ggrepel: 7 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>ggrepel: 7 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span>
<span class="r-plt img"><img src="PLOT.locus-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

