<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Use <em>awk</em> to query locus subsets. — query_by_coordinates_merged • echolocatoR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Use <em>awk</em> to query locus subsets. — query_by_coordinates_merged" />
<meta property="og:description" content="Search full summary stats file by genomics coordinates.
To be used when CHR and POS are merged into one column (e.g. chr12:12209944)." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/full_pipeline_vignette.html">Full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Use <em>awk</em> to query locus subsets.</h1>
    <small class="dont-index">Source: <a href='https://github.com/RajLabMSSM/echolocatoR/blob/master/R/query.R'><code>R/query.R</code></a></small>
    <div class="hidden name"><code>query_by_coordinates_merged.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Search full summary stats file by genomics coordinates.
To be used when <strong>CHR</strong> and <strong>POS</strong> are merged into one column (e.g. chr12:12209944).</p>
    </div>

    <pre class="usage"><span class='fu'>query_by_coordinates_merged</span>(
  <span class='no'>top_SNPs</span>,
  <span class='no'>fullSS_path</span>,
  <span class='no'>subset_path</span>,
  <span class='no'>locus</span>,
  <span class='no'>chrom_col</span>,
  <span class='kw'>file_sep</span> <span class='kw'>=</span> <span class='st'>" "</span>,
  <span class='kw'>location_sep</span> <span class='kw'>=</span> <span class='st'>":"</span>,
  <span class='no'>min_POS</span>,
  <span class='no'>max_POS</span>,
  <span class='no'>bp_distance</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>top_SNPs</th>
      <td><p>A data.frame with the genomic coordinates of the lead SNP for each locus.
The lead SNP will be used as the center of the window when extracting subset from the full GWAS/QTL summary statistics file.
Only one SNP per <strong>Locus</strong> should be included.
At minimum, <code>top_SNPs</code> should include the following columns:</p><dl'>
<dt><em>Locus</em></dt><dd><p>A unique name for each locus. Often, loci are named after a relevant gene (e.g. LRRK2) or based on the name/coordinates of the lead SNP (e.g. locus_chr12_40734202)</p></dd>
<dt><em>CHR</em></dt><dd><p>The chromosome that the SNP is on. Can be "chr12" or "12" format.</p></dd>
<dt><em>POS</em></dt><dd><p>The genomic position of the SNP (in basepairs)</p></dd>

</dl></td>
    </tr>
    <tr>
      <th>fullSS_path</th>
      <td><p>Path to the full summary statistics file (GWAS or QTL) that you want to fine-map.</p></td>
    </tr>
    <tr>
      <th>chrom_col</th>
      <td><p>Name of the chromosome column in the full summary stats file.
Can be "chr1" or "1" format.
(<em>default: ="CHR"</em>)</p></td>
    </tr>
    <tr>
      <th>file_sep</th>
      <td><p>The separator in the full summary stats file.
This parameter is only necessary if <code>query_by!="tabix"</code>.</p></td>
    </tr>
    <tr>
      <th>location_sep</th>
      <td><p>The separator character when <strong>CHR</strong> and <strong>POS</strong> are merged into one column (e.g. ":" when formatted like chr12:12209944).</p></td>
    </tr>
    <tr>
      <th>min_POS</th>
      <td><p>Manually set the minimum genomic position for your locus subset.
<code>min_POS</code> can clip the window size set by <code>bp_distance</code>.
Only use this parameter when fine-mapping one locus at a time.</p></td>
    </tr>
    <tr>
      <th>max_POS</th>
      <td><p>Manually set the maximum genomic position for your locus subset.
<code>max_POS</code> can clip the window size set by <code>bp_distance</code>.
Only use this parameter when fine-mapping one locus at a time.</p></td>
    </tr>
    <tr>
      <th>bp_distance</th>
      <td><p>The width of the window size you want each locus to be.
For example, if <code>bp_distance=500000</code> then the locus will span 500kb from the lead SNP in either direction,
resulting in a locus that is ~1Mb long (depending on the dataset).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other query functions: 
<code><a href='TABIX.convert_file.html'>TABIX.convert_file</a>()</code>,
<code><a href='TABIX.query.html'>TABIX.query</a>()</code>,
<code><a href='TABIX.html'>TABIX</a>()</code>,
<code><a href='extract_SNP_subset.html'>extract_SNP_subset</a>()</code>,
<code><a href='import_topSNPs.html'>import_topSNPs</a>()</code>,
<code><a href='query_by_coordinates.html'>query_by_coordinates</a>()</code>,
<code><a href='query_by_gene.html'>query_by_gene</a>()</code>,
<code><a href='query_by_probe.html'>query_by_probe</a>()</code>,
<code><a href='query_fullSS.html'>query_fullSS</a>()</code>,
<code><a href='query_handler.html'>query_handler</a>()</code></p></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


