<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Use awk to query locus subsets. — query_by_coordinates_merged • echolocatoR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Use awk to query locus subsets. — query_by_coordinates_merged"><meta property="og:description" content="Search full summary stats file by genomics coordinates.
To be used when CHR and POS are merged into one column (e.g. chr12:12209944)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echolocatoR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Use <em>awk</em> to query locus subsets.</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/HEAD/R/query.R" class="external-link"><code>R/query.R</code></a></small>
    <div class="hidden name"><code>query_by_coordinates_merged.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Search full summary stats file by genomics coordinates.
To be used when <strong>CHR</strong> and <strong>POS</strong> are merged into one column (e.g. chr12:12209944).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">query_by_coordinates_merged</span><span class="op">(</span>
  <span class="va">top_SNPs</span>,
  <span class="va">fullSS_path</span>,
  <span class="va">subset_path</span>,
  <span class="va">locus</span>,
  <span class="va">chrom_col</span>,
  file_sep <span class="op">=</span> <span class="st">" "</span>,
  location_sep <span class="op">=</span> <span class="st">":"</span>,
  <span class="va">min_POS</span>,
  <span class="va">max_POS</span>,
  <span class="va">bp_distance</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>top_SNPs</dt>
<dd><p>A data.frame with the genomic coordinates of the lead SNP for each locus.
The lead SNP will be used as the center of the window when extracting subset from the full GWAS/QTL summary statistics file.
Only one SNP per <strong>Locus</strong> should be included.
At minimum, <code>top_SNPs</code> should include the following columns:</p><dl><dt><em>Locus</em></dt>
<dd><p>A unique name for each locus. Often, loci are named after a relevant gene (e.g. LRRK2) or based on the name/coordinates of the lead SNP (e.g. locus_chr12_40734202)</p></dd>

<dt><em>CHR</em></dt>
<dd><p>The chromosome that the SNP is on. Can be "chr12" or "12" format.</p></dd>

<dt><em>POS</em></dt>
<dd><p>The genomic position of the SNP (in basepairs)</p></dd>


</dl></dd>
<dt>fullSS_path</dt>
<dd><p>Path to the full summary statistics file (GWAS or QTL) that you want to fine-map.
It is usually best to provide the absolute path rather than the relative path.</p></dd>
<dt>chrom_col</dt>
<dd><p>Name of the chromosome column in the full summary stats file.
Can be "chr1" or "1" format.
(<em>default: ="CHR"</em>)</p></dd>
<dt>file_sep</dt>
<dd><p>The separator in the full summary stats file.
This parameter is only necessary if <code>query_by!="tabix"</code>.</p></dd>
<dt>location_sep</dt>
<dd><p>The separator character when <strong>CHR</strong> and <strong>POS</strong> are merged into one column (e.g. ":" when formatted like chr12:12209944).</p></dd>
<dt>min_POS</dt>
<dd><p>Manually set the minimum genomic position for your locus subset.
<code>min_POS</code> can clip the window size set by <code>bp_distance</code>.
Can also be a list of positions (one for each locus) (e.g. <code>min_POS=top_SNPs$min_POS</code>).</p></dd>
<dt>max_POS</dt>
<dd><p>Manually set the maximum genomic position for your locus subset.
<code>max_POS</code> can clip the window size set by <code>bp_distance</code>.
Can also be a list of positions (one for each locus) (e.g. <code>max_POS=top_SNPs$max_POS</code>).</p></dd>
<dt>bp_distance</dt>
<dd><p>The width of the window size you want each locus to be.
For example, if <code>bp_distance=500000</code> then the locus will span 500kb from the lead SNP in either direction,
resulting in a locus that is ~1Mb long (depending on the dataset).</p></dd>
</dl></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other query functions: 
<code><a href="TABIX.convert_file.html">TABIX.convert_file</a>()</code>,
<code><a href="TABIX.query.html">TABIX.query</a>()</code>,
<code><a href="TABIX.html">TABIX</a>()</code>,
<code><a href="extract_SNP_subset.html">extract_SNP_subset</a>()</code>,
<code><a href="import_topSNPs.html">import_topSNPs</a>()</code>,
<code><a href="query_by_coordinates.html">query_by_coordinates</a>()</code>,
<code><a href="query_by_gene.html">query_by_gene</a>()</code>,
<code><a href="query_by_probe.html">query_by_probe</a>()</code>,
<code><a href="query_fullSS.html">query_fullSS</a>()</code>,
<code><a href="query_handler.html">query_handler</a>()</code></p></div>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

