<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Fine-map with SUSIE — SUSIE • echolocatoR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Fine-map with SUSIE — SUSIE" />
<meta property="og:description" content="Sum of Single Effects (SuSiE): Iterative Bayesian Step-wise Selection" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/full_pipeline_vignette.html">Full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Fine-map with SUSIE</h1>
    <small class="dont-index">Source: <a href='https://github.com/RajLabMSSM/echolocatoR/blob/master/R/SUSIE.R'><code>R/SUSIE.R</code></a></small>
    <div class="hidden name"><code>SUSIE.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Sum of Single Effects (SuSiE): Iterative Bayesian Step-wise Selection</p>
    </div>

    <pre class="usage"><span class='fu'>SUSIE</span><span class='op'>(</span>
  <span class='va'>subset_DT</span>,
  <span class='va'>LD_matrix</span>,
  dataset_type <span class='op'>=</span> <span class='st'>"GWAS"</span>,
  max_causal <span class='op'>=</span> <span class='fl'>5</span>,
  sample_size <span class='op'>=</span> <span class='cn'>NULL</span>,
  prior_weights <span class='op'>=</span> <span class='cn'>NULL</span>,
  PP_threshold <span class='op'>=</span> <span class='fl'>0.95</span>,
  scaled_prior_variance <span class='op'>=</span> <span class='fl'>0.001</span>,
  estimate_residual_variance <span class='op'>=</span> <span class='cn'>F</span>,
  estimate_prior_variance <span class='op'>=</span> <span class='cn'>T</span>,
  residual_variance <span class='op'>=</span> <span class='cn'>NULL</span>,
  max_iter <span class='op'>=</span> <span class='fl'>100</span>,
  estimate_prior_method <span class='op'>=</span> <span class='st'>"optim"</span>,
  manual_var_y <span class='op'>=</span> <span class='cn'>F</span>,
  rescale_priors <span class='op'>=</span> <span class='cn'>T</span>,
  plot_track_fit <span class='op'>=</span> <span class='cn'>F</span>,
  return_all_CS <span class='op'>=</span> <span class='cn'>T</span>,
  verbose <span class='op'>=</span> <span class='cn'>T</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dataset_type</th>
      <td><p>The kind dataset you're fine-mapping (e.g. GWAS, eQTL, tQTL).
This will also be used when creating the subdirectory where your results will be stored
(e.g. <em>Data/&lt;dataset_type&gt;/Kunkle_2019</em>).</p></td>
    </tr>
    <tr>
      <th>max_causal</th>
      <td><p>The maximum number of non-zero effects (and thus causal variants).</p></td>
    </tr>
    <tr>
      <th>sample_size</th>
      <td><p>The overall sample size of the study.
If none is given, and <strong>N_cases</strong> and <strong>N_controls</strong> columns are present,
then sample_size is inferred to be:  <code>max(N_cases) + max(N_controls)</code>.</p></td>
    </tr>
    <tr>
      <th>prior_weights</th>
      <td><p>a p vector of prior probability that each element is non-zero</p></td>
    </tr>
    <tr>
      <th>PP_threshold</th>
      <td><p>The minimum fine-mapped posterior probability for a SNP to be considered part of a Credible Set.
For example, <code>PP_threshold=.95</code> means that all Credible Set SNPs will be 95% Credible Set SNPs.</p></td>
    </tr>
    <tr>
      <th>scaled_prior_variance</th>
      <td><p>the scaled prior variance (vector of length L, or scalar. In latter case gets repeated L times)</p></td>
    </tr>
    <tr>
      <th>estimate_residual_variance</th>
      <td><p>indicates whether to estimate residual variance</p></td>
    </tr>
    <tr>
      <th>estimate_prior_variance</th>
      <td><p>indicates whether to estimate prior</p></td>
    </tr>
    <tr>
      <th>residual_variance</th>
      <td><p>the residual variance (defaults to variance of y)</p></td>
    </tr>
    <tr>
      <th>max_iter</th>
      <td><p>maximum number of iterations to perform</p></td>
    </tr>
    <tr>
      <th>estimate_prior_method</th>
      <td><p>The method used for estimating prior variance.</p></td>
    </tr>
    <tr>
      <th>rescale_priors</th>
      <td><p>If prior probabiltities are supplied,
rescale them from 0-1 (i.e. <code>rescaled_priors = priors / sum(priors)</code>).</p></td>
    </tr>
    <tr>
      <th>plot_track_fit</th>
      <td><p>Record each iteration and make a GIF of the
fine-mapping algorithm learning the causal variants.
<strong>WARNING!:</strong> Making this plot can take a long time if there's many iterations.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>if true outputs some progress messages</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p><a href='https://stephenslab.github.io/susieR/'>GitHub</a>
<a href='https://rss.onlinelibrary.wiley.com/doi/full/10.1111/rssb.12388'>Publication</a></p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p><strong>Notes on convergence:</strong>
<span class="pkg">susieR</span> will often give the warning: <code>IBSS algorithm did not converge in 100 iterations!</code>.
This means the results might not necessarily be reliable.
There's several things you can try to avoid this:</p><ul>
<li><p>Make sure <code>susieR</code> is up-to-date: <code><a href='https://rdrr.io/pkg/devtools/man/remote-reexports.html'> devtools::install_github("stephenslab/susieR@0.9.0")</a></code></p></li>
<li><p>Increase <code>max_causal</code> (e.g. 5 =&gt; 10).</p></li>
<li><p>Increase <code>max_iter</code> (e.g. 100 =&gt; 1000), though this will take longer.</p></li>
<li><p>Decrease the locus window size, which will also speed up the algorithm but potentially miss causal variants far from the lead SNP.</p></li>
</ul><p>Changing <code>estimate_prior_method</code> does not seem to affect covergence warnings.</p>
<p><strong>Notes on variance:</strong>
<a href='https://github.com/stephenslab/susieR/issues/90'>GitHub Issue</a>
If <code>estimate_residual_variance=TRUE</code> <em>without</em> providing <code>var_y</code>
<em>and</em> <code>L&gt;1</code>, <span class="pkg">susieR</span> will throw error:
<code>Estimating residual variance failed: the estimated value is negative</code>
Running <span class="pkg">susieR</span> with <code>var_y = var(b)</code> provides <em>exactly</em> the same results.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>"BST1"</span><span class='op'>)</span>; <span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>"LD_matrix"</span><span class='op'>)</span>;
<span class='co'># LD_matrix &lt;- readRDS("~/Desktop/Fine_Mapping/Data/GWAS/Nalls23andMe_2019/BST1/plink/UKB_LD.RDS")</span>
<span class='va'>finemap_DT</span> <span class='op'>&lt;-</span> <span class='fu'>SUSIE</span><span class='op'>(</span>subset_DT<span class='op'>=</span><span class='va'>BST1</span>, LD_matrix<span class='op'>=</span><span class='va'>LD_matrix</span>, estimate_residual_variance<span class='op'>=</span><span class='cn'>T</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "+ Preparing sample_size (N) column"
#&gt; [1] "++ Computing effective sample size."
#&gt; [1] "+ SUSIE:: sample_size= 216621"
#&gt; [1] "+ SUSIE:: max_causal = 5"
#&gt; [1] "+ Subsetting LD matrix and finemap_dat to common SNPs..."
#&gt; [1] "+ LD:: Removing unnamed rows/cols"
#&gt; [1] "+ LD:: Replacing NAs with 0"
#&gt; [1] "+ LD_matrix = 95 SNPs."
#&gt; [1] "+ finemap_dat = 6216 SNPs."
#&gt; [1] "+ 95 SNPs in common."
#&gt; [1] "+ SUSIE:: Using `susie_suff_stat()` from susieR v0.9.0"
#&gt; [1] "+ SUSIE:: Extracting Credible Sets..."</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


