<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Fine-map with SUSIE — SUSIE • echolocatoR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Fine-map with SUSIE — SUSIE"><meta property="og:description" content="Sum of Single Effects (SuSiE): Iterative Bayesian Step-wise Selection"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echolocatoR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/hexSticker.html">hexSticker</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Fine-map with SUSIE</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echolocatoR/blob/HEAD/R/SUSIE.R" class="external-link"><code>R/SUSIE.R</code></a></small>
    <div class="hidden name"><code>SUSIE.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Sum of Single Effects (SuSiE): Iterative Bayesian Step-wise Selection</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">SUSIE</span><span class="op">(</span>
  <span class="va">subset_DT</span>,
  <span class="va">LD_matrix</span>,
  dataset_type <span class="op">=</span> <span class="st">"GWAS"</span>,
  max_causal <span class="op">=</span> <span class="fl">5</span>,
  sample_size <span class="op">=</span> <span class="cn">NULL</span>,
  prior_weights <span class="op">=</span> <span class="cn">NULL</span>,
  PP_threshold <span class="op">=</span> <span class="fl">0.95</span>,
  scaled_prior_variance <span class="op">=</span> <span class="fl">0.001</span>,
  estimate_residual_variance <span class="op">=</span> <span class="cn">F</span>,
  estimate_prior_variance <span class="op">=</span> <span class="cn">T</span>,
  residual_variance <span class="op">=</span> <span class="cn">NULL</span>,
  max_iter <span class="op">=</span> <span class="fl">100</span>,
  estimate_prior_method <span class="op">=</span> <span class="st">"optim"</span>,
  manual_var_y <span class="op">=</span> <span class="cn">F</span>,
  rescale_priors <span class="op">=</span> <span class="cn">T</span>,
  plot_track_fit <span class="op">=</span> <span class="cn">F</span>,
  return_all_CS <span class="op">=</span> <span class="cn">T</span>,
  verbose <span class="op">=</span> <span class="cn">T</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://stephenslab.github.io/susieR/" class="external-link">GitHub</a>
<a href="https://rss.onlinelibrary.wiley.com/doi/full/10.1111/rssb.12388" class="external-link">Publication</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dataset_type</dt>
<dd><p>The kind dataset you're fine-mapping (e.g. GWAS, eQTL, tQTL).
This will also be used when creating the subdirectory where your results will be stored
(e.g. <em>Data/&lt;dataset_type&gt;/Kunkle_2019</em>).</p></dd>
<dt>max_causal</dt>
<dd><p>The maximum number of non-zero effects (and thus causal variants).</p></dd>
<dt>sample_size</dt>
<dd><p>The overall sample size of the study.
If none is given, and <strong>N_cases</strong> and <strong>N_controls</strong> columns are present,
then sample_size is inferred to be:  <code>max(N_cases) + max(N_controls)</code>.</p></dd>
<dt>prior_weights</dt>
<dd><p>A vector of length p, in which each entry
gives the prior probability that corresponding column of X has a
nonzero effect on the outcome, y.</p></dd>
<dt>PP_threshold</dt>
<dd><p>The minimum fine-mapped posterior probability for a SNP to be considered part of a Credible Set.
For example, <code>PP_threshold=.95</code> means that all Credible Set SNPs will be 95% Credible Set SNPs.</p></dd>
<dt>scaled_prior_variance</dt>
<dd><p>The prior variance, divided by
<code>var(y)</code> (or by <code>(1/(n-1))yty</code> for
<code>susie_suff_stat</code>); that is, the prior variance of each
non-zero element of b is <code>var(y) * scaled_prior_variance</code>. The
value provided should be either a scalar or a vector of length
<code>L</code>. If <code>estimate_prior_variance = TRUE</code>, this provides
initial estimates of the prior variances.</p></dd>
<dt>estimate_residual_variance</dt>
<dd><p>If
<code>estimate_residual_variance = TRUE</code>, the residual variance is
estimated, using <code>residual_variance</code> as an initial value. If
<code>estimate_residual_variance = FALSE</code>, the residual variance is
fixed to the value supplied by <code>residual_variance</code>.</p></dd>
<dt>estimate_prior_variance</dt>
<dd><p>If <code>estimate_prior_variance =
TRUE</code>, the prior variance is estimated (this is a separate
parameter for each of the L effects). If provided,
<code>scaled_prior_variance</code> is then used as an initial value for
the optimization. When <code>estimate_prior_variance = FALSE</code>, the
prior variance for each of the L effects is determined by the
value supplied to <code>scaled_prior_variance</code>.</p></dd>
<dt>residual_variance</dt>
<dd><p>Variance of the residual. If
<code>estimate_residual_variance = TRUE</code>, this value provides the
initial estimate of the residual variance. By default, it is set to
<code>var(y)</code> in <code>susie</code> and <code>(1/(n-1))yty</code> in
<code>susie_suff_stat</code>.</p></dd>
<dt>max_iter</dt>
<dd><p>Maximum number of IBSS iterations to perform.</p></dd>
<dt>estimate_prior_method</dt>
<dd><p>The method used for estimating prior
variance. When <code>estimate_prior_method = "simple"</code> is used, the
likelihood at the specified prior variance is compared to the
likelihood at a variance of zero, and the setting with the larger
likelihood is retained.</p></dd>
<dt>rescale_priors</dt>
<dd><p>If prior probabiltities are supplied,
rescale them from 0-1 (i.e. <code>rescaled_priors = priors / sum(priors)</code>).</p></dd>
<dt>plot_track_fit</dt>
<dd><p>Record each iteration and make a GIF of the
fine-mapping algorithm learning the causal variants.
<strong>WARNING!:</strong> Making this plot can take a long time if there's many iterations.</p></dd>
<dt>verbose</dt>
<dd><p>If <code>verbose = TRUE</code>, the algorithm's progress,
and a summary of the optimization settings, are printed to the
console.</p></dd>
</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p><strong>Notes on convergence:</strong>
<span class="pkg">susieR</span> will often give the warning: <code>IBSS algorithm did not converge in 100 iterations!</code>.
This means the results might not necessarily be reliable.
There's several things you can try to avoid this:</p><ul><li><p>Make sure <code>susieR</code> is up-to-date: <code> devtools::install_github("stephenslab/susieR@0.9.0")</code></p></li>
<li><p>Increase <code>max_causal</code> (e.g. 5 =&gt; 10).</p></li>
<li><p>Increase <code>max_iter</code> (e.g. 100 =&gt; 1000), though this will take longer.</p></li>
<li><p>Decrease the locus window size, which will also speed up the algorithm but potentially miss causal variants far from the lead SNP.</p></li>
</ul><p>Changing <code>estimate_prior_method</code> does not seem to affect covergence warnings.</p>
<p><strong>Notes on variance:</strong>
<a href="https://github.com/stephenslab/susieR/issues/90" class="external-link">GitHub Issue</a>
If <code>estimate_residual_variance=TRUE</code> <em>without</em> providing <code>var_y</code>
<em>and</em> <code>L&gt;1</code>, <span class="pkg">susieR</span> will throw error:
<code>Estimating residual variance failed: the estimated value is negative</code>
Running <span class="pkg">susieR</span> with <code>var_y = var(b)</code> provides <em>exactly</em> the same results.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"BST1"</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"LD_matrix"</span><span class="op">)</span>;</span>
<span class="r-in"><span class="co"># LD_matrix &lt;- readRDS("~/Desktop/Fine_Mapping/Data/GWAS/Nalls23andMe_2019/BST1/plink/UKB_LD.RDS")</span></span>
<span class="r-in"><span class="va">finemap_DT</span> <span class="op">&lt;-</span> <span class="fu">SUSIE</span><span class="op">(</span>subset_DT<span class="op">=</span><span class="va">BST1</span>, LD_matrix<span class="op">=</span><span class="va">LD_matrix</span>, estimate_residual_variance<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Preparing sample_size (N) column"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "++ Computing effective sample size."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ SUSIE:: sample_size= 216621"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ SUSIE:: max_causal = 5"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ Subsetting LD matrix and finemap_dat to common SNPs..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ LD:: Removing unnamed rows/cols"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ LD:: Replacing NAs with 0"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ LD_matrix = 95 SNPs."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ finemap_dat = 6216 SNPs."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ 95 SNPs in common."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ SUSIE:: Using `susie_suff_stat()` from susieR v0.11.92"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "+ SUSIE:: Extracting Credible Sets..."</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

