<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Covert and query — TABIX • echolocatoR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Covert and query — TABIX" />
<meta property="og:description" content="If it is not tabix format already
(determined by checking for a .tbi file of the same name in the same directory),
the full summary statistics file is converted into tabix format for super fast querying.
A query is then made using the min/max genomic positions to extract a locus-specific summary stats file." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/PD_loci_vignette.html">PD loci</a>
    </li>
    <li>
      <a href="../articles/QTL_pipeline_vignette.html">QTL pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/full_pipeline_vignette.html">Full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting vignette</a>
    </li>
    <li>
      <a href="../articles/summarise_vignette.html">Summarise vignette</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Covert and query</h1>
    <small class="dont-index">Source: <a href='https://github.com/RajLabMSSM/echolocatoR/blob/master/R/tabix.R'><code>R/tabix.R</code></a></small>
    <div class="hidden name"><code>TABIX.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>If it is not tabix format already
(determined by checking for a .tbi file of the same name in the same directory),
the full summary statistics file is converted into tabix format for super fast querying.
A query is then made using the min/max genomic positions to extract a locus-specific summary stats file.</p>
    </div>

    <pre class="usage"><span class='fu'>TABIX</span><span class='op'>(</span>
  <span class='va'>fullSS_path</span>,
  study_dir <span class='op'>=</span> <span class='cn'>NULL</span>,
  subset_path <span class='op'>=</span> <span class='cn'>NULL</span>,
  is_tabix <span class='op'>=</span> <span class='cn'>F</span>,
  chrom_col <span class='op'>=</span> <span class='st'>"CHR"</span>,
  chrom_type <span class='op'>=</span> <span class='cn'>NULL</span>,
  position_col <span class='op'>=</span> <span class='st'>"POS"</span>,
  <span class='va'>min_POS</span>,
  <span class='va'>max_POS</span>,
  <span class='va'>chrom</span>,
  save_subset <span class='op'>=</span> <span class='cn'>T</span>,
  nThread <span class='op'>=</span> <span class='fl'>1</span>,
  conda_env <span class='op'>=</span> <span class='st'>"echoR"</span>,
  verbose <span class='op'>=</span> <span class='cn'>T</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>fullSS_path</th>
      <td><p>Path to the full summary statistics file (GWAS or QTL) that you want to fine-map.
It is usually best to provide the absolute path rather than the relative path.</p></td>
    </tr>
    <tr>
      <th>chrom_col</th>
      <td><p>Name of the chromosome column in the full summary stats file.
Can be "chr1" or "1" format.
(<em>default: ="CHR"</em>)</p></td>
    </tr>
    <tr>
      <th>position_col</th>
      <td><p>Name of the genomic position column in the full summary stats file.
Must be in units of basepairs.
(<em>default: ="POS"</em>)</p></td>
    </tr>
    <tr>
      <th>min_POS</th>
      <td><p>Manually set the minimum genomic position for your locus subset.
<code>min_POS</code> can clip the window size set by <code>bp_distance</code>.
Can also be a list of positions (one for each locus) (e.g. <code>min_POS=top_SNPs$min_POS</code>).</p></td>
    </tr>
    <tr>
      <th>max_POS</th>
      <td><p>Manually set the maximum genomic position for your locus subset.
<code>max_POS</code> can clip the window size set by <code>bp_distance</code>.
Can also be a list of positions (one for each locus) (e.g. <code>max_POS=top_SNPs$max_POS</code>).</p></td>
    </tr>
    <tr>
      <th>conda_env</th>
      <td><p>The name of a conda environment to use.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Whether <span class="pkg">echolocatoR</span> should be verbose or silent.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>data.table of locus subset summary statistics</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other query functions: 
<code><a href='TABIX.convert_file.html'>TABIX.convert_file</a>()</code>,
<code><a href='TABIX.query.html'>TABIX.query</a>()</code>,
<code><a href='extract_SNP_subset.html'>extract_SNP_subset</a>()</code>,
<code><a href='import_topSNPs.html'>import_topSNPs</a>()</code>,
<code><a href='query_by_coordinates_merged.html'>query_by_coordinates_merged</a>()</code>,
<code><a href='query_by_coordinates.html'>query_by_coordinates</a>()</code>,
<code><a href='query_by_gene.html'>query_by_gene</a>()</code>,
<code><a href='query_by_probe.html'>query_by_probe</a>()</code>,
<code><a href='query_fullSS.html'>query_fullSS</a>()</code>,
<code><a href='query_handler.html'>query_handler</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>"locus_dir"</span><span class='op'>)</span>; <span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>"Nalls_top_SNPs"</span><span class='op'>)</span>
<span class='va'>fullSS_path</span> <span class='op'>&lt;-</span> <span class='st'>"./results/GWAS/Nalls23andMe_2019/_genome_wide/nallsEtAl2019_allSamples_allVariants.mod.txt.gz"</span>
<span class='va'>top_SNPs</span> <span class='op'>&lt;-</span> <span class='fu'><a href='import_topSNPs.html'>import_topSNPs</a></span><span class='op'>(</span>topSS <span class='op'>=</span> <span class='va'>Nalls_top_SNPs</span>, chrom_col <span class='op'>=</span> <span class='st'>"CHR"</span>, position_col <span class='op'>=</span> <span class='st'>"BP"</span>, snp_col<span class='op'>=</span><span class='st'>"SNP"</span>, pval_col<span class='op'>=</span><span class='st'>"P, all studies"</span>, effect_col<span class='op'>=</span><span class='st'>"Beta, all studies"</span>, gene_col<span class='op'>=</span><span class='st'>"Nearest Gene"</span>, group_by_locus <span class='op'>=</span> <span class='cn'>T</span>,locus_col <span class='op'>=</span> <span class='st'>"Nearest Gene"</span><span class='op'>)</span>
<span class='va'>top_SNPs_BST1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/subset.html'>subset</a></span><span class='op'>(</span><span class='va'>top_SNPs</span>, <span class='va'>Locus</span><span class='op'>==</span><span class='st'>'BST1'</span><span class='op'>)</span>
<span class='va'>bp_distance</span> <span class='op'>&lt;-</span> <span class='fl'>1e+06</span>
<span class='va'>min_POS</span> <span class='op'>&lt;-</span> <span class='va'>top_SNPs_BST1</span><span class='op'>$</span><span class='va'>POS</span> <span class='op'>-</span> <span class='va'>bp_distance</span>
<span class='va'>max_POS</span> <span class='op'>&lt;-</span> <span class='va'>top_SNPs_BST1</span><span class='op'>$</span><span class='va'>POS</span> <span class='op'>+</span> <span class='va'>bp_distance</span>
<span class='va'>subset_file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>results_path</span>,<span class='st'>"BST1_Nalls23andMe_2019_subset.tsv.gz"</span><span class='op'>)</span>
<span class='va'>dat</span> <span class='op'>&lt;-</span> <span class='fu'>TABIX</span><span class='op'>(</span>fullSS_path<span class='op'>=</span><span class='va'>fullSS_path</span>, subset_path<span class='op'>=</span><span class='st'>"auto"</span>, min_POS<span class='op'>=</span><span class='va'>min_POS</span>, max_POS<span class='op'>=</span><span class='va'>max_POS</span>, chrom<span class='op'>=</span><span class='va'>top_SNPs_BST1</span><span class='op'>$</span><span class='va'>CHR</span><span class='op'>)</span>
<span class='op'>}</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


