<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Conduct enrichment tests for each annotation — XGR.iterate_enrichment • echolocatoR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Conduct enrichment tests for each annotation — XGR.iterate_enrichment" />
<meta property="og:description" content="XGR uses a binomial enrichment tests for each annotation." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echolocatoR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/full_pipeline_vignette.html">echolocatoR: full pipeline vignette</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">echolocatoR: plotting vignette</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RajLabMSSM/echolocatoR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Conduct enrichment tests for each annotation</h1>
    <small class="dont-index">Source: <a href='https://github.com/RajLabMSSM/echolocatoR/blob/master/R/XGR.R'><code>R/XGR.R</code></a></small>
    <div class="hidden name"><code>XGR.iterate_enrichment.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>XGR uses a binomial enrichment tests for each annotation.</p>
    </div>

    <pre class="usage"><span class='fu'>XGR.iterate_enrichment</span>(
  <span class='no'>subset_DT</span>,
  <span class='kw'>foreground_filter</span> <span class='kw'>=</span> <span class='st'>"Consensus_SNP"</span>,
  <span class='kw'>background_filter</span> <span class='kw'>=</span> <span class='st'>"leadSNP"</span>,
  <span class='kw'>lib.selections</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"ENCODE_TFBS_ClusteredV3_CellTypes"</span>,
    <span class='st'>"ENCODE_DNaseI_ClusteredV3_CellTypes"</span>, <span class='st'>"Broad_Histone"</span>, <span class='st'>"FANTOM5_Enhancer"</span>,
    <span class='st'>"Segment_Combined_Gm12878"</span>, <span class='st'>"TFBS_Conserved"</span>, <span class='st'>"ReMap_PublicAndEncode_TFBS"</span>,
    <span class='st'>"Blueprint_VenousBlood_Histone"</span>, <span class='st'>"Blueprint_DNaseI"</span>, <span class='st'>"FANTOM5_CAT_Cell"</span>,
    <span class='st'>"FANTOM5_CAT_MESH"</span>, <span class='st'>"GWAScatalog_alltraits"</span>),
  <span class='kw'>save_path</span> <span class='kw'>=</span> <span class='no'>F</span>,
  <span class='kw'>nCores</span> <span class='kw'>=</span> <span class='fl'>4</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>subset_DT</th>
      <td><p>Data.frame with at least the following columns:</p><dl'>
<dt>SNP</dt><dd><p>SNP RSID</p></dd>
<dt>CHR</dt><dd><p>chromosome</p></dd>
<dt>POS</dt><dd><p>position</p></dd>

</dl></td>
    </tr>
    <tr>
      <th>foreground_filter</th>
      <td><p>Specify foreground by filtering SNPs in <code>subset_DT</code>.
Write filter as a string (or <code>NULL</code> to include all SNPs).</p></td>
    </tr>
    <tr>
      <th>background_filter</th>
      <td><p>Specify background by filtering SNPs in <code>subset_DT</code>.
Write filter as a string (or <code>NULL</code> to include all SNPs).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other XGR: 
<code><a href='DT_to_GRanges.html'>DT_to_GRanges</a>()</code>,
<code><a href='GRanges_to_BED.html'>GRanges_to_BED</a>()</code>,
<code><a href='XGR.download_and_standardize.html'>XGR.download_and_standardize</a>()</code>,
<code><a href='XGR.filter_assays.html'>XGR.filter_assays</a>()</code>,
<code><a href='XGR.filter_sources.html'>XGR.filter_sources</a>()</code>,
<code><a href='XGR.import_annotations.html'>XGR.import_annotations</a>()</code>,
<code><a href='XGR.iterate_overlap.html'>XGR.iterate_overlap</a>()</code>,
<code><a href='XGR.merge_and_process.html'>XGR.merge_and_process</a>()</code>,
<code><a href='XGR.plot_enrichment.html'>XGR.plot_enrichment</a>()</code>,
<code><a href='XGR.plot_peaks.html'>XGR.plot_peaks</a>()</code>,
<code><a href='XGR.prepare_foreground_background.html'>XGR.prepare_foreground_background</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"merged_DT"</span>)
<span class='no'>enrich_res</span> <span class='kw'>&lt;-</span> <span class='fu'>XGR.iterate_enrichment</span>(<span class='kw'>subset_DT</span><span class='kw'>=</span><span class='no'>merged_DT</span>, <span class='kw'>foreground_filter</span> <span class='kw'>=</span> <span class='st'>"Consensus_SNP"</span>, <span class='kw'>background_filter</span> <span class='kw'>=</span> <span class='st'>"leadSNP"</span>, <span class='kw'>lib.selections</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"ENCODE_TFBS_ClusteredV3_CellTypes"</span>), <span class='kw'>nCores</span><span class='kw'>=</span><span class='fl'>1</span>)</div><div class='output co'>#&gt; [1] "XGR:: 224 SNPs in foreground."
#&gt; [1] "XGR:: 81 SNPs in background"
#&gt; [1] "XGR:: Testing enrichment:  ENCODE_TFBS_ClusteredV3_CellTypes"</div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:13</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>'ENCODE_TFBS_ClusteredV3_CellTypes' (from http://galahad.well.ox.ac.uk/bigdata/ENCODE_TFBS_ClusteredV3_CellTypes.RData) has been loaded into the working environment (at 2020-06-26 12:24:32)</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:32</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 19 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:32</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:32</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:32</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:32) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:34) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:35) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (2)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 50</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:36</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 4 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:36</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:36) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:36) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:36) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:36) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:36) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:36) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:38) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:40) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (3)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 59</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:40</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 4 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:40</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:40) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:40) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:40) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:40) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:40) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:40) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:40) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (2)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 8</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:41</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:41</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:41</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:41</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:41</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:41</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:41</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:41</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:41) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:42) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:42</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:42</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:42) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:42) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:42) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:42) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:42) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:42) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:43) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:43) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 24</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:43</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:43</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:43) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:43) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:43) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:43) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:43) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:43) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:44</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:44) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:45</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:45) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:46</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:46</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:46</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:46</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:46</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:46</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:46</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:46</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:46) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 8</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:47</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:47</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:47</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:47</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 5</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:47</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:47</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:47) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:48</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:48</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 5</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:48</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:48</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (2)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:48</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:48</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:48</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:48</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:48) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 7</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:49</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:49</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 3</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:49</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:49</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (2)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:49</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:49</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:49) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:50</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:50</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:50</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:50</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:50</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:50</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:50</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:50</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:50) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:51) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:54) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (4)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 76</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:58</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 8 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:58</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:58</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:58</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 6</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:58</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:58</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:58) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:59</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:59</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:24:59</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:24:59</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:24:59) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:01) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:03) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 55</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:03</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 4 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:03</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:03) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:03) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:03) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:03) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:03) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:04) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:07) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (2)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 100</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:10</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 7 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:10</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:10</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:10</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:10) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 5</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:11</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:11</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 5</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:11</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:11</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:11) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:12</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:12) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:13</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:13</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:13</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:13</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:13</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:13</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:13</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:13</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:13) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 5</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:14</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:14</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:14</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:14</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:14</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:14</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:14</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:14</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (2)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 4</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:14</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:14</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:14) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 5</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:15</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:15</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:15</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:15</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:15</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:15</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:15</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:15</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:15</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:15</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:15) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:16</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:16</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:16</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:16</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:16</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:16</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:16</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:16</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:16) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:17</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:17</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:17</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:17</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 4</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:17</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:17</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:17</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:17</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:17) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:18</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:18</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:18</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:18</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:18</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:18</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:18) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 5</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:19</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:19</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:19</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:19</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 4</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:19</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:19</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 4</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:19</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:19</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:19) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 3</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:20</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:20</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (1)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:20</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:20</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:20</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:20</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 2</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:20</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:20</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:20) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:21</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 1 secs</span></div><div class='output co'>#&gt; <span class='message'>Start at 2020-06-26 12:25:21</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>First, import the files formatted as 'data.frame' (2020-06-26 12:25:21) ...</span></div><div class='output co'>#&gt; <span class='message'>	import the data file (2020-06-26 12:25:21) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>	import the annotation file (2020-06-26 12:25:21) ...</span></div><div class='output co'>#&gt; <span class='message'>		The file 'annotation.file' is not provided, so built-in RData will be used instead!</span></div><div class='output co'>#&gt; <span class='message'>	import the background file (2020-06-26 12:25:21) ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>Second, construct GenomicRanges object (2020-06-26 12:25:21) ...</span></div><div class='output co'>#&gt; <span class='message'>Third, define the background (2020-06-26 12:25:21) ...</span></div><div class='output co'>#&gt; <span class='message'>	the given background regions but restricted to the annotatable are used as the background (2020-06-26 12:25:21) ...</span></div><div class='output co'>#&gt; <span class='message'>Forth, perform enrichment analysis at 'bases' resolution with 'one-tail' p-values (2020-06-26 12:25:21) ...</span></div><div class='output co'>#&gt; <span class='message'>	the number of bases: data (0)</span></div><div class='output co'>#&gt; <span class='message'>	the number of annotations: 1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>End at 2020-06-26 12:25:21</span></div><div class='output co'>#&gt; <span class='message'>Runtime in total is: 0 secs</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


